---
title: "MoTrPAC R Package Workshop"
author: "Nicole Gay"
date: "2023-01-05"
output: 
  rmarkdown::html_document:
    toc: true
    code_folding: show 
    toc_depth: 4
    toc_float:
      collapsed: false
      smooth_scroll: false
    theme: sandstone
    highlight: tango
---

```{r setup}
# MoTrPAC R packages 
library(MotrpacRatTraining6mo)
library(MotrpacRatTraining6moData)

# for pathway enrichment
library(foreach)
library(gprofiler2)
library(doParallel)
library(IHW)

# for plotting
library(ggplot2)

knitr::opts_chunk$set(echo = TRUE)
```

## Background 

### About these R package 
These package provides functions to fetch, explore, and reproduce the processed 
data and downstream analysis results presented in the main paper for the first 
large-scale multi-omic multi-tissue endurance exercise training study conducted 
in young adult rats by the Molecular Transducers of Physical Activity Consortium 
(MoTrPAC). Find the [preprint on bioRxiv](https://www.biorxiv.org/content/10.1101/2022.09.21.508770v2).

While functions in [MotrpacRatTraining6moData](https://motrpac.github.io/MotrpacRatTraining6moData) 
can be used by themselves, they were primarily written to analyze data in the 
[MotrpacRatTraining6moData](https://motrpac.github.io/MotrpacRatTraining6moData)
R package. 

**Note:** Data objects from the [MotrpacRatTraining6moData](https://motrpac.github.io/MotrpacRatTraining6moData) 
package are indicated as variable names in all caps, e.g. `PHENO`, `TRNSCRPT_LIVER_RAW_COUNTS`. 

**Tip:** To learn more about any data object or function, use `?` to retrieve the documentation,
e.g., `?METAB_FEATURE_ID_MAP`, `?load_sample_data`. Note that both `MotrpacRatTraining6mo` and 
`MotrpacRatTraining6moData` must be installed and loaded with `library()` for this to work. 

### About MoTrPAC 
MoTrPAC is a national research consortium designed to discover and perform 
preliminary characterization of the range of molecular transducers (the 
"molecular map") that underlie the effects of physical activity in humans. 
The program's goal is to study the molecular changes that occur during and after 
exercise and ultimately to advance the understanding of how physical activity 
improves and preserves health. The six-year program is the largest targeted NIH 
investment of funds into the mechanisms of how physical activity improves health 
and prevents disease. See [motrpac.org](https://www.motrpac.org/) and 
[motrpac-data.org](https://motrpac-data.org/) for more details. 

### Study design 
Details of the experimental design can be found in the [supplementary methods of the bioRxiv preprint](https://www.biorxiv.org/content/biorxiv/early/2022/10/05/2022.09.21.508770/DC1/embed/media-1.pdf?download=true). 
Briefly, 6-month-old young adult rats 
were subjected to progressive endurance exercise training
for 1, 2, 4, or 8 weeks, with tissues collected 48 hours after the last training bout. 
Sex-matched sedentary, untrained rats were used as controls. Whole blood, plasma, 
and 18 solid tissues were analyzed using genomics, proteomics, metabolomics, 
and protein immunoassay technologies, with most assays performed in a subset of these tissues. 
Depending on the assay, between 3 and 6 replicates per sex per time point were
analyzed. 

### Tissue and assay abbreviations  
It is important to be aware of the tissue and assay abbreviations because they are used
to name data objects and define arguments for many functions. 

#### Tissues
* **ADRNL:** adrenal gland  
* **BAT:** brown adipose tissue  
* **BLOOD:** whole blood  
* **COLON:** colon  
* **CORTEX:** cerebral cortex  
* **HEART:** heart  
* **HIPPOC:** hippocampus  
* **HYPOTH:** hypothalamus  
* **KIDNEY:** kidney  
* **LIVER:** liver  
* **LUNG:** lung  
* **OVARY:** ovaries (female gonads)  
* **PLASMA:** plasma from blood
* **SKM-GN:** gastrocnemius (leg skeletal muscle)  
* **SKM-VL:** vastus lateralis (leg skeletal muscle) 
* **SMLINT:** small intestine  
* **SPLEEN:** spleen  
* **TESTES:** testes (male gonads)  
* **VENACV:** vena cava  
* **WAT-SC:** subcutaneous white adipose tissue  

#### Assays/omes
* **ACETYL:** acetylproteomics; protein site acetylation  
* **ATAC:** chromatin accessibility, ATAC-seq data  
* **IMMUNO:** multiplexed immunoassays (cytokines and hormones)  
* **METAB:** metabolomics and lipidomics  
* **METHYL:** DNA methylation, RRBS data   
* **PHOSPHO:** phosphoproteomics; protein site phosphorylation  
* **PROT:** global proteomics; protein abundance  
* **TRNSCRPT:** transcriptomics, RNA-Seq data  
* **UBIQ:** ubiquitylome; protein site ubiquitination  

The vectors of abbreviations are also available in `TISSUE_ABBREV` and `ASSAY_ABBREV`: 
```{r}
TISSUE_ABBREV
ASSAY_ABBREV
```

### Guiding question for this tutorial

Let's see how we can use these R packages to identify an interesting shared
multi-omic signal in the heart (HEART) and gastrocnemius (SKM-GN) muscle tissues. 

## Sample-level data

First, let's look at a set of functions used to easily load data from the 
`MotrpacRatTraining6moData` R package. 

Use `load_sample_data()` to load sample-level data for a specific ome and tissue. 
```{r load_sample_data}
?load_sample_data

# heart protein expression 
heart_prot = load_sample_data("HEART", "PROT")

# heart immunoassay data
heart_immuno = load_sample_data("HEART", "IMMUNO")

# gastroc gene expression
gastroc_rna = load_sample_data("SKM-GN", "TRNSCRPT")

# gastroc metabolomics
gastroc_metab = load_sample_data("SKM-GN", "METAB")
```

Let's look at the format of these data frames.
```{r}
head(heart_prot)
```

The `feature` column is non-NA only for training-regulated features, which provides 
a convenient way of identifying this subset of features. 

Sometimes it is more convenient to have all-numeric data frames. We can easily
convert these data frames to this format with `df_to_numeric()`. 
```{r df_to_numeric}
head(df_to_numeric(heart_prot, rownames = "feature_ID"))
```

Note that `load_sample_data()` also prints a message with the name of the object
in `MotrpacRatTraining6moData` that is being returned. Therefore, instead of using
`load_sample_data()`, we could call the object directly. 
```{r}
head(PROT_HEART_NORM_DATA)
```

Let's plot the sample-level data for a single differential feature.
```{r, out.width="50%", out.height="50%", fig.show="hold"}
head(heart_prot[!is.na(heart_prot$feature),])
plot_feature_normalized_data(feature="PROT;HEART;XP_003754359.1", 
                             add_gene_symbol = TRUE)
plot_feature_normalized_data(feature="PROT;HEART;XP_003754359.1", 
                             add_gene_symbol = TRUE,
                             facet_by_sex = TRUE)
```

We can also use `combine_normalized_data()` to load normalized data from multiple omes or tissues
simultaneously. For example, let's load all non-epigenetic data from the gastroc.
```{r gastroc data}
all_gastroc = combine_normalized_data(tissues="SKM-GN", include_epigen = FALSE)
head(all_gastroc)
# how many features?
nrow(all_gastroc)
# how many differential features?
nrow(all_gastroc[!is.na(all_gastroc$feature),])
# what omes?
table(all_gastroc$assay[!is.na(all_gastroc$feature)])
```

Note that column names are now PIDs instead of vial labels. This is because 
measurements from multiple vial labels (samples) from the same PID (animal) are included in the same column. 

## Principal component analysis 

Now that we know how to get the sample-level data, let's perform PCA with the heart 
and gastroc data. PCA is best used on one data type at a time, so let's use gene expression data. 
```{r pca}
# load data
heart = load_sample_data(tissue = "HEART", assay = "TRNSCRPT", exclude_outliers = TRUE)
gastroc = load_sample_data(tissue = "SKM-GN", assay = "TRNSCRPT", exclude_outliers = TRUE)

# convert to numeric
heart = df_to_numeric(heart)
gastroc = df_to_numeric(gastroc)

# merge
merged = merge(heart, gastroc, by="row.names")
merged = df_to_numeric(merged, rownames="Row.names")

# perform PCA 
pca = prcomp(t(merged), center = TRUE)

# merge PCs with PHENO
pcs = as.data.frame(pca$x)
pheno = unique(PHENO[,c("viallabel","group","sex","tissue")])
head(pheno)
data_for_plotting = merge(pheno, pcs, by.x="viallabel", by.y="row.names")

# minimal
ggplot(data_for_plotting, aes(x=PC1, PC2, colour=tissue)) +
  geom_point()

# prettier
ggplot(data_for_plotting, aes(x=PC1, y=PC2, fill=tissue, shape=sex)) + 
  geom_point(size=3) +
  scale_fill_manual(values=TISSUE_COLORS) +
  scale_shape_manual(values=c(male=21, female=24)) +
  theme_bw() +
  guides(fill=guide_legend(override.aes = list(shape=21))) +
  labs(title="PCA of heart and gastroc RNA-seq data") +
  theme(plot.title = element_text(hjust=0.5))
```

Now let's perform PCA for a single tissue and label samples by training group. 
```{r}
# perform PCA 
pca = prcomp(t(gastroc), center = TRUE)

# merge PCs with PHENO
pcs = as.data.frame(pca$x)
pheno = unique(PHENO[,c("viallabel","group","sex","tissue")])
data_for_plotting = merge(pheno, pcs, by.x="viallabel", by.y="row.names")

# minimal
ggplot(data_for_plotting, aes(x=PC1, PC2)) +
  geom_point()

# prettier
ggplot(data_for_plotting, aes(x=PC1, y=PC2, fill=group, shape=sex)) + 
  geom_point(size=3) +
  scale_fill_manual(values=GROUP_COLORS) +
  scale_shape_manual(values=c(male=21, female=24)) +
  theme_bw() +
  guides(fill=guide_legend(override.aes = list(shape=21))) +
  labs(title="PCA of gastroc RNA-seq data") +
  theme(plot.title = element_text(hjust=0.5))
```

## Differential analysis results 

Next, let's load the differential analysis results. Recall there are two types of differential analysis results:
*timewise* and *training*. We use the adjusted p-value from the *training* differential
analysis to identify training-regulated features (`selection_fdr`); we use the summary statistics from the 
*timewise* differential analysis to quantify time- and sex- specific training effects. The 
results below are the *timewise* results **and** adjusted p-values from the *training* results. 
Lesser-used additional *training* summary statistics are available through `load_training_da()`. 

First, let's consider training-regulated features only. These are conveniently provided 
in `TRAINING_REGULATED_FEATURES`, which we can easily subset to any data set of interest. 
```{r}
head(TRAINING_REGULATED_FEATURES)
heart_gastroc_sig = TRAINING_REGULATED_FEATURES[TRAINING_REGULATED_FEATURES$tissue %in% c("HEART","SKM-GN"),]
# how many differential features?
nrow(heart_gastroc_sig)
# what tissues and omes?
table(heart_gastroc_sig$tissue, heart_gastroc_sig$assay)
```

We can also load the full set of differential analysis results. This consumes some
memory, so let's skip this for now. 
```{r all da, eval=F}
# for a single tissue
heart_da = combine_da_results(tissues="HEART")
# for multiple tissues
heart_gastroc_da = combine_da_results(tissues=c("SKM-GN","HEART"))
```

Plot the differential analysis results for a single training-regulated feature.
```{r, out.width="50%", out.height="50%", fig.show="hold"}
# plot the first differential feature
head(heart_gastroc_sig)
plot_feature_logfc(feature = "ACETYL;HEART;NP_001003673.1_K477k", facet_by_sex = TRUE, add_gene_symbol = TRUE)
# how does this compare to the sample-level data?
plot_feature_normalized_data(feature = "ACETYL;HEART;NP_001003673.1_K477k", facet_by_sex = TRUE, add_gene_symbol = TRUE)
```

## Explore graphical clustering results

Across all tissues and omes, we had over 30,000 training-regulated features. We 
used a graphical approach to identify groups of features with similar temporal dynamics. 
See more details in the ["Get Started" tutorial](https://motrpac.github.io/MotrpacRatTraining6mo/articles/MotrpacRatTraining6mo.html#bayesian-graphical-clustering). 

The graph representation of the `repfdr` results are provided in `GRAPH_COMPONENTS` (node and edge lists) and 
`GRAPH_STATES` (data frame representation). Features are specified in the format "[ASSAY_ABBREV];[TISSUE_ABBREV];[feature_ID]",
e.g. "PHOSPHO;SKM-GN;NP_001006973.1_S34s", "METAB;SKM-GN;1-methylnicotinamide". 

### Plot graphs of differential feature trajectories

Let's start exploring the heart and gastroc graphical clustering results
features by plotting the graphs themselves. 

Start by plotting graphs for gastroc and heart separately. 
```{r, fig.width=9, fig.height=6}
# gastroc
get_tree_plot_for_tissue(tissues=c("SKM-GN"))
get_tree_plot_for_tissue(tissues=c("SKM-GN"),
                         max_trajectories = 5)
get_tree_plot_for_tissue(tissues=c("SKM-GN"),
                         max_trajectories = 3,
                         parallel_edges_by_ome = TRUE,
                         curvature = 1,
                         edge_alpha_range = c(0.5, 1))
get_tree_plot_for_tissue(tissues=c("SKM-GN"),
                         omes=c("PROT","PHOSPHO","TRNSCRPT"),
                         max_trajectories = 3,
                         parallel_edges_by_ome = TRUE,
                         curvature = 0.5,
                         edge_alpha_range = c(0.5, 1))

# heart
get_tree_plot_for_tissue(tissues=c("HEART"),
                         max_trajectories = 3,
                         parallel_edges_by_ome = TRUE,
                         curvature = 1,
                         edge_alpha_range = c(0.5, 1))
get_tree_plot_for_tissue(tissues=c("HEART"),
                         omes=c("PROT","PHOSPHO","TRNSCRPT","METAB","ACETYL"),
                         max_trajectories = 3,
                         parallel_edges_by_ome = TRUE,
                         curvature = 0.8,
                         edge_alpha_range = c(0.5, 1))
```

Now plot both heart and gastroc features together. For the sake of visualization, 
let's limit it to popular omes in both tissues. 
```{r, fig.width=9, fig.height=6}
# edges by tissue
get_tree_plot_for_tissue(tissues=c("SKM-GN","HEART"),
                         omes=c("PROT","PHOSPHO","TRNSCRPT","METAB"),
                         max_trajectories = 3,
                         parallel_edges_by_tissue = TRUE,
                         curvature = 0.5,
                         edge_alpha_range = c(0.5, 1))
# edges by ome
get_tree_plot_for_tissue(tissues=c("SKM-GN","HEART"),
                         omes=c("PROT","PHOSPHO","TRNSCRPT","METAB"),
                         max_trajectories = 3,
                         parallel_edges_by_ome = TRUE,
                         curvature = 0.8,
                         edge_alpha_range = c(0.5, 1))
```

There's a lot going on in each of these three trajectories. **We will focus on these
three paths in the heart and gastroc for the rest of this workshop**. 

### Get features underlying graphical clusters 

Let's determine the features that belong to each of these paths. This can be 
done one of a few ways. 

#### Use `extract_tissue_sets()`

`extract_tissue_sets()` gives us features in the `k` largest nodes, edges, and paths
for a given tissue or tissues. We already know that the 3 paths shown above were the 3
largest paths in the heart and gastroc. 
```{r}
clusters = extract_tissue_sets(tissues=c("HEART","SKM-GN"), k=3)
# limit to paths only
names(clusters)
clusters = clusters[grepl("->", names(clusters))]
names(clusters)
lapply(clusters, head)
```

This list format is fine, but we can also convert it to a data frame format if
that is easier to work with. 
```{r}
clusters_df = check_cluster_res_format(clusters)
head(clusters_df)
table(clusters_df$cluster, clusters_df$tissue)
```

#### Use `extract_main_clusters()`

Similarly, `extract_main_clusters()` returns the subset of graphical clusters for which pathway 
enrichment was performed for the landscape manuscript, 
namely the 2 largest nodes, 2 largest edges, 10 largest non-null paths, and all 8-week nodes.  
```{r extract main clusters}
clusters = extract_main_clusters()
names(clusters)
```

#### Subset `GRAPH_STATES`

Perhaps the easiest way to get features that belong to a specific graphical cluster
is to subset `GRAPH_STATES`. 
In addition to subsetting the `path` column as shown here, 
we can use the `state_*w` columns to select features in any node, edge, or path. 
```{r graphical clustering results}
head(GRAPH_STATES)

clusters = list()
paths = c("1w_F1_M1->2w_F1_M1->4w_F1_M1->8w_F1_M1", "1w_F-1_M-1->2w_F-1_M-1->4w_F-1_M-1->8w_F-1_M-1", "1w_F0_M1->2w_F0_M1->4w_F1_M1->8w_F1_M1")
tissues = c("SKM-GN", "HEART")
for (path in paths){
  for(t in tissues){
    name = paste0(t, ":", path)
    clusters[[name]] = as.vector(na.omit(GRAPH_STATES$feature[GRAPH_STATES$path == path & 
                                                                GRAPH_STATES$tissue == t]))
  } 
}
lapply(clusters, length)
```

Let's move forward with this list of features per cluster. We'll rename them
to make them easier to keep track of. 
```{r}
MY_CLUSTERS = clusters
MY_CLUSTER_DF = check_cluster_res_format(clusters)
```

### Plot features in graphical clusters 

We can visualize the composition of these clusters in terms of tissues and omes. 
```{r, fig.width=9, fig.height=3}
plot_features_per_cluster(MY_CLUSTERS) # could also use MY_CLUSTER_DF as input
```

We can also look at the trajectories of these features' abundances over the 
training time course. Let's look at a single cluster.   
```{r}
plot_feature_trajectories(MY_CLUSTERS$`SKM-GN:1w_F1_M1->2w_F1_M1->4w_F1_M1->8w_F1_M1`, 
                          title="SKM-GN:1w_F1_M1->2w_F1_M1->4w_F1_M1->8w_F1_M1")
```

## Pathway enrichment

One important tool we use to interpret the biology underlying graphical clusters
is pathway enrichment. In this section, we look at functions 
to reproduce the pathway enrichment results in the manuscript and generate our own.

### Explore existing results 

For the manuscript, we performed pathway enrichment for the 2 largest nodes, 
2 largest edges, 10 largest non-null paths, and all 8-week nodes from the graphical 
representation of training-regulated features in each tissue. This set of 
graphical clusters can be extracted using `extract_main_clusters()`. 
Pathway enrichment results were adjusted over *all* tests using 
[IHW](https://www.nature.com/articles/nmeth.3885) with tissue as a covariate. 
These pathway enrichment results are available in `GRAPH_PW_ENRICH`. 

Because our 3 paths of interest are among the 10 largest paths in both tissues, the
pathway enrichment results for these features are already available in `GRAPH_PW_ENRICH`.
Let's take a look. 
```{r GRAPH_PW_ENRICH}
head(GRAPH_PW_ENRICH)
writeLines(colnames(GRAPH_PW_ENRICH))
```

The `cluster` column in `GRAPH_PW_ENRICH` prepends the cluster name with the tissue abbreviation.
Let's get all significant heart and gastroc enrichment results for these three paths. 
```{r}
enrich_res = GRAPH_PW_ENRICH[GRAPH_PW_ENRICH$cluster %in% names(MY_CLUSTERS) & GRAPH_PW_ENRICH$adj_p_value < 0.1,]
table(enrich_res$cluster, enrich_res$ome)
```

### Perform pathway enrichment 

Before we get to visualizing these pathway enrichments, let's look at ways to perform
our own pathway enrichment analyses. 

#### Enrichment with sets of MoTrPAC features 

If we want to perform enrichment for MoTrPAC features, use `cluster_pathway_enrichment()`. 
`cluster_pathway_enrichment()` is a wrapper for 
[gprofiler2::gost()](https://rdrr.io/cran/gprofiler2/man/gost.html)
and [FELLA::enrich()](https://rdrr.io/bioc/FELLA/man/enrich-funs.html), 
where metabolomics pathway enrichment is performed for KEGG pathways, and gene-centric
pathway enrichment can be performed for any set of 
[pathway databases accepted by g:Profiler](https://rdrr.io/cran/gprofiler2/man/gost.html). 

Here we will regenerate the pathway enrichment results for our clusters of interest (`MY_CLUSTERS`). 
In this case, clusters are determined using the graphical clustering approach
described above, but in practice the clusters could be identified using any approach (e.g., k-means, fuzzy C-means). 

Keep in mind these are multi-omic graphical clusters, 
and pathway enrichment is performed for each set of features per cluster per tissue per ome. 
We can run tests in parallel by specifying `num_cores`, but since this sometimes has
unexpected behavior, let's just use one core today, which is the default. 
```{r cluster_pathway_enrichment}
res = cluster_pathway_enrichment(MY_CLUSTERS, 
                                 databases = c("REAC", "KEGG")) 
```

We get a few warnings and messages. 

* First, we get a warning that pathway enrichment will not be 
performed with metabolites because `kegg_db_destination=NULL` (default parameter).   
* Second, we are told that the analysis will be run on a single core. If you are 
performing pathway enrichment for more than a few clusters, the runtime will be 
dramatically reduced if you parallelize the analysis on more than one core. You can do this 
by specifying the `num_cores` argument. `cluster_pathway_enrichment()` internally uses 
`doParallel::registerDoParallel(num_cores)` to register more than one core.  
* We get a few messages about METHYL or ATAC-seq features not being specified 
in the background (`GENE_UNIVERSES$ensembl_gene` by default) and are therefore excluded
from the analysis. This is because we decided to define ATAC-seq (ATAC) and RRBS (METHYL) universes 
according to the *expressed genes* in each tissue, i.e., using the RNA-seq (TRNSCRPT) data.
Therefore, there are some differential ATAC/METHYL features where the corresponding 
gene is not regulated by training. 

Let's look at how many significantly enriched pathways we get at 5% FDR.
```{r}
sig = res[res$adj_p_value < 0.05, c("cluster","tissue","ome","term_name","adj_p_value","intersection")]
table(sig$cluster, sig$ome)
```

Note that we get different numbers of significant enrichments compared to when we look at `GRAPH_PW_ENRICH`.
`GRAPH_PW_ENRICH` was generated more than a year ago, and these pathways databases have changed slightly
since then. The multiple testing adjustment procedures are also different - here, 
we are adjusting over a different set of tests than when we applied IHW to all tests in `GRAPH_PW_ENRICH`. 

#### Enrichment with single list of genes 

If you want to perform enrichment for a single list of genes, use `gene_pathway_enrichment()`.
While `cluster_pathway_enrichment()` is more specific for our data because of the
input feature format it expects, `gene_pathway_enrichment()` is a simple wrapper for 
`gprofiler2::gost()`. 

Let's perform pathway enrichment for a single set of feature in our clusters of interest. 
This will also show you how to map features to genes using `FEATURE_TO_GENE_FILT`. 
```{r gene_pathway_enrichment}
features = MY_CLUSTER_DF$feature_ID[MY_CLUSTER_DF$cluster == "SKM-GN:1w_F1_M1->2w_F1_M1->4w_F1_M1->8w_F1_M1" &
                                      MY_CLUSTER_DF$ome == "PROT"]
input = FEATURE_TO_GENE_FILT$gene_symbol[FEATURE_TO_GENE_FILT$feature_ID %in% features]
background = GENE_UNIVERSES$gene_symbol$PROT$`SKM-GN`

res = gene_pathway_enrichment(input, background)
res[res$BH_adj_p_value < 0.1, c("term_name", "BH_adj_p_value", "intersection")]
```

#### Enrichment with custom pathways 

If you want to query pathways not available through `gProfiler`,
`custom_cluster_pathway_enrichment()` can be used to perform pathway enrichment 
for custom pathway annotations. Here we show an example with the three clusters of 
gastroc features and [Mitocarta](https://www.broadinstitute.org/mitocarta/mitocarta30-inventory-mammalian-mitochondrial-proteins-and-pathways) pathways. 

Note that features in the input must still 
be in the format "[ASSAY_ABBREV];[TISSUE_ABBREV];[feature_ID]",
e.g. "PHOSPHO;SKM-GN;NP_001006973.1_S34s", "METAB;SKM-GN;1-methylnicotinamide",
where feature IDs are included in the supplied `feature_to_gene` map 
(`FEATURE_TO_GENE` by default). 

The `universe`/background must also be specified using the same kind of gene identifier 
as the pathway annotation (rat gene symbols in this case). Here, for simplicity, we are
going to assume that the gene backgrounds for Mitocarta are the same as other pathway databases. 
In practice, since Mitocarta only considers mitochondria-associated features, we would
want to further restrict our gene backgrounds to the list of genes in the Mitocarta database,
which is what we did for the mitochondrial companion paper. 
```{r custom pw enrich}
# download and format Mitocarta pathways
download.file("https://personal.broadinstitute.org/scalvo/MitoCarta3.0/Human.MitoPathways3.0.gmx", destfile="~/Downloads/Human.MitoPathways3.0.gmx")
mitocarta = read.table("~/Downloads/Human.MitoPathways3.0.gmx", sep="\t", header=TRUE, skip=1)
# each column includes pathway members (human gene symbols)
# for each column, convert human gene smybols to rat gene symbols
mitocarta_members = lapply(mitocarta, function(x){
  human_genes = as.vector(na.omit(x[x!=""]))
  # convert to rat
  rat_genes = unique(na.omit(RAT_TO_HUMAN_GENE$RAT_SYMBOL[RAT_TO_HUMAN_GENE$HUMAN_ORTHOLOG_SYMBOL %in% human_genes]))
  rat_genes
})

# select our gastroc clusters
clust = MY_CLUSTERS[grepl("SKM-GN", names(MY_CLUSTERS))]
clust_df = check_cluster_res_format(clust)

# perform pathway enrichment 
res = custom_cluster_pathway_enrichment(clust_df,
                                        pathway_member_list = mitocarta_members,
                                        gene_identifier_type = "gene_symbol", 
                                        universe = GENE_UNIVERSES$gene_symbol,
                                        min_pw_set_size = 5)

# look at significant results at 5% FDR
res[res$adj_p_value < 0.05, c("cluster","ome","term_name","intersection","ome","computed_p_value","adj_p_value")]
```

## Interactive networks of pathway enrichments

Okay, so now we know how to perform pathway enrichment. Great! Now let's get to my
favorite part - how to visualize large numbers of significant pathway enrichments. 

### Approach

After performing pathway enrichment for our graphical clusters of interest, we
observed that many clusters had many dozens of significantly enriched pathways. 
In order to make these results more digestible, we leveraged
[visNetwork](https://cran.r-project.org/web/packages/visNetwork/vignettes/Introduction-to-visNetwork.html)
to construct interactive networks of pathway enrichments. These networks are 
conceptually similar to those made by the [EnrichmentMap](https://apps.cytoscape.org/apps/enrichmentmap)
Cytoscape module, but it's easier to generate them and explore the underlying data. 

Briefly, each node in the network is a pathway, and edges connect highly similar
pathways, i.e., those whose enrichments are driven by highly similar sets of genes
across omes. This groups together highly similar pathway enrichments, which makes
it easier to digest large numbers of results. 

In our implementation, groups of similar pathway enrichments are color-coded, and 
group labels are inferred from the term names and parents of the pathways in the group. 
Larger nodes indicate that the pathway was significantly enriched in multiple 
datasets; thicker edges indicate higher similarity between the pathways. 

Best of all, these networks are interactive! Use your cursor to zoom and drag,
and hover over nodes and edges to see more information about the underlying data, 
namely the genes and datasets driving the pathway enrichment. 

Note that you could run this function with any pathway enrichment results, 
as long as the columns of the input
include "adj_p_value", "ome", "tissue", "intersection", "computed_p_value", "term_name", and "term_id" 
and the columns of `feature_to_gene` include "gene_symbol", "ensembl_gene". 
`feature$ensembl_gene` can be a dummy column if `intersection_id_type = "gene_symbol"`.

### Examples 

#### Single tissue

Let's focus on a single trajectory, first separately in each tissue. While it is possible to 
[programatically render vizNetwork objects](https://motrpac.github.io/MotrpacRatTraining6mo/articles/MotrpacRatTraining6mo.html#printing-networks-within-a-loop), 
it is not trivial, so let's keep it simple here.  
```{r viznetwork gastroc, fig.width=8}
names(MY_CLUSTERS)
gastroc_res = GRAPH_PW_ENRICH[GRAPH_PW_ENRICH$cluster == "SKM-GN:1w_F1_M1->2w_F1_M1->4w_F1_M1->8w_F1_M1",]

enrichment_network_vis(gastroc_res)

enrichment_network_vis(gastroc_res, 
                       title = "SKM-GN:1w_F1_M1->2w_F1_M1->4w_F1_M1->8w_F1_M1", 
                       add_group_label_nodes = TRUE)
```

```{r viznetwork heart, fig.width=8}
heart_res = GRAPH_PW_ENRICH[GRAPH_PW_ENRICH$cluster == "HEART:1w_F1_M1->2w_F1_M1->4w_F1_M1->8w_F1_M1",]
enrichment_network_vis(heart_res, 
                       title = "HEART:1w_F1_M1->2w_F1_M1->4w_F1_M1->8w_F1_M1", 
                       add_group_label_nodes = TRUE)
```

#### Multiple tissues 

Now let's examine pathways that are enriched in the same graphical cluster in 
multiple tissues (heart and gastroc in our case). 
If we are specifically interested in pathways that are significantly enriched in
multiple tissues, we can set `multitissue_pathways_only` to `TRUE`. This removes 
any pathways that are not enriched in more than one tissue (in at least one ome). 
```{r viznetwork2, fig.width=8}
enrich = GRAPH_PW_ENRICH[GRAPH_PW_ENRICH$cluster %in% c("HEART:1w_F1_M1->2w_F1_M1->4w_F1_M1->8w_F1_M1",
                                                        "SKM-GN:1w_F1_M1->2w_F1_M1->4w_F1_M1->8w_F1_M1"),]
enrichment_network_vis(enrich, 
                       multitissue_pathways_only = TRUE,
                       add_group_label_nodes = TRUE,
                       title = "'All up' path in heart and gastroc - shared enrichments")
```
