<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Analyze genome tiles — analyze_tile • MotrpacRatTraining6mo</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Analyze genome tiles — analyze_tile"><meta name="description" content="A wrapper function that analyzes a tile of loci in the genome to obtain correlated clusters."><meta property="og:description" content="A wrapper function that analyzes a tile of loci in the genome to obtain correlated clusters."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MotrpacRatTraining6mo</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.6.6</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/MotrpacRatTraining6mo.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/key_metabolites.html">Key metabolites</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/MoTrPAC/MotrpacRatTraining6mo/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Analyze genome tiles</h1>
      <small class="dont-index">Source: <a href="https://github.com/MoTrPAC/MotrpacRatTraining6mo/blob/main/R/rrbs_pipeline.R" class="external-link"><code>R/rrbs_pipeline.R</code></a></small>
      <div class="d-none name"><code>analyze_tile.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>A wrapper function that analyzes a tile of loci in the genome to obtain correlated clusters.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">analyze_tile</span><span class="op">(</span></span>
<span>  <span class="va">tile_name</span>,</span>
<span>  <span class="va">tile_l</span>,</span>
<span>  <span class="va">M</span>,</span>
<span>  min_cor <span class="op">=</span> <span class="fl">0.7</span>,</span>
<span>  inflations <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">2.5</span>, <span class="fl">2</span>, <span class="fl">1.5</span><span class="op">)</span>,</span>
<span>  plotcorr <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-tile-name">tile_name<a class="anchor" aria-label="anchor" href="#arg-tile-name"></a></dt>
<dd><p>A character. The name of the current window for analysis.</p></dd>


<dt id="arg-tile-l">tile_l<a class="anchor" aria-label="anchor" href="#arg-tile-l"></a></dt>
<dd><p>A named list. tile_name must be an item in the list. Each entry contains the M matrix row indices of the loci of the window.</p></dd>


<dt id="arg-m">M<a class="anchor" aria-label="anchor" href="#arg-m"></a></dt>
<dd><p>A numeric matrix. Contains the "M" methylation values. Rows are loci, columns are samples.</p></dd>


<dt id="arg-min-cor">min_cor<a class="anchor" aria-label="anchor" href="#arg-min-cor"></a></dt>
<dd><p>A number. The minimal correlation to be considered for connecting a pair of loci.</p></dd>


<dt id="arg-inflations">inflations<a class="anchor" aria-label="anchor" href="#arg-inflations"></a></dt>
<dd><p>A numeric vector. Internal parameters of the MCL algorithm, see Details.</p></dd>


<dt id="arg-plotcorr">plotcorr<a class="anchor" aria-label="anchor" href="#arg-plotcorr"></a></dt>
<dd><p>A logical. TRUE: plot the correlation matrix of the window (useful for analysis of specific windows).</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A character vector. Names are loci (M matrix rows), entries are the names of the loci clusters.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function implements clustering analysis of a specific window in the genome.
We observed that often adjacent loci in RRBS data may manifest low correlations. This
function can be used for identified a set of highly correlated loci within a window, which
can then be used to extract new genomic features for downstream analysis.
RRBS data have two columns per sample: "Un" (unmethylated) and "Me"(methylated). This function
takes the "M matrix" which contains the integrated values of these columns (see examples). This matrix
is used for computing the correlations of the loci, and then we use the Markov Clustering algorithm (MCL)
for identifying homogeneous clusters within a window.</p>
<p>MCL (Markov clustering) details from <a href="https://www.micans.org/mcl/intro.html" class="external-link">https://www.micans.org/mcl/intro.html</a>.
<strong>MCL:</strong> Expansion coincides with taking the power of a stochastic matrix
using the normal matrix product (i.e. matrix squaring).
Inflation corresponds with taking the Hadamard power of a matrix
(taking powers entrywise), followed by a scaling step, such that the resulting matrix is stochastic again,
i.e. the matrix elements (on each column) correspond to probability values.
<strong>Inflation parameter:</strong> strengthen intra-region connections and promote cluster homogeneity.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># Raw data by tissue are available as RData files in Google Cloud Storage with the following URLs:   </span></span></span>
<span class="r-in"><span><span class="co"># Brown adipose:</span></span></span>
<span class="r-in"><span><span class="co"># https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_BAT_RAW_DATA.rda</span></span></span>
<span class="r-in"><span><span class="co"># Heart: </span></span></span>
<span class="r-in"><span><span class="co"># https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_HEART_RAW_DATA.rda</span></span></span>
<span class="r-in"><span><span class="co"># Hippocampus: </span></span></span>
<span class="r-in"><span><span class="co"># https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_HIPPOC_RAW_DATA.rda</span></span></span>
<span class="r-in"><span><span class="co"># Kidney: </span></span></span>
<span class="r-in"><span><span class="co"># https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_KIDNEY_RAW_DATA.rda</span></span></span>
<span class="r-in"><span><span class="co"># Lung: </span></span></span>
<span class="r-in"><span><span class="co"># https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_LUNG_RAW_DATA.rda</span></span></span>
<span class="r-in"><span><span class="co"># Liver: </span></span></span>
<span class="r-in"><span><span class="co"># https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_LIVER_RAW_DATA.rda</span></span></span>
<span class="r-in"><span><span class="co"># Gastrocnemius: </span></span></span>
<span class="r-in"><span><span class="co"># https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_SKMGN_RAW_DATA.rda</span></span></span>
<span class="r-in"><span><span class="co"># White adipose: </span></span></span>
<span class="r-in"><span><span class="co"># https://storage.googleapis.com/motrpac-rat-training-6mo-extdata/METHYL_WATSC_RAW_DATA.rda</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># load_methyl_raw_data() can be used to download data for a tissue, e.g.,</span></span></span>
<span class="r-in"><span><span class="co"># download the gastrocnemius data and load the data object into this session:</span></span></span>
<span class="r-in"><span><span class="va">yall</span> <span class="op">=</span> <span class="fu"><a href="load_methyl_raw_data.html">load_methyl_raw_data</a></span><span class="op">(</span><span class="st">"SKM-GN"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># remove control samples</span></span></span>
<span class="r-in"><span><span class="va">is_sample</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"^9"</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">yall</span><span class="op">)</span>,perl<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">yall</span> <span class="op">=</span> <span class="va">yall</span><span class="op">[</span>,<span class="va">is_sample</span><span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Filtering unassembled chromosomes</span></span></span>
<span class="r-in"><span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="cn">TRUE</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">yall</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">Chr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">yall</span><span class="op">$</span><span class="va">genes</span><span class="op">$</span><span class="va">Chr</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">keep</span><span class="op">[</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"random"</span>,<span class="va">Chr</span><span class="op">)</span> <span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="va">keep</span><span class="op">[</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"chrUn"</span>,<span class="va">Chr</span><span class="op">)</span> <span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="co"># remove non-chr ones</span></span></span>
<span class="r-in"><span><span class="va">keep</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"chr"</span>,<span class="va">Chr</span><span class="op">)</span> <span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="co"># remove M chromosome (otherwise we get error when assigning annotation below)</span></span></span>
<span class="r-in"><span><span class="va">keep</span><span class="op">[</span><span class="va">Chr</span><span class="op">==</span><span class="st">"chrM"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="co"># keep.lib.sizes=FALSE causes the library sizes to be recomputed</span></span></span>
<span class="r-in"><span><span class="va">yall</span> <span class="op">&lt;-</span> <span class="va">yall</span><span class="op">[</span><span class="va">keep</span>,, keep.lib.sizes<span class="op">=</span><span class="cn">FALSE</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="co"># rat genome chromosomes:</span></span></span>
<span class="r-in"><span><span class="va">ChrNames</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span>,<span class="st">"X"</span>,<span class="st">"Y"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">yall</span><span class="op">$</span><span class="va">genes</span><span class="op">$</span><span class="va">Chr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">yall</span><span class="op">$</span><span class="va">genes</span><span class="op">$</span><span class="va">Chr</span>, levels<span class="op">=</span><span class="va">ChrNames</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">o</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">yall</span><span class="op">$</span><span class="va">genes</span><span class="op">$</span><span class="va">Chr</span>, <span class="va">yall</span><span class="op">$</span><span class="va">genes</span><span class="op">$</span><span class="va">Locus</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">yall</span> <span class="op">&lt;-</span> <span class="va">yall</span><span class="op">[</span><span class="va">o</span>,<span class="op">]</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Counts matrix dim before low counts filter:"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">yall</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/gc.html" class="external-link">gc</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Remove low count features</span></span></span>
<span class="r-in"><span><span class="co"># The analysis needs to be restricted to CpG sites that have enough coverage for the</span></span></span>
<span class="r-in"><span><span class="co"># methylation level to be measurable in a meaningful way at that site. </span></span></span>
<span class="r-in"><span><span class="co"># As a conservative rule of thumb, we require a site to have a total count</span></span></span>
<span class="r-in"><span><span class="co"># (both methylated and unmethylated) of at least 8 in every sample.</span></span></span>
<span class="r-in"><span><span class="va">Coverage</span> <span class="op">&lt;-</span> <span class="va">yall</span><span class="op">$</span><span class="va">counts</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"-Me"</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">yall</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span><span class="op">]</span> <span class="op">+</span> </span></span>
<span class="r-in"><span>     <span class="va">yall</span><span class="op">$</span><span class="va">counts</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"-Un"</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">yall</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">min_count_coverage</span> <span class="op">=</span> <span class="fl">10</span></span></span>
<span class="r-in"><span><span class="va">min_per_samples</span> <span class="op">=</span> <span class="fl">1</span></span></span>
<span class="r-in"><span><span class="co"># see description of min_count_coverage and min_per_samples above</span></span></span>
<span class="r-in"><span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">Coverage</span> <span class="op">&gt;=</span> <span class="va">min_count_coverage</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="va">min_per_samples</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">Coverage</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># filter the data</span></span></span>
<span class="r-in"><span><span class="va">yall</span> <span class="op">&lt;-</span> <span class="va">yall</span><span class="op">[</span><span class="va">keep</span>,, keep.lib.sizes<span class="op">=</span><span class="cn">FALSE</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Counts matrix dim after low counts filter:"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">yall</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># get locations, gene ids, etc</span></span></span>
<span class="r-in"><span><span class="va">TSS</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/nearestTSS.html" class="external-link">nearestTSS</a></span><span class="op">(</span><span class="va">yall</span><span class="op">$</span><span class="va">genes</span><span class="op">$</span><span class="va">Chr</span>, <span class="va">yall</span><span class="op">$</span><span class="va">genes</span><span class="op">$</span><span class="va">Locus</span>, species<span class="op">=</span><span class="st">"Rn"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">yall</span><span class="op">$</span><span class="va">genes</span><span class="op">$</span><span class="va">EntrezID</span> <span class="op">&lt;-</span> <span class="va">TSS</span><span class="op">$</span><span class="va">gene_id</span></span></span>
<span class="r-in"><span><span class="va">yall</span><span class="op">$</span><span class="va">genes</span><span class="op">$</span><span class="va">Symbol</span> <span class="op">&lt;-</span> <span class="va">TSS</span><span class="op">$</span><span class="va">symbol</span></span></span>
<span class="r-in"><span><span class="va">yall</span><span class="op">$</span><span class="va">genes</span><span class="op">$</span><span class="va">Strand</span> <span class="op">&lt;-</span> <span class="va">TSS</span><span class="op">$</span><span class="va">strand</span></span></span>
<span class="r-in"><span><span class="va">yall</span><span class="op">$</span><span class="va">genes</span><span class="op">$</span><span class="va">Distance</span> <span class="op">&lt;-</span> <span class="va">TSS</span><span class="op">$</span><span class="va">distance</span></span></span>
<span class="r-in"><span><span class="va">yall</span><span class="op">$</span><span class="va">genes</span><span class="op">$</span><span class="va">Width</span> <span class="op">&lt;-</span> <span class="va">TSS</span><span class="op">$</span><span class="va">width</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Now we are ready to cluster the genome</span></span></span>
<span class="r-in"><span><span class="va">wsize</span> <span class="op">=</span> <span class="fl">500</span></span></span>
<span class="r-in"><span><span class="co"># step 1: define the genome-level tiles</span></span></span>
<span class="r-in"><span><span class="va">chrs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">yall</span><span class="op">$</span><span class="va">genes</span><span class="op">$</span><span class="va">Chr</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pos</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">yall</span><span class="op">$</span><span class="va">genes</span><span class="op">$</span><span class="va">Locus</span><span class="op">/</span><span class="va">wsize</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">tiles</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">chrs</span>,<span class="va">pos</span>,sep<span class="op">=</span><span class="st">"-"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">tiles</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">yall</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># step 2: add the M matrix for correlations</span></span></span>
<span class="r-in"><span><span class="va">Me</span> <span class="op">&lt;-</span> <span class="va">yall</span><span class="op">$</span><span class="va">counts</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"-Me"</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">yall</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">Un</span> <span class="op">&lt;-</span> <span class="va">yall</span><span class="op">$</span><span class="va">counts</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"-Un"</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">yall</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">M</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">Me</span> <span class="op">+</span> <span class="fl">2</span><span class="op">)</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">Un</span> <span class="op">+</span> <span class="fl">2</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># step 3: analyze each tile</span></span></span>
<span class="r-in"><span><span class="va">tile_l</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">M</span><span class="op">)</span>,<span class="va">tiles</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">tile_l</span> <span class="op">=</span> <span class="va">tile_l</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">tiles</span><span class="op">)</span><span class="op">]</span> <span class="co"># keep the genome order</span></span></span>
<span class="r-in"><span><span class="co"># parallel comp</span></span></span>
<span class="r-in"><span><span class="kw">if</span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"parallel"</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span></span>
<span class="r-in"><span>  <span class="va">tile_clusters</span> <span class="op">=</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/copy.html" class="external-link">copy</a></span><span class="op">(</span><span class="va">tiles</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="kw">for</span><span class="op">(</span><span class="va">chr</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">chrs</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">chr</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">curr_chr_inds</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">chr</span>,<span class="st">"-"</span><span class="op">)</span>,<span class="va">tiles</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">chr_tiles</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">tiles</span><span class="op">[</span><span class="va">curr_chr_inds</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">chr_tile_l</span> <span class="op">=</span> <span class="va">tile_l</span><span class="op">[</span><span class="va">chr_tiles</span><span class="op">]</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="op">{</span></span></span>
<span class="r-in"><span>      <span class="va">chr_tile_clusters</span> <span class="op">=</span> <span class="fu">parallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/parallel/mclapply.html" class="external-link">mclapply</a></span><span class="op">(</span><span class="va">chr_tiles</span>,</span></span>
<span class="r-in"><span>                                             <span class="va">analyze_tile</span>,</span></span>
<span class="r-in"><span>                                             tile_l <span class="op">=</span> <span class="va">chr_tile_l</span>,</span></span>
<span class="r-in"><span>                                             M <span class="op">=</span> <span class="va">M</span>, </span></span>
<span class="r-in"><span>                                             mc.cores <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span></span>
<span class="r-in"><span>      <span class="va">chr_tile_clusters</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">chr_tile_clusters</span><span class="op">)</span></span></span>
<span class="r-in"><span>      <span class="va">tile_clusters</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">chr_tile_clusters</span><span class="op">)</span><span class="op">]</span> <span class="op">=</span> <span class="va">chr_tile_clusters</span><span class="op">}</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/base/gc.html" class="external-link">gc</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">}</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span><span class="va">yall</span> <span class="op">=</span> <span class="fu"><a href="merge_sites_by_clusters.html">merge_sites_by_clusters</a></span><span class="op">(</span><span class="va">yall</span>,<span class="va">tile_clusters</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Data normalization</span></span></span>
<span class="r-in"><span><span class="co"># A key difference between BS-seq and other sequencing data is that the pair of libraries </span></span></span>
<span class="r-in"><span><span class="co"># holding the methylated and unmethylated reads for a particular sample are treated as a unit.</span></span></span>
<span class="r-in"><span><span class="co"># To ensure that the methylated and unmethylated reads for the same sample are treated on the</span></span></span>
<span class="r-in"><span><span class="co"># same scale, we need to set the library sizes to be equal for each pair of libraries. </span></span></span>
<span class="r-in"><span><span class="co"># We set the library sizes for each sample to be the average of the total read counts for </span></span></span>
<span class="r-in"><span><span class="co"># the methylated and unmethylated libraries.</span></span></span>
<span class="r-in"><span><span class="co"># Other normalization methods developed for RNA-seq data, </span></span></span>
<span class="r-in"><span><span class="co"># such as TMM, are not required for BS-seq data.</span></span></span>
<span class="r-in"><span><span class="va">TotalLibSize</span> <span class="op">&lt;-</span> <span class="va">yall</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">lib.size</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"-Me"</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">yall</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span><span class="op">]</span> <span class="op">+</span> </span></span>
<span class="r-in"><span>  <span class="va">yall</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">lib.size</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"-Un"</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">yall</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">yall</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">lib.size</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="va">TotalLibSize</span>, each<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Nicole Gay, David Amar, Pierre Jean Beltran, MoTrPAC Study Group, NIH Common Fund, National Institute of Diabetes and Digestive and Kidney Diseases (NIDDK), National Institute of Arthritis and Musculoskeletal Diseases (NIAMS), National Institute on Aging (NIA), National Institute of Biomedical Imaging and Bioengineering.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

