<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Combine normalized sample-level data — combine_normalized_data • MotrpacRatTraining6mo</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Combine normalized sample-level data — combine_normalized_data"><meta name="description" content="Combine normalized sample-level data from the specified tissues and omes(s)/assay(s).
If no tissues or omes are specified, all data is returned.
In order to combine data from different tissues and data types, sample-specific vial labels
are converted to animal-specific Participant IDs (PIDs)."><meta property="og:description" content="Combine normalized sample-level data from the specified tissues and omes(s)/assay(s).
If no tissues or omes are specified, all data is returned.
In order to combine data from different tissues and data types, sample-specific vial labels
are converted to animal-specific Participant IDs (PIDs)."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MotrpacRatTraining6mo</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.6.6</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/MotrpacRatTraining6mo.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/key_metabolites.html">Key metabolites</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/MoTrPAC/MotrpacRatTraining6mo/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Combine normalized sample-level data</h1>
      <small class="dont-index">Source: <a href="https://github.com/MoTrPAC/MotrpacRatTraining6mo/blob/main/R/fetch_prep_data.R" class="external-link"><code>R/fetch_prep_data.R</code></a></small>
      <div class="d-none name"><code>combine_normalized_data.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Combine normalized sample-level data from the specified tissues and omes(s)/assay(s).
If no tissues or omes are specified, all data is returned.
In order to combine data from different tissues and data types, sample-specific vial labels
are converted to animal-specific Participant IDs (PIDs).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">combine_normalized_data</span><span class="op">(</span></span>
<span>  tissues <span class="op">=</span> <span class="fu">MotrpacRatTraining6moData</span><span class="fu">::</span><span class="va"><a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/TISSUE_ABBREV.html">TISSUE_ABBREV</a></span>,</span>
<span>  assays <span class="op">=</span> <span class="fu">MotrpacRatTraining6moData</span><span class="fu">::</span><span class="va"><a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/ASSAY_ABBREV.html">ASSAY_ABBREV</a></span>,</span>
<span>  include_epigen <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  scratchdir <span class="op">=</span> <span class="st">"."</span>,</span>
<span>  training_regulated_only <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  exclude_outliers <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  nrows <span class="op">=</span> <span class="cn">Inf</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-tissues">tissues<a class="anchor" aria-label="anchor" href="#arg-tissues"></a></dt>
<dd><p>optional character vector of tissue abbreviations,
one of <a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/TISSUE_ABBREV.html">MotrpacRatTraining6moData::TISSUE_ABBREV</a>.</p></dd>


<dt id="arg-assays">assays<a class="anchor" aria-label="anchor" href="#arg-assays"></a></dt>
<dd><p>optional character vector of assay abbreviations,
one of <a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/ASSAY_ABBREV.html">MotrpacRatTraining6moData::ASSAY_ABBREV</a></p></dd>


<dt id="arg-include-epigen">include_epigen<a class="anchor" aria-label="anchor" href="#arg-include-epigen"></a></dt>
<dd><p>bool, whether to include the full ATAC or METHYL
differential analysis results from Google Cloud Storage.
Only relevant if <code>assays</code> includes "ATAC" or "METHYL".
<code>FALSE</code> by default.</p></dd>


<dt id="arg-scratchdir">scratchdir<a class="anchor" aria-label="anchor" href="#arg-scratchdir"></a></dt>
<dd><p>character, local directory in which to download data from the web.
Current working directory by default. Only relevant if <code>assays</code> includes "ATAC" or "METHYL".</p></dd>


<dt id="arg-training-regulated-only">training_regulated_only<a class="anchor" aria-label="anchor" href="#arg-training-regulated-only"></a></dt>
<dd><p>bool, whether to filter features down to those training-regulated at 5% FDR</p></dd>


<dt id="arg-exclude-outliers">exclude_outliers<a class="anchor" aria-label="anchor" href="#arg-exclude-outliers"></a></dt>
<dd><p>bool, whether to remove sample outliers specified by <a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/OUTLIERS.html">MotrpacRatTraining6moData::OUTLIERS</a></p></dd>


<dt id="arg-nrows">nrows<a class="anchor" aria-label="anchor" href="#arg-nrows"></a></dt>
<dd><p>integer, number of rows to return from each dataset. Defaults to Inf.
Useful to return a subset of a large data frame for tests.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>data frame with features in rows and Participant IDs (PIDs) in columns</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="load_sample_data.html">load_sample_data()</a></code>, <code><a href="viallabel_to_pid.html">viallabel_to_pid()</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># Return all normalized RNA-seq data</span></span></span>
<span class="r-in"><span><span class="va">data</span> <span class="op">=</span> <span class="fu">combine_normalized_data</span><span class="op">(</span>assays <span class="op">=</span> <span class="st">"TRNSCRPT"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Return all normalized proteomics data. Exclude outliers </span></span></span>
<span class="r-in"><span><span class="va">data</span> <span class="op">=</span> <span class="fu">combine_normalized_data</span><span class="op">(</span>assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PROT"</span>,<span class="st">"UBIQ"</span>,<span class="st">"PHOSPHO"</span>,<span class="st">"ACETYL"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                               exclude_outliers <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PROT_CORTEX_NORM_DATA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PROT_HEART_NORM_DATA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PROT_KIDNEY_NORM_DATA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PROT_LIVER_NORM_DATA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PROT_LUNG_NORM_DATA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PROT_SKMGN_NORM_DATA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PROT_WATSC_NORM_DATA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> UBIQ_HEART_NORM_DATA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> UBIQ_LIVER_NORM_DATA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PHOSPHO_CORTEX_NORM_DATA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PHOSPHO_HEART_NORM_DATA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PHOSPHO_KIDNEY_NORM_DATA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PHOSPHO_LIVER_NORM_DATA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PHOSPHO_LUNG_NORM_DATA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PHOSPHO_SKMGN_NORM_DATA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> PHOSPHO_WATSC_NORM_DATA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ACETYL_HEART_NORM_DATA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ACETYL_LIVER_NORM_DATA</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># Return normalized ATAC-seq data for training-regulated features </span></span></span>
<span class="r-in"><span><span class="va">data</span> <span class="op">=</span> <span class="fu">combine_normalized_data</span><span class="op">(</span>assays <span class="op">=</span> <span class="st">"ATAC"</span>, training_regulated_only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Return normalized ATAC-seq data for the first 1000 features in each tissue </span></span></span>
<span class="r-in"><span><span class="va">data</span> <span class="op">=</span> <span class="fu">combine_normalized_data</span><span class="op">(</span>assays <span class="op">=</span> <span class="st">"ATAC"</span>, </span></span>
<span class="r-in"><span>                               nrows <span class="op">=</span> <span class="fl">1000</span>, </span></span>
<span class="r-in"><span>                               scratchdir <span class="op">=</span> <span class="st">"/tmp"</span>, </span></span>
<span class="r-in"><span>                               include_epigen <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Return all normalized metabolomics data </span></span></span>
<span class="r-in"><span><span class="va">data</span> <span class="op">=</span> <span class="fu">combine_normalized_data</span><span class="op">(</span>assays <span class="op">=</span> <span class="st">"METAB"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> METAB ADRNL normalized data from METAB_NORM_DATA_FLAT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> METAB BAT normalized data from METAB_NORM_DATA_FLAT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> METAB BLOOD normalized data from METAB_NORM_DATA_FLAT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> METAB COLON normalized data from METAB_NORM_DATA_FLAT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> METAB CORTEX normalized data from METAB_NORM_DATA_FLAT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> METAB HEART normalized data from METAB_NORM_DATA_FLAT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> METAB HIPPOC normalized data from METAB_NORM_DATA_FLAT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> METAB HYPOTH normalized data from METAB_NORM_DATA_FLAT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> METAB KIDNEY normalized data from METAB_NORM_DATA_FLAT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> METAB LIVER normalized data from METAB_NORM_DATA_FLAT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> METAB LUNG normalized data from METAB_NORM_DATA_FLAT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> METAB OVARY normalized data from METAB_NORM_DATA_FLAT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> METAB PLASMA normalized data from METAB_NORM_DATA_FLAT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> METAB SKM-GN normalized data from METAB_NORM_DATA_FLAT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> METAB SKM-VL normalized data from METAB_NORM_DATA_FLAT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> METAB SMLINT normalized data from METAB_NORM_DATA_FLAT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> METAB SPLEEN normalized data from METAB_NORM_DATA_FLAT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> METAB TESTES normalized data from METAB_NORM_DATA_FLAT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> METAB VENACV normalized data from METAB_NORM_DATA_FLAT</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> METAB WAT-SC normalized data from METAB_NORM_DATA_FLAT</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Nicole Gay, David Amar, Pierre Jean Beltran, MoTrPAC Study Group, NIH Common Fund, National Institute of Diabetes and Digestive and Kidney Diseases (NIDDK), National Institute of Arthritis and Musculoskeletal Diseases (NIAMS), National Institute on Aging (NIA), National Institute of Biomedical Imaging and Bioengineering.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

