<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Combine differential analysis results from the specified tissues and omes(s)/assay(s).
If no tissues or omes are specified, all differential analysis results are returned."><title>Combine differential analysis results — combine_da_results • MotrpacRatTraining6mo</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Combine differential analysis results — combine_da_results"><meta property="og:description" content="Combine differential analysis results from the specified tissues and omes(s)/assay(s).
If no tissues or omes are specified, all differential analysis results are returned."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MotrpacRatTraining6mo</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.4.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/MotrpacRatTraining6mo.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/MoTrPAC/MotrpacRatTraining6mo/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Combine differential analysis results</h1>
      <small class="dont-index">Source: <a href="https://github.com/MoTrPAC/MotrpacRatTraining6mo/blob/HEAD/R/fetch_prep_data.R" class="external-link"><code>R/fetch_prep_data.R</code></a></small>
      <div class="d-none name"><code>combine_da_results.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Combine differential analysis results from the specified tissues and omes(s)/assay(s).
If no tissues or omes are specified, all differential analysis results are returned.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">combine_da_results</span><span class="op">(</span></span>
<span>  tissues <span class="op">=</span> <span class="fu">MotrpacRatTraining6moData</span><span class="fu">::</span><span class="va"><a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/TISSUE_ABBREV.html">TISSUE_ABBREV</a></span>,</span>
<span>  assays <span class="op">=</span> <span class="fu">MotrpacRatTraining6moData</span><span class="fu">::</span><span class="va"><a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/ASSAY_ABBREV.html">ASSAY_ABBREV</a></span>,</span>
<span>  metareg <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  include_epigen <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  scratchdir <span class="op">=</span> <span class="st">"."</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>tissues</dt>
<dd><p>optional character vector of tissue abbreviations,
one of <a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/TISSUE_ABBREV.html">MotrpacRatTraining6moData::TISSUE_ABBREV</a>.</p></dd>


<dt>assays</dt>
<dd><p>optional character vector of assay abbreviations,
one of <a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/ASSAY_ABBREV.html">MotrpacRatTraining6moData::ASSAY_ABBREV</a></p></dd>


<dt>metareg</dt>
<dd><p>bool, whether to use the meta-regression results for METAB.
Use the upstream differential analysis results if <code>FALSE</code>.
<code>TRUE</code> by default.</p></dd>


<dt>include_epigen</dt>
<dd><p>bool, whether to include the full ATAC or METHYL
differential analysis results from Google Cloud Storage.
Only relevant if <code>assays</code> includes "ATAC" or "METHYL".
<code>FALSE</code> by default.</p></dd>


<dt>scratchdir</dt>
<dd><p>character, local directory in which to download data from the web.
Current working directory by default. Only relevant if <code>assays</code> includes "ATAC" or "METHYL".</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>data frame. Depending on the specified assays, some of these columns may not be included:</p><dl><dt><code>feature</code></dt>
<dd><p>character, unique feature identifier in the format '<a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/ASSAY_ABBREV.html">MotrpacRatTraining6moData::ASSAY_ABBREV</a>;<a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/TISSUE_ABBREV.html">MotrpacRatTraining6moData::TISSUE_ABBREV</a>;feature_ID' only for training-regulated features at 5% IHW FDR. For redundant differential features, 'feature_ID' is prepended with the specific platform to make unique identifiers. See <a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/REPEATED_FEATURES.html">MotrpacRatTraining6moData::REPEATED_FEATURES</a> for details.</p></dd>

<dt><code>assay</code></dt>
<dd><p>character, assay abbreviation, one of <a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/ASSAY_ABBREV.html">MotrpacRatTraining6moData::ASSAY_ABBREV</a></p></dd>

<dt><code>assay_code</code></dt>
<dd><p>character, assay code used in data release. See <a href="https://rdrr.io/pkg/MotrpacBicQC/man/assay_codes.html" class="external-link">MotrpacBicQC::assay_codes</a>.</p></dd>

<dt><code>tissue</code></dt>
<dd><p>character, tissue abbreviation, one of <a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/TISSUE_ABBREV.html">MotrpacRatTraining6moData::TISSUE_ABBREV</a></p></dd>

<dt><code>tissue_code</code></dt>
<dd><p>character, tissue code used in data release. See <a href="https://rdrr.io/pkg/MotrpacBicQC/man/bic_animal_tissue_code.html" class="external-link">MotrpacBicQC::bic_animal_tissue_code</a>.</p></dd>

<dt><code>feature_ID</code></dt>
<dd><p>character, MoTrPAC feature identifier</p></dd>

<dt><code>sex</code></dt>
<dd><p>character, one of 'male' or 'female'</p></dd>

<dt><code>comparison_group</code></dt>
<dd><p>character, time point of trained animals compared to the sex-matched sedentary control animals, one of '1w', '2w', '4w', '8w'</p></dd>

<dt><code>p_value</code></dt>
<dd><p>double, unadjusted p-value for the difference between 'comparison_group' and the group of sex-matched sedentary control animals</p></dd>

<dt><code>adj_p_value</code></dt>
<dd><p>double, adjusted p-value from 'p_value' column. P-values are BY-adjusted across all datasets within a given assay/ome.</p></dd>

<dt><code>logFC</code></dt>
<dd><p>double, log fold-change where the numerator is 'comparison_group', e.g., '1w', and the denominator is the group of sex-matched sedentary control animals</p></dd>

<dt><code>logFC_se</code></dt>
<dd><p>double, standard error of the log fold-change</p></dd>

<dt><code>tscore</code></dt>
<dd><p>double, t statistic</p></dd>

<dt><code>covariates</code></dt>
<dd><p>character, comma-separated list of adjustment variables or NA</p></dd>

<dt><code>numNAs</code></dt>
<dd><p>integer, number of missing values in the current 'comparison_group' and sex-matched sedentary control animals</p></dd>

<dt><code>comparison_average_intensity</code></dt>
<dd><p>double, average intensity among the replicates in 'comparison_group'</p></dd>

<dt><code>reference_average_intensity</code></dt>
<dd><p>double, average intensity among the replicates in the group of sex-matched sedentary control animals</p></dd>

<dt><code>selection_fdr</code></dt>
<dd><p>double, adjusted training p-value used to select training-regulated analytes. P-values are IHW-adjusted across all datasets within a given assay with tissue as a covariate.</p></dd>

<dt><code>dataset</code></dt>
<dd><p>character, immune panel, metabolomics platform, or ATAC-seq dataset name</p></dd>

<dt><code>site</code></dt>
<dd><p>character, Chemical Analysis Site (CAS) name. METAB only</p></dd>

<dt><code>is_targeted</code></dt>
<dd><p>logical, is this a targeted platform? METAB only</p></dd>

<dt><code>metabolite_refmet</code></dt>
<dd><p>character, RefMet name of metabolite. METAB only</p></dd>

<dt><code>cv</code></dt>
<dd><p>double, feature coefficient of variation in the dataset. METAB only</p></dd>

<dt><code>metabolite</code></dt>
<dd><p>character, name of metabolite as appears in the CAS's data. METAB only</p></dd>

<dt><code>control_cv</code></dt>
<dd><p>double, feature coefficient of variation in the dataset. METAB only</p></dd>

<dt><code>mz</code></dt>
<dd><p>double, mass over charge. METAB only</p></dd>

<dt><code>rt</code></dt>
<dd><p>double, retention time. METAB only</p></dd>

<dt><code>neutral_mass</code></dt>
<dd><p>double, neutral mass. METAB only</p></dd>

<dt><code>meta_reg_het_p</code></dt>
<dd><p>double, for metabolites with multiple measurements, the meta-regression heterogeneity p-value, where a smaller p-value indicates more disagreement between platforms. One value per feature (not per training group). METAB only</p></dd>

<dt><code>meta_reg_pvalue</code></dt>
<dd><p>double, for metabolites with multiple measurements, the meta-regression p-value. One value per feature (not per training group). METAB only</p></dd>

<dt><code>shrunk_logFC</code></dt>
<dd><p>double, log fold-change with shrinkage applied</p></dd>

<dt><code>shrunk_logFC_se</code></dt>
<dd><p>double, standard error of the shrunken log fold-change</p></dd>

<dt><code>zscore</code></dt>
<dd><p>double, z statistic</p></dd>

<dt><code>removed_samples</code></dt>
<dd><p>character, comma-separated list of vial labels excluded from differential analysis or NA</p></dd>

<dt><code>comparison_average_intensity_se</code></dt>
<dd><p>double, standard error of 'comparison_average_intensity'</p></dd>

<dt><code>reference_average_intensity_se</code></dt>
<dd><p>double, standard error of 'comparison_average_intensity'</p></dd>

<dt><code>Chr</code></dt>
<dd><p>integer, chromosome. METHYL only</p></dd>

<dt><code>Locus</code></dt>
<dd><p>character, base pair range of feature. METHYL only</p></dd>

<dt><code>EntrezID</code></dt>
<dd><p>character, Entrez ID of closest gene. METHYL only</p></dd>

<dt><code>Symbol</code></dt>
<dd><p>character, gene symbol of closest gene. METHYL only</p></dd>


</dl></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="co"># Return all non-epigenetic differential analysis results, </span></span></span>
<span class="r-in"><span><span class="co"># including meta-regression results for metabolomics</span></span></span>
<span class="r-in"><span><span class="va">data</span> <span class="op">=</span> <span class="fu">combine_da_results</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Return all global proteomics differential analysis results</span></span></span>
<span class="r-in"><span><span class="va">data</span> <span class="op">=</span> <span class="fu">combine_da_results</span><span class="op">(</span>assays<span class="op">=</span><span class="st">"PROT"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Return METHYL and ATAC differential analysis results for gastrocnemius </span></span></span>
<span class="r-in"><span><span class="va">data</span> <span class="op">=</span> <span class="fu">combine_da_results</span><span class="op">(</span>tissues<span class="op">=</span><span class="st">"SKM-GN"</span>, </span></span>
<span class="r-in"><span>                          assays<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ATAC"</span>,<span class="st">"METHYL"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                          include_epigen<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Nicole Gay, David Amar, Pierre Jean Beltran, MoTrPAC Study Group, NIH Common Fund, National Institute of Diabetes and Digestive and Kidney Diseases (NIDDK), National Institute of Arthritis and Musculoskeletal Diseases (NIAMS), National Institute on Aging (NIA), National Institute of Biomedical Imaging and Bioengineering.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

