<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MotrpacRatTraining6mo">
<title>Tutorial for MotrpacRatTraining6mo R package • MotrpacRatTraining6mo</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Tutorial for MotrpacRatTraining6mo R package">
<meta property="og:description" content="MotrpacRatTraining6mo">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MotrpacRatTraining6mo</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/MotrpacRatTraining6mo.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/MoTrPAC/MotrpacRatTraining6mo/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="MotrpacRatTraining6mo_files/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="MotrpacRatTraining6mo_files/vis-9.1.0/vis-network.min.css" rel="stylesheet">
<script src="MotrpacRatTraining6mo_files/vis-9.1.0/vis-network.min.js"></script><script src="MotrpacRatTraining6mo_files/visNetwork-binding-2.1.0/visNetwork.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Tutorial for MotrpacRatTraining6mo R package</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/MoTrPAC/MotrpacRatTraining6mo/blob/HEAD/vignettes/MotrpacRatTraining6mo.Rmd" class="external-link"><code>vignettes/MotrpacRatTraining6mo.Rmd</code></a></small>
      <div class="d-none name"><code>MotrpacRatTraining6mo.Rmd</code></div>
    </div>

    
    
<style>
div.blue {
  background-color: #e6f0ff; 
  border-radius: 10px; 
  padding-top: 20px; 
  padding-right: 20px; 
  padding-left: 20px; 
  padding-bottom: 10px; 
  margin-bottom: 20px; 
}
div.yellow {
  background-color: #fff5e6; 
  border-radius: 10px; 
  padding-top: 20px; 
  padding-right: 20px; 
  padding-left: 20px; 
  padding-bottom: 10px; 
  margin-bottom: 20px; 
}
#TOC {
  top: 1%;
  opacity: 0.5;
}
#TOC:hover {
  opacity: 1;
}
</style>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<div class="yellow">
<p>This vignette is under construction. If something is unclear or
obviously missing, please submit an issue <a href="https://github.com/MoTrPAC/MotrpacRatTraining6mo/issues" class="external-link">here</a>.</p>
</div>
<div class="section level3">
<h3 id="about-this-package">About this package<a class="anchor" aria-label="anchor" href="#about-this-package"></a>
</h3>
<p>This package provides functions to fetch, explore, and reproduce the
processed data and downstream analysis results presented in the main
paper for the first large-scale multi-omic multi-tissue endurance
exercise training study conducted in young adult rats by the Molecular
Transducers of Physical Activity Consortium (MoTrPAC). A <a href="https://www.biorxiv.org/" class="external-link">bioRxiv</a> link to the corresponding
preprint will be added <a href="https://motrpac.github.io/MotrpacRatTraining6mo/#citing-motrpac-data">here</a>
shortly. <strong>We highly recommend skimming the preprint before using
this package as it provides important context and much greater detail
than we can provide here.</strong></p>
<div class="blue">
<p>While some of the functions in this package can be used by
themselves, they were primarily written to analyze data in the <a href="https://motrpac.github.io/MotrpacRatTraining6moData">MotrpacRatTraining6moData</a>
R package. See examples of how these data can be analyzed
<em>without</em> this package in the <a href="https://motrpac.github.io/MotrpacRatTraining6moData/articles/MotrpacRatTraining6moData.html">MotrpacRatTraining6moData
vignette</a>.</p>
</div>
</div>
<div class="section level3">
<h3 id="about-motrpac">About MoTrPAC<a class="anchor" aria-label="anchor" href="#about-motrpac"></a>
</h3>
<p>MoTrPAC is a national research consortium designed to discover and
perform preliminary characterization of the range of molecular
transducers (the “molecular map”) that underlie the effects of physical
activity in humans. The program’s goal is to study the molecular changes
that occur during and after exercise and ultimately to advance the
understanding of how physical activity improves and preserves health.
The six-year program is the largest targeted NIH investment of funds
into the mechanisms of how physical activity improves health and
prevents disease. See <a href="https://www.motrpac.org/" class="external-link">motrpac.org</a>
and <a href="https://motrpac-data.org/" class="external-link">motrpac-data.org</a> for more
details.</p>
</div>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>Install this package with <code>devtools</code>:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">"devtools"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"MoTrPAC/MotrpacRatTraining6mo"</span><span class="op">)</span></span></code></pre></div>
<p><a href="https://motrpac.github.io/MotrpacRatTraining6moData">MotrpacRatTraining6moData</a>
is a critical dependency for this package. If the above command has
issues installing this dependency, follow <a href="https://motrpac.github.io/MotrpacRatTraining6moData/index.html#installation">these
instructions</a> to install <a href="https://motrpac.github.io/MotrpacRatTraining6moData">MotrpacRatTraining6moData</a>
separately before re-running the above command to install this package
and its other dependencies.</p>
<p>Once we install this package, we can load the library.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MoTrPAC/MotrpacRatTraining6mo" class="external-link">MotrpacRatTraining6mo</a></span><span class="op">)</span></span></code></pre></div>
<p>For the sake of easily loading and looking up documentation for data
objects, let’s also load the <a href="https://motrpac.github.io/MotrpacRatTraining6moData">MotrpacRatTraining6moData</a>
package.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MoTrPAC/MotrpacRatTraining6moData" class="external-link">MotrpacRatTraining6moData</a></span><span class="op">)</span></span></code></pre></div>
<p>We will also load several suggested packages, which are required to
run some of the examples in this vignette.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">suggests</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gprofiler2"</span>,<span class="st">"IHW"</span>,<span class="st">"FELLA"</span>,<span class="st">"foreach"</span>,<span class="st">"KEGGREST"</span>,<span class="st">"repfdr"</span><span class="op">)</span></span>
<span><span class="va">need_to_install</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">p</span> <span class="kw">in</span> <span class="va">suggests</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="va">p</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">need_to_install</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">need_to_install</span>,<span class="va">p</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span><span class="op">}</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">need_to_install</span><span class="op">)</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"The following packages must be installed to run examples in this vignette:\n %s"</span>,</span>
<span>               <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">need_to_install</span>, collapse<span class="op">=</span><span class="st">", "</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<div class="blue">
<p><strong>Note:</strong> Data objects from the <a href="https://motrpac.github.io/MotrpacRatTraining6moData">MotrpacRatTraining6moData</a>
package are indicated as variable names in all caps,
e.g. <code>PHENO</code>, <code>TRNSCRPT_LIVER_RAW_COUNTS</code>.</p>
</div>
<div class="blue">
<p><strong>Tip:</strong> To learn more about any data object or
function, use <code>?</code> to retrieve the documentation, e.g., <a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/METAB_FEATURE_ID_MAP.html"><code>?METAB_FEATURE_ID_MAP</code></a>,
<code><a href="../reference/load_sample_data.html">?load_sample_data</a></code>. Note that both
<code>MotrpacRatTraining6mo</code> and
<code>MotrpacRatTraining6moData</code> must be installed and loaded with
<code><a href="https://rdrr.io/r/base/library.html" class="external-link">library()</a></code> for this to work.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="study-design">Study design<a class="anchor" aria-label="anchor" href="#study-design"></a>
</h2>
<p>Details of the experimental design can be found in the supplementary
methods of the <a href="https://motrpac.github.io/MotrpacRatTraining6mo/#citing-motrpac-data">bioRxiv
preprint</a>. Briefly, 6-month-old young adult rats were subjected to
progressive endurance exercise training for 1, 2, 4, or 8 weeks, with
tissues collected 48 hours after the last training bout. Sex-matched
sedentary, untrained rats were used as controls. Whole blood, plasma,
and 18 solid tissues were analyzed using genomics, proteomics,
metabolomics, and protein immunoassay technologies, with most assays
performed in a subset of these tissues. Depending on the assay, between
3 and 5 replicates per sex per time point were analyzed.</p>
</div>
<div class="section level2">
<h2 id="tissue-and-assay-abbreviations">Tissue and assay abbreviations<a class="anchor" aria-label="anchor" href="#tissue-and-assay-abbreviations"></a>
</h2>
<p>It is important to be aware of the tissue and assay abbreviations
because they are used to name data objects and define arguments for many
functions. The vectors of abbreviations are also available in
<code>TISSUE_ABBREV</code> and <code>ASSAY_ABBREV</code>.</p>
<div class="section level3">
<h3 id="tissues">Tissues<a class="anchor" aria-label="anchor" href="#tissues"></a>
</h3>
<ul>
<li>
<strong>ADRNL:</strong> adrenal gland<br>
</li>
<li>
<strong>BAT:</strong> brown adipose tissue<br>
</li>
<li>
<strong>BLOOD:</strong> whole blood<br>
</li>
<li>
<strong>COLON:</strong> colon<br>
</li>
<li>
<strong>CORTEX:</strong> cerebral cortex<br>
</li>
<li>
<strong>HEART:</strong> heart<br>
</li>
<li>
<strong>HIPPOC:</strong> hippocampus<br>
</li>
<li>
<strong>HYPOTH:</strong> hypothalamus<br>
</li>
<li>
<strong>KIDNEY:</strong> kidney<br>
</li>
<li>
<strong>LIVER:</strong> liver<br>
</li>
<li>
<strong>LUNG:</strong> lung<br>
</li>
<li>
<strong>OVARY:</strong> ovaries (female gonads)<br>
</li>
<li>
<strong>PLASMA:</strong> plasma from blood</li>
<li>
<strong>SKM-GN:</strong> gastrocnemius (leg skeletal muscle)<br>
</li>
<li>
<strong>SKM-VL:</strong> vastus lateralis (leg skeletal muscle)</li>
<li>
<strong>SMLINT:</strong> small intestine<br>
</li>
<li>
<strong>SPLEEN:</strong> spleen<br>
</li>
<li>
<strong>TESTES:</strong> testes (male gonads)<br>
</li>
<li>
<strong>VENACV:</strong> vena cava<br>
</li>
<li>
<strong>WAT-SC:</strong> subcutaneous white adipose tissue</li>
</ul>
</div>
<div class="section level3">
<h3 id="assaysomes">Assays/omes<a class="anchor" aria-label="anchor" href="#assaysomes"></a>
</h3>
<ul>
<li>
<strong>ACETYL:</strong> acetylproteomics; protein site
acetylation<br>
</li>
<li>
<strong>ATAC:</strong> chromatin accessibility, ATAC-seq data<br>
</li>
<li>
<strong>IMMUNO:</strong> multiplexed immunoassays (cytokines and
hormones)<br>
</li>
<li>
<strong>METAB:</strong> metabolomics and lipidomics<br>
</li>
<li>
<strong>METHYL:</strong> DNA methylation, RRBS data<br>
</li>
<li>
<strong>PHOSPHO:</strong> phosphoproteomics; protein site
phosphorylation<br>
</li>
<li>
<strong>PROT:</strong> global proteomics; protein abundance<br>
</li>
<li>
<strong>TRNSCRPT:</strong> transcriptomics, RNA-Seq data<br>
</li>
<li>
<strong>UBIQ:</strong> ubiquitylome; protein site
ubiquitination</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="data-in-motrpacrattraining6modata">Data in MotrpacRatTraining6moData<a class="anchor" aria-label="anchor" href="#data-in-motrpacrattraining6modata"></a>
</h2>
<p>While some of the functions in this package can be used by
themselves, they were primarily written to analyze data in the <a href="https://motrpac.github.io/MotrpacRatTraining6moData/">MotrpacRatTraining6moData</a>
R package.</p>
<p>Here is a brief summary of the kinds of data included in <a href="https://motrpac.github.io/MotrpacRatTraining6moData/">MotrpacRatTraining6moData</a>:</p>
<ul>
<li>Assay, tissue, sex, and training group abbreviations, codes, colors,
and order used in plots</li>
<li>Phenotypic data, <code>PHENO</code>
</li>
<li>Mapping between various feature identifiers, i.e.,
<code>FEATURE_TO_GENE</code>, <code>RAT_TO_HUMAN_GENE</code><br>
</li>
<li>Ome-specific feature annotation, i.e.,
<code>METAB_FEATURE_ID_MAP</code>, <code>METHYL_FEATURE_ANNOT</code>
(GCP only), <code>ATAC_FEATURE_ANNOT</code> (GCP only)</li>
<li>Ome-specific sample-level metadata, i.e.,
<code>TRNSCRPT_META</code>, <code>ATAC_META</code>,
<code>METHYL_META</code>, <code>IMMUNO_META</code>,
<code>PHOSPHO_META</code>, <code>PROT_META</code>,
<code>ACETYL_META</code>, <code>UBIQ_META</code><br>
</li>
<li>Raw counts for RNA-Seq (TRNSCRPT), ATAC-Seq (ATAC), and RRBS
(METHYL) data, e.g., <code>TRNSCRPT_LIVER_RAW_COUNTS</code>. Note that
epigenetic data (ATAC and METHYL) must be downloaded from Google Cloud
Storage.<br>
</li>
<li>Normalized sample-level data, e.g.,
<code>TRNSCRPT_SKMGN_NORM_DATA</code><br>
</li>
<li>Differential analysis results, e.g., <code>HEART_PROT_DA</code> (see
more details below)<br>
</li>
<li>Sample outliers excluded from differential analysis,
<code>OUTLIERS</code><br>
</li>
<li>Table of training-regulated features at 5% FDR,
<code>TRAINING_REGULATED_FEATURES</code><br>
</li>
<li>Bayesian graphical analysis inputs and results (see more details
below)<br>
</li>
<li>Pathway enrichment of main graphical clusters,
<code>GRAPH_PW_ENRICH</code>
</li>
</ul>
<p>For more details about the <a href="https://motrpac.github.io/MotrpacRatTraining6moData">MotrpacRatTraining6moData</a>
R package, see its <a href="https://motrpac.github.io/MotrpacRatTraining6moData/articles/MotrpacRatTraining6moData.html">vignette</a>.
The remainder of this vignette will focus on functions available in <a href="https://motrpac.github.io/MotrpacRatTraining6mo">MotrpacRatTraining6mo</a>.</p>
<p><a href="https://motrpac.github.io/MotrpacRatTraining6mo">MotrpacRatTraining6mo</a>
includes several functions to easily fetch and compile data from <a href="https://motrpac.github.io/MotrpacRatTraining6moData">MotrpacRatTraining6moData</a>.</p>
<p>Use <code><a href="../reference/load_sample_data.html">load_sample_data()</a></code> to load sample-level data for a
specific ome and tissue. Here we fetch various forms of the sample-level
RNA-Seq (TRNSCRPT) data for brown adipose tissue (BAT) as an
example.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load RNA-seq raw counts for brown adipose tissue </span></span>
<span><span class="va">data</span> <span class="op">=</span> <span class="fu"><a href="../reference/load_sample_data.html">load_sample_data</a></span><span class="op">(</span><span class="st">"BAT"</span>, <span class="st">"TRNSCRPT"</span>, normalized <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; TRNSCRPT_BAT_RAW_COUNTS</span></span>
<span></span>
<span><span class="co"># Load the normalized RNA-seq data for brown adipose tissue instead</span></span>
<span><span class="va">data</span> <span class="op">=</span> <span class="fu"><a href="../reference/load_sample_data.html">load_sample_data</a></span><span class="op">(</span><span class="st">"BAT"</span>, <span class="st">"TRNSCRPT"</span><span class="op">)</span></span>
<span><span class="co">#&gt; TRNSCRPT_BAT_NORM_DATA</span></span>
<span></span>
<span><span class="co"># Load the normalized RNA-seq data for brown adipose tissue, but exclude sample outliers </span></span>
<span><span class="va">data</span> <span class="op">=</span> <span class="fu"><a href="../reference/load_sample_data.html">load_sample_data</a></span><span class="op">(</span><span class="st">"BAT"</span>, <span class="st">"TRNSCRPT"</span>, exclude_outliers <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; TRNSCRPT_BAT_NORM_DATA</span></span>
<span></span>
<span><span class="co"># Load the normalized RNA-seq data for brown adipose tissue for training-regulated features only</span></span>
<span><span class="va">data</span> <span class="op">=</span> <span class="fu"><a href="../reference/load_sample_data.html">load_sample_data</a></span><span class="op">(</span><span class="st">"BAT"</span>, <span class="st">"TRNSCRPT"</span>, training_regulated_only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; TRNSCRPT_BAT_NORM_DATA</span></span></code></pre></div>
<p><code><a href="../reference/load_sample_data.html">load_sample_data()</a></code> will tell you if the specified
dataset doesn’t exist.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">=</span> <span class="fu"><a href="../reference/load_sample_data.html">load_sample_data</a></span><span class="op">(</span><span class="st">"VENACV"</span>, <span class="st">"PROT"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in load_sample_data("VENACV", "PROT"): No data returned for tissue</span></span>
<span><span class="co">#&gt; VENACV and assay PROT with current arguments.</span></span></code></pre></div>
<p><code><a href="../reference/load_sample_data.html">load_sample_data()</a></code> will also download epigenetic data
from Google Cloud Storage.</p>
<div class="yellow">
<p><strong>Note:</strong> Epigenetic data require substantially more
memory than other omes.</p>
</div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load ATAC-seq raw counts for hippocampus, excluding outliers </span></span>
<span><span class="va">data</span> <span class="op">=</span> <span class="fu"><a href="../reference/load_sample_data.html">load_sample_data</a></span><span class="op">(</span><span class="st">"HIPPOC"</span>, </span>
<span>                        <span class="st">"ATAC"</span>, </span>
<span>                        exclude_outliers <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                        normalized <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                        scratchdir <span class="op">=</span> <span class="st">"/tmp"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ATAC_HIPPOC_RAW_COUNTS</span></span></code></pre></div>
<p><code><a href="../reference/combine_normalized_data.html">combine_normalized_data()</a></code> is a wrapper for
<code><a href="../reference/load_sample_data.html">load_sample_data()</a></code> that returns combined sample-level
normalized data for multiple datasets. Note that the sample-specific
vial labels used as column names for most of the sample-level data are
replaced with rat-specific participant IDs (PIDs) to allow measurements
from multiple datasets for the same animal to be concatenated.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Return all normalized RNA-seq data</span></span>
<span><span class="va">data</span> <span class="op">=</span> <span class="fu"><a href="../reference/combine_normalized_data.html">combine_normalized_data</a></span><span class="op">(</span>assays <span class="op">=</span> <span class="st">"TRNSCRPT"</span><span class="op">)</span></span>
<span><span class="co">#&gt; TRNSCRPT_ADRNL_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_BAT_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_BLOOD_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_COLON_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_CORTEX_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_HEART_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_HIPPOC_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_HYPOTH_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_KIDNEY_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_LIVER_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_LUNG_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_OVARY_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_SKMGN_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_SKMVL_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_SMLINT_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_SPLEEN_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_TESTES_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_VENACV_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_WATSC_NORM_DATA</span></span>
<span></span>
<span><span class="co"># Return all normalized proteomics data. Exclude outliers </span></span>
<span><span class="va">data</span> <span class="op">=</span> <span class="fu"><a href="../reference/combine_normalized_data.html">combine_normalized_data</a></span><span class="op">(</span>assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PROT"</span>,<span class="st">"UBIQ"</span>,<span class="st">"PHOSPHO"</span>,<span class="st">"ACETYL"</span><span class="op">)</span>,</span>
<span>                               exclude_outliers <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; PROT_CORTEX_NORM_DATA</span></span>
<span><span class="co">#&gt; PROT_HEART_NORM_DATA</span></span>
<span><span class="co">#&gt; PROT_KIDNEY_NORM_DATA</span></span>
<span><span class="co">#&gt; PROT_LIVER_NORM_DATA</span></span>
<span><span class="co">#&gt; PROT_LUNG_NORM_DATA</span></span>
<span><span class="co">#&gt; PROT_SKMGN_NORM_DATA</span></span>
<span><span class="co">#&gt; PROT_WATSC_NORM_DATA</span></span>
<span><span class="co">#&gt; UBIQ_HEART_NORM_DATA</span></span>
<span><span class="co">#&gt; UBIQ_LIVER_NORM_DATA</span></span>
<span><span class="co">#&gt; PHOSPHO_CORTEX_NORM_DATA</span></span>
<span><span class="co">#&gt; PHOSPHO_HEART_NORM_DATA</span></span>
<span><span class="co">#&gt; PHOSPHO_KIDNEY_NORM_DATA</span></span>
<span><span class="co">#&gt; PHOSPHO_LIVER_NORM_DATA</span></span>
<span><span class="co">#&gt; PHOSPHO_LUNG_NORM_DATA</span></span>
<span><span class="co">#&gt; PHOSPHO_SKMGN_NORM_DATA</span></span>
<span><span class="co">#&gt; PHOSPHO_WATSC_NORM_DATA</span></span>
<span><span class="co">#&gt; ACETYL_HEART_NORM_DATA</span></span>
<span><span class="co">#&gt; ACETYL_LIVER_NORM_DATA</span></span>
<span></span>
<span><span class="co"># Return normalized ATAC-seq data for training-regulated features </span></span>
<span><span class="va">data</span> <span class="op">=</span> <span class="fu"><a href="../reference/combine_normalized_data.html">combine_normalized_data</a></span><span class="op">(</span>assays <span class="op">=</span> <span class="st">"ATAC"</span>, </span>
<span>                               training_regulated_only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; ATAC_BAT_NORM_DATA_05FDR</span></span>
<span><span class="co">#&gt; ATAC_HEART_NORM_DATA_05FDR</span></span>
<span><span class="co">#&gt; ATAC_HIPPOC_NORM_DATA_05FDR</span></span>
<span><span class="co">#&gt; ATAC_KIDNEY_NORM_DATA_05FDR</span></span>
<span><span class="co">#&gt; ATAC_LIVER_NORM_DATA_05FDR</span></span>
<span><span class="co">#&gt; ATAC_LUNG_NORM_DATA_05FDR</span></span>
<span><span class="co">#&gt; ATAC_SKMGN_NORM_DATA_05FDR</span></span>
<span><span class="co">#&gt; ATAC_WATSC_NORM_DATA_05FDR</span></span>
<span></span>
<span><span class="co"># Return all non-epigenetic data</span></span>
<span><span class="co"># Note that the "include_epigen" argument is FALSE by default</span></span>
<span><span class="va">data</span> <span class="op">=</span> <span class="fu"><a href="../reference/combine_normalized_data.html">combine_normalized_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in combine_normalized_data(): 'include_epigen' is FALSE. Excluding ATAC</span></span>
<span><span class="co">#&gt; and METHYL data.</span></span>
<span><span class="co">#&gt; ACETYL_HEART_NORM_DATA</span></span>
<span><span class="co">#&gt; ACETYL_LIVER_NORM_DATA</span></span>
<span><span class="co">#&gt; IMMUNO ADRNL normalized data from IMMUNO_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; IMMUNO BAT normalized data from IMMUNO_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; IMMUNO BLOOD normalized data from IMMUNO_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; IMMUNO COLON normalized data from IMMUNO_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; IMMUNO CORTEX normalized data from IMMUNO_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; IMMUNO HEART normalized data from IMMUNO_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; IMMUNO HIPPOC normalized data from IMMUNO_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; IMMUNO HYPOTH normalized data from IMMUNO_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; IMMUNO KIDNEY normalized data from IMMUNO_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; IMMUNO LIVER normalized data from IMMUNO_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; IMMUNO LUNG normalized data from IMMUNO_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; IMMUNO OVARY normalized data from IMMUNO_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; IMMUNO PLASMA normalized data from IMMUNO_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; IMMUNO SKM-GN normalized data from IMMUNO_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; IMMUNO SKM-VL normalized data from IMMUNO_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; IMMUNO SMLINT normalized data from IMMUNO_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; IMMUNO SPLEEN normalized data from IMMUNO_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; IMMUNO TESTES normalized data from IMMUNO_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; IMMUNO VENACV normalized data from IMMUNO_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; IMMUNO WAT-SC normalized data from IMMUNO_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; METAB ADRNL normalized data from METAB_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; METAB BAT normalized data from METAB_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; METAB BLOOD normalized data from METAB_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; METAB COLON normalized data from METAB_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; METAB CORTEX normalized data from METAB_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; METAB HEART normalized data from METAB_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; METAB HIPPOC normalized data from METAB_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; METAB HYPOTH normalized data from METAB_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; METAB KIDNEY normalized data from METAB_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; METAB LIVER normalized data from METAB_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; METAB LUNG normalized data from METAB_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; METAB OVARY normalized data from METAB_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; METAB PLASMA normalized data from METAB_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; METAB SKM-GN normalized data from METAB_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; METAB SKM-VL normalized data from METAB_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; METAB SMLINT normalized data from METAB_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; METAB SPLEEN normalized data from METAB_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; METAB TESTES normalized data from METAB_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; METAB VENACV normalized data from METAB_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; METAB WAT-SC normalized data from METAB_NORM_DATA_FLAT</span></span>
<span><span class="co">#&gt; PHOSPHO_CORTEX_NORM_DATA</span></span>
<span><span class="co">#&gt; PHOSPHO_HEART_NORM_DATA</span></span>
<span><span class="co">#&gt; PHOSPHO_KIDNEY_NORM_DATA</span></span>
<span><span class="co">#&gt; PHOSPHO_LIVER_NORM_DATA</span></span>
<span><span class="co">#&gt; PHOSPHO_LUNG_NORM_DATA</span></span>
<span><span class="co">#&gt; PHOSPHO_SKMGN_NORM_DATA</span></span>
<span><span class="co">#&gt; PHOSPHO_WATSC_NORM_DATA</span></span>
<span><span class="co">#&gt; PROT_CORTEX_NORM_DATA</span></span>
<span><span class="co">#&gt; PROT_HEART_NORM_DATA</span></span>
<span><span class="co">#&gt; PROT_KIDNEY_NORM_DATA</span></span>
<span><span class="co">#&gt; PROT_LIVER_NORM_DATA</span></span>
<span><span class="co">#&gt; PROT_LUNG_NORM_DATA</span></span>
<span><span class="co">#&gt; PROT_SKMGN_NORM_DATA</span></span>
<span><span class="co">#&gt; PROT_WATSC_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_ADRNL_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_BAT_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_BLOOD_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_COLON_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_CORTEX_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_HEART_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_HIPPOC_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_HYPOTH_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_KIDNEY_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_LIVER_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_LUNG_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_OVARY_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_SKMGN_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_SKMVL_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_SMLINT_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_SPLEEN_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_TESTES_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_VENACV_NORM_DATA</span></span>
<span><span class="co">#&gt; TRNSCRPT_WATSC_NORM_DATA</span></span>
<span><span class="co">#&gt; UBIQ_HEART_NORM_DATA</span></span>
<span><span class="co">#&gt; UBIQ_LIVER_NORM_DATA</span></span></code></pre></div>
<p>Similarly, <code><a href="../reference/combine_da_results.html">combine_da_results()</a></code> concatenates
differential analysis results for multiple datasets.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Return all non-epigenetic differential analysis results, </span></span>
<span><span class="co"># including meta-regression results for metabolomics</span></span>
<span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="../reference/combine_da_results.html">combine_da_results</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in combine_da_results(): 'include_epigen' is FALSE. Excluding ATAC and</span></span>
<span><span class="co">#&gt; METHYL results.</span></span>
<span><span class="co">#&gt; ACETYL_HEART_DA</span></span>
<span><span class="co">#&gt; ACETYL_LIVER_DA</span></span>
<span><span class="co">#&gt; IMMUNO_ADRNL_DA</span></span>
<span><span class="co">#&gt; IMMUNO_BAT_DA</span></span>
<span><span class="co">#&gt; IMMUNO_COLON_DA</span></span>
<span><span class="co">#&gt; IMMUNO_CORTEX_DA</span></span>
<span><span class="co">#&gt; IMMUNO_HEART_DA</span></span>
<span><span class="co">#&gt; IMMUNO_HIPPOC_DA</span></span>
<span><span class="co">#&gt; IMMUNO_KIDNEY_DA</span></span>
<span><span class="co">#&gt; IMMUNO_LIVER_DA</span></span>
<span><span class="co">#&gt; IMMUNO_LUNG_DA</span></span>
<span><span class="co">#&gt; IMMUNO_OVARY_DA</span></span>
<span><span class="co">#&gt; IMMUNO_PLASMA_DA</span></span>
<span><span class="co">#&gt; IMMUNO_SKMGN_DA</span></span>
<span><span class="co">#&gt; IMMUNO_SKMVL_DA</span></span>
<span><span class="co">#&gt; IMMUNO_SMLINT_DA</span></span>
<span><span class="co">#&gt; IMMUNO_SPLEEN_DA</span></span>
<span><span class="co">#&gt; IMMUNO_TESTES_DA</span></span>
<span><span class="co">#&gt; IMMUNO_WATSC_DA</span></span>
<span><span class="co">#&gt; METAB_ADRNL_DA_METAREG</span></span>
<span><span class="co">#&gt; METAB_BAT_DA_METAREG</span></span>
<span><span class="co">#&gt; METAB_COLON_DA_METAREG</span></span>
<span><span class="co">#&gt; METAB_CORTEX_DA_METAREG</span></span>
<span><span class="co">#&gt; METAB_HEART_DA_METAREG</span></span>
<span><span class="co">#&gt; METAB_HIPPOC_DA_METAREG</span></span>
<span><span class="co">#&gt; METAB_HYPOTH_DA_METAREG</span></span>
<span><span class="co">#&gt; METAB_KIDNEY_DA_METAREG</span></span>
<span><span class="co">#&gt; METAB_LIVER_DA_METAREG</span></span>
<span><span class="co">#&gt; METAB_LUNG_DA_METAREG</span></span>
<span><span class="co">#&gt; METAB_OVARY_DA_METAREG</span></span>
<span><span class="co">#&gt; METAB_PLASMA_DA_METAREG</span></span>
<span><span class="co">#&gt; METAB_SKMGN_DA_METAREG</span></span>
<span><span class="co">#&gt; METAB_SKMVL_DA_METAREG</span></span>
<span><span class="co">#&gt; METAB_SMLINT_DA_METAREG</span></span>
<span><span class="co">#&gt; METAB_SPLEEN_DA_METAREG</span></span>
<span><span class="co">#&gt; METAB_TESTES_DA_METAREG</span></span>
<span><span class="co">#&gt; METAB_VENACV_DA_METAREG</span></span>
<span><span class="co">#&gt; METAB_WATSC_DA_METAREG</span></span>
<span><span class="co">#&gt; PHOSPHO_CORTEX_DA</span></span>
<span><span class="co">#&gt; PHOSPHO_HEART_DA</span></span>
<span><span class="co">#&gt; PHOSPHO_KIDNEY_DA</span></span>
<span><span class="co">#&gt; PHOSPHO_LIVER_DA</span></span>
<span><span class="co">#&gt; PHOSPHO_LUNG_DA</span></span>
<span><span class="co">#&gt; PHOSPHO_SKMGN_DA</span></span>
<span><span class="co">#&gt; PHOSPHO_WATSC_DA</span></span>
<span><span class="co">#&gt; PROT_CORTEX_DA</span></span>
<span><span class="co">#&gt; PROT_HEART_DA</span></span>
<span><span class="co">#&gt; PROT_KIDNEY_DA</span></span>
<span><span class="co">#&gt; PROT_LIVER_DA</span></span>
<span><span class="co">#&gt; PROT_LUNG_DA</span></span>
<span><span class="co">#&gt; PROT_SKMGN_DA</span></span>
<span><span class="co">#&gt; PROT_WATSC_DA</span></span>
<span><span class="co">#&gt; TRNSCRPT_ADRNL_DA</span></span>
<span><span class="co">#&gt; TRNSCRPT_BAT_DA</span></span>
<span><span class="co">#&gt; TRNSCRPT_BLOOD_DA</span></span>
<span><span class="co">#&gt; TRNSCRPT_COLON_DA</span></span>
<span><span class="co">#&gt; TRNSCRPT_CORTEX_DA</span></span>
<span><span class="co">#&gt; TRNSCRPT_HEART_DA</span></span>
<span><span class="co">#&gt; TRNSCRPT_HIPPOC_DA</span></span>
<span><span class="co">#&gt; TRNSCRPT_HYPOTH_DA</span></span>
<span><span class="co">#&gt; TRNSCRPT_KIDNEY_DA</span></span>
<span><span class="co">#&gt; TRNSCRPT_LIVER_DA</span></span>
<span><span class="co">#&gt; TRNSCRPT_LUNG_DA</span></span>
<span><span class="co">#&gt; TRNSCRPT_OVARY_DA</span></span>
<span><span class="co">#&gt; TRNSCRPT_SKMGN_DA</span></span>
<span><span class="co">#&gt; TRNSCRPT_SKMVL_DA</span></span>
<span><span class="co">#&gt; TRNSCRPT_SMLINT_DA</span></span>
<span><span class="co">#&gt; TRNSCRPT_SPLEEN_DA</span></span>
<span><span class="co">#&gt; TRNSCRPT_TESTES_DA</span></span>
<span><span class="co">#&gt; TRNSCRPT_VENACV_DA</span></span>
<span><span class="co">#&gt; TRNSCRPT_WATSC_DA</span></span>
<span><span class="co">#&gt; UBIQ_HEART_DA</span></span>
<span><span class="co">#&gt; UBIQ_LIVER_DA</span></span>
<span></span>
<span><span class="co"># Return METHYL and ATAC differential analysis results for gastrocnemius </span></span>
<span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="../reference/combine_da_results.html">combine_da_results</a></span><span class="op">(</span>tissues<span class="op">=</span><span class="st">"SKM-GN"</span>, </span>
<span>                         assays<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ATAC"</span>,<span class="st">"METHYL"</span><span class="op">)</span>,</span>
<span>                         include_epigen<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; ATAC_SKMGN_DA</span></span>
<span><span class="co">#&gt; METHYL_SKMGN_DA</span></span></code></pre></div>
<p><code><a href="../reference/transcript_prep_data.html">transcript_prep_data()</a></code> and <code><a href="../reference/atac_prep_data.html">atac_prep_data()</a></code>
are functions used within the differential analysis functions for
TRNSCRPT and ATAC data. They collect the filtered raw counts, normalized
sample-level data, phenotypic data, and ome-specific sample metadata,
covariates, and outliers associated with a given tissue.</p>
<p>A few additional functions are provided to fetch epigenetic data from
Google Cloud Storage:</p>
<ul>
<li>
<strong>load_methyl_raw_data():</strong> Return the unfiltered raw
counts for a given tissue.<br>
</li>
<li>
<strong>load_methyl_feature_annotation():</strong> Return the METHYL
feature annotation file.<br>
</li>
<li>
<strong>load_atac_feature_annotation():</strong> Return the ATAC
feature annotation file.<br>
</li>
<li>
<strong>get_rdata_from_url():</strong> If you know the URL for a
specific RData file in GCS, you can use this function to return the
object using the <code>url</code> argument. Full URLs for all of the
epigenetic data are available in the <a href="https://motrpac.github.io/MotrpacRatTraining6moData/#access-epigenomics-data-through-google-cloud-storage">README
for the MotrpacRatTraining6moData package</a>.</li>
</ul>
<p>Finally, <code><a href="../reference/list_available_data.html">list_available_data()</a></code> returns a list of all of
the available data objects in the specified package.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/list_available_data.html">list_available_data</a></span><span class="op">(</span><span class="st">"MotrpacRatTraining6moData"</span><span class="op">)</span></span>
<span><span class="co">#&gt;   [1] "ACETYL_HEART_DA"               "ACETYL_HEART_NORM_DATA"       </span></span>
<span><span class="co">#&gt;   [3] "ACETYL_LIVER_DA"               "ACETYL_LIVER_NORM_DATA"       </span></span>
<span><span class="co">#&gt;   [5] "ACETYL_META"                   "ASSAY_ABBREV"                 </span></span>
<span><span class="co">#&gt;   [7] "ASSAY_ABBREV_TO_CODE"          "ASSAY_CODE_TO_ABBREV"         </span></span>
<span><span class="co">#&gt;   [9] "ASSAY_COLORS"                  "ASSAY_ORDER"                  </span></span>
<span><span class="co">#&gt;  [11] "ATAC_BAT_NORM_DATA_05FDR"      "ATAC_HEART_NORM_DATA_05FDR"   </span></span>
<span><span class="co">#&gt;  [13] "ATAC_HIPPOC_NORM_DATA_05FDR"   "ATAC_KIDNEY_NORM_DATA_05FDR"  </span></span>
<span><span class="co">#&gt;  [15] "ATAC_LIVER_NORM_DATA_05FDR"    "ATAC_LUNG_NORM_DATA_05FDR"    </span></span>
<span><span class="co">#&gt;  [17] "ATAC_META"                     "ATAC_SKMGN_NORM_DATA_05FDR"   </span></span>
<span><span class="co">#&gt;  [19] "ATAC_WATSC_NORM_DATA_05FDR"    "FEATURE_TO_GENE"              </span></span>
<span><span class="co">#&gt;  [21] "GENE_UNIVERSES"                "GRAPH_COMPONENTS"             </span></span>
<span><span class="co">#&gt;  [23] "GRAPH_PW_ENRICH"               "GRAPH_STATES"                 </span></span>
<span><span class="co">#&gt;  [25] "GROUP_COLORS"                  "IMMUNO_ADRNL_DA"              </span></span>
<span><span class="co">#&gt;  [27] "IMMUNO_BAT_DA"                 "IMMUNO_COLON_DA"              </span></span>
<span><span class="co">#&gt;  [29] "IMMUNO_CORTEX_DA"              "IMMUNO_HEART_DA"              </span></span>
<span><span class="co">#&gt;  [31] "IMMUNO_HIPPOC_DA"              "IMMUNO_KIDNEY_DA"             </span></span>
<span><span class="co">#&gt;  [33] "IMMUNO_LIVER_DA"               "IMMUNO_LUNG_DA"               </span></span>
<span><span class="co">#&gt;  [35] "IMMUNO_META"                   "IMMUNO_NORM_DATA_FLAT"        </span></span>
<span><span class="co">#&gt;  [37] "IMMUNO_NORM_DATA_NESTED"       "IMMUNO_OVARY_DA"              </span></span>
<span><span class="co">#&gt;  [39] "IMMUNO_PLASMA_DA"              "IMMUNO_SKMGN_DA"              </span></span>
<span><span class="co">#&gt;  [41] "IMMUNO_SKMVL_DA"               "IMMUNO_SMLINT_DA"             </span></span>
<span><span class="co">#&gt;  [43] "IMMUNO_SPLEEN_DA"              "IMMUNO_TESTES_DA"             </span></span>
<span><span class="co">#&gt;  [45] "IMMUNO_WATSC_DA"               "METAB_ADRNL_DA"               </span></span>
<span><span class="co">#&gt;  [47] "METAB_ADRNL_DA_METAREG"        "METAB_BAT_DA"                 </span></span>
<span><span class="co">#&gt;  [49] "METAB_BAT_DA_METAREG"          "METAB_COLON_DA"               </span></span>
<span><span class="co">#&gt;  [51] "METAB_COLON_DA_METAREG"        "METAB_CORTEX_DA"              </span></span>
<span><span class="co">#&gt;  [53] "METAB_CORTEX_DA_METAREG"       "METAB_FEATURE_ID_MAP"         </span></span>
<span><span class="co">#&gt;  [55] "METAB_HEART_DA"                "METAB_HEART_DA_METAREG"       </span></span>
<span><span class="co">#&gt;  [57] "METAB_HIPPOC_DA"               "METAB_HIPPOC_DA_METAREG"      </span></span>
<span><span class="co">#&gt;  [59] "METAB_HYPOTH_DA"               "METAB_HYPOTH_DA_METAREG"      </span></span>
<span><span class="co">#&gt;  [61] "METAB_KIDNEY_DA"               "METAB_KIDNEY_DA_METAREG"      </span></span>
<span><span class="co">#&gt;  [63] "METAB_LIVER_DA"                "METAB_LIVER_DA_METAREG"       </span></span>
<span><span class="co">#&gt;  [65] "METAB_LUNG_DA"                 "METAB_LUNG_DA_METAREG"        </span></span>
<span><span class="co">#&gt;  [67] "METAB_NORM_DATA_FLAT"          "METAB_NORM_DATA_NESTED"       </span></span>
<span><span class="co">#&gt;  [69] "METAB_OVARY_DA"                "METAB_OVARY_DA_METAREG"       </span></span>
<span><span class="co">#&gt;  [71] "METAB_PLASMA_DA"               "METAB_PLASMA_DA_METAREG"      </span></span>
<span><span class="co">#&gt;  [73] "METAB_SKMGN_DA"                "METAB_SKMGN_DA_METAREG"       </span></span>
<span><span class="co">#&gt;  [75] "METAB_SKMVL_DA"                "METAB_SKMVL_DA_METAREG"       </span></span>
<span><span class="co">#&gt;  [77] "METAB_SMLINT_DA"               "METAB_SMLINT_DA_METAREG"      </span></span>
<span><span class="co">#&gt;  [79] "METAB_SPLEEN_DA"               "METAB_SPLEEN_DA_METAREG"      </span></span>
<span><span class="co">#&gt;  [81] "METAB_TESTES_DA"               "METAB_TESTES_DA_METAREG"      </span></span>
<span><span class="co">#&gt;  [83] "METAB_VENACV_DA"               "METAB_VENACV_DA_METAREG"      </span></span>
<span><span class="co">#&gt;  [85] "METAB_WATSC_DA"                "METAB_WATSC_DA_METAREG"       </span></span>
<span><span class="co">#&gt;  [87] "METHYL_BAT_NORM_DATA_05FDR"    "METHYL_HEART_NORM_DATA_05FDR" </span></span>
<span><span class="co">#&gt;  [89] "METHYL_HIPPOC_NORM_DATA_05FDR" "METHYL_KIDNEY_NORM_DATA_05FDR"</span></span>
<span><span class="co">#&gt;  [91] "METHYL_LIVER_NORM_DATA_05FDR"  "METHYL_LUNG_NORM_DATA_05FDR"  </span></span>
<span><span class="co">#&gt;  [93] "METHYL_META"                   "METHYL_SKMGN_NORM_DATA_05FDR" </span></span>
<span><span class="co">#&gt;  [95] "METHYL_WATSC_NORM_DATA_05FDR"  "OUTLIERS"                     </span></span>
<span><span class="co">#&gt;  [97] "PATHWAY_PARENTS"               "PHENO"                        </span></span>
<span><span class="co">#&gt;  [99] "PHOSPHO_CORTEX_DA"             "PHOSPHO_CORTEX_NORM_DATA"     </span></span>
<span><span class="co">#&gt; [101] "PHOSPHO_HEART_DA"              "PHOSPHO_HEART_NORM_DATA"      </span></span>
<span><span class="co">#&gt; [103] "PHOSPHO_KIDNEY_DA"             "PHOSPHO_KIDNEY_NORM_DATA"     </span></span>
<span><span class="co">#&gt; [105] "PHOSPHO_LIVER_DA"              "PHOSPHO_LIVER_NORM_DATA"      </span></span>
<span><span class="co">#&gt; [107] "PHOSPHO_LUNG_DA"               "PHOSPHO_LUNG_NORM_DATA"       </span></span>
<span><span class="co">#&gt; [109] "PHOSPHO_META"                  "PHOSPHO_SKMGN_DA"             </span></span>
<span><span class="co">#&gt; [111] "PHOSPHO_SKMGN_NORM_DATA"       "PHOSPHO_WATSC_DA"             </span></span>
<span><span class="co">#&gt; [113] "PHOSPHO_WATSC_NORM_DATA"       "PROT_CORTEX_DA"               </span></span>
<span><span class="co">#&gt; [115] "PROT_CORTEX_NORM_DATA"         "PROT_HEART_DA"                </span></span>
<span><span class="co">#&gt; [117] "PROT_HEART_NORM_DATA"          "PROT_KIDNEY_DA"               </span></span>
<span><span class="co">#&gt; [119] "PROT_KIDNEY_NORM_DATA"         "PROT_LIVER_DA"                </span></span>
<span><span class="co">#&gt; [121] "PROT_LIVER_NORM_DATA"          "PROT_LUNG_DA"                 </span></span>
<span><span class="co">#&gt; [123] "PROT_LUNG_NORM_DATA"           "PROT_META"                    </span></span>
<span><span class="co">#&gt; [125] "PROT_SKMGN_DA"                 "PROT_SKMGN_NORM_DATA"         </span></span>
<span><span class="co">#&gt; [127] "PROT_WATSC_DA"                 "PROT_WATSC_NORM_DATA"         </span></span>
<span><span class="co">#&gt; [129] "RAT_TO_HUMAN_GENE"             "REPEATED_FEATURES"            </span></span>
<span><span class="co">#&gt; [131] "REPFDR_INPUTS"                 "REPFDR_RES"                   </span></span>
<span><span class="co">#&gt; [133] "SEX_COLORS"                    "TISSUE_ABBREV"                </span></span>
<span><span class="co">#&gt; [135] "TISSUE_ABBREV_TO_CODE"         "TISSUE_CODE_TO_ABBREV"        </span></span>
<span><span class="co">#&gt; [137] "TISSUE_COLORS"                 "TISSUE_ORDER"                 </span></span>
<span><span class="co">#&gt; [139] "TRAINING_REGULATED_FEATURES"   "TRNSCRPT_ADRNL_DA"            </span></span>
<span><span class="co">#&gt; [141] "TRNSCRPT_ADRNL_NORM_DATA"      "TRNSCRPT_ADRNL_RAW_COUNTS"    </span></span>
<span><span class="co">#&gt; [143] "TRNSCRPT_BAT_DA"               "TRNSCRPT_BAT_NORM_DATA"       </span></span>
<span><span class="co">#&gt; [145] "TRNSCRPT_BAT_RAW_COUNTS"       "TRNSCRPT_BLOOD_DA"            </span></span>
<span><span class="co">#&gt; [147] "TRNSCRPT_BLOOD_NORM_DATA"      "TRNSCRPT_BLOOD_RAW_COUNTS"    </span></span>
<span><span class="co">#&gt; [149] "TRNSCRPT_COLON_DA"             "TRNSCRPT_COLON_NORM_DATA"     </span></span>
<span><span class="co">#&gt; [151] "TRNSCRPT_COLON_RAW_COUNTS"     "TRNSCRPT_CORTEX_DA"           </span></span>
<span><span class="co">#&gt; [153] "TRNSCRPT_CORTEX_NORM_DATA"     "TRNSCRPT_CORTEX_RAW_COUNTS"   </span></span>
<span><span class="co">#&gt; [155] "TRNSCRPT_HEART_DA"             "TRNSCRPT_HEART_NORM_DATA"     </span></span>
<span><span class="co">#&gt; [157] "TRNSCRPT_HEART_RAW_COUNTS"     "TRNSCRPT_HIPPOC_DA"           </span></span>
<span><span class="co">#&gt; [159] "TRNSCRPT_HIPPOC_NORM_DATA"     "TRNSCRPT_HIPPOC_RAW_COUNTS"   </span></span>
<span><span class="co">#&gt; [161] "TRNSCRPT_HYPOTH_DA"            "TRNSCRPT_HYPOTH_NORM_DATA"    </span></span>
<span><span class="co">#&gt; [163] "TRNSCRPT_HYPOTH_RAW_COUNTS"    "TRNSCRPT_KIDNEY_DA"           </span></span>
<span><span class="co">#&gt; [165] "TRNSCRPT_KIDNEY_NORM_DATA"     "TRNSCRPT_KIDNEY_RAW_COUNTS"   </span></span>
<span><span class="co">#&gt; [167] "TRNSCRPT_LIVER_DA"             "TRNSCRPT_LIVER_NORM_DATA"     </span></span>
<span><span class="co">#&gt; [169] "TRNSCRPT_LIVER_RAW_COUNTS"     "TRNSCRPT_LUNG_DA"             </span></span>
<span><span class="co">#&gt; [171] "TRNSCRPT_LUNG_NORM_DATA"       "TRNSCRPT_LUNG_RAW_COUNTS"     </span></span>
<span><span class="co">#&gt; [173] "TRNSCRPT_META"                 "TRNSCRPT_OVARY_DA"            </span></span>
<span><span class="co">#&gt; [175] "TRNSCRPT_OVARY_NORM_DATA"      "TRNSCRPT_OVARY_RAW_COUNTS"    </span></span>
<span><span class="co">#&gt; [177] "TRNSCRPT_SKMGN_DA"             "TRNSCRPT_SKMGN_NORM_DATA"     </span></span>
<span><span class="co">#&gt; [179] "TRNSCRPT_SKMGN_RAW_COUNTS"     "TRNSCRPT_SKMVL_DA"            </span></span>
<span><span class="co">#&gt; [181] "TRNSCRPT_SKMVL_NORM_DATA"      "TRNSCRPT_SKMVL_RAW_COUNTS"    </span></span>
<span><span class="co">#&gt; [183] "TRNSCRPT_SMLINT_DA"            "TRNSCRPT_SMLINT_NORM_DATA"    </span></span>
<span><span class="co">#&gt; [185] "TRNSCRPT_SMLINT_RAW_COUNTS"    "TRNSCRPT_SPLEEN_DA"           </span></span>
<span><span class="co">#&gt; [187] "TRNSCRPT_SPLEEN_NORM_DATA"     "TRNSCRPT_SPLEEN_RAW_COUNTS"   </span></span>
<span><span class="co">#&gt; [189] "TRNSCRPT_TESTES_DA"            "TRNSCRPT_TESTES_NORM_DATA"    </span></span>
<span><span class="co">#&gt; [191] "TRNSCRPT_TESTES_RAW_COUNTS"    "TRNSCRPT_VENACV_DA"           </span></span>
<span><span class="co">#&gt; [193] "TRNSCRPT_VENACV_NORM_DATA"     "TRNSCRPT_VENACV_RAW_COUNTS"   </span></span>
<span><span class="co">#&gt; [195] "TRNSCRPT_WATSC_DA"             "TRNSCRPT_WATSC_NORM_DATA"     </span></span>
<span><span class="co">#&gt; [197] "TRNSCRPT_WATSC_RAW_COUNTS"     "UBIQ_HEART_DA"                </span></span>
<span><span class="co">#&gt; [199] "UBIQ_HEART_NORM_DATA"          "UBIQ_LIVER_DA"                </span></span>
<span><span class="co">#&gt; [201] "UBIQ_LIVER_NORM_DATA"          "UBIQ_META"</span></span></code></pre></div>
<p>If the <a href="https://motrpac.github.io/MotrpacRatTraining6moData/">MotrpacRatTraining6moData</a>
library is loaded, you can learn more about any of these data objects
with <code>?</code>, e.g., <a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/TISSUE_COLORS.html"><code>?TISSUE_COLORS</code></a>,
and you can load data objects into your environment using
<code><a href="https://rdrr.io/r/utils/data.html" class="external-link">data()</a></code>, e.g.,
<code>data(TRAINING_REGULATED_FEATURES)</code>.</p>
</div>
<div class="section level2">
<h2 id="differential-analysis">Differential analysis<a class="anchor" aria-label="anchor" href="#differential-analysis"></a>
</h2>
<p>More details about the differential analysis methods are available in
the supplementary methods of the <a href="https://motrpac.github.io/MotrpacRatTraining6mo/#citing-motrpac-data">bioRxiv
preprint</a>. Simply put, the <em>training</em> differential analysis
considers all training groups for each sex (sedentary controls and 4
training time points) to determine if the analyte significantly changes
in either sex at any point during the training time course. The adjusted
p-values from this analysis were used to determine the set of analytes
that are regulated by endurance exercise training at 5% FDR, referred to
as the <em>training-regulated features</em>. The <em>timewise</em>
differential analysis performs pairwise contrasts between trained
animals at each time point (1, 2, 4, or 8 weeks) and the sex-matched
sedentary control animals. This gives us sex- and time- specific
p-values and effect sizes, referred to as the <em>timewise summary
statistics</em>.</p>
<p>We provide the following functions to replicate our <em>timewise</em>
and <em>training</em> differential analysis results for each dataset.
Look at the corresponding documentation for each function for more
details, e.g. <code><a href="../reference/transcript_timewise_da.html">?transcript_timewise_da</a></code>.</p>
<ul>
<li>Proteomics (ACETYL, PHOSPHO, PROT, UBIQ)
<ul>
<li>
<code><a href="../reference/proteomics_timewise_da.html">proteomics_timewise_da()</a></code><br>
</li>
<li>
<code><a href="../reference/proteomics_training_da.html">proteomics_training_da()</a></code><br>
</li>
</ul>
</li>
<li>ATAC
<ul>
<li>
<code><a href="../reference/atac_training_da.html">atac_training_da()</a></code><br>
</li>
<li>
<code><a href="../reference/atac_timewise_da.html">atac_timewise_da()</a></code><br>
</li>
</ul>
</li>
<li>IMMUNO
<ul>
<li>
<code><a href="../reference/immuno_timewise_da.html">immuno_timewise_da()</a></code><br>
</li>
<li>
<code><a href="../reference/immuno_training_da.html">immuno_training_da()</a></code><br>
</li>
</ul>
</li>
<li>METAB
<ul>
<li>
<code><a href="../reference/metab_timewise_da.html">metab_timewise_da()</a></code><br>
</li>
<li>
<code><a href="../reference/metab_training_da.html">metab_training_da()</a></code><br>
</li>
</ul>
</li>
<li>METHYL
<ul>
<li>
<code><a href="../reference/rrbs_differential_analysis.html">rrbs_differential_analysis()</a></code><br>
</li>
</ul>
</li>
<li>TRNSCRPT
<ul>
<li>
<code><a href="../reference/transcript_timewise_da.html">transcript_timewise_da()</a></code><br>
</li>
<li><code><a href="../reference/transcript_training_da.html">transcript_training_da()</a></code></li>
</ul>
</li>
</ul>
<div class="yellow">
<p><strong>Note:</strong> The functions to reproduce the metabolomics
meta-regression for metabolites with multiple measurements have not yet
been added.</p>
</div>
<p>Here, we replicate the protein acetylation timewise differential
analysis results provided in <a href="https://motrpac.github.io/MotrpacRatTraining6moData/">MotrpacRatTraining6moData</a>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">timewiseList</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">tissue</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HEART"</span>,<span class="st">"LIVER"</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">timewiseList</span><span class="op">[[</span><span class="va">tissue</span><span class="op">]</span><span class="op">]</span> <span class="op">=</span> <span class="fu"><a href="../reference/proteomics_timewise_da.html">proteomics_timewise_da</a></span><span class="op">(</span><span class="st">"ACETYL"</span>, <span class="va">tissue</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="co">#&gt; ACETYL_HEART_NORM_DATA</span></span>
<span><span class="co">#&gt; Warning: Partial NA coefficients for 157 probe(s)</span></span>
<span><span class="co">#&gt; ACETYL_LIVER_NORM_DATA</span></span>
<span><span class="co">#&gt; Warning: Partial NA coefficients for 355 probe(s)</span></span>
<span><span class="va">timewise</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="st">"rbind"</span>, <span class="va">timewiseList</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># merge with version of results in MotrpacRatTraining6moData</span></span>
<span><span class="va">original</span> <span class="op">=</span> <span class="fu"><a href="../reference/combine_da_results.html">combine_da_results</a></span><span class="op">(</span>assays<span class="op">=</span><span class="st">"ACETYL"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ACETYL_HEART_DA</span></span>
<span><span class="co">#&gt; ACETYL_LIVER_DA</span></span>
<span><span class="va">merged</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">original</span>, <span class="va">timewise</span>, </span>
<span>               by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"feature_ID"</span>,<span class="st">"assay"</span>,<span class="st">"assay_code"</span>,<span class="st">"tissue"</span>,<span class="st">"tissue_code"</span>,<span class="st">"sex"</span>,<span class="st">"comparison_group"</span><span class="op">)</span>,</span>
<span>               suffixes<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"_orginal"</span>,<span class="st">"_reproduced"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">merged</span><span class="op">$</span><span class="va">p_value_orginal</span><span class="op">)</span>, </span>
<span>     <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">merged</span><span class="op">$</span><span class="va">p_value_reproduced</span><span class="op">)</span>, </span>
<span>     type<span class="op">=</span><span class="st">"p"</span>, cex<span class="op">=</span><span class="fl">0.5</span>,</span>
<span>     xlab<span class="op">=</span><span class="st">"Original timewise p-value (-log10)"</span>,</span>
<span>     ylab<span class="op">=</span><span class="st">"Reproduced timewise p-value (-log10)"</span>,</span>
<span>     main<span class="op">=</span><span class="st">"Timewise DA results for ACETYL data"</span><span class="op">)</span></span></code></pre></div>
<p><img src="MotrpacRatTraining6mo_files/figure-html/replicate%20prot-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="bayesian-graphical-clustering">Bayesian graphical clustering<a class="anchor" aria-label="anchor" href="#bayesian-graphical-clustering"></a>
</h2>
<div class="section level3">
<h3 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h3>
<p>After performing differential analysis between trained animals and
sedentary control animals to identify training-regulated features across
datasets, we wanted to characterize groups of these features that had
similar trajectories over the training time course and compare them
within and between tissues.</p>
<p>To do this, we clustered the 34,244 training-regulated features with
complete timewise summary statistics (that is, from all four training
time points in both sexes) using their timewise z-scores with the
expectation-maximization (EM) clustering algorithm of <a href="https://pubmed.ncbi.nlm.nih.gov/25012182/" class="external-link">repfdr</a>. This
algorithm assumes that each timewise z-score can take one of three
latent states: -1 for down-regulation, 1 for up-regulation, and 0 for
null (i.e., unchanged relative to the sedentary control group). The
algorithm computes the posterior probabilities for each combination of
states (-1/down, 1/up, or 0/null) over all experimental groups (8
groups: 1-, 2-, 4-, and 8-week time points in females and males). We
used these posterior probabilities to effectively assign continuous
z-scores to simplified states. For example, a feature with the following
z-scores would be assigned to the corresponding states with high
probability:</p>
<table style="width:100%;" class="table">
<colgroup>
<col width="9%">
<col width="12%">
<col width="9%">
<col width="12%">
<col width="9%">
<col width="12%">
<col width="9%">
<col width="12%">
<col width="9%">
</colgroup>
<thead><tr class="header">
<th></th>
<th align="center">Female 1w</th>
<th align="center">Male 1w</th>
<th align="center">Female 2w</th>
<th align="center">Male 2w</th>
<th align="center">Female 4w</th>
<th align="center">Male 4w</th>
<th align="center">Female 8w</th>
<th align="center">Male 8w</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>z-score</td>
<td align="center">-5.7</td>
<td align="center">0.1</td>
<td align="center">0.65</td>
<td align="center">-0.06</td>
<td align="center">0.09</td>
<td align="center">6.7</td>
<td align="center">0.21</td>
<td align="center">5</td>
</tr>
<tr class="even">
<td>state</td>
<td align="center">-1</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">1</td>
<td align="center">0</td>
<td align="center">1</td>
</tr>
</tbody>
</table>
<p>We then combine the male and female states for a given time point
using the following notation:</p>
<table class="table">
<thead><tr class="header">
<th></th>
<th align="center">1 week</th>
<th align="center">2 weeks</th>
<th align="center">4 weeks</th>
<th align="center">8 weeks</th>
</tr></thead>
<tbody><tr class="odd">
<td>notation</td>
<td align="center"><code>1w_F-1_M0</code></td>
<td align="center"><code>2w_F0_M0</code></td>
<td align="center"><code>4w_F0_M1</code></td>
<td align="center"><code>8w_F0_M1</code></td>
</tr></tbody>
</table>
<p>Descriptively, this feature is down-regulated in females
(<code>F-1</code>) at 1 week of training (<code>1w</code>), null in both
sexes (<code>F0_M0</code>) at 2 weeks of training (<code>2w</code>), and
up-regulated in males only (<code>F0_M1</code>) at 4 and 8 weeks of
training.</p>
<p>One major advantage of this approach is that we can use the time
points and discrete states to construct a graph, where each point (or
“node”) in the graph represents a state at a given time point (e.g.,
<code>1w_F-1_M0</code>). Then for each feature of interest, we can draw
a path through the graph to represent its trajectory over the training
time course. We can use the paths and nodes in the graph to define
clusters of features with similar temporal behavior and characterize the
underlying biology.</p>
<p>For example, we can represent the complete path of the above feature
over the training time course as follows:
<code>1w_F-1_M0-&gt;2w_F0_M0-&gt;4w_F0_M1-&gt;8w_F0_M1</code>. This path
is made by connecting the <code>1w_F-1_M0</code> node to the
<code>2w_F0_M0</code> node to the <code>4w_F0_M1</code> node to the
<code>8w_F0_M1</code> node. The connection between a single pair of
consecutive nodes is called an edge. For example, the
<code>1w_F-1_M0---2w_F0_M0</code> edge represents features that are
down-regulated only in females after 1 week of training and null in both
sexes after 2 weeks of training. The
<code>1w_F-1_M0-&gt;2w_F0_M0-&gt;4w_F0_M1-&gt;8w_F0_M1</code> path is
constructed from 3 edges, each represented by <code>-&gt;</code> in the
notation.</p>
<p>We herein refer to our approach of using <code>repfdr</code> to
construct a graph of the dynamic multi-omic responses to endurance
exercise training across tissues as <em>graphical clustering</em>. The
following sections describe how to reproduce our graphical clustering
analysis, run <code>repfdr</code> on your own data, and explore and
visualize the nodes, edges, and paths (collectively referred to as
clusters) in the resulting graph. For additional details and advantages
of this approach over traditional clustering methods, see the
supplementary methods in the <a href="https://motrpac.github.io/MotrpacRatTraining6mo/#citing-motrpac-data">bioRxiv
preprint</a>. If you are more interested in exploring the existing
pathway enrichment results, skip to <a href="#vizEnrich">Visualization
of pathway enrichment results</a>.</p>
</div>
<div class="section level3">
<h3 id="re-run-repfdr-to-determine-states">(Re-)run <code>repfdr</code> to determine states<a class="anchor" aria-label="anchor" href="#re-run-repfdr-to-determine-states"></a>
</h3>
<p><code>REPFDR_INPUTS</code> provides the inputs for our analysis with
<code>repfdr</code>, which can be replicated with
<code><a href="../reference/bayesian_graphical_clustering.html">bayesian_graphical_clustering()</a></code> as shown below. The raw
<code>repfdr</code> results are provided in <code>REPFDR_RES</code>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">REPFDR_INPUTS</span><span class="op">)</span></span>
<span><span class="va">zscores</span> <span class="op">=</span> <span class="va">REPFDR_INPUTS</span><span class="op">$</span><span class="va">zs_smoothed</span></span>
<span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="../reference/bayesian_graphical_clustering.html">bayesian_graphical_clustering</a></span><span class="op">(</span><span class="va">zscores</span><span class="op">)</span></span></code></pre></div>
<p><code><a href="../reference/bayesian_graphical_clustering.html">bayesian_graphical_clustering()</a></code> is a study-specific
wrapper for <code><a href="../reference/repfdr_wrapper.html">repfdr_wrapper()</a></code>. <code><a href="../reference/repfdr_wrapper.html">repfdr_wrapper()</a></code>
can be used to analyze any set of z-scores. Here we show an example with
simulated data.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load required library for this function</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/barakbri/repfdr" class="external-link">repfdr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Simulate data with a single null cluster</span></span>
<span><span class="co"># Toy experiment with 8 groups </span></span>
<span><span class="va">zcolnames</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Exp"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">)</span></span>
<span><span class="va">zscores</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fl">80000</span><span class="op">)</span>,ncol<span class="op">=</span><span class="fl">8</span>,dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10000</span>,<span class="va">zcolnames</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># Add a cluster with a strong signal,</span></span>
<span><span class="co"># i.e. assign a z-score of ~5 to the first 4 groups for the first 500 features</span></span>
<span><span class="va">zscores</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">500</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span> <span class="op">=</span> <span class="va">zscores</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">500</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span> <span class="op">+</span> <span class="fl">5</span></span>
<span><span class="co"># Run the repfdr wrapper</span></span>
<span><span class="va">repfdr_results</span> <span class="op">=</span> <span class="fu"><a href="../reference/repfdr_wrapper.html">repfdr_wrapper</a></span><span class="op">(</span><span class="va">zscores</span>, df<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="co">#&gt; EM iteration 2</span></span>
<span><span class="co">#&gt; Converged after 1 EM iterations.EM iteration 2</span></span>
<span><span class="co">#&gt; EM iteration 3</span></span>
<span><span class="co">#&gt; EM iteration 4</span></span>
<span><span class="co">#&gt; EM iteration 5</span></span>
<span><span class="co">#&gt; EM iteration 6</span></span>
<span><span class="co">#&gt; EM iteration 7</span></span>
<span><span class="co">#&gt; EM iteration 8</span></span>
<span><span class="co">#&gt; EM iteration 9</span></span>
<span><span class="co">#&gt; EM iteration 10</span></span>
<span><span class="co">#&gt; EM iteration 11</span></span>
<span><span class="co">#&gt; EM iteration 12</span></span>
<span><span class="co">#&gt; EM iteration 13</span></span>
<span><span class="co">#&gt; EM iteration 14</span></span>
<span><span class="co">#&gt; EM iteration 15</span></span>
<span><span class="co">#&gt; EM iteration 16</span></span>
<span><span class="co">#&gt; EM iteration 17</span></span>
<span><span class="co">#&gt; EM iteration 18</span></span>
<span><span class="co">#&gt; EM iteration 19</span></span>
<span><span class="co">#&gt; EM iteration 20</span></span>
<span><span class="co">#&gt; EM iteration 21</span></span>
<span><span class="co">#&gt; EM iteration 22</span></span>
<span><span class="co">#&gt; EM iteration 23</span></span>
<span><span class="co">#&gt; EM iteration 24</span></span>
<span><span class="co">#&gt; EM iteration 25</span></span>
<span><span class="co">#&gt; EM iteration 26</span></span>
<span><span class="co">#&gt; EM iteration 27</span></span>
<span><span class="co">#&gt; EM iteration 28</span></span>
<span><span class="co">#&gt; EM iteration 29</span></span>
<span><span class="co">#&gt; EM iteration 30</span></span>
<span><span class="co">#&gt; EM iteration 31</span></span>
<span><span class="co">#&gt; EM iteration 32</span></span>
<span><span class="co">#&gt; EM iteration 33</span></span>
<span><span class="co">#&gt; EM iteration 34</span></span>
<span><span class="co">#&gt; EM iteration 35</span></span>
<span><span class="co">#&gt; EM iteration 36</span></span>
<span><span class="co">#&gt; EM iteration 37</span></span>
<span><span class="co">#&gt; EM iteration 38</span></span>
<span><span class="co">#&gt; EM iteration 39</span></span>
<span><span class="co">#&gt; EM iteration 40</span></span>
<span><span class="co">#&gt; EM iteration 41</span></span>
<span><span class="co">#&gt; EM iteration 42</span></span>
<span><span class="co">#&gt; EM iteration 43</span></span>
<span><span class="co">#&gt; EM iteration 44</span></span>
<span><span class="co">#&gt; EM iteration 45</span></span>
<span><span class="co">#&gt; EM iteration 46</span></span>
<span><span class="co">#&gt; EM iteration 47</span></span>
<span><span class="co">#&gt; EM iteration 48</span></span>
<span><span class="co">#&gt; EM iteration 49</span></span>
<span><span class="co">#&gt; EM iteration 50</span></span>
<span><span class="co">#&gt; EM iteration 51</span></span>
<span><span class="co">#&gt; EM iteration 52</span></span>
<span><span class="co">#&gt; EM iteration 53</span></span>
<span><span class="co">#&gt; EM iteration 54</span></span>
<span><span class="co">#&gt; EM iteration 55</span></span>
<span><span class="co">#&gt; EM iteration 56</span></span>
<span><span class="co">#&gt; EM iteration 57</span></span>
<span><span class="co">#&gt; EM iteration 58</span></span>
<span><span class="co">#&gt; EM iteration 59</span></span>
<span><span class="co">#&gt; EM iteration 60</span></span>
<span><span class="co">#&gt; EM iteration 61</span></span>
<span><span class="co">#&gt; EM iteration 62</span></span>
<span><span class="co">#&gt; EM iteration 63</span></span>
<span><span class="co">#&gt; EM iteration 64</span></span>
<span><span class="co">#&gt; EM iteration 65</span></span>
<span><span class="co">#&gt; EM iteration 66</span></span>
<span><span class="co">#&gt; EM iteration 67</span></span>
<span><span class="co">#&gt; EM iteration 68</span></span>
<span><span class="co">#&gt; EM iteration 69</span></span>
<span><span class="co">#&gt; EM iteration 70</span></span>
<span><span class="co">#&gt; EM iteration 71</span></span>
<span><span class="co">#&gt; EM iteration 72</span></span>
<span><span class="co">#&gt; Converged after 71 EM iterations.</span></span>
<span></span>
<span><span class="co"># Now we expect the first 500 features to belong to cluster 11110000 (up-regulated in the first 4 groups)</span></span>
<span><span class="co"># and the remaining 9500 to belong to cluster 00000000 (null in all groups)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">repfdr_results</span><span class="op">$</span><span class="va">repfdr_cluster_posteriors</span><span class="op">[</span>,<span class="st">"00000000"</span><span class="op">]</span>, </span>
<span>     breaks<span class="op">=</span><span class="fl">20</span>, </span>
<span>     main<span class="op">=</span><span class="st">"Posterior probability of null cluster for all features"</span><span class="op">)</span></span></code></pre></div>
<p><img src="MotrpacRatTraining6mo_files/figure-html/repfdr_wrapper-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">repfdr_results</span><span class="op">$</span><span class="va">repfdr_cluster_posteriors</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">500</span>,<span class="st">"11110000"</span><span class="op">]</span>, </span>
<span>     breaks<span class="op">=</span><span class="fl">20</span>, </span>
<span>     main<span class="op">=</span><span class="st">"Posterior probability of 11110000 cluster for first 500 features"</span><span class="op">)</span></span></code></pre></div>
<p><img src="MotrpacRatTraining6mo_files/figure-html/repfdr_wrapper-2.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="explore-graphical-clustering-results">Explore graphical clustering results<a class="anchor" aria-label="anchor" href="#explore-graphical-clustering-results"></a>
</h3>
<p>The graph representation of the <code>repfdr</code> results are
provided in <code>GRAPH_COMPONENTS</code> (node and edge lists) and
<code>GRAPH_STATES</code> (data frame representation).</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GRAPH_STATES</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">GRAPH_STATES</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                                             feature    ome</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001003673.1_K477k ACETYL;HEART;NP_001003673.1_K477k ACETYL</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004072.2_K77k   ACETYL;HEART;NP_001004072.2_K77k ACETYL</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004072.2_K83k   ACETYL;HEART;NP_001004072.2_K83k ACETYL</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004085.1_K186k ACETYL;HEART;NP_001004085.1_K186k ACETYL</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004085.1_K261k ACETYL;HEART;NP_001004085.1_K261k ACETYL</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004085.1_K268k ACETYL;HEART;NP_001004085.1_K268k ACETYL</span></span>
<span><span class="co">#&gt;                                   tissue           feature_ID state_1w state_2w</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001003673.1_K477k  HEART NP_001003673.1_K477k    F0_M0    F0_M0</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004072.2_K77k   HEART  NP_001004072.2_K77k    F0_M0    F0_M0</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004072.2_K83k   HEART  NP_001004072.2_K83k  F-1_M-1  F-1_M-1</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004085.1_K186k  HEART NP_001004085.1_K186k    F0_M1    F0_M1</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004085.1_K261k  HEART NP_001004085.1_K261k    F0_M1    F0_M1</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004085.1_K268k  HEART NP_001004085.1_K268k    F0_M1    F0_M1</span></span>
<span><span class="co">#&gt;                                   state_4w state_8w</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001003673.1_K477k    F0_M0  F-1_M-1</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004072.2_K77k     F1_M0    F1_M0</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004072.2_K83k    F0_M-1    F0_M0</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004085.1_K186k    F0_M0   F0_M-1</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004085.1_K261k    F0_M0   F0_M-1</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004085.1_K268k    F0_M0   F0_M-1</span></span>
<span><span class="co">#&gt;                                                                          path</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001003673.1_K477k    1w_F0_M0-&gt;2w_F0_M0-&gt;4w_F0_M0-&gt;8w_F-1_M-1</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004072.2_K77k       1w_F0_M0-&gt;2w_F0_M0-&gt;4w_F1_M0-&gt;8w_F1_M0</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004072.2_K83k  1w_F-1_M-1-&gt;2w_F-1_M-1-&gt;4w_F0_M-1-&gt;8w_F0_M0</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004085.1_K186k     1w_F0_M1-&gt;2w_F0_M1-&gt;4w_F0_M0-&gt;8w_F0_M-1</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004085.1_K261k     1w_F0_M1-&gt;2w_F0_M1-&gt;4w_F0_M0-&gt;8w_F0_M-1</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004085.1_K268k     1w_F0_M1-&gt;2w_F0_M1-&gt;4w_F0_M0-&gt;8w_F0_M-1</span></span>
<span><span class="co">#&gt;                                                                         tissue_path</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001003673.1_K477k    HEART:1w_F0_M0-&gt;2w_F0_M0-&gt;4w_F0_M0-&gt;8w_F-1_M-1</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004072.2_K77k       HEART:1w_F0_M0-&gt;2w_F0_M0-&gt;4w_F1_M0-&gt;8w_F1_M0</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004072.2_K83k  HEART:1w_F-1_M-1-&gt;2w_F-1_M-1-&gt;4w_F0_M-1-&gt;8w_F0_M0</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004085.1_K186k     HEART:1w_F0_M1-&gt;2w_F0_M1-&gt;4w_F0_M0-&gt;8w_F0_M-1</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004085.1_K261k     HEART:1w_F0_M1-&gt;2w_F0_M1-&gt;4w_F0_M0-&gt;8w_F0_M-1</span></span>
<span><span class="co">#&gt; ACETYL;HEART;NP_001004085.1_K268k     HEART:1w_F0_M1-&gt;2w_F0_M1-&gt;4w_F0_M0-&gt;8w_F0_M-1</span></span></code></pre></div>
<div class="blue">
<p><strong>Note:</strong> Features are specified in the format
“[ASSAY_ABBREV];[TISSUE_ABBREV];[feature_ID]”,
e.g. “PHOSPHO;SKM-GN;NP_001006973.1_S34s”,
“METAB;SKM-GN;1-methylnicotinamide”</p>
</div>
<p>We can extract node, edge, and path feature sets from a given tissue
using <code><a href="../reference/extract_tissue_sets.html">extract_tissue_sets()</a></code>. Here we extract
training-regulated features in the three (<code>k=3</code>) largest
nodes, edges, and paths in the gastrocnemius
(<code>tissues="SKM-GN"</code>) that have at least 20 features
(<code>min_size=20</code>). We also add the feature lists for all 8-week
nodes (<code>add_week8=TRUE</code>), which are generally of interest
because the 8-week time point represents the final trained state.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gastroc_sets</span> <span class="op">=</span> <span class="fu"><a href="../reference/extract_tissue_sets.html">extract_tissue_sets</a></span><span class="op">(</span>tissues<span class="op">=</span><span class="st">"SKM-GN"</span>,</span>
<span>                                   k<span class="op">=</span><span class="fl">3</span>,</span>
<span>                                   min_size<span class="op">=</span><span class="fl">20</span>,</span>
<span>                                   add_week8<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gastroc_sets</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "8w_F1_M1"                                      </span></span>
<span><span class="co">#&gt;  [2] "8w_F-1_M-1"                                    </span></span>
<span><span class="co">#&gt;  [3] "4w_F1_M1"                                      </span></span>
<span><span class="co">#&gt;  [4] "8w_F-1_M0"                                     </span></span>
<span><span class="co">#&gt;  [5] "8w_F0_M-1"                                     </span></span>
<span><span class="co">#&gt;  [6] "8w_F0_M0"                                      </span></span>
<span><span class="co">#&gt;  [7] "8w_F0_M1"                                      </span></span>
<span><span class="co">#&gt;  [8] "8w_F1_M0"                                      </span></span>
<span><span class="co">#&gt;  [9] "4w_F1_M1---8w_F1_M1"                           </span></span>
<span><span class="co">#&gt; [10] "1w_F0_M0---2w_F0_M0"                           </span></span>
<span><span class="co">#&gt; [11] "4w_F-1_M-1---8w_F-1_M-1"                       </span></span>
<span><span class="co">#&gt; [12] "1w_F1_M1-&gt;2w_F1_M1-&gt;4w_F1_M1-&gt;8w_F1_M1"        </span></span>
<span><span class="co">#&gt; [13] "1w_F-1_M-1-&gt;2w_F-1_M-1-&gt;4w_F-1_M-1-&gt;8w_F-1_M-1"</span></span>
<span><span class="co">#&gt; [14] "1w_F0_M1-&gt;2w_F0_M1-&gt;4w_F1_M1-&gt;8w_F1_M1"</span></span>
<span></span>
<span><span class="co"># look at the list of features in the "1w_F1_M1-&gt;2w_F1_M1-&gt;4w_F1_M1-&gt;8w_F1_M1" path</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gastroc_sets</span><span class="op">$</span><span class="va">`1w_F1_M1-&gt;2w_F1_M1-&gt;4w_F1_M1-&gt;8w_F1_M1`</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "ATAC;SKM-GN;chr15:101579022-101579761"</span></span>
<span><span class="co">#&gt; [2] "ATAC;SKM-GN;chr1:221194280-221195125" </span></span>
<span><span class="co">#&gt; [3] "ATAC;SKM-GN;chr1:221199295-221200124" </span></span>
<span><span class="co">#&gt; [4] "ATAC;SKM-GN;chr20:34579127-34579734"  </span></span>
<span><span class="co">#&gt; [5] "ATAC;SKM-GN;chr5:114013062-114014681" </span></span>
<span><span class="co">#&gt; [6] "ATAC;SKM-GN;chrX:119151643-119154560"</span></span></code></pre></div>
<p><code><a href="../reference/get_all_trajectories.html">get_all_trajectories()</a></code> returns all non-empty
<em>paths</em> in the results.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">paths</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_all_trajectories.html">get_all_trajectories</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">paths</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "1w_F-1_M1-&gt;2w_F-1_M0-&gt;4w_F-1_M0-&gt;8w_F-1_M0"</span></span>
<span><span class="co">#&gt; [2] "1w_F0_M1-&gt;2w_F0_M1-&gt;4w_F0_M1-&gt;8w_F1_M1"    </span></span>
<span><span class="co">#&gt; [3] "1w_F1_M1-&gt;2w_F1_M1-&gt;4w_F1_M1-&gt;8w_F1_M1"    </span></span>
<span><span class="co">#&gt; [4] "1w_F0_M0-&gt;2w_F0_M0-&gt;4w_F0_M0-&gt;8w_F1_M1"    </span></span>
<span><span class="co">#&gt; [5] "1w_F0_M0-&gt;2w_F0_M0-&gt;4w_F0_M0-&gt;8w_F-1_M-1"  </span></span>
<span><span class="co">#&gt; [6] "1w_F0_M1-&gt;2w_F0_M1-&gt;4w_F0_M1-&gt;8w_F0_M1"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">paths</span>, <span class="va">length</span><span class="op">)</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fl">100</span><span class="op">)</span></span></code></pre></div>
<p><img src="MotrpacRatTraining6mo_files/figure-html/get%20all%20trajectories-1.png" width="700" style="display: block; margin: auto;"></p>
<p><code><a href="../reference/extract_main_clusters.html">extract_main_clusters()</a></code> returns the subset of graphical
clusters for which pathway enrichment was performed for the <a href="https://motrpac.github.io/MotrpacRatTraining6mo/#citing-motrpac-data">preprint</a>,
namely the 2 largest nodes, 2 largest edges, 10 largest non-null paths,
and all 8-week nodes from the graphical representation of
training-regulated features in each tissue.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clusters</span> <span class="op">=</span> <span class="fu"><a href="../reference/extract_main_clusters.html">extract_main_clusters</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>To extract the training-regulated features associated with a specific
graphical cluster, tissue, ome, etc., we can subset
<code>GRAPH_STATES</code>.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GRAPH_STATES</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># if you want features in this path in LUNG:</span></span>
<span><span class="va">clust</span> <span class="op">=</span> <span class="st">"LUNG:1w_F1_M-1-&gt;2w_F1_M-1-&gt;4w_F0_M-1-&gt;8w_F-1_M-1"</span></span>
<span><span class="va">features_of_interest</span> <span class="op">=</span> <span class="fu">stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html" class="external-link">na.omit</a></span><span class="op">(</span><span class="va">GRAPH_STATES</span><span class="op">$</span><span class="va">feature</span><span class="op">[</span><span class="va">GRAPH_STATES</span><span class="op">$</span><span class="va">tissue_path</span> <span class="op">==</span> <span class="va">clust</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="va">features_of_interest</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "PHOSPHO;LUNG;NP_001012023.1_S447s"     </span></span>
<span><span class="co">#&gt;  [2] "PHOSPHO;LUNG;NP_001013238.2_S1594s"    </span></span>
<span><span class="co">#&gt;  [3] "PHOSPHO;LUNG;NP_001020859.1_S486sS489s"</span></span>
<span><span class="co">#&gt;  [4] "PHOSPHO;LUNG;NP_001028859.1_S133s"     </span></span>
<span><span class="co">#&gt;  [5] "PHOSPHO;LUNG;NP_001099225.2_S621s"     </span></span>
<span><span class="co">#&gt;  [6] "PHOSPHO;LUNG;NP_001099344.2_S1816s"    </span></span>
<span><span class="co">#&gt;  [7] "PHOSPHO;LUNG;NP_001164067.1_T456t"     </span></span>
<span><span class="co">#&gt;  [8] "PHOSPHO;LUNG;NP_001178760.1_S542s"     </span></span>
<span><span class="co">#&gt;  [9] "PHOSPHO;LUNG;NP_001178775.1_S270s"     </span></span>
<span><span class="co">#&gt; [10] "PHOSPHO;LUNG;NP_001178880.1_S5423s"    </span></span>
<span><span class="co">#&gt; [11] "PHOSPHO;LUNG;NP_036849.1_T422t"        </span></span>
<span><span class="co">#&gt; [12] "PHOSPHO;LUNG;NP_037349.1_S512s"        </span></span>
<span><span class="co">#&gt; [13] "PHOSPHO;LUNG;NP_058862.1_S63s"         </span></span>
<span><span class="co">#&gt; [14] "PHOSPHO;LUNG;NP_647546.1_S288s"        </span></span>
<span><span class="co">#&gt; [15] "PHOSPHO;LUNG;NP_665709.2_S702s"        </span></span>
<span><span class="co">#&gt; [16] "PHOSPHO;LUNG;XP_001058477.1_S2558s"    </span></span>
<span><span class="co">#&gt; [17] "PHOSPHO;LUNG;XP_002726994.2_S307s"     </span></span>
<span><span class="co">#&gt; [18] "PHOSPHO;LUNG;XP_003752547.1_S411s"     </span></span>
<span><span class="co">#&gt; [19] "PHOSPHO;LUNG;XP_003752548.1_S411s"     </span></span>
<span><span class="co">#&gt; [20] "PHOSPHO;LUNG;XP_003754609.1_S1206s"    </span></span>
<span><span class="co">#&gt; [21] "PHOSPHO;LUNG;XP_006226323.1_Y618y"     </span></span>
<span><span class="co">#&gt; [22] "PHOSPHO;LUNG;XP_006235089.1_S434s"     </span></span>
<span><span class="co">#&gt; [23] "PHOSPHO;LUNG;XP_008766933.1_S1807s"    </span></span>
<span><span class="co">#&gt; [24] "PHOSPHO;LUNG;XP_017443701.1_S938s"     </span></span>
<span><span class="co">#&gt; [25] "PHOSPHO;LUNG;XP_017444244.1_S5362s"    </span></span>
<span><span class="co">#&gt; [26] "PHOSPHO;LUNG;XP_017454714.1_S1250s"    </span></span>
<span><span class="co">#&gt; [27] "PHOSPHO;LUNG;XP_017454716.1_S1225s"    </span></span>
<span><span class="co">#&gt; [28] "PHOSPHO;LUNG;XP_017457486.1_S406s"     </span></span>
<span><span class="co">#&gt; [29] "PHOSPHO;LUNG;XP_017457774.1_S758s"     </span></span>
<span><span class="co">#&gt; [30] "PROT;LUNG;XP_003750594.1"</span></span>
<span></span>
<span><span class="co"># if you want features in this path in multiple tissues:</span></span>
<span><span class="va">clust</span> <span class="op">=</span> <span class="st">"1w_F1_M1-&gt;2w_F1_M1-&gt;4w_F1_M1-&gt;8w_F1_M1"</span></span>
<span><span class="va">tissues</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HEART"</span>,<span class="st">"SKM-GN"</span>,<span class="st">"SKM-VL"</span><span class="op">)</span></span>
<span><span class="va">features_of_interest</span> <span class="op">=</span> <span class="fu">stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html" class="external-link">na.omit</a></span><span class="op">(</span><span class="va">GRAPH_STATES</span><span class="op">$</span><span class="va">feature</span><span class="op">[</span><span class="va">GRAPH_STATES</span><span class="op">$</span><span class="va">path</span> <span class="op">==</span> <span class="va">clust</span> <span class="op">&amp;</span> <span class="va">GRAPH_STATES</span><span class="op">$</span><span class="va">tissue</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">tissues</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">features_of_interest</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 685</span></span>
<span><span class="co"># look at a table of omes and tissues represented </span></span>
<span><span class="va">tissues</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">features_of_interest</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/unname.html" class="external-link">unname</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">x</span>, <span class="st">";"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="va">omes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">features_of_interest</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/unname.html" class="external-link">unname</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">x</span>, <span class="st">";"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">tissues</span>, <span class="va">omes</span><span class="op">)</span></span>
<span><span class="co">#&gt;         omes</span></span>
<span><span class="co">#&gt; tissues  ACETYL ATAC METAB PHOSPHO PROT TRNSCRPT UBIQ</span></span>
<span><span class="co">#&gt;   HEART      50    0    35      52  108      110    2</span></span>
<span><span class="co">#&gt;   SKM-GN      0    6    14      55  130       44    0</span></span>
<span><span class="co">#&gt;   SKM-VL      0    0     2       0    0       77    0</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="plot-graph-of-differential-features">Plot graph of differential features<a class="anchor" aria-label="anchor" href="#plot-graph-of-differential-features"></a>
</h3>
<p>We can use <code><a href="../reference/get_tree_plot_for_tissue.html">get_tree_plot_for_tissue()</a></code> to draw the
graphical representation of a given set of training-regulated
features.</p>
<p>First, let’s look at the graph of all differential features in the
liver. The sizes of the edges and nodes are proportional to the number
of training-regulated features that were assigned to the corresponding
states. Nodes are arranged in rows by states (e.g., “F only up”) and in
columns by training time points. The node at the intersection of “F only
up” and “week 1” represents features that are null in males and
up-regulated in females after one week of endurance exercise training
(i.e., <code>1w_F1_M0</code> in the notation above).</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_tree_plot_for_tissue.html">get_tree_plot_for_tissue</a></span><span class="op">(</span></span>
<span>  tissues<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"LIVER"</span><span class="op">)</span>,</span>
<span>  min_size <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Using `sugiyama` as default layout</span></span></code></pre></div>
<p><img src="MotrpacRatTraining6mo_files/figure-html/tree%20plot%201-1.png" width="960" style="display: block; margin: auto;"></p>
<p>This is messy, so let’s look at just the 5 largest paths instead.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_tree_plot_for_tissue.html">get_tree_plot_for_tissue</a></span><span class="op">(</span></span>
<span>  tissues<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"LIVER"</span><span class="op">)</span>,</span>
<span>  max_trajectories <span class="op">=</span> <span class="fl">5</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Using `sugiyama` as default layout</span></span></code></pre></div>
<p><img src="MotrpacRatTraining6mo_files/figure-html/tree%20plot%202-1.png" width="960" style="display: block; margin: auto;"></p>
<p>Further, let’s split these edges by data type (assay/ome) to see what
features are represented. We can change the curvature and alpha range to
make the edges easier to see.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_tree_plot_for_tissue.html">get_tree_plot_for_tissue</a></span><span class="op">(</span></span>
<span>  tissues<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"LIVER"</span><span class="op">)</span>,</span>
<span>  max_trajectories <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  parallel_edges_by_ome <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  curvature <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  edge_alpha_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.5</span>,<span class="fl">1</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Using `sugiyama` as default layout</span></span></code></pre></div>
<p><img src="MotrpacRatTraining6mo_files/figure-html/tree%20plot%203-1.png" width="960" style="display: block; margin: auto;"></p>
<p>This is too busy with edges split by ome. Let’s look at just the
largest path and only RNA-Seq and proteomics assays.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_tree_plot_for_tissue.html">get_tree_plot_for_tissue</a></span><span class="op">(</span></span>
<span>  omes<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PROT"</span>,<span class="st">"ACETYL"</span>,<span class="st">"UBIQ"</span>,<span class="st">"PHOSPHO"</span>,<span class="st">"TRNSCRPT"</span><span class="op">)</span>,</span>
<span>  tissues<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"LIVER"</span><span class="op">)</span>,</span>
<span>  max_trajectories <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  parallel_edges_by_ome <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  curvature <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  edge_alpha_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.5</span>,<span class="fl">1</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Using `sugiyama` as default layout</span></span></code></pre></div>
<p><img src="MotrpacRatTraining6mo_files/figure-html/tree%20plot%204-1.png" width="960" style="display: block; margin: auto;"></p>
<p>From this plot, we can see that most of the differential features
that follow this path in the liver are protein acetlysites. This path
represents features that are up-regulated only in males (“M only up”)
through the first four weeks of training and then in both males and
females by eight weeks of training (“Both up”).</p>
<p>We can also look at multi-tissue graphs. For example, let’s look at
the three largest paths of training-regulated transcripts in the three
muscle tissues. Now instead of splitting edges by ome, we can split them
by tissue.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_tree_plot_for_tissue.html">get_tree_plot_for_tissue</a></span><span class="op">(</span></span>
<span>  tissues<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SKM-GN"</span>,<span class="st">"HEART"</span>,<span class="st">"SKM-VL"</span><span class="op">)</span>,</span>
<span>  omes<span class="op">=</span><span class="st">"TRNSCRPT"</span>,</span>
<span>  parallel_edges_by_tissue <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  max_trajectories <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  edge_alpha_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.5</span>,<span class="fl">1</span><span class="op">)</span>,</span>
<span>  curvature <span class="op">=</span> <span class="fl">0.6</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Using `sugiyama` as default layout</span></span></code></pre></div>
<p><img src="MotrpacRatTraining6mo_files/figure-html/tree%20plot%205-1.png" width="960" style="display: block; margin: auto;"></p>
<p>From this plot, we can see that a large number of training-regulated
transcripts are up-regulated in both sexes at all four training time
points in all three muscle tissues. Perhaps some interesting shared
biology is represented in muscle features that follow this path.</p>
<p>To visualize the representation of features from different tissues
and/or omes for a given set of clusters, we can also use
<code><a href="../reference/plot_features_per_cluster.html">plot_features_per_cluster()</a></code>. Here, we’ll consider the 10
largest paths of training-regulated features in the gastrocnemius (lower
leg skeletal muscle).</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get top 10 largest paths, nodes, edges in gastrocnemius</span></span>
<span><span class="co"># Exclude additional 8-week nodes</span></span>
<span><span class="va">clusters</span> <span class="op">=</span> <span class="fu"><a href="../reference/extract_tissue_sets.html">extract_tissue_sets</a></span><span class="op">(</span><span class="st">"SKM-GN"</span>, k<span class="op">=</span><span class="fl">10</span>, add_week8<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co"># Select paths only</span></span>
<span><span class="va">clusters</span> <span class="op">=</span> <span class="va">clusters</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"-&gt;"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co"># Plot distribution of features</span></span>
<span><span class="fu"><a href="../reference/plot_features_per_cluster.html">plot_features_per_cluster</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">)</span></span></code></pre></div>
<p><img src="MotrpacRatTraining6mo_files/figure-html/plot_features_per_cluster-1.png" width="864" style="display: block; margin: auto;"></p>
<p>From this plot, we see that the majority of training-regulated
features in the gastrocnemius that are up-regulated in both sexes at all
training time points
(<code>1w_F1_M1-&gt;2w_F1_M1-&gt;4w_F1_M1-&gt;8w_F1_M1</code>, top row)
are proteins, followed by protein phosphosites. We also see a cluster
comprised mostly of protein phosphosites, where these features are
down-regulated in females at all training time points and also
down-regulated in males at 8 weeks of training
(<code>1w_F-1_M0-&gt;2w_F-1_M0-&gt;4w_F-1_M0-&gt;8w_F-1_M-1</code>). In
the next section, we show how pathway enrichment can be used to begin
interpreting the biology underlying a graphical cluster (node, edge, or
path) of interest.</p>
</div>
</div>
<div class="section level2">
<h2 id="pathway-enrichment-of-clusters">Pathway enrichment of clusters<a class="anchor" aria-label="anchor" href="#pathway-enrichment-of-clusters"></a>
</h2>
<p>One important tool we used to interpret the biology underlying
graphical clusters was pathway enrichment. In this section, we describe
functions provided in this package to reproduce our pathway enrichment
analysis or run your own. If you are more interested in exploring the
existing pathway enrichment results, skip to <a href="#vizEnrich">Visualization of pathway enrichment results</a>.</p>
<div class="section level3">
<h3 id="approach">Approach<a class="anchor" aria-label="anchor" href="#approach"></a>
</h3>
<p>We performed pathway enrichment for the 2 largest nodes, 2 largest
edges, 10 largest non-null paths, and all 8-week nodes from the
graphical representation of training-regulated features in each tissue.
This set of graphical clusters can be extracted using
<code><a href="../reference/extract_main_clusters.html">extract_main_clusters()</a></code>.</p>
<p>For each tissue-specific cluster, we performed pathway enrichment
separately for the features corresponding to each ome. For gene-centric
omes, i.e., all but metabolomics, we mapped features to genes and
queried <a href="https://pubmed.ncbi.nlm.nih.gov/33125081/" class="external-link">KEGG</a> and
<a href="https://pubmed.ncbi.nlm.nih.gov/31691815/" class="external-link">REACTOME</a>
pathways using <a href="https://pubmed.ncbi.nlm.nih.gov/33564394/" class="external-link">gprofiler2</a>. For
metabolomics, we mapped metabolites to KEGG IDs and used <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2487-5" class="external-link">FELLA</a>
to query KEGG pathways. The feature-to-gene and metabolite-to-KEGG ID
mappings are provided in <code>FEATURE_TO_GENE</code>.</p>
<p>For each enrichment test, we specified the background/universe as the
set of genes or metabolite IDs that were tested for differential
analysis in that tissue. These backgrounds are provided in
<code>GENE_UNIVERSES</code>. If you are performing your own pathway
enrichment analyses, it is important to specify the appropriate
background to avoid artificially small p-values.</p>
<p>Pathway enrichment results were adjusted over <em>all</em> tests
using <a href="https://www.nature.com/articles/nmeth.3885" class="external-link">IHW</a> with
tissue as a covariate.</p>
<p>Pathway enrichment results for graphical clusters of interest
presented in the <a href="https://motrpac.github.io/MotrpacRatTraining6mo/#citing-motrpac-data">preprint</a>
are provided in <code>GRAPH_PW_ENRICH</code>.</p>
</div>
<div class="section level3">
<h3 id="enrichment-with-standard-pws">Enrichment with standard PWs<a class="anchor" aria-label="anchor" href="#enrichment-with-standard-pws"></a>
</h3>
<p><code><a href="../reference/cluster_pathway_enrichment.html">cluster_pathway_enrichment()</a></code> is a wrapper for <a href="https://rdrr.io/cran/gprofiler2/man/gost.html" class="external-link">gprofiler2::gost()</a>
and <a href="https://rdrr.io/bioc/FELLA/man/enrich-funs.html" class="external-link">FELLA::enrich()</a>,
where metabolomics pathway enrichment is performed for KEGG pathways,
and gene-centric pathway enrichment can be performed for any set of <a href="https://rdrr.io/cran/gprofiler2/man/gost.html" class="external-link">pathway databases
accepted by g:Profiler</a>.</p>
<p>Here we show an example of performing pathway enrichment on two
clusters of training-regulated features, one from each of two tissues.
In this case, clusters are determined using the graphical clustering
approach described above, but in practice the clusters could be
identified using any approach (e.g., k-means, fuzzy C-means).</p>
<p>First, extract the clusters of interest. Here <code>paths</code> is a
named list, where the names are the names of the cluster and the values
are vectors of the corresponding training-regulated features.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># extract the top path from gastrocnemius and heart </span></span>
<span><span class="va">paths</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">tissue</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SKM-GN"</span>,<span class="st">"HEART"</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># extract top node, edge, path</span></span>
<span>  <span class="va">clusters</span> <span class="op">=</span> <span class="fu"><a href="../reference/extract_tissue_sets.html">extract_tissue_sets</a></span><span class="op">(</span><span class="va">tissue</span>, k<span class="op">=</span><span class="fl">1</span>, add_week8<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span>  <span class="co"># select the path only, which contains "-&gt;"</span></span>
<span>  <span class="va">top_path</span> <span class="op">=</span> <span class="va">clusters</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"-&gt;"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span>  <span class="co"># add to list </span></span>
<span>  <span class="va">paths</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">tissue</span>, <span class="st">":"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">top_path</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">]</span><span class="op">]</span> <span class="op">=</span> <span class="va">top_path</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">paths</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "SKM-GN:1w_F1_M1-&gt;2w_F1_M1-&gt;4w_F1_M1-&gt;8w_F1_M1"</span></span>
<span><span class="co">#&gt; [2] "HEART:1w_F1_M1-&gt;2w_F1_M1-&gt;4w_F1_M1-&gt;8w_F1_M1"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">paths</span>, <span class="va">head</span><span class="op">)</span></span>
<span><span class="co">#&gt; $`SKM-GN:1w_F1_M1-&gt;2w_F1_M1-&gt;4w_F1_M1-&gt;8w_F1_M1`</span></span>
<span><span class="co">#&gt; [1] "ATAC;SKM-GN;chr15:101579022-101579761"</span></span>
<span><span class="co">#&gt; [2] "ATAC;SKM-GN;chr1:221194280-221195125" </span></span>
<span><span class="co">#&gt; [3] "ATAC;SKM-GN;chr1:221199295-221200124" </span></span>
<span><span class="co">#&gt; [4] "ATAC;SKM-GN;chr20:34579127-34579734"  </span></span>
<span><span class="co">#&gt; [5] "ATAC;SKM-GN;chr5:114013062-114014681" </span></span>
<span><span class="co">#&gt; [6] "ATAC;SKM-GN;chrX:119151643-119154560" </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`HEART:1w_F1_M1-&gt;2w_F1_M1-&gt;4w_F1_M1-&gt;8w_F1_M1`</span></span>
<span><span class="co">#&gt; [1] "ACETYL;HEART;NP_001004220.1_K116k" "ACETYL;HEART;NP_001006973.1_K401k"</span></span>
<span><span class="co">#&gt; [3] "ACETYL;HEART;NP_001007621.1_K354k" "ACETYL;HEART;NP_001014183.1_K193k"</span></span>
<span><span class="co">#&gt; [5] "ACETYL;HEART;NP_001014183.1_K199k" "ACETYL;HEART;NP_001030123.1_K581k"</span></span></code></pre></div>
<p>Now perform pathway enrichment. First, we will run the function
without specifying <code>kegg_db_destination</code>, which will force it
to ignore metabolites. By default, KEGG and REACTOME pathways are
queried.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load required libraries for this function</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">gprofiler2</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RevolutionAnalytics/foreach" class="external-link">foreach</a></span><span class="op">)</span></span>
<span><span class="co"># to learn more about the arguments and default parameters, look at `?cluster_pathway_enrichment`</span></span>
<span><span class="va">res1</span> <span class="op">=</span> <span class="fu"><a href="../reference/cluster_pathway_enrichment.html">cluster_pathway_enrichment</a></span><span class="op">(</span><span class="va">paths</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in cluster_pathway_enrichment(paths): It looks like you have metabolites in your clustering result, but you did not provide a path for the KEGG database ('kegg_db_destination'). Pathway enrichment will be skipped for metabolites.</span></span>
<span><span class="co">#&gt; Number of cores not defined with 'num_cores' argument. Setting the number of cores to 1. Explicity define 'num_cores' to take advantage of parallelization. If you aren't sure how many cores are available to you, try 'length(parallel::mcaffinity()).'</span></span>
<span><span class="co">#&gt; Running enrichments for 11 gene sets on 1 core(s). This may take a while...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 1 out of 6 identifiers in SKM-GN:1w_F1_M1-&gt;2w_F1_M1-&gt;4w_F1_M1-&gt;8w_F1_M1 ATAC SKM-GN are not in the current universe. Removing these features.</span></span>
<span><span class="co">#&gt; [1] "ENSRNOG00000052048"</span></span>
<span><span class="co">#&gt; Done.</span></span>
<span><span class="co">#&gt; We recommend that you supply (many) more than 1000 p-values for meaningful data-driven hypothesis weighting results.</span></span></code></pre></div>
<p>We get a few warnings and messages.</p>
<ul>
<li>First, we get a warning that pathway enrichment will not be
performed with metabolites because <code>kegg_db_destination=NULL</code>
(default parameter).<br>
</li>
<li>Second, we are told that the analysis will be run on a single core.
If you are performing pathway enrichment for more than a few clusters,
the runtime will be dramatically reduced if you parallelize the analysis
on more than one core. You can do this by specifying the
<code>num_cores</code> argument.
<code><a href="../reference/cluster_pathway_enrichment.html">cluster_pathway_enrichment()</a></code> internally uses
<code>doParallel::registerDoParallel(num_cores)</code> to register more
than one core.<br>
</li>
<li>Third, we are told that one of the gastrocnemius ATAC-seq features
is not specified in the background
(<code>GENE_UNIVERSES$ensembl_gene</code> by default) and is therefore
excluded from the analysis. This is because we decided to define
ATAC-seq (ATAC) and RRBS (METHYL) universes according to the
<em>expressed genes</em> in each tissue, i.e., using the RNA-seq
(TRNSCRPT) data. Therefore, there are some differential ATAC/METHYL
features where the corresponding gene is not regulated by training. This
is one such case.<br>
</li>
<li>Finally, we get a message from <code><a href="https://rdrr.io/pkg/IHW/man/ihw.default.html" class="external-link">IHW::ihw()</a></code>: “We
recommend that you supply (many) more than 1000 p-values for meaningful
data-driven hypothesis weighting results.” With this few tests, one
could reasonably perform BH-adjustment across all tests instead of using
IHW to consider distributions of p-values from each tissue.</li>
</ul>
<p>Let’s look at how many significantly enriched pathways we get at 5%
FDR.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">res1</span><span class="op">[</span><span class="va">res1</span><span class="op">$</span><span class="va">adj_p_value</span> <span class="op">&lt;</span> <span class="fl">0.05</span>,<span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 103</span></span>
<span><span class="co"># split by tissue and ome</span></span>
<span><span class="va">significant</span> <span class="op">=</span> <span class="va">res1</span><span class="op">[</span><span class="va">res1</span><span class="op">$</span><span class="va">adj_p_value</span> <span class="op">&lt;</span> <span class="fl">0.05</span>,<span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">significant</span><span class="op">$</span><span class="va">tissue</span>, <span class="va">significant</span><span class="op">$</span><span class="va">ome</span><span class="op">)</span></span>
<span><span class="co">#&gt;         </span></span>
<span><span class="co">#&gt;          ACETYL PHOSPHO PROT TRNSCRPT</span></span>
<span><span class="co">#&gt;   HEART      12      17   25        8</span></span>
<span><span class="co">#&gt;   SKM-GN      0       0   29       12</span></span></code></pre></div>
<p>Now let’s perform pathway enrichment on the same clusters, this time
including metabolites in the analysis by specifying
<code>kegg_db_destination</code>. In this case,
<code><a href="../reference/cluster_pathway_enrichment.html">cluster_pathway_enrichment()</a></code> calls the
<code><a href="../reference/run_fella.html">run_fella()</a></code> helper function, which is a convenient <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2487-5" class="external-link">FELLA</a>
wrapper to perform KEGG pathway enrichment with metabolites.</p>
<p>If you have not yet installed the KEGG database, e.g., with
<code><a href="../reference/make_kegg_db.html">make_kegg_db()</a></code>, <code>kegg_db_destination</code> must point
to a directory that <strong>does not yet exist</strong> and will be
where the KEGG database needed to run <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2487-5" class="external-link">FELLA</a>
is installed. If necessary, <code><a href="../reference/make_kegg_db.html">make_kegg_db()</a></code> is called within
<code><a href="../reference/cluster_pathway_enrichment.html">cluster_pathway_enrichment()</a></code> to install the KEGG database.
If you have already installed the KEGG database,
<code>kegg_db_destination</code> should point to its parent directory.
For this example, assume the KEGG database has already been installed in
<code>~/kegg/20220919</code>.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load required libraries for pathway enrichment with metabolites</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">FELLA</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">KEGGREST</span><span class="op">)</span></span>
<span><span class="co"># to learn more about the arguments and default parameters, look at `?cluster_pathway_enrichment`</span></span>
<span><span class="va">res2</span> <span class="op">=</span> <span class="fu"><a href="../reference/cluster_pathway_enrichment.html">cluster_pathway_enrichment</a></span><span class="op">(</span><span class="va">paths</span>,</span>
<span>                                  kegg_db_destination <span class="op">=</span> <span class="st">"~/kegg/20220919"</span>,</span>
<span>                                  fella_method <span class="op">=</span> <span class="st">"hypergeom"</span><span class="op">)</span> <span class="co"># "hypergeom" or "diffusion"</span></span>
<span><span class="co">#&gt; KEGG database already exists:</span></span>
<span><span class="co">#&gt;  ~/kegg/20220919</span></span>
<span><span class="co">#&gt; Loading KEGG graph data...</span></span>
<span><span class="co">#&gt; Done.</span></span>
<span><span class="co">#&gt; Loading hypergeom data...</span></span>
<span><span class="co">#&gt; Loading matrix...</span></span>
<span><span class="co">#&gt; Done.</span></span>
<span><span class="co">#&gt; Loading diffusion data...</span></span>
<span><span class="co">#&gt; Loading matrix...</span></span>
<span><span class="co">#&gt; Done.</span></span>
<span><span class="co">#&gt; Loading rowSums...</span></span>
<span><span class="co">#&gt; Done.</span></span>
<span><span class="co">#&gt; Loading pagerank data...</span></span>
<span><span class="co">#&gt; Loading matrix...</span></span>
<span><span class="co">#&gt; 'pagerank.matrix.RData' not loaded. Simulated permutations may execute slower for pagerank.</span></span>
<span><span class="co">#&gt; Done.</span></span>
<span><span class="co">#&gt; Loading rowSums...</span></span>
<span><span class="co">#&gt; Done.</span></span>
<span><span class="co">#&gt; Data successfully loaded.</span></span>
<span><span class="co">#&gt; Number of cores not defined with 'num_cores' argument. Setting the number of cores to 1. Explicity define 'num_cores' to take advantage of parallelization. If you aren't sure how many cores are available to you, try 'length(parallel::mcaffinity()).'</span></span>
<span><span class="co">#&gt; Running enrichments for 11 gene sets on 1 core(s). This may take a while...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 1 out of 6 identifiers in SKM-GN:1w_F1_M1-&gt;2w_F1_M1-&gt;4w_F1_M1-&gt;8w_F1_M1 ATAC SKM-GN are not in the current universe. Removing these features.</span></span>
<span><span class="co">#&gt; [1] "ENSRNOG00000052048"</span></span>
<span><span class="co">#&gt; Running hypergeom...</span></span>
<span><span class="co">#&gt; Starting hypergeometric p-values calculation...</span></span>
<span><span class="co">#&gt; Done.</span></span>
<span><span class="co">#&gt; Warning in defineCompounds(compounds = compounds, compoundsBackground =</span></span>
<span><span class="co">#&gt; compoundsBackground, : Some compounds were introduced as affected but they</span></span>
<span><span class="co">#&gt; do not belong to the background. These compounds will be excluded from the</span></span>
<span><span class="co">#&gt; analysis. Use 'getExcluded' to see them.</span></span>
<span><span class="co">#&gt; Running hypergeom...</span></span>
<span><span class="co">#&gt; Starting hypergeometric p-values calculation...</span></span>
<span><span class="co">#&gt; Done.</span></span>
<span><span class="co">#&gt; Done.</span></span>
<span><span class="co">#&gt; We recommend that you supply (many) more than 1000 p-values for meaningful data-driven hypothesis weighting results.</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">res2</span><span class="op">[</span><span class="va">res2</span><span class="op">$</span><span class="va">adj_p_value</span> <span class="op">&lt;</span> <span class="fl">0.05</span>,<span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 94</span></span>
<span><span class="co"># split by tissue and ome</span></span>
<span><span class="va">significant</span> <span class="op">=</span> <span class="va">res2</span><span class="op">[</span><span class="va">res2</span><span class="op">$</span><span class="va">adj_p_value</span> <span class="op">&lt;</span> <span class="fl">0.05</span>,<span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">significant</span><span class="op">$</span><span class="va">tissue</span>, <span class="va">significant</span><span class="op">$</span><span class="va">ome</span><span class="op">)</span></span>
<span><span class="co">#&gt;         </span></span>
<span><span class="co">#&gt;          ACETYL METAB PHOSPHO PROT TRNSCRPT</span></span>
<span><span class="co">#&gt;   HEART      11     2      15   24        7</span></span>
<span><span class="co">#&gt;   SKM-GN      0     0       0   27        8</span></span></code></pre></div>
<p>Note that the number of significantly enriched pathways for some
non-METAB omes is less than in the previous example. This is a necessary
consequence of performing multiple testing adjustment with more
tests.</p>
</div>
<div class="section level3">
<h3 id="enrichment-with-custom-pws">Enrichment with custom PWs<a class="anchor" aria-label="anchor" href="#enrichment-with-custom-pws"></a>
</h3>
<p>If you want to query pathways not available through
<code>gProfiler</code>, <code><a href="../reference/custom_cluster_pathway_enrichment.html">custom_cluster_pathway_enrichment()</a></code>
can be used to perform pathway enrichment for custom pathway
annotations. Here we show an example with a single cluster of features
and a single annotated pathway.</p>
<p>Note that features in the input must still be in the format
“[ASSAY_ABBREV];[TISSUE_ABBREV];[feature_ID]”,
e.g. “PHOSPHO;SKM-GN;NP_001006973.1_S34s”,
“METAB;SKM-GN;1-methylnicotinamide”, where feature IDs are included in
the supplied <code>feature_to_gene</code> map
(<code>FEATURE_TO_GENE</code> by default). The
<code>universe</code>/background must also be specified using the same
kind of gene identifier as the pathway annotation (rat gene symbols in
this case).</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Use graphical clusters as an example</span></span>
<span><span class="va">cluster_res</span> <span class="op">=</span> <span class="fu"><a href="../reference/extract_main_clusters.html">extract_main_clusters</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co"># Pick a single graphical cluster</span></span>
<span><span class="co"># Gastrocnemius features up-regulated in both males and females at 8 weeks of training</span></span>
<span><span class="va">cluster_res</span> <span class="op">=</span> <span class="va">cluster_res</span><span class="op">[</span><span class="va">cluster_res</span><span class="op">$</span><span class="va">cluster</span> <span class="op">==</span> <span class="st">"SKM-GN:8w_F1_M1"</span>,<span class="op">]</span></span>
<span></span>
<span><span class="co"># Make a toy pathway member list with human gene symbols</span></span>
<span><span class="va">pathways</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"TCA cycle"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'ACO2'</span>,<span class="st">'CS'</span>,<span class="st">'FH'</span>,<span class="st">'MDH1'</span>,<span class="st">'OGDH'</span>,<span class="st">'PDHA1'</span>,<span class="st">'PDHA2'</span>,<span class="st">'SDHC'</span>,<span class="st">'SUCLG1'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Convert human gene symbols to rat gene symbols</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">RAT_TO_HUMAN_GENE</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">pw</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pathways</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">newmembers</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="kw">for</span> <span class="op">(</span><span class="va">m</span> <span class="kw">in</span> <span class="va">pathways</span><span class="op">[[</span><span class="va">pw</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="co"># get rat symbol</span></span>
<span>    <span class="va">rat</span> <span class="op">=</span> <span class="va">RAT_TO_HUMAN_GENE</span><span class="op">$</span><span class="va">RAT_SYMBOL</span><span class="op">[</span><span class="va">RAT_TO_HUMAN_GENE</span><span class="op">$</span><span class="va">HUMAN_ORTHOLOG_SYMBOL</span> <span class="op">==</span> <span class="va">m</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span>    <span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">rat</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>      <span class="va">newmembers</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">newmembers</span>, <span class="va">rat</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="va">pathways</span><span class="op">[[</span><span class="va">pw</span><span class="op">]</span><span class="op">]</span> <span class="op">=</span> <span class="va">newmembers</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Perform pathway enrichment</span></span>
<span><span class="co"># Note that the type of gene identifier used to specify the background</span></span>
<span><span class="co"># must match the type of gene identifier used in the custom pathway annotation </span></span>
<span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="../reference/custom_cluster_pathway_enrichment.html">custom_cluster_pathway_enrichment</a></span><span class="op">(</span><span class="va">cluster_res</span>,</span>
<span>                                        pathway_member_list <span class="op">=</span> <span class="va">pathways</span>,</span>
<span>                                        gene_identifier_type <span class="op">=</span> <span class="st">"gene_symbol"</span>, </span>
<span>                                        universe <span class="op">=</span> <span class="va">GENE_UNIVERSES</span><span class="op">$</span><span class="va">gene_symbol</span>,</span>
<span>                                        min_pw_set_size <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="co">#&gt; Number of cores not defined with 'num_cores' argument. Setting the number of cores to 1. Explicity define 'num_cores' to take advantage of parallelization. If you aren't sure how many cores are available to you, try 'length(parallel::mcaffinity()).'</span></span>
<span><span class="co">#&gt; Running enrichments for 6 gene sets on 1 core(s). This may take a while...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 6 out of 38 identifiers in SKM-GN:8w_F1_M1 ATAC SKM-GN are not in the current universe. Removing these features.</span></span>
<span><span class="co">#&gt; [1] "Edn2"           "AABR07030921.1" "U6"             "AABR07004761.1"</span></span>
<span><span class="co">#&gt; [5] "AABR07062157.1" "Y_RNA"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 2 out of 5 identifiers in SKM-GN:8w_F1_M1 METHYL SKM-GN are not in the current universe. Removing these features.</span></span>
<span><span class="co">#&gt; [1] "Mir196a"        "AABR07002473.1"</span></span>
<span><span class="co">#&gt; Done.</span></span>
<span><span class="va">res</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"term_name"</span>,<span class="st">"intersection"</span>,<span class="st">"tissue"</span>,<span class="st">"ome"</span>,<span class="st">"computed_p_value"</span>,<span class="st">"adj_p_value"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">#&gt;   term_name                 intersection tissue     ome computed_p_value</span></span>
<span><span class="co">#&gt; 1 TCA cycle                      Cs,Mdh1 SKM-GN PHOSPHO     1.306742e-02</span></span>
<span><span class="co">#&gt; 2 TCA cycle Cs,Fh,Mdh1,Ogdh,Pdha1,Suclg1 SKM-GN    PROT     2.885974e-06</span></span>
<span><span class="co">#&gt;    adj_p_value</span></span>
<span><span class="co">#&gt; 1 1.306742e-02</span></span>
<span><span class="co">#&gt; 2 5.771947e-06</span></span></code></pre></div>
<p>Note we get one result per tissue and ome combination.</p>
</div>
</div>
<div class="section level2">
<h2 id="vizEnrich">Visualization of PW enrichments<a class="anchor" aria-label="anchor" href="#vizEnrich"></a>
</h2>
<div class="section level3">
<h3 id="approach-1">Approach<a class="anchor" aria-label="anchor" href="#approach-1"></a>
</h3>
<p>After performing pathway enrichment for our graphical clusters of
interest, we observed that many clusters had many dozens of
significantly enriched pathways. In order to make these results more
digestible, we leveraged <a href="https://cran.r-project.org/web/packages/visNetwork/vignettes/Introduction-to-visNetwork.html" class="external-link">visNetwork</a>
to construct interactive networks of pathway enrichments. These networks
are conceptually similar to those made by the <a href="https://apps.cytoscape.org/apps/enrichmentmap" class="external-link">EnrichmentMap</a>
Cytoscape module, but it’s easier to generate them and explore the
underlying data.</p>
<p>Briefly, each node in the network is a pathway, and edges connect
highly similar pathways, i.e., those whose enrichments are driven by
highly similar sets of genes across omes. This groups together highly
similar pathway enrichments, which makes it easier to digest large
numbers of results.</p>
<p>In our implementation, groups of similar pathway enrichments are
color-coded, and group labels are inferred from the term names and
parents of the pathways in the group. Larger nodes indicate that the
pathway was significantly enriched in multiple datasets; thicker edges
indicate higher similarity between the pathways.</p>
<p>Best of all, these networks are interactive! Use your cursor to zoom
and drag, and hover over nodes and edges to see more information about
the underlying data, namely the genes and datasets driving the pathway
enrichment.</p>
</div>
<div class="section level3">
<h3 id="minimal-example-single-tissue">Minimal example: single tissue<a class="anchor" aria-label="anchor" href="#minimal-example-single-tissue"></a>
</h3>
<p>Here we show the interactive network of pathway enrichments
corresponding to training-regulated features that are up-regulated in
the gastrocnemius in both sexes at the 8-week time point,
i.e. <code>SKM-GN:8w_F1_M1</code>.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract pathway enrichments for this cluster</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GRAPH_PW_ENRICH</span><span class="op">)</span></span>
<span><span class="va">pw</span> <span class="op">=</span> <span class="va">GRAPH_PW_ENRICH</span><span class="op">[</span><span class="va">GRAPH_PW_ENRICH</span><span class="op">$</span><span class="va">cluster</span> <span class="op">==</span> <span class="st">"SKM-GN:8w_F1_M1"</span>,<span class="op">]</span></span>
<span><span class="fu"><a href="../reference/enrichment_network_vis.html">enrichment_network_vis</a></span><span class="op">(</span><span class="va">pw</span><span class="op">)</span></span>
<span><span class="co">#&gt; Formatting inputs...</span></span>
<span><span class="co">#&gt; Subsetting feature-to-gene map...</span></span>
<span><span class="co">#&gt; Calculating similarity metric between pairs of enriched pathways...</span></span>
<span><span class="co">#&gt; Constructing graph...</span></span>
<span><span class="co">#&gt; Elapsed time: 5.865s</span></span></code></pre></div>
<div id="htmlwidget-8591b3e96091637f6f56" style="width:960px;height:432.632880098888px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-8591b3e96091637f6f56">{"x":{"nodes":{"id":["KEGG:00051","KEGG:04137","REAC:R-RNO-70171","KEGG:00564","KEGG:00565","KEGG:04217","KEGG:04977","KEGG:04714","KEGG:05415","KEGG:00190","KEGG:04932","REAC:R-RNO-3371571","REAC:R-RNO-6799198","REAC:R-RNO-611105","REAC:R-RNO-163200","REAC:R-RNO-3371497","REAC:R-RNO-3371556","REAC:R-RNO-1428517","KEGG:00020","KEGG:01200","KEGG:05012","KEGG:05020","KEGG:04723","KEGG:00640","KEGG:00280","KEGG:01212","KEGG:00630","KEGG:03320","KEGG:00620","KEGG:00061","KEGG:00650","KEGG:01210","KEGG:00071","KEGG:00062","REAC:R-RNO-5389840","REAC:R-RNO-5368287","REAC:R-RNO-5419276","REAC:R-RNO-71406","REAC:R-RNO-71403","REAC:R-RNO-8978868","REAC:R-RNO-77289","REAC:R-RNO-389661","REAC:R-RNO-3371568","REAC:R-RNO-556833","REAC:R-RNO-70895","REAC:R-RNO-71291","REAC:R-RNO-70268","REAC:R-RNO-1592230","REAC:R-RNO-5362517","KEGG:04260","KEGG:04670","KEGG:00250","REAC:R-RNO-163210","REAC:R-RNO-8949613","REAC:R-RNO-9663891","REAC:R-RNO-5205647"],"value":[100,100,100,100,100,100,100,400,900,900,900,400,900,900,900,400,100,900,100,100,100,100,400,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,400,100,100,100,100,100,100,100,100],"title":["<b>Fructose and mannose metabolism<\/b> <br>(Carbohydrate metabolism)<br><b>P-val:<\/b> 0.00079<br><b>ATAC:SKM-GN:<\/b> Pfkfb1, Pfkfb3","<b>Mitophagy - animal<\/b> (Transport and <br>catabolism)<br><b>P-val:<\/b> 0.0036<br><b>ATAC:SKM-GN:<\/b> Jun, Ubc","<b>Glycolysis<\/b> (Glucose metabolism)<br><b>P-val:<\/b> 0.0027<br><b>ATAC:SKM-GN:<\/b> Pfkfb1, Pfkfb3","<b>Glycerophospholipid metabolism<\/b> (Lipid <br>metabolism)<br><b>P-val:<\/b> 0.00011<br><b>METAB:SKM-GN:<\/b> unknown (METAB)","<b>Ether lipid metabolism<\/b> (Lipid metabolism)<br><b>P-val:<\/b> 8.2e-05<br><b>METAB:SKM-GN:<\/b> unknown (METAB)","<b>Necroptosis<\/b> (Cell growth and death)<br><b>P-val:<\/b> 0.00064<br><b>METAB:SKM-GN:<\/b> unknown (METAB)","<b>Vitamin digestion and absorption<\/b> <br>(Digestive system)<br><b>P-val:<\/b> 0.0016<br><b>METAB:SKM-GN:<\/b> unknown (METAB)","<b>Thermogenesis<\/b> (Environmental adaptation)<br><b>P-val:<\/b> 9.5e-19<br><b>PHOSPHO:SKM-GN:<\/b> Acsl1, Atp5f1a, Atp5f1c,<br> Ndufa7, Ndufb10, Ndufs4, Ndufs7, Ndufv1, Prkab1,<br> Prkag2, Taf3<br><b>PROT:SKM-GN:<\/b> Acsl1, Atp5f1a, Atp5f1b,<br> Atp5f1c, Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Coa5, Coa7,<br> Cox17, Cox19, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6, Ndufa7,<br> Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7, Ndufaf8,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufs8, Ndufv1, Ndufv2, Ndufv3, Prkag2, Sdha,<br> Sdhb, Taf3, Uqcr10","<b>Diabetic cardiomyopathy<\/b> (Cardiovascular <br>disease)<br><b>P-val:<\/b> 1e-35<br><b>PHOSPHO:SKM-GN:<\/b> Atp5f1a, Atp5f1c, Camk2g,<br> Gys1, Ndufa7, Ndufb10, Ndufs4, Ndufs7, Ndufv1,<br> Slc2a4, Taf3<br><b>PROT:SKM-GN:<\/b> Ace, Atp5f1a, Atp5f1b,<br> Atp5f1c, Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b,<br> Cox6b1, Irs1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7,<br> Ndufs8, Ndufv1, Ndufv2, Ndufv3, Pdha1, Pdhb, Ppif,<br> Sdha, Sdhb, Slc2a4, Taf3, Uqcr10<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Cox4i1, Cox5a,<br> Cox6b1, Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b,<br> Mt-atp6, Mt-co1, Mt-co3, Mt-cyb, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa12, Ndufa3, Ndufa6, Ndufb11,<br> Ndufb4, Ndufb9, Ndufs3, Ndufv1, Tbc1d4, Uqcrq","<b>Oxidative phosphorylation<\/b> (Energy <br>metabolism)<br><b>P-val:<\/b> 6.2e-44<br><b>PHOSPHO:SKM-GN:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufb10, Ndufs4, Ndufs7, Ndufv1, Taf3<br><b>PROT:SKM-GN:<\/b> Atp5f1a, Atp5f1b, Atp5f1c,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox17, Cox5a,<br> Cox5b, Cox6b1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufs8,<br> Ndufv1, Ndufv2, Ndufv3, Ppa2, Sdha, Sdhb, Taf3,<br> Uqcr10<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Cox4i1,<br> Cox5a, Cox6b1, Cox7a1, Cox7a2, Cox7a2l2, Cox7b,<br> Cox8b, Mt-atp6, Mt-co1, Mt-co3, Mt-cyb, Mt-nd1,<br> Mt-nd2, Mt-nd5, Mt-nd6, Ndufa12, Ndufa3, Ndufa6,<br> Ndufb11, Ndufb4, Ndufb9, Ndufs3, Ndufv1, Uqcrq","<b>Non-alcoholic fatty liver disease<\/b> <br>(Endocrine and metabolic disease)<br><b>P-val:<\/b> 2.4e-28<br><b>PHOSPHO:SKM-GN:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Prkab1, Prkag2<br><b>PROT:SKM-GN:<\/b> Cox5a, Cox5b, Cox6b1, Irs1,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa7, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufs8, Ndufv1,<br> Ndufv2, Ndufv3, Prkag2, Sdha, Sdhb, Uqcr10<br><b>TRNSCRPT:SKM-GN:<\/b> Cox4i1, Cox5a, Cox6b1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Il6r, Mt-co1,<br> Mt-co3, Mt-cyb, Ndufa12, Ndufa3, Ndufa6, Ndufb11,<br> Ndufb4, Ndufb9, Ndufs3, Ndufv1, Uqcrq","<b>HSF1-dependent transactivation<\/b> (Cellular <br>response to heat stress)<br><b>P-val:<\/b> 1.2e-07<br><b>PHOSPHO:SKM-GN:<\/b> Camk2g, Hsp90aa1, Hspa1a,<br> Hspa1b<br><b>PROT:SKM-GN:<\/b> Dnajb1, Fkbp4, Hsp90aa1,<br> Hspa1a, Hspa1b, Hspb8","<b>Complex I biogenesis<\/b> (Respiratory electron <br>transport)<br><b>P-val:<\/b> 9.9e-33<br><b>PHOSPHO:SKM-GN:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1<br><b>PROT:SKM-GN:<\/b> Acad9, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa7, Ndufa8, Ndufa9,<br> Ndufaf3, Ndufaf4, Ndufaf7, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufv1, Ndufv2,<br> Ndufv3, Ndufv3-ps1<br><b>TRNSCRPT:SKM-GN:<\/b> Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa12, Ndufa3, Ndufa6, Ndufb11, Ndufb4,<br> Ndufb9, Ndufs3, Ndufv1","<b>Respiratory electron transport<\/b> <br>(Respiratory electron transport, ATP synthesis by <br>chemiosmotic coupling, and heat production by uncoupling <br>proteins.)<br><b>P-val:<\/b> 7.9e-34<br><b>PHOSPHO:SKM-GN:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1<br><b>PROT:SKM-GN:<\/b> Acad9, Etfa, Etfb, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa6, Ndufa7, Ndufa8,<br> Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufv1,<br> Ndufv2, Ndufv3, Ndufv3-ps1<br><b>TRNSCRPT:SKM-GN:<\/b> Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa12, Ndufa3, Ndufa6, Ndufb11, Ndufb4,<br> Ndufb9, Ndufs3, Ndufv1","<b>Respiratory electron transport, ATP synthesis <br>by chemiosmotic coupling, and heat production by <br>uncoupling proteins.<\/b> (The citric acid (TCA) <br>cycle and respiratory electron transport)<br><b>P-val:<\/b> 1.6e-35<br><b>PHOSPHO:SKM-GN:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1<br><b>PROT:SKM-GN:<\/b> Acad9, Atp5pf, Atp5po, Etfa,<br> Etfb, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa7, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufv1, Ndufv2, Ndufv3, Ndufv3-ps1<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Mt-atp6,<br> Mt-nd1, Mt-nd2, Mt-nd5, Mt-nd6, Ndufa12, Ndufa3,<br> Ndufa6, Ndufb11, Ndufb4, Ndufb9, Ndufs3, Ndufv1","<b>HSP90 chaperone cycle for steroid hormone <br>receptors (SHR)<\/b> (Cellular responses to stress)<br><b>P-val:<\/b> 7.5e-06<br><b>PHOSPHO:SKM-GN:<\/b> Dync1li2, Hsp90aa1, Hspa1a,<br> Hspa1b<br><b>PROT:SKM-GN:<\/b> Dnaja1, Dnajb1, Fkbp4, Fkbp5,<br> Hsp90aa1, Hspa1a, Hspa1b, Stip1","<b>Cellular response to heat stress<\/b> (Cellular <br>responses to stress)<br><b>P-val:<\/b> 0.001<br><b>PHOSPHO:SKM-GN:<\/b> Bag3, Camk2g, Hsp90aa1,<br> Hspa1a, Hspa1b","<b>The citric acid (TCA) cycle and respiratory <br>electron transport<\/b> (Metabolism)<br><b>P-val:<\/b> 5e-44<br><b>PHOSPHO:SKM-GN:<\/b> Dld, Ndufa7, Ndufb10,<br> Ndufs4, Ndufs7, Ndufv1<br><b>PROT:SKM-GN:<\/b> Acad9, Atp5pf, Atp5po, Bsg,<br> Dlat, Dld, Etfa, Etfb, Fh, Idh3b, L2hgdh, Mdh2, Me3,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa7, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufv1, Ndufv2, Ndufv3, Ndufv3-ps1, Ogdh, Pdhb,<br> Pdhx, Sdha, Sdhb, Sucla2, Suclg2<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Me3,<br> Mt-atp6, Mt-nd1, Mt-nd2, Mt-nd5, Mt-nd6, Ndufa12,<br> Ndufa3, Ndufa6, Ndufb11, Ndufb4, Ndufb9, Ndufs3,<br> Ndufv1, Pdhx","<b>Citrate cycle (TCA cycle)<\/b> (Carbohydrate <br>metabolism)<br><b>P-val:<\/b> 5.5e-16<br><b>PROT:SKM-GN:<\/b> Cs, Dlat, Dld, Dlst, Fh,<br> Idh3a, Idh3b, Idh3g, Mdh1, Mdh2, Ogdh, Pdha1, Pdhb,<br> Sdha, Sdhb, Sucla2, Suclg1, Suclg2","<b>Carbon metabolism<\/b> (NULL)<br><b>P-val:<\/b> 5.7e-14<br><b>PROT:SKM-GN:<\/b> Acads, Acss2, Cs, Dlat, Dld,<br> Dlst, Fh, Gcsh, Got1, Got2, Hibch, Hk2, Idh3a,<br> Idh3b, Idh3g, Mcee, Mdh1, Mdh2, Me3, Mmut, Ogdh,<br> Pdha1, Pdhb, Sdha, Sdhb, Shmt2, Sucla2, Suclg1, Suclg2","<b>Parkinson disease<\/b> (Neurodegenerative <br>disease)<br><b>P-val:<\/b> 1.3e-10<br><b>PROT:SKM-GN:<\/b> Atp5f1a, Atp5f1b, Atp5f1c,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b,<br> Cox6b1, Htra2, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufs8,<br> Ndufv1, Ndufv2, Ndufv3, Ppif, Sdha, Sdhb, Taf3,<br> Trap1, Uqcr10","<b>Prion disease<\/b> (Neurodegenerative disease)<br><b>P-val:<\/b> 1.1e-09<br><b>PROT:SKM-GN:<\/b> Atp5f1a, Atp5f1b, Atp5f1c,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b,<br> Cox6b1, Hspa1a, Hspa1b, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufs8, Ndufv1, Ndufv2, Ndufv3, Ppif, Sdha,<br> Sdhb, Stip1, Taf3, Uqcr10","<b>Retrograde endocannabinoid signaling<\/b> <br>(Nervous system)<br><b>P-val:<\/b> 4.3e-16<br><b>PROT:SKM-GN:<\/b> Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufs8, Ndufv1, Ndufv2, Ndufv3<br><b>TRNSCRPT:SKM-GN:<\/b> Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa12, Ndufa3, Ndufa6, Ndufb11, Ndufb4,<br> Ndufb9, Ndufs3, Ndufv1","<b>Propanoate metabolism<\/b> (Carbohydrate <br>metabolism)<br><b>P-val:<\/b> 6.2e-08<br><b>PROT:SKM-GN:<\/b> Acads, Acss2, Bckdha, Bckdhb,<br> Dld, Hibch, Mcee, Mlycd, Mmut, Sucla2, Suclg1,<br> Suclg2","<b>Valine, leucine and isoleucine degradation<\/b> <br>(Amino acid metabolism)<br><b>P-val:<\/b> 1.5e-07<br><b>PROT:SKM-GN:<\/b> Acaa2, Acadm, Acads, Bckdha,<br> Bckdhb, Dld, Hadh, Hibadh, Hibch, Hsd17b10, Mcee,<br> Mmut, Oxct1","<b>Fatty acid metabolism<\/b> (NULL)<br><b>P-val:<\/b> 2.3e-06<br><b>PROT:SKM-GN:<\/b> Acaa2, Acadl, Acadm, Acads,<br> Acadvl, Acsl1, Cbr4, Fasn, Hadh, Hsd17b8, Mecr,<br> Oxsm","<b>Glyoxylate and dicarboxylate metabolism<\/b> <br>(Carbohydrate metabolism)<br><b>P-val:<\/b> 3.7e-06<br><b>PROT:SKM-GN:<\/b> Acss2, Cs, Dld, Gcsh, Mcee,<br> Mdh1, Mdh2, Mmut, Shmt2","<b>PPAR signaling pathway<\/b> (Endocrine system)<br><b>P-val:<\/b> 3.3e-05<br><b>PROT:SKM-GN:<\/b> Acadl, Acadm, Acsl1, Fabp4,<br> Fabp5, Gk, Me3, Plin2, Plin4, Plin5, Sorbs1","<b>Pyruvate metabolism<\/b> (Carbohydrate <br>metabolism)<br><b>P-val:<\/b> 3.5e-05<br><b>PROT:SKM-GN:<\/b> Acss2, Dlat, Dld, Fh, Ldhd,<br> Mdh1, Mdh2, Me3, Pdha1, Pdhb","<b>Fatty acid biosynthesis<\/b> (Lipid metabolism)<br><b>P-val:<\/b> 9.1e-05<br><b>PROT:SKM-GN:<\/b> Acsl1, Cbr4, Fasn, Hsd17b8,<br> Mecr, Oxsm","<b>Butanoate metabolism<\/b> (Carbohydrate <br>metabolism)<br><b>P-val:<\/b> 9.1e-05<br><b>PROT:SKM-GN:<\/b> Acads, Aldh5a1, Bdh1, Hadh,<br> L2hgdh, Oxct1","<b>2-Oxocarboxylic acid metabolism<\/b> (NULL)<br><b>P-val:<\/b> 9.1e-05<br><b>PROT:SKM-GN:<\/b> Cs, Got1, Got2, Idh3a, Idh3b,<br> Idh3g","<b>Fatty acid degradation<\/b> (Lipid metabolism)<br><b>P-val:<\/b> 0.00013<br><b>PROT:SKM-GN:<\/b> Acaa2, Acadl, Acadm, Acads,<br> Acadvl, Acsl1, Eci2, Eci3, Gcdh, Hadh","<b>Fatty acid elongation<\/b> (Lipid metabolism)<br><b>P-val:<\/b> 0.0014<br><b>PROT:SKM-GN:<\/b> Acaa2, Acot2, Hadh, Mecr,<br> Them4","<b>Mitochondrial translation elongation<\/b> <br>(Mitochondrial translation)<br><b>P-val:<\/b> 5.1e-14<br><b>PROT:SKM-GN:<\/b> Dap3, Gfm1, Mrpl12, Mrpl16,<br> Mrpl17, mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55,<br> Mrps12, Mrps15, Mrps16, Mrps17, Mrps21, Mrps22,<br> Mrps23, Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Mitochondrial translation<\/b> (Translation)<br><b>P-val:<\/b> 1e-13<br><b>PROT:SKM-GN:<\/b> Dap3, Gfm1, Mrpl12, Mrpl16,<br> Mrpl17, mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55,<br> Mrps12, Mrps15, Mrps16, Mrps17, Mrps21, Mrps22,<br> Mrps23, Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Mitochondrial translation termination<\/b> <br>(Mitochondrial translation)<br><b>P-val:<\/b> 5.9e-13<br><b>PROT:SKM-GN:<\/b> Dap3, Mrpl12, Mrpl16, Mrpl17,<br> mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55, Mrps12,<br> Mrps15, Mrps16, Mrps17, Mrps21, Mrps22, Mrps23,<br> Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Pyruvate metabolism and Citric Acid (TCA) <br>cycle<\/b> (The citric acid (TCA) cycle and respiratory <br>electron transport)<br><b>P-val:<\/b> 2.5e-10<br><b>PROT:SKM-GN:<\/b> Bsg, Dlat, Dld, Fh, Idh3b,<br> L2hgdh, Mdh2, Me3, Ogdh, Pdhb, Pdhx, Sdha, Sdhb,<br> Sucla2, Suclg2","<b>Citric acid cycle (TCA cycle)<\/b> (Pyruvate <br>metabolism and Citric Acid (TCA) cycle)<br><b>P-val:<\/b> 7.5e-09<br><b>PROT:SKM-GN:<\/b> Dld, Fh, Idh3b, Mdh2, Me3,<br> Ogdh, Sdha, Sdhb, Sucla2, Suclg2","<b>Fatty acid metabolism<\/b> (Metabolism of <br>lipids)<br><b>P-val:<\/b> 1.2e-08<br><b>PROT:SKM-GN:<\/b> Acad10, Acadl, Acot11, Acot13,<br> Acot9, Acsl1, Cbr4, Eci2, Eci3, Fasn, Hadh,<br> Hsd17b8, Mcee, Mlycd, Mmut, Phyh, Prkag2, Ptges2,<br> Them4","<b>Mitochondrial Fatty Acid Beta-Oxidation<\/b> <br>(Fatty acid metabolism)<br><b>P-val:<\/b> 2.3e-06<br><b>PROT:SKM-GN:<\/b> Acad10, Acadl, Acot11, Acot13,<br> Acot9, Hadh, Mcee, Mmut, Them4","<b>Glyoxylate metabolism and glycine <br>degradation<\/b> (Metabolism of amino acids and derivatives)<br><b>P-val:<\/b> 6.5e-06<br><b>PROT:SKM-GN:<\/b> Dlat, Dld, Gcsh, Got2, Ogdh,<br> Pdhb, Pdhx","<b>Attenuation phase<\/b> (HSF1-dependent <br>transactivation)<br><b>P-val:<\/b> 3.1e-05<br><b>PROT:SKM-GN:<\/b> Dnajb1, Fkbp4, Hsp90aa1,<br> Hspa1a, Hspa1b","<b>Metabolism of lipids<\/b> (Metabolism)<br><b>P-val:<\/b> 0.00027<br><b>PROT:SKM-GN:<\/b> Acad10, Acadl, Acot11, Acot13,<br> Acot9, Acp6, Acsl1, Cbr4, Eci2, Eci3, Fabp5,<br> Fasn, Fdxr, Gpd1l, Hadh, Hsd17b8, Lpin1, Mcee, Mlycd,<br> Mmut, Oxct1, Phyh, Plin3, Prkag2, Ptges2, Them4","<b>Branched-chain amino acid catabolism<\/b> <br>(Metabolism of amino acids and derivatives)<br><b>P-val:<\/b> 0.00026<br><b>PROT:SKM-GN:<\/b> Bckdk, Dld, Hibadh, Hibch,<br> Ppm1k","<b>Metabolism of amino acids and derivatives<\/b> <br>(Metabolism)<br><b>P-val:<\/b> 0.00049<br><b>PROT:SKM-GN:<\/b> Bckdk, Dlat, Dld, Ethe1, Gcdh,<br> Gcsh, Got1, Got2, Hibadh, Hibch, Oat, Ogdh, Pdhb,<br> Pdhx, Ppm1k","<b>Pyruvate metabolism<\/b> (Pyruvate metabolism <br>and Citric Acid (TCA) cycle)<br><b>P-val:<\/b> 0.0014<br><b>PROT:SKM-GN:<\/b> Bsg, Dlat, Dld, Pdhb, Pdhx","<b>Mitochondrial biogenesis<\/b> (Organelle <br>biogenesis and maintenance)<br><b>P-val:<\/b> 3.7e-05<br><b>PROT:SKM-GN:<\/b> Acss2, Atp5pf, Atp5po, Sirt3,<br> Sod2<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Signaling by Retinoic Acid<\/b> (Signaling by <br>Nuclear Receptors)<br><b>P-val:<\/b> 0.0035<br><b>PROT:SKM-GN:<\/b> Dlat, Dld, Fabp5, Pdhb, Pdhx","<b>Cardiac muscle contraction<\/b> (Circulatory <br>system)<br><b>P-val:<\/b> 1.5e-11<br><b>TRNSCRPT:SKM-GN:<\/b> Atp1a2, Cox4i1, Cox5a,<br> Cox6b1, Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b,<br> Mt-co1, Mt-co3, Mt-cyb, Uqcrq","<b>Leukocyte transendothelial migration<\/b> <br>(Immune system)<br><b>P-val:<\/b> 0.0024<br><b>TRNSCRPT:SKM-GN:<\/b> Cdh5, Cxcl12, Jam2,<br> Rapgef4, Vcl","<b>Alanine, aspartate and glutamate <br>metabolism<\/b> (Amino acid metabolism)<br><b>P-val:<\/b> 0.0029<br><b>TRNSCRPT:SKM-GN:<\/b> Aldh5a1, Got1, Got2","<b>Formation of ATP by chemiosmotic coupling<\/b> <br>(Respiratory electron transport, ATP synthesis by <br>chemiosmotic coupling, and heat production by <br>uncoupling proteins.)<br><b>P-val:<\/b> 0.00031<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Cristae formation<\/b> (Mitochondrial <br>biogenesis)<br><b>P-val:<\/b> 0.00031<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Selective autophagy<\/b> (Macroautophagy)<br><b>P-val:<\/b> 0.0014<br><b>TRNSCRPT:SKM-GN:<\/b> Map1lc3a, Map1lc3b, Src","<b>Mitophagy<\/b> (Selective autophagy)<br><b>P-val:<\/b> 0.0014<br><b>TRNSCRPT:SKM-GN:<\/b> Map1lc3a, Map1lc3b, Src"],"color.highlight.background":["yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow"],"color.border":["black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black"],"shadow":[false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],"physics":[false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],"borderWidth":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],"borderWidthSelected":[3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3],"pathway_subclass":["Carbohydrate metabolism","Transport and catabolism","Glucose metabolism","Lipid metabolism","Lipid metabolism","Cell growth and death","Digestive system","Environmental adaptation","Cardiovascular disease","Energy metabolism","Endocrine and metabolic disease","Cellular response to heat stress","Respiratory electron transport","Respiratory electron transport, ATP synthesis by chemiosmotic coupling, and heat production by uncoupling proteins.","The citric acid (TCA) cycle and respiratory electron transport","Cellular responses to stress","Cellular responses to stress","Metabolism","Carbohydrate metabolism","NULL","Neurodegenerative disease","Neurodegenerative disease","Nervous system","Carbohydrate metabolism","Amino acid metabolism","NULL","Carbohydrate metabolism","Endocrine system","Carbohydrate metabolism","Lipid metabolism","Carbohydrate metabolism","NULL","Lipid metabolism","Lipid metabolism","Mitochondrial translation","Translation","Mitochondrial translation","The citric acid (TCA) cycle and respiratory electron transport","Pyruvate metabolism and Citric Acid (TCA) cycle","Metabolism of lipids","Fatty acid metabolism","Metabolism of amino acids and derivatives","HSF1-dependent transactivation","Metabolism","Metabolism of amino acids and derivatives","Metabolism","Pyruvate metabolism and Citric Acid (TCA) cycle","Organelle biogenesis and maintenance","Signaling by Nuclear Receptors","Circulatory system","Immune system","Amino acid metabolism","Respiratory electron transport, ATP synthesis by chemiosmotic coupling, and heat production by uncoupling proteins.","Mitochondrial biogenesis","Macroautophagy","Selective autophagy"],"group":["1: Carbohydrate metabolism (1/2)","0: Carbohydrate metabolism (1/6)","1: Carbohydrate metabolism (1/2)","3: Lipid metabolism (2/2)","3: Lipid metabolism (2/2)","0: Carbohydrate metabolism (1/6)","0: Carbohydrate metabolism (1/6)","6: Neurodegenerative disease (2/15)","6: Neurodegenerative disease (2/15)","6: Neurodegenerative disease (2/15)","6: Neurodegenerative disease (2/15)","7: Cellular responses to stress (2/4)","6: Neurodegenerative disease (2/15)","6: Neurodegenerative disease (2/15)","6: Neurodegenerative disease (2/15)","7: Cellular responses to stress (2/4)","7: Cellular responses to stress (2/4)","6: Neurodegenerative disease (2/15)","8: Carbohydrate metabolism (4/13)","8: Carbohydrate metabolism (4/13)","6: Neurodegenerative disease (2/15)","6: Neurodegenerative disease (2/15)","6: Neurodegenerative disease (2/15)","8: Carbohydrate metabolism (4/13)","8: Carbohydrate metabolism (4/13)","9: Lipid metabolism (3/6)","8: Carbohydrate metabolism (4/13)","0: Carbohydrate metabolism (1/6)","8: Carbohydrate metabolism (4/13)","9: Lipid metabolism (3/6)","0: Carbohydrate metabolism (1/6)","8: Carbohydrate metabolism (4/13)","9: Lipid metabolism (3/6)","9: Lipid metabolism (3/6)","12: Mitochondrial translation (2/3)","12: Mitochondrial translation (2/3)","12: Mitochondrial translation (2/3)","8: Carbohydrate metabolism (4/13)","8: Carbohydrate metabolism (4/13)","9: Lipid metabolism (3/6)","9: Lipid metabolism (3/6)","8: Carbohydrate metabolism (4/13)","7: Cellular responses to stress (2/4)","9: Lipid metabolism (3/6)","8: Carbohydrate metabolism (4/13)","8: Carbohydrate metabolism (4/13)","8: Carbohydrate metabolism (4/13)","6: Neurodegenerative disease (2/15)","8: Carbohydrate metabolism (4/13)","6: Neurodegenerative disease (2/15)","0: Carbohydrate metabolism (1/6)","8: Carbohydrate metabolism (4/13)","6: Neurodegenerative disease (2/15)","6: Neurodegenerative disease (2/15)","14: Macroautophagy (1/2)","14: Macroautophagy (1/2)"],"group_number":["1","0","1","3","3","0","0","6","6","6","6","7","6","6","6","7","7","6","8","8","6","6","6","8","8","9","8","0","8","9","0","8","9","9","12","12","12","8","8","9","9","8","7","9","8","8","8","6","8","6","0","8","6","6","14","14"],"color.background":["#8DD3C7","#FFFFB3","#8DD3C7","#BEBADA","#BEBADA","#FFFFB3","#FFFFB3","#FB8072","#FB8072","#FB8072","#FB8072","#80B1D3","#FB8072","#FB8072","#FB8072","#80B1D3","#80B1D3","#FB8072","#FDB462","#FDB462","#FB8072","#FB8072","#FB8072","#FDB462","#FDB462","#B3DE69","#FDB462","#FFFFB3","#FDB462","#B3DE69","#FFFFB3","#FDB462","#B3DE69","#B3DE69","#FCCDE5","#FCCDE5","#FCCDE5","#FDB462","#FDB462","#B3DE69","#B3DE69","#FDB462","#80B1D3","#B3DE69","#FDB462","#FDB462","#FDB462","#FB8072","#FDB462","#FB8072","#FFFFB3","#FDB462","#FB8072","#FB8072","#D9D9D9","#D9D9D9"],"color.highlight.border":["#8DD3C7","#FFFFB3","#8DD3C7","#BEBADA","#BEBADA","#FFFFB3","#FFFFB3","#FB8072","#FB8072","#FB8072","#FB8072","#80B1D3","#FB8072","#FB8072","#FB8072","#80B1D3","#80B1D3","#FB8072","#FDB462","#FDB462","#FB8072","#FB8072","#FB8072","#FDB462","#FDB462","#B3DE69","#FDB462","#FFFFB3","#FDB462","#B3DE69","#FFFFB3","#FDB462","#B3DE69","#B3DE69","#FCCDE5","#FCCDE5","#FCCDE5","#FDB462","#FDB462","#B3DE69","#B3DE69","#FDB462","#80B1D3","#B3DE69","#FDB462","#FDB462","#FDB462","#FB8072","#FDB462","#FB8072","#FFFFB3","#FDB462","#FB8072","#FB8072","#D9D9D9","#D9D9D9"],"x":[-0.11026967701262,-0.982861196625032,-0.0574475172782052,-1,-0.907816591517361,0.172208487605309,-0.321089439480344,-0.397983806881813,-0.425887580187571,-0.357376847671733,-0.48435699534822,0.66327124446585,-0.254595885009632,-0.297698652326017,-0.275367407565433,0.743314369574789,0.647907398410176,-0.0883022248998029,0.282255493202449,0.469040364820352,-0.276043682293548,-0.359697169745887,-0.349203157223405,0.718108437330633,0.75295604907962,-0.641467915662642,0.740015514147539,-0.991251062621519,0.34994770256412,-0.474778551903129,0.486021932172035,0.550804757336745,-0.638404926742745,-0.813770650786042,0.198329969084393,0.139857018718515,0.273983601856271,0.196194210193302,0.223013448043693,-0.571792836110292,-0.539222376524774,0.247130484104703,0.567153847677941,-0.426981323057316,0.566724570214044,0.41699846687339,0.183814627978322,-0.162992896749402,0.156381795173481,-0.634707363896377,-0.134351283106032,0.644011003654512,-0.244345075790664,-0.15333839620616,0.94762434366122,1],"y":[-0.831989638823115,-0.322034854898085,-0.977035579458035,0.212166899784794,0.0973110459401187,-0.380406150393419,-1,0.559513618258743,0.450140193024031,0.625366334587401,0.503712084742127,-0.635015153057943,0.448640605952243,0.388354889791006,0.608883245712365,-0.533095546075824,-0.443289277591293,0.526167125843414,0.541381686953654,0.484897537485571,0.521748875236027,0.404439088054035,0.495314542956331,0.262613346317959,0.0512281523182951,-0.457842372043466,0.591205847785105,-0.104884450043368,0.39805548109104,-0.515853133903684,-0.91810063252879,0.619193734759778,-0.270419131308773,-0.507546831302382,-0.868755929988269,-0.715716737338843,-0.742736958905191,0.517027314965256,0.665440489174645,-0.634483206144121,-0.799628018055225,0.408100049120196,-0.543907823621292,-0.681491067855321,0.0343043043155746,0.267806407980252,0.319757766218906,1,0.395637887910044,0.560186094837954,-0.38251101638016,0.409907514985502,0.809194285672352,0.785631370206028,-0.24750305742147,-0.109983309507107]},"edges":{"from":["REAC:R-RNO-1428517","KEGG:01212","KEGG:01212","KEGG:01212","KEGG:04714","KEGG:05415","KEGG:01210","REAC:R-RNO-71291","KEGG:00640","KEGG:00564","KEGG:00020","KEGG:01200","KEGG:01200","KEGG:01200","KEGG:00020","KEGG:00020","KEGG:01200","KEGG:00190","KEGG:04932","KEGG:05415","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05012","KEGG:05020","KEGG:05415","REAC:R-RNO-1428517","REAC:R-RNO-163200","REAC:R-RNO-611105","REAC:R-RNO-6799198","KEGG:00190","KEGG:04714","KEGG:05415","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05415","REAC:R-RNO-1428517","REAC:R-RNO-163200","REAC:R-RNO-611105","REAC:R-RNO-6799198","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05012","KEGG:05415","REAC:R-RNO-1428517","REAC:R-RNO-163200","REAC:R-RNO-611105","REAC:R-RNO-6799198","KEGG:04714","KEGG:00190","KEGG:04714","KEGG:05415","REAC:R-RNO-163200","REAC:R-RNO-611105","REAC:R-RNO-6799198","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05415","REAC:R-RNO-611105","REAC:R-RNO-6799198","KEGG:00190","REAC:R-RNO-1428517","REAC:R-RNO-1592230","REAC:R-RNO-163200","REAC:R-RNO-3371571","REAC:R-RNO-3371497","REAC:R-RNO-3371571","REAC:R-RNO-3371497","REAC:R-RNO-3371556","REAC:R-RNO-3371571","KEGG:00020","KEGG:01200","REAC:R-RNO-1428517","REAC:R-RNO-71406","REAC:R-RNO-9663891","KEGG:00020","KEGG:00620","REAC:R-RNO-1428517","REAC:R-RNO-389661","REAC:R-RNO-70268","REAC:R-RNO-71291","REAC:R-RNO-71406","REAC:R-RNO-5389840","REAC:R-RNO-5368287","REAC:R-RNO-5389840","KEGG:00061","REAC:R-RNO-77289","REAC:R-RNO-8978868","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05415","REAC:R-RNO-6799198","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05415","KEGG:00051","KEGG:00020","KEGG:00620","REAC:R-RNO-1428517","REAC:R-RNO-389661","REAC:R-RNO-71291","REAC:R-RNO-71406","KEGG:00280","KEGG:01200","REAC:R-RNO-389661","REAC:R-RNO-70895","KEGG:00020","KEGG:01200","REAC:R-RNO-1428517","REAC:R-RNO-71406","KEGG:00020","KEGG:00620","KEGG:01200","REAC:R-RNO-1428517","REAC:R-RNO-8978868","KEGG:00190","REAC:R-RNO-1428517","REAC:R-RNO-1592230","REAC:R-RNO-163200","REAC:R-RNO-163210","KEGG:00061","KEGG:01212"],"to":["KEGG:00020","KEGG:00061","KEGG:00062","KEGG:00071","KEGG:00190","KEGG:00190","KEGG:00250","KEGG:00250","KEGG:00280","KEGG:00565","KEGG:00620","KEGG:00620","KEGG:00630","KEGG:00640","KEGG:01200","KEGG:01210","KEGG:01210","KEGG:04260","KEGG:04260","KEGG:04260","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04932","KEGG:04932","KEGG:04932","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05415","REAC:R-RNO-1428517","REAC:R-RNO-1428517","REAC:R-RNO-1428517","REAC:R-RNO-1428517","REAC:R-RNO-1428517","REAC:R-RNO-1428517","REAC:R-RNO-163200","REAC:R-RNO-163200","REAC:R-RNO-163200","REAC:R-RNO-163200","REAC:R-RNO-163200","REAC:R-RNO-163200","REAC:R-RNO-163210","REAC:R-RNO-163210","REAC:R-RNO-163210","REAC:R-RNO-163210","REAC:R-RNO-3371497","REAC:R-RNO-3371556","REAC:R-RNO-3371556","REAC:R-RNO-3371568","REAC:R-RNO-3371568","REAC:R-RNO-3371568","REAC:R-RNO-389661","REAC:R-RNO-389661","REAC:R-RNO-389661","REAC:R-RNO-389661","REAC:R-RNO-5205647","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5368287","REAC:R-RNO-5419276","REAC:R-RNO-5419276","REAC:R-RNO-556833","REAC:R-RNO-556833","REAC:R-RNO-556833","REAC:R-RNO-611105","REAC:R-RNO-611105","REAC:R-RNO-611105","REAC:R-RNO-611105","REAC:R-RNO-611105","REAC:R-RNO-6799198","REAC:R-RNO-6799198","REAC:R-RNO-6799198","REAC:R-RNO-6799198","REAC:R-RNO-70171","REAC:R-RNO-70268","REAC:R-RNO-70268","REAC:R-RNO-70268","REAC:R-RNO-70268","REAC:R-RNO-70268","REAC:R-RNO-70268","REAC:R-RNO-70895","REAC:R-RNO-71291","REAC:R-RNO-71291","REAC:R-RNO-71291","REAC:R-RNO-71403","REAC:R-RNO-71403","REAC:R-RNO-71403","REAC:R-RNO-71403","REAC:R-RNO-71406","REAC:R-RNO-71406","REAC:R-RNO-71406","REAC:R-RNO-71406","REAC:R-RNO-77289","REAC:R-RNO-8949613","REAC:R-RNO-8949613","REAC:R-RNO-8949613","REAC:R-RNO-8949613","REAC:R-RNO-8949613","REAC:R-RNO-8978868","REAC:R-RNO-8978868"],"value":[19.7132616487455,37.5,20.3571428571429,31.9444444444444,29.9709724238026,39.4301470588235,23.8095238095238,19.7916666666667,24.3055555555556,18.75,25.8333333333333,30,32.7586206896552,25.78125,40.5172413793103,21.6666666666667,30.1724137931034,27.827380952381,29.1666666666667,27.3521505376344,33.3664772727273,25.2403846153846,30.2027925531915,27.2529069767442,27.2529069767442,32.1593237704918,29.0948275862069,32.703488372093,34.5394736842105,35.4166666666667,34.2526964560863,25.9673808762392,33.8068181818182,37.1408045977011,40.1041666666667,30.3197064989518,37.1031746031746,22.0833333333333,21.5277777777778,19.6759259259259,19.9786324786325,37.1408045977011,40.1041666666667,30.3197064989518,45.9795321637427,37.1031746031746,22.0833333333333,21.5277777777778,19.6759259259259,19.9786324786325,27.8432914046122,22.0211161387632,20.125786163522,21.1180124223602,37.7896613190731,34.9308300395257,33.75,26.2443438914027,20.5487033523087,18.7618595825427,24.2851307189542,40.8967391304348,39.375,26.3636363636364,26.3636363636364,34.375,26.875,25,25,34.7222222222222,40,29.1666666666667,43.75,19.047619047619,26.4285714285714,20.0501253132832,25.2100840336134,50,18.75,21.25,21.7857142857143,32.5,36.6666666666667,26.25,26.25,50,49.0740740740741,49.0740740740741,20.3703703703704,34,43,23.2023411371237,20.3113485630066,20.6235697940503,22.3897734231476,42.3214285714286,24.1251803751804,21.1423444976077,21.4772727272727,23.2625164690382,50,18.75,21.25,27.2727272727273,32.5,26.25,33.3333333333333,20,18.8888888888889,36.6666666666667,33.3333333333333,34.3421052631579,33.6206896551724,29.5454545454545,41.6666666666667,30.8333333333333,22.8947368421053,29.375,31.8181818181818,37.5,26.3636363636364,26.3636363636364,34.375,26.875,50,21.6666666666667,18.75],"title":["<b>Similarity score:<\/b> 0.39<br><b>Intersection:<\/b> Dld, Dlat, Fh, Idh3b, Mdh2,<br> Ogdh, Pdhb, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.75<br><b>Intersection:<\/b> Acsl1, Cbr4, Fasn, Hsd17b8,<br> Mecr, Oxsm","<b>Similarity score:<\/b> 0.41<br><b>Intersection:<\/b> Acaa2, Hadh, Mecr","<b>Similarity score:<\/b> 0.64<br><b>Intersection:<\/b> Acaa2, Acadl, Acadm, Acads,<br> Acadvl, Acsl1, Hadh","<b>Similarity score:<\/b> 0.6<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufb10, Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox17, Cox5a, Cox5b,<br> Cox6b1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa4, Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Sdha,<br> Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.79<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufb10, Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb,<br> Uqcr10, Atp5mc1, Cox4i1, Cox7a1, Cox7a2, Cox7a2l2,<br> Cox7b, Cox8b, Mt-atp6, Mt-co1, Mt-co3, Mt-cyb,<br> Mt-nd1, Mt-nd2, Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4,<br> Uqcrq","<b>Similarity score:<\/b> 0.48<br><b>Intersection:<\/b> Got1, Got2","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Got1, Got2","<b>Similarity score:<\/b> 0.49<br><b>Intersection:<\/b> Acads, Bckdha, Bckdhb, Dld,<br> Hibch, Mcee, Mmut","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> unknown (METAB)","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Dlat, Dld, Fh, Mdh1, Mdh2,<br> Pdha1, Pdhb","<b>Similarity score:<\/b> 0.6<br><b>Intersection:<\/b> Acss2, Dlat, Dld, Fh, Mdh1,<br> Mdh2, Me3, Pdha1, Pdhb","<b>Similarity score:<\/b> 0.66<br><b>Intersection:<\/b> Acss2, Cs, Dld, Gcsh, Mcee,<br> Mdh1, Mdh2, Mmut, Shmt2","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Acads, Acss2, Dld, Hibch,<br> Mcee, Mmut, Sucla2, Suclg1, Suclg2","<b>Similarity score:<\/b> 0.81<br><b>Intersection:<\/b> Cs, Dlat, Dld, Dlst, Fh,<br> Idh3a, Idh3b, Idh3g, Mdh1, Mdh2, Ogdh, Pdha1, Pdhb,<br> Sdha, Sdhb, Sucla2, Suclg1, Suclg2","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Cs, Idh3a, Idh3b, Idh3g","<b>Similarity score:<\/b> 0.6<br><b>Intersection:<\/b> Cs, Got1, Got2, Idh3a, Idh3b,<br> Idh3g","<b>Similarity score:<\/b> 0.56<br><b>Intersection:<\/b> Cox5a, Cox6b1, Cox4i1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Mt-co1, Mt-co3,<br> Mt-cyb, Uqcrq","<b>Similarity score:<\/b> 0.58<br><b>Intersection:<\/b> Cox5a, Cox6b1, Cox4i1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Mt-co1, Mt-co3,<br> Mt-cyb, Uqcrq","<b>Similarity score:<\/b> 0.55<br><b>Intersection:<\/b> Cox5a, Cox6b1, Cox4i1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Mt-co1, Mt-co3,<br> Mt-cyb, Uqcrq","<b>Similarity score:<\/b> 0.67<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Mt-nd1,<br> Mt-nd2, Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.5<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.6<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Ndufa3,<br> Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.55<br><b>Intersection:<\/b> Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufs8, Ndufv1, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.55<br><b>Intersection:<\/b> Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufs8, Ndufv1, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.64<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Mt-nd1,<br> Mt-nd2, Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.58<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.65<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.71<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6, Ndufa8,<br> Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufs8,<br> Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10, Cox4i1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Mt-co1, Mt-co3,<br> Mt-cyb, Ndufa3, Ndufb11, Ndufb4, Uqcrq","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Prkab1, Prkag2, Cox5a, Cox5b, Cox6b1,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb,<br> Uqcr10","<b>Similarity score:<\/b> 0.68<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Cox5a, Cox5b, Cox6b1, Irs1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6, Ndufa8,<br> Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10, Cox4i1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Mt-co1,<br> Mt-co3, Mt-cyb, Ndufa3, Ndufb11, Ndufb4, Uqcrq","<b>Similarity score:<\/b> 0.74<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.8<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.61<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Cox5a, Cox5b, Cox6b1, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa4, Ndufa6, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufs8, Ndufv2,<br> Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.74<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Ppif, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.44<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3, Sdha, Sdhb","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.39<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.74<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.8<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.61<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Cox5a, Cox5b, Cox6b1, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa4, Ndufa6, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufs8, Ndufv2,<br> Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.92<br><b>Intersection:<\/b> Atp5f1a, Atp5f1b, Atp5f1c,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b,<br> Cox6b1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa7, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufs8,<br> Ndufv1, Ndufv2, Ndufv3, Ppif, Sdha, Sdhb, Taf3, Uqcr10","<b>Similarity score:<\/b> 0.74<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Ppif, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.44<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3, Sdha, Sdhb","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.39<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.56<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufb10, Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb,<br> Uqcr10","<b>Similarity score:<\/b> 0.44<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3, Sdha,<br> Sdhb, Atp5mc1, Atp5mf, Mt-atp6, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3,<br> Ndufaf4, Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Sdha, Sdhb","<b>Similarity score:<\/b> 0.42<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3, Pdhb,<br> Sdha, Sdhb, Atp5mc1, Mt-atp6, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.76<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Atp5pf, Atp5po, Etfa, Etfb,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6, Ndufa8,<br> Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3,<br> Ndufv3-ps1, Atp5mc1, Atp5mf, Mt-atp6, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.7<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Etfa, Etfb, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3,<br> Ndufaf4, Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Ndufv3-ps1, Mt-nd1,<br> Mt-nd2, Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.68<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4,<br> Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufv2, Ndufv3, Ndufv3-ps1, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3,<br> Atp5mc1, Atp5mf, Mt-atp6, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.41<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3,<br> Ndufaf4, Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Ndufa3, Ndufb11,<br> Ndufb4","<b>Similarity score:<\/b> 0.49<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3,<br> Atp5mc1, Mt-atp6, Mt-nd1, Mt-nd2, Mt-nd5, Mt-nd6,<br> Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.82<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Etfa, Etfb, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3,<br> Ndufaf4, Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Ndufv3-ps1, Mt-nd1,<br> Mt-nd2, Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.79<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4,<br> Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufv2, Ndufv3, Ndufv3-ps1, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.53<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.53<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.54<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.5<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b,<br> Dnajb1, Fkbp4","<b>Similarity score:<\/b> 0.5<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Camk2g, Hsp90aa1, Hspa1a,<br> Hspa1b","<b>Similarity score:<\/b> 0.8<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b,<br> Dnajb1, Fkbp4","<b>Similarity score:<\/b> 0.58<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b","<b>Similarity score:<\/b> 0.88<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b,<br> Dnajb1, Fkbp4","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Dlat, Dld, Ogdh, Pdhb","<b>Similarity score:<\/b> 0.53<br><b>Intersection:<\/b> Dlat, Dld, Gcsh, Got2, Ogdh,<br> Pdhb","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Dld, Dlat, Ogdh, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.5<br><b>Intersection:<\/b> Dlat, Dld, Ogdh, Pdhb, Pdhx","<b>Similarity score:<\/b> 1<br><b>Intersection:<\/b> Map1lc3a, Map1lc3b, Src","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Dlat, Dld, Pdhb","<b>Similarity score:<\/b> 0.42<br><b>Intersection:<\/b> Dlat, Dld, Pdhb","<b>Similarity score:<\/b> 0.44<br><b>Intersection:<\/b> Dld, Dlat, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.65<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.73<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 1<br><b>Intersection:<\/b> Dap3, Gfm1, Mrpl12, Mrpl16,<br> Mrpl17, mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55,<br> Mrps12, Mrps15, Mrps16, Mrps17, Mrps21, Mrps22,<br> Mrps23, Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Similarity score:<\/b> 0.98<br><b>Intersection:<\/b> Dap3, Mrpl12, Mrpl16, Mrpl17,<br> mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55, Mrps12,<br> Mrps15, Mrps16, Mrps17, Mrps21, Mrps22, Mrps23,<br> Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Similarity score:<\/b> 0.98<br><b>Intersection:<\/b> Dap3, Mrpl12, Mrpl16, Mrpl17,<br> mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55, Mrps12,<br> Mrps15, Mrps16, Mrps17, Mrps21, Mrps22, Mrps23,<br> Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Similarity score:<\/b> 0.41<br><b>Intersection:<\/b> Acsl1, Cbr4, Fasn, Hsd17b8","<b>Similarity score:<\/b> 0.68<br><b>Intersection:<\/b> Acad10, Acadl, Acot11,<br> Acot13, Acot9, Hadh, Mcee, Mmut, Them4","<b>Similarity score:<\/b> 0.86<br><b>Intersection:<\/b> Acad10, Acadl, Acot11,<br> Acot13, Acot9, Acsl1, Cbr4, Eci2, Eci3, Fasn, Hadh,<br> Hsd17b8, Mcee, Mlycd, Mmut, Phyh, Prkag2, Ptges2,<br> Them4","<b>Similarity score:<\/b> 0.46<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.41<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2,<br> Ndufv3","<b>Similarity score:<\/b> 0.41<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Ndufa3, Ndufb11,<br> Ndufb4","<b>Similarity score:<\/b> 0.45<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.85<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4,<br> Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufv2, Ndufv3, Ndufv3-ps1, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.48<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.42<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2,<br> Ndufv3","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Ndufa3, Ndufb11,<br> Ndufb4","<b>Similarity score:<\/b> 0.47<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 1<br><b>Intersection:<\/b> Pfkfb1, Pfkfb3","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Dlat, Dld, Pdhb","<b>Similarity score:<\/b> 0.42<br><b>Intersection:<\/b> Dlat, Dld, Pdhb","<b>Similarity score:<\/b> 0.55<br><b>Intersection:<\/b> Dld, Bsg, Dlat, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.65<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.67<br><b>Intersection:<\/b> Bsg, Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Dld, Hibadh, Hibch","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Dlat, Dld, Gcsh, Got1, Got2,<br> Hibch, Ogdh, Pdhb","<b>Similarity score:<\/b> 0.73<br><b>Intersection:<\/b> Dlat, Dld, Gcsh, Got2, Ogdh,<br> Pdhb, Pdhx","<b>Similarity score:<\/b> 0.67<br><b>Intersection:<\/b> Bckdk, Dld, Hibadh, Hibch,<br> Ppm1k","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Dld, Fh, Idh3b, Mdh2, Ogdh,<br> Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.67<br><b>Intersection:<\/b> Dld, Fh, Idh3b, Mdh2, Me3,<br> Ogdh, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.59<br><b>Intersection:<\/b> Dld, Fh, Idh3b, Mdh2, Me3,<br> Ogdh, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.83<br><b>Intersection:<\/b> Dld, Fh, Idh3b, Mdh2, Me3,<br> Ogdh, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.62<br><b>Intersection:<\/b> Dlat, Dld, Fh, Idh3b, Mdh2,<br> Ogdh, Pdhb, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.46<br><b>Intersection:<\/b> Dlat, Dld, Fh, Mdh2, Me3,<br> Pdhb","<b>Similarity score:<\/b> 0.59<br><b>Intersection:<\/b> Dlat, Dld, Fh, Idh3b, Mdh2,<br> Me3, Ogdh, Pdhb, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.64<br><b>Intersection:<\/b> Dld, Bsg, Dlat, Fh, Idh3b,<br> L2hgdh, Mdh2, Me3, Ogdh, Pdhb, Pdhx, Sdha, Sdhb,<br> Sucla2, Suclg2","<b>Similarity score:<\/b> 0.75<br><b>Intersection:<\/b> Acad10, Acadl, Acot11,<br> Acot13, Acot9, Hadh, Mcee, Mmut, Them4","<b>Similarity score:<\/b> 0.53<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.53<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.54<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 1<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Acsl1, Cbr4, Fasn, Hsd17b8","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Acadl, Acsl1, Cbr4, Fasn,<br> Hadh, Hsd17b8"],"color.color":["#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484"],"physics":[false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],"color.highlight":["#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454"],"labelHighlightBold":[true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot","physics":false},"manipulation":{"enabled":false},"interaction":{"hideEdgesOnZoom":true,"tooltipDelay":10,"zoomSpeed":1},"edges":{"smooth":false},"physics":{"stabilization":false}},"groups":["1: Carbohydrate metabolism (1/2)","0: Carbohydrate metabolism (1/6)","3: Lipid metabolism (2/2)","6: Neurodegenerative disease (2/15)","7: Cellular responses to stress (2/4)","8: Carbohydrate metabolism (4/13)","9: Lipid metabolism (3/6)","12: Mitochondrial translation (2/3)","14: Macroautophagy (1/2)"],"width":null,"height":null,"idselection":{"enabled":false},"byselection":{"enabled":false},"main":null,"submain":null,"footer":null,"background":"rgba(0, 0, 0, 0)","legend":{"width":0.3,"useGroups":false,"position":"left","ncol":1,"stepX":100,"stepY":40,"zoom":false,"nodes":{"label":["0: Carbohydrate metabolism (1/6)","1: Carbohydrate metabolism (1/2)","3: Lipid metabolism (2/2)","6: Neurodegenerative disease (2/15)","7: Cellular responses to stress (2/4)","8: Carbohydrate metabolism (4/13)","9: Lipid metabolism (3/6)","12: Mitochondrial translation (2/3)","14: Macroautophagy (1/2)"],"shape":["box","box","box","box","box","box","box","box","box"],"color":["#FFFFB3","#8DD3C7","#BEBADA","#FB8072","#80B1D3","#FDB462","#B3DE69","#FCCDE5","#D9D9D9"],"physics":[false,false,false,false,false,false,false,false,false]},"nodesToDataframe":true},"tooltipStay":null,"tooltipStyle":"position: fixed;visibility:hidden;padding: 5px;white-space: nowrap;font-family: verdana;font-size:14px;font-color:#000000;background-color: #f5f4ed;-moz-border-radius: 3px;-webkit-border-radius: 3px;border-radius: 3px;border: 1px solid #808074;box-shadow: 3px 3px 10px rgba(0, 0, 0, 0.2);","igraphlayout":{"type":"square"}},"evals":[],"jsHooks":[]}</script><p>Note that you could run this function with any pathway enrichment
results, as long as the columns of <code>pw_enrich_res</code> include
“adj_p_value”, “ome”, “tissue”, “intersection”, “computed_p_value”,
“term_name”, and “term_id” and the columns of
<code>feature_to_gene</code> include “gene_symbol”, “ensembl_gene”.
<code>feature$ensembl_gene</code> can be a dummy column if
`intersection_id_type = “gene_symbol”.</p>
</div>
<div class="section level3">
<h3 id="example-with-multiple-tissues">Example with multiple tissues<a class="anchor" aria-label="anchor" href="#example-with-multiple-tissues"></a>
</h3>
<p>Now let’s examine pathways that are enriched in the same graphical
cluster for multiple tissues. For this example, we’ll consider features
that are up-regulated in both males and females at the 8-week training
time point (<code>8w_F1_M1</code>) in any of our three muscle tissues
(heart, gastrocnemius, vastus lateralis).</p>
<p>If we are specifically interested in pathways that are significantly
enriched in multiple tissues, we can set
<code>multitissue_pathways_only</code> to <code>TRUE</code>. This
removes any pathways that are not enriched in more than one tissue (in
at least one ome).</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pw</span> <span class="op">=</span> <span class="va">GRAPH_PW_ENRICH</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">":8w_F1_M1"</span>, <span class="va">GRAPH_PW_ENRICH</span><span class="op">$</span><span class="va">cluster</span><span class="op">)</span> <span class="op">&amp;</span> </span>
<span>                       <span class="va">GRAPH_PW_ENRICH</span><span class="op">$</span><span class="va">tissue</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HEART"</span>,<span class="st">"SKM-GN"</span>,<span class="st">"SKM-VL"</span><span class="op">)</span>,<span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="../reference/enrichment_network_vis.html">enrichment_network_vis</a></span><span class="op">(</span><span class="va">pw</span>, multitissue_pathways_only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Formatting inputs...</span></span>
<span><span class="co">#&gt; Subsetting feature-to-gene map...</span></span>
<span><span class="co">#&gt; Calculating similarity metric between pairs of enriched pathways...</span></span>
<span><span class="co">#&gt; Constructing graph...</span></span>
<span><span class="co">#&gt; Elapsed time: 4.266s</span></span></code></pre></div>
<div id="htmlwidget-7f00fd9105fb14885aa3" style="width:960px;height:432.632880098888px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-7f00fd9105fb14885aa3">{"x":{"nodes":{"id":["KEGG:01200","KEGG:00020","KEGG:00640","KEGG:01212","KEGG:00280","KEGG:00071","KEGG:01210","KEGG:00630","KEGG:00620","KEGG:01230","KEGG:00650","REAC:R-RNO-1428517","REAC:R-RNO-71406","REAC:R-RNO-71403","REAC:R-RNO-77289","REAC:R-RNO-8978868","REAC:R-RNO-70895","REAC:R-RNO-70268","REAC:R-RNO-389661","REAC:R-RNO-163200","KEGG:04217","REAC:R-RNO-1592230","KEGG:00330","REAC:R-RNO-3371571","REAC:R-RNO-3371556","REAC:R-RNO-3371497","KEGG:00250","KEGG:00051","KEGG:03320","REAC:R-RNO-3371568","REAC:R-RNO-70171","KEGG:05415","KEGG:00190","KEGG:04137","KEGG:04723","KEGG:00220","REAC:R-RNO-611105","REAC:R-RNO-6799198","KEGG:04977","KEGG:04932","REAC:R-RNO-5368287","REAC:R-RNO-5419276","KEGG:04260","REAC:R-RNO-163210","REAC:R-RNO-8949613"],"value":[3600,2500,1600,900,900,900,2500,900,900,2500,900,3600,900,900,900,900,400,900,900,3600,900,1600,400,2500,1600,2500,400,400,400,400,400,2500,2500,400,1600,400,2500,2500,400,1600,400,400,400,400,400],"title":["<b>Carbon metabolism<\/b> (NULL)<br><b>P-val:<\/b> 4e-45<br><b>ACETYL:HEART:<\/b> Acads, Aco2, Acox1, Cs, Dlat,<br> Dld, Echs1, Fh, Hibch, Idh1, Idh2, Idh3a, Mdh1,<br> Mmut, Pdhb, Pkm, Sucla2, Suclg2<br><b>PHOSPHO:HEART:<\/b> Acox3, Dld, Eno1, Eno3,<br> Glyctk, Got1, Idh2, Mdh1, Pdha1, Pkm, Prps1<br><b>PROT:HEART:<\/b> Acox1, Acox3, Aldoa, Cs, Dld,<br> Echs1, Eno3, Gapdh, Glud1, Got2, Hk2, Idh1, Idh3a,<br> Idh3b, Idh3g, Mcee, Pdha1, Pfkp, Pgam1, Pgk1, Pkm,<br> Prps1, Tpi1<br><b>TRNSCRPT:HEART:<\/b> Dld, Eno3, Got1, Got2, Gpt,<br> Hk2, Pkm<br><b>PROT:SKM-GN:<\/b> Acads, Acss2, Cs, Dlat, Dld,<br> Dlst, Fh, Gcsh, Got1, Got2, Hibch, Hk2, Idh3a,<br> Idh3b, Idh3g, Mcee, Mdh1, Mdh2, Me3, Mmut, Ogdh,<br> Pdha1, Pdhb, Sdha, Sdhb, Shmt2, Sucla2, Suclg1, Suclg2<br><b>TRNSCRPT:SKM-VL:<\/b> Acads, Cs, Dlst, Got1,<br> Got2, Gpt, Idh2, Idh3b, Mcee, Mdh1, Sdhb","<b>Citrate cycle (TCA cycle)<\/b> (Carbohydrate <br>metabolism)<br><b>P-val:<\/b> 1.5e-32<br><b>ACETYL:HEART:<\/b> Aco2, Cs, Dlat, Dld, Fh,<br> Idh1, Idh2, Idh3a, Mdh1, Pdhb, Sucla2, Suclg2<br><b>PHOSPHO:HEART:<\/b> Dld, Idh2, Mdh1, Pdha1<br><b>PROT:HEART:<\/b> Cs, Dld, Idh1, Idh3a, Idh3b,<br> Idh3g, Pdha1<br><b>PROT:SKM-GN:<\/b> Cs, Dlat, Dld, Dlst, Fh,<br> Idh3a, Idh3b, Idh3g, Mdh1, Mdh2, Ogdh, Pdha1, Pdhb,<br> Sdha, Sdhb, Sucla2, Suclg1, Suclg2<br><b>TRNSCRPT:SKM-VL:<\/b> Cs, Dlst, Idh2, Idh3b,<br> Mdh1, Sdhb","<b>Propanoate metabolism<\/b> (Carbohydrate <br>metabolism)<br><b>P-val:<\/b> 1.2e-16<br><b>ACETYL:HEART:<\/b> Acads, Acox1, Dbt, Dld,<br> Echs1, Hibch, Mmut, Sucla2, Suclg2<br><b>PROT:HEART:<\/b> Acox1, Acox3, Dld, Echs1,<br> Ehhadh, Mcee, Mlycd<br><b>TRNSCRPT:HEART:<\/b> Bckdhb, Dbt, Dld, Ldhb<br><b>PROT:SKM-GN:<\/b> Acads, Acss2, Bckdha, Bckdhb,<br> Dld, Hibch, Mcee, Mlycd, Mmut, Sucla2, Suclg1,<br> Suclg2","<b>Fatty acid metabolism<\/b> (NULL)<br><b>P-val:<\/b> 3.6e-12<br><b>ACETYL:HEART:<\/b> Acaa2, Acads, Acadvl, Acox1,<br> Echs1, Hadh, Hadhb, Hsd17b4, Scp2<br><b>PROT:HEART:<\/b> Acadvl, Acox1, Acox3, Echs1,<br> Ehhadh, Hsd17b4<br><b>PROT:SKM-GN:<\/b> Acaa2, Acadl, Acadm, Acads,<br> Acadvl, Acsl1, Cbr4, Fasn, Hadh, Hsd17b8, Mecr,<br> Oxsm","<b>Valine, leucine and isoleucine degradation<\/b> <br>(Amino acid metabolism)<br><b>P-val:<\/b> 6.4e-13<br><b>ACETYL:HEART:<\/b> Acaa2, Acads, Dbt, Dld,<br> Echs1, Hadh, Hadhb, Hibadh, Hibch, Mmut<br><b>PROT:HEART:<\/b> Aox3, Dld, Echs1, Ehhadh,<br> Hsd17b10, Mcee<br><b>PROT:SKM-GN:<\/b> Acaa2, Acadm, Acads, Bckdha,<br> Bckdhb, Dld, Hadh, Hibadh, Hibch, Hsd17b10, Mcee,<br> Mmut, Oxct1","<b>Fatty acid degradation<\/b> (Lipid metabolism)<br><b>P-val:<\/b> 7.2e-11<br><b>ACETYL:HEART:<\/b> Acaa2, Acads, Acadvl, Acox1,<br> Echs1, Eci1, Hadh, Hadhb<br><b>PROT:HEART:<\/b> Acadvl, Acox1, Acox3, Adh1,<br> Adh6, Echs1, Eci2, Eci3, Ehhadh<br><b>PROT:SKM-GN:<\/b> Acaa2, Acadl, Acadm, Acads,<br> Acadvl, Acsl1, Eci2, Eci3, Gcdh, Hadh","<b>2-Oxocarboxylic acid metabolism<\/b> (NULL)<br><b>P-val:<\/b> 2.2e-17<br><b>ACETYL:HEART:<\/b> Aco2, Cs, Idh1, Idh2, Idh3a<br><b>PROT:HEART:<\/b> Cs, Got2, Idh1, Idh3a, Idh3b,<br> Idh3g<br><b>TRNSCRPT:HEART:<\/b> Got1, Got2, Gpt<br><b>PROT:SKM-GN:<\/b> Cs, Got1, Got2, Idh3a, Idh3b,<br> Idh3g<br><b>TRNSCRPT:SKM-VL:<\/b> Cs, Got1, Got2, Gpt, Idh2,<br> Idh3b","<b>Glyoxylate and dicarboxylate metabolism<\/b> <br>(Carbohydrate metabolism)<br><b>P-val:<\/b> 1.1e-09<br><b>ACETYL:HEART:<\/b> Aco2, Cs, Dld, Mdh1, Mmut<br><b>PHOSPHO:HEART:<\/b> Dld, Glyctk, Mdh1<br><b>PROT:SKM-GN:<\/b> Acss2, Cs, Dld, Gcsh, Mcee,<br> Mdh1, Mdh2, Mmut, Shmt2","<b>Pyruvate metabolism<\/b> (Carbohydrate <br>metabolism)<br><b>P-val:<\/b> 4.3e-09<br><b>ACETYL:HEART:<\/b> Dlat, Dld, Fh, Mdh1, Pdhb,<br> Pkm<br><b>PHOSPHO:HEART:<\/b> Dld, Mdh1, Pdha1, Pkm<br><b>PROT:SKM-GN:<\/b> Acss2, Dlat, Dld, Fh, Ldhd,<br> Mdh1, Mdh2, Me3, Pdha1, Pdhb","<b>Biosynthesis of amino acids<\/b> (NULL)<br><b>P-val:<\/b> 1.3e-18<br><b>ACETYL:HEART:<\/b> Aco2, Cs, Idh1, Idh2, Idh3a,<br> Pkm<br><b>PHOSPHO:HEART:<\/b> Eno1, Eno3, Got1, Idh2, Pkm,<br> Prps1<br><b>PROT:HEART:<\/b> Aldoa, Cs, Eno3, Gapdh, Got2,<br> Idh1, Idh3a, Idh3b, Idh3g, Pfkp, Pgam1, Pgk1, Pkm,<br> Prps1, Pycr2, Tpi1<br><b>TRNSCRPT:HEART:<\/b> Eno3, Got1, Got2, Gpt, Pkm<br><b>TRNSCRPT:SKM-VL:<\/b> Cs, Got1, Got2, Gpt, Idh2,<br> Idh3b","<b>Butanoate metabolism<\/b> (Carbohydrate <br>metabolism)<br><b>P-val:<\/b> 8.1e-08<br><b>ACETYL:HEART:<\/b> Acads, Aldh5a1, Echs1, Hadh<br><b>PROT:HEART:<\/b> Aldh5a1, Bdh1, Echs1, Ehhadh<br><b>PROT:SKM-GN:<\/b> Acads, Aldh5a1, Bdh1, Hadh,<br> L2hgdh, Oxct1","<b>The citric acid (TCA) cycle and respiratory <br>electron transport<\/b> (Metabolism)<br><b>P-val:<\/b> 1.6e-94<br><b>ACETYL:HEART:<\/b> Acad9, Aco2, Atp5po, Dlat,<br> Dld, Etfb, Fh, Idh2, Ndufa2, Ndufb3, Ndufs1, Ndufv1,<br> Pdhb, Pdpr, Sucla2, Suclg2<br><b>TRNSCRPT:HEART:<\/b> Bsg, Coq10a, Dld, Ldhb,<br> Mt-nd1, Mt-nd2, Mt-nd4, Mt-nd5, Ndufa12, Ndufb4<br><b>PHOSPHO:SKM-GN:<\/b> Dld, Ndufa7, Ndufb10,<br> Ndufs4, Ndufs7, Ndufv1<br><b>PROT:SKM-GN:<\/b> Acad9, Atp5pf, Atp5po, Bsg,<br> Dlat, Dld, Etfa, Etfb, Fh, Idh3b, L2hgdh, Mdh2, Me3,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa7, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufv1, Ndufv2, Ndufv3, Ndufv3-ps1, Ogdh, Pdhb,<br> Pdhx, Sdha, Sdhb, Sucla2, Suclg2<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Me3,<br> Mt-atp6, Mt-nd1, Mt-nd2, Mt-nd5, Mt-nd6, Ndufa12,<br> Ndufa3, Ndufa6, Ndufb11, Ndufb4, Ndufb9, Ndufs3,<br> Ndufv1, Pdhx<br><b>TRNSCRPT:SKM-VL:<\/b> Atp5mc1, Atp5mf, Atp5pb,<br> Atp5pd, Atp5pf, Atp5po, Dmac2l, Etfb, Idh2, Idh3b,<br> Mt-atp6, Mt-nd1, Mt-nd2, Mt-nd4, Mt-nd5, Mt-nd6,<br> Ndufa12, Ndufa2, Ndufa3, Ndufa5, Ndufa6, Ndufa7,<br> Ndufa9, Ndufab1, Ndufb10, Ndufb11, Ndufb4, Ndufb5,<br> Ndufb6, Ndufb7, Ndufb8, Ndufb9, Ndufc1, Ndufc2,<br> Ndufs3, Ndufs4, Ndufs5, Ndufv1, Nnt, Pdhx, Sdhb,<br> Slc16a1","<b>Pyruvate metabolism and Citric Acid (TCA) <br>cycle<\/b> (The citric acid (TCA) cycle and respiratory <br>electron transport)<br><b>P-val:<\/b> 3.5e-17<br><b>ACETYL:HEART:<\/b> Aco2, Dlat, Dld, Fh, Idh2,<br> Pdhb, Pdpr, Sucla2, Suclg2<br><b>PROT:SKM-GN:<\/b> Bsg, Dlat, Dld, Fh, Idh3b,<br> L2hgdh, Mdh2, Me3, Ogdh, Pdhb, Pdhx, Sdha, Sdhb,<br> Sucla2, Suclg2<br><b>TRNSCRPT:SKM-VL:<\/b> Idh2, Idh3b, Nnt, Pdhx,<br> Sdhb, Slc16a1","<b>Citric acid cycle (TCA cycle)<\/b> (Pyruvate <br>metabolism and Citric Acid (TCA) cycle)<br><b>P-val:<\/b> 7.2e-14<br><b>ACETYL:HEART:<\/b> Aco2, Dld, Fh, Idh2, Sucla2,<br> Suclg2<br><b>PROT:SKM-GN:<\/b> Dld, Fh, Idh3b, Mdh2, Me3,<br> Ogdh, Sdha, Sdhb, Sucla2, Suclg2<br><b>TRNSCRPT:SKM-VL:<\/b> Idh2, Idh3b, Nnt, Sdhb","<b>Mitochondrial Fatty Acid Beta-Oxidation<\/b> <br>(Fatty acid metabolism)<br><b>P-val:<\/b> 1.9e-11<br><b>ACETYL:HEART:<\/b> Acsf2, Echs1, Hadh, Hadhb,<br> Mmut<br><b>PROT:HEART:<\/b> Acad11, Acot11, Dbi, Echs1,<br> Mcee<br><b>PROT:SKM-GN:<\/b> Acad10, Acadl, Acot11, Acot13,<br> Acot9, Hadh, Mcee, Mmut, Them4","<b>Fatty acid metabolism<\/b> (Metabolism of <br>lipids)<br><b>P-val:<\/b> 1.9e-15<br><b>ACETYL:HEART:<\/b> Acsf2, Echs1, Hadh, Hadhb,<br> Hsd17b4, Mmut<br><b>PROT:HEART:<\/b> Acad11, Acot11, Dbi, Echs1,<br> Eci2, Eci3, Ehhadh, Hacl1, Hsd17b4, Mcee, Mlycd,<br> Phyh, Prkag2, Ptges2<br><b>PROT:SKM-GN:<\/b> Acad10, Acadl, Acot11, Acot13,<br> Acot9, Acsl1, Cbr4, Eci2, Eci3, Fasn, Hadh,<br> Hsd17b8, Mcee, Mlycd, Mmut, Phyh, Prkag2, Ptges2,<br> Them4","<b>Branched-chain amino acid catabolism<\/b> <br>(Metabolism of amino acids and derivatives)<br><b>P-val:<\/b> 3.2e-06<br><b>ACETYL:HEART:<\/b> Dbt, Dld, Hibadh, Hibch<br><b>PROT:SKM-GN:<\/b> Bckdk, Dld, Hibadh, Hibch,<br> Ppm1k","<b>Pyruvate metabolism<\/b> (Pyruvate metabolism <br>and Citric Acid (TCA) cycle)<br><b>P-val:<\/b> 5.2e-07<br><b>ACETYL:HEART:<\/b> Dlat, Dld, Pdhb, Pdpr<br><b>TRNSCRPT:HEART:<\/b> Bsg, Dld, Ldhb<br><b>PROT:SKM-GN:<\/b> Bsg, Dlat, Dld, Pdhb, Pdhx","<b>Glyoxylate metabolism and glycine <br>degradation<\/b> (Metabolism of amino acids and derivatives)<br><b>P-val:<\/b> 6.7e-09<br><b>ACETYL:HEART:<\/b> Dbt, Dlat, Dld, Pdhb<br><b>TRNSCRPT:HEART:<\/b> Dbt, Dld, Got2<br><b>PROT:SKM-GN:<\/b> Dlat, Dld, Gcsh, Got2, Ogdh,<br> Pdhb, Pdhx","<b>Respiratory electron transport, ATP synthesis <br>by chemiosmotic coupling, and heat production by <br>uncoupling proteins.<\/b> (The citric acid (TCA) <br>cycle and respiratory electron transport)<br><b>P-val:<\/b> 3.9e-79<br><b>ACETYL:HEART:<\/b> Acad9, Atp5po, Etfb, Ndufa2,<br> Ndufb3, Ndufs1, Ndufv1<br><b>TRNSCRPT:HEART:<\/b> Coq10a, Mt-nd1, Mt-nd2,<br> Mt-nd4, Mt-nd5, Ndufa12, Ndufb4<br><b>PHOSPHO:SKM-GN:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1<br><b>PROT:SKM-GN:<\/b> Acad9, Atp5pf, Atp5po, Etfa,<br> Etfb, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa7, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufv1, Ndufv2, Ndufv3, Ndufv3-ps1<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Mt-atp6,<br> Mt-nd1, Mt-nd2, Mt-nd5, Mt-nd6, Ndufa12, Ndufa3,<br> Ndufa6, Ndufb11, Ndufb4, Ndufb9, Ndufs3, Ndufv1<br><b>TRNSCRPT:SKM-VL:<\/b> Atp5mc1, Atp5mf, Atp5pb,<br> Atp5pd, Atp5pf, Atp5po, Dmac2l, Etfb, Mt-atp6,<br> Mt-nd1, Mt-nd2, Mt-nd4, Mt-nd5, Mt-nd6, Ndufa12,<br> Ndufa2, Ndufa3, Ndufa5, Ndufa6, Ndufa7, Ndufa9,<br> Ndufab1, Ndufb10, Ndufb11, Ndufb4, Ndufb5, Ndufb6,<br> Ndufb7, Ndufb8, Ndufb9, Ndufc1, Ndufc2, Ndufs3, Ndufs4,<br> Ndufs5, Ndufv1","<b>Necroptosis<\/b> (Cell growth and death)<br><b>P-val:<\/b> 1.7e-06<br><b>METAB:HEART:<\/b> unknown (METAB)<br><b>TRNSCRPT:HEART:<\/b> Ftl1, Jak1, Mapk9, Pygm, Slc25a5, Sqstm1<br><b>METAB:SKM-GN:<\/b> unknown (METAB)","<b>Mitochondrial biogenesis<\/b> (Organelle <br>biogenesis and maintenance)<br><b>P-val:<\/b> 2.6e-17<br><b>METHYL:HEART:<\/b> Sirt4<br><b>PROT:SKM-GN:<\/b> Acss2, Atp5pf, Atp5po, Sirt3,<br> Sod2<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Mt-atp6<br><b>TRNSCRPT:SKM-VL:<\/b> Atp5mc1, Atp5mf, Atp5pb,<br> Atp5pd, Atp5pf, Atp5po, Dmac2l, Idh2, Mt-atp6,<br> Sod2","<b>Arginine and proline metabolism<\/b> (Amino <br>acid metabolism)<br><b>P-val:<\/b> 4.6e-05<br><b>PHOSPHO:HEART:<\/b> Ckb, Ckm, Got1<br><b>TRNSCRPT:SKM-VL:<\/b> Ckmt1, Got1, Got2, Nos3,<br> Prodh1","<b>HSF1-dependent transactivation<\/b> (Cellular <br>response to heat stress)<br><b>P-val:<\/b> 1.3e-19<br><b>PHOSPHO:HEART:<\/b> Cryab, Hsp90aa1, Hspa1a,<br> Hspa1b<br><b>PROT:HEART:<\/b> Camk2d, Cryab, Hsp90aa1,<br> Hspa1a, Hspa1b, Hspa2, Hspb8<br><b>UBIQ:HEART:<\/b> Hspa1a, Hspa1b<br><b>PHOSPHO:SKM-GN:<\/b> Camk2g, Hsp90aa1, Hspa1a,<br> Hspa1b<br><b>PROT:SKM-GN:<\/b> Dnajb1, Fkbp4, Hsp90aa1,<br> Hspa1a, Hspa1b, Hspb8","<b>Cellular response to heat stress<\/b> (Cellular <br>responses to stress)<br><b>P-val:<\/b> 3.4e-12<br><b>PHOSPHO:HEART:<\/b> Cryab, Hsp90aa1, Hspa1a,<br> Hspa1b, Hsph1<br><b>PROT:HEART:<\/b> Bag3, Camk2d, Cryab, Hikeshi,<br> Hsp90aa1, Hspa1a, Hspa1b, Hspa2, Hspb8, Hsph1,<br> St13<br><b>UBIQ:HEART:<\/b> Hspa1a, Hspa1b<br><b>PHOSPHO:SKM-GN:<\/b> Bag3, Camk2g, Hsp90aa1,<br> Hspa1a, Hspa1b","<b>HSP90 chaperone cycle for steroid hormone <br>receptors (SHR)<\/b> (Cellular responses to stress)<br><b>P-val:<\/b> 2.6e-11<br><b>PHOSPHO:HEART:<\/b> Hsp90aa1, Hspa1a, Hspa1b<br><b>PROT:HEART:<\/b> Fkbp5, Hsp90aa1, Hspa1a,<br> Hspa1b, Hspa2, Stip1<br><b>UBIQ:HEART:<\/b> Hspa1a, Hspa1b<br><b>PHOSPHO:SKM-GN:<\/b> Dync1li2, Hsp90aa1, Hspa1a,<br> Hspa1b<br><b>PROT:SKM-GN:<\/b> Dnaja1, Dnajb1, Fkbp4, Fkbp5,<br> Hsp90aa1, Hspa1a, Hspa1b, Stip1","<b>Alanine, aspartate and glutamate <br>metabolism<\/b> (Amino acid metabolism)<br><b>P-val:<\/b> 3.7e-05<br><b>PROT:HEART:<\/b> Aldh5a1, Gls, Glud1, Got2, Nit2<br><b>TRNSCRPT:SKM-GN:<\/b> Aldh5a1, Got1, Got2","<b>Fructose and mannose metabolism<\/b> <br>(Carbohydrate metabolism)<br><b>P-val:<\/b> 2.7e-05<br><b>PROT:HEART:<\/b> Aldoa, Hk2, Mpi, Pfkp, Tpi1<br><b>ATAC:SKM-GN:<\/b> Pfkfb1, Pfkfb3","<b>PPAR signaling pathway<\/b> (Endocrine system)<br><b>P-val:<\/b> 3.8e-06<br><b>PROT:HEART:<\/b> Acox1, Acox3, Cyp27a1, Dbi,<br> Ehhadh, Sorbs1<br><b>PROT:SKM-GN:<\/b> Acadl, Acadm, Acsl1, Fabp4,<br> Fabp5, Gk, Me3, Plin2, Plin4, Plin5, Sorbs1","<b>Attenuation phase<\/b> (HSF1-dependent <br>transactivation)<br><b>P-val:<\/b> 2.5e-08<br><b>PROT:HEART:<\/b> Hsp90aa1, Hspa1a, Hspa1b, Hspa2<br><b>PROT:SKM-GN:<\/b> Dnajb1, Fkbp4, Hsp90aa1,<br> Hspa1a, Hspa1b","<b>Glycolysis<\/b> (Glucose metabolism)<br><b>P-val:<\/b> 1.1e-05<br><b>PROT:HEART:<\/b> Aldoa, Eno3, Gapdh, Hk2, Pgam1,<br> Pgk1, Tpi1<br><b>ATAC:SKM-GN:<\/b> Pfkfb1, Pfkfb3","<b>Diabetic cardiomyopathy<\/b> (Cardiovascular <br>disease)<br><b>P-val:<\/b> 7.9e-92<br><b>TRNSCRPT:HEART:<\/b> Atp5f1a, Cox7a1, Mapk9,<br> Mt-nd1, Mt-nd2, Mt-nd4, Mt-nd5, Ndufa12, Ndufb4,<br> Pdk2, Pln, Ryr2, Slc25a5, Slc2a4<br><b>PHOSPHO:SKM-GN:<\/b> Atp5f1a, Atp5f1c, Camk2g,<br> Gys1, Ndufa7, Ndufb10, Ndufs4, Ndufs7, Ndufv1,<br> Slc2a4, Taf3<br><b>PROT:SKM-GN:<\/b> Ace, Atp5f1a, Atp5f1b,<br> Atp5f1c, Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b,<br> Cox6b1, Irs1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7,<br> Ndufs8, Ndufv1, Ndufv2, Ndufv3, Pdha1, Pdhb, Ppif,<br> Sdha, Sdhb, Slc2a4, Taf3, Uqcr10<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Cox4i1, Cox5a,<br> Cox6b1, Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b,<br> Mt-atp6, Mt-co1, Mt-co3, Mt-cyb, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa12, Ndufa3, Ndufa6, Ndufb11,<br> Ndufb4, Ndufb9, Ndufs3, Ndufv1, Tbc1d4, Uqcrq<br><b>TRNSCRPT:SKM-VL:<\/b> Atp5f1a, Atp5f1c, Atp5f1d,<br> Atp5mc1, Atp5mc3, Atp5pb, Atp5pd, Atp5pf, Atp5po,<br> Cox4i1, Cox5a, Cox5b, Cox6b1, Cox6c, Cox7a1,<br> Cox7a2, Cox7a2l2, Cox7b, Cox7c, Cox8b, Mpc1, Mpc2,<br> Mt-atp6, Mt-co1, Mt-co3, Mt-cyb, Mt-nd1, Mt-nd2,<br> Mt-nd4, Mt-nd5, Mt-nd6, Ndufa12, Ndufa2, Ndufa3,<br> Ndufa4, Ndufa5, Ndufa6, Ndufa7, Ndufa9, Ndufab1,<br> Ndufb10, Ndufb11, Ndufb4, Ndufb5, Ndufb6, Ndufb7,<br> Ndufb8, Ndufb9, Ndufc1, Ndufc2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs8, Ndufv1, Nos3, Sdhb, Slc2a4, Taf3, Uqcr10,<br> Uqcr11, Uqcrb, Uqcrc1, Uqcrq","<b>Oxidative phosphorylation<\/b> (Energy <br>metabolism)<br><b>P-val:<\/b> 1.1e-113<br><b>TRNSCRPT:HEART:<\/b> Atp5f1a, Atp6v1g2, Cox7a1,<br> Mt-nd1, Mt-nd2, Mt-nd4, Mt-nd5, Ndufa12, Ndufb4<br><b>PHOSPHO:SKM-GN:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufb10, Ndufs4, Ndufs7, Ndufv1, Taf3<br><b>PROT:SKM-GN:<\/b> Atp5f1a, Atp5f1b, Atp5f1c,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox17, Cox5a,<br> Cox5b, Cox6b1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufs8,<br> Ndufv1, Ndufv2, Ndufv3, Ppa2, Sdha, Sdhb, Taf3,<br> Uqcr10<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Cox4i1,<br> Cox5a, Cox6b1, Cox7a1, Cox7a2, Cox7a2l2, Cox7b,<br> Cox8b, Mt-atp6, Mt-co1, Mt-co3, Mt-cyb, Mt-nd1,<br> Mt-nd2, Mt-nd5, Mt-nd6, Ndufa12, Ndufa3, Ndufa6,<br> Ndufb11, Ndufb4, Ndufb9, Ndufs3, Ndufv1, Uqcrq<br><b>TRNSCRPT:SKM-VL:<\/b> Atp5f1a, Atp5f1c, Atp5f1d,<br> Atp5mc1, Atp5mc3, Atp5mf, Atp5pb, Atp5pd, Atp5pf,<br> Atp5po, Atp6v1g2, Cox15, Cox17, Cox4i1, Cox5a,<br> Cox5b, Cox6b1, Cox6c, Cox7a1, Cox7a2, Cox7a2l2,<br> Cox7b, Cox7c, Cox8b, Mt-atp6, Mt-co1, Mt-co3, Mt-cyb,<br> Mt-nd1, Mt-nd2, Mt-nd4, Mt-nd5, Mt-nd6, Ndufa12,<br> Ndufa2, Ndufa3, Ndufa4, Ndufa5, Ndufa6, Ndufa7,<br> Ndufa9, Ndufab1, Ndufb10, Ndufb11, Ndufb4, Ndufb5,<br> Ndufb6, Ndufb7, Ndufb8, Ndufb9, Ndufc1, Ndufc2,<br> Ndufs3, Ndufs4, Ndufs5, Ndufs8, Ndufv1, Sdhb, Taf3,<br> Uqcr10, Uqcr11, Uqcrb, Uqcrc1, Uqcrq","<b>Mitophagy - animal<\/b> (Transport and <br>catabolism)<br><b>P-val:<\/b> 4.5e-05<br><b>TRNSCRPT:HEART:<\/b> Atf4, Bnip3, Mapk9, Mfn1,<br> Optn, Sqstm1<br><b>ATAC:SKM-GN:<\/b> Jun, Ubc","<b>Retrograde endocannabinoid signaling<\/b> <br>(Nervous system)<br><b>P-val:<\/b> 1.6e-40<br><b>TRNSCRPT:HEART:<\/b> Gng5, Mapk9, Mt-nd1,<br> Mt-nd2, Mt-nd4, Mt-nd5, Ndufa12, Ndufb4<br><b>PROT:SKM-GN:<\/b> Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufs8, Ndufv1, Ndufv2, Ndufv3<br><b>TRNSCRPT:SKM-GN:<\/b> Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa12, Ndufa3, Ndufa6, Ndufb11, Ndufb4,<br> Ndufb9, Ndufs3, Ndufv1<br><b>TRNSCRPT:SKM-VL:<\/b> Gabrr1, Gng5, Mt-nd1,<br> Mt-nd2, Mt-nd4, Mt-nd5, Mt-nd6, Ndufa12, Ndufa2,<br> Ndufa3, Ndufa4, Ndufa5, Ndufa6, Ndufa7, Ndufa9,<br> Ndufab1, Ndufb10, Ndufb11, Ndufb4, Ndufb5, Ndufb6,<br> Ndufb7, Ndufb8, Ndufb9, Ndufc1, Ndufc2, Ndufs3, Ndufs4,<br> Ndufs5, Ndufs8, Ndufv1","<b>Arginine biosynthesis<\/b> (Amino acid <br>metabolism)<br><b>P-val:<\/b> 1.5e-05<br><b>TRNSCRPT:HEART:<\/b> Got1, Got2, Gpt<br><b>TRNSCRPT:SKM-VL:<\/b> Got1, Got2, Gpt, Nos3","<b>Respiratory electron transport<\/b> <br>(Respiratory electron transport, ATP synthesis by <br>chemiosmotic coupling, and heat production by uncoupling <br>proteins.)<br><b>P-val:<\/b> 1.8e-67<br><b>TRNSCRPT:HEART:<\/b> Coq10a, Mt-nd1, Mt-nd2,<br> Mt-nd4, Mt-nd5, Ndufa12, Ndufb4<br><b>PHOSPHO:SKM-GN:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1<br><b>PROT:SKM-GN:<\/b> Acad9, Etfa, Etfb, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa6, Ndufa7, Ndufa8,<br> Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufv1,<br> Ndufv2, Ndufv3, Ndufv3-ps1<br><b>TRNSCRPT:SKM-GN:<\/b> Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa12, Ndufa3, Ndufa6, Ndufb11, Ndufb4,<br> Ndufb9, Ndufs3, Ndufv1<br><b>TRNSCRPT:SKM-VL:<\/b> Etfb, Mt-nd1, Mt-nd2,<br> Mt-nd4, Mt-nd5, Mt-nd6, Ndufa12, Ndufa2, Ndufa3,<br> Ndufa5, Ndufa6, Ndufa7, Ndufa9, Ndufab1, Ndufb10,<br> Ndufb11, Ndufb4, Ndufb5, Ndufb6, Ndufb7, Ndufb8,<br> Ndufb9, Ndufc1, Ndufc2, Ndufs3, Ndufs4, Ndufs5, Ndufv1","<b>Complex I biogenesis<\/b> (Respiratory electron <br>transport)<br><b>P-val:<\/b> 7.4e-65<br><b>TRNSCRPT:HEART:<\/b> Mt-nd1, Mt-nd2, Mt-nd4,<br> Mt-nd5, Ndufa12, Ndufb4<br><b>PHOSPHO:SKM-GN:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1<br><b>PROT:SKM-GN:<\/b> Acad9, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa7, Ndufa8, Ndufa9,<br> Ndufaf3, Ndufaf4, Ndufaf7, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufv1, Ndufv2,<br> Ndufv3, Ndufv3-ps1<br><b>TRNSCRPT:SKM-GN:<\/b> Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa12, Ndufa3, Ndufa6, Ndufb11, Ndufb4,<br> Ndufb9, Ndufs3, Ndufv1<br><b>TRNSCRPT:SKM-VL:<\/b> Mt-nd1, Mt-nd2, Mt-nd4,<br> Mt-nd5, Mt-nd6, Ndufa12, Ndufa2, Ndufa3, Ndufa5,<br> Ndufa6, Ndufa7, Ndufa9, Ndufab1, Ndufb10, Ndufb11,<br> Ndufb4, Ndufb5, Ndufb6, Ndufb7, Ndufb8, Ndufb9,<br> Ndufc1, Ndufc2, Ndufs3, Ndufs4, Ndufs5, Ndufv1","<b>Vitamin digestion and absorption<\/b> <br>(Digestive system)<br><b>P-val:<\/b> 3.3e-05<br><b>METAB:SKM-GN:<\/b> unknown (METAB)<br><b>METAB:SKM-VL:<\/b> unknown (METAB)","<b>Non-alcoholic fatty liver disease<\/b> <br>(Endocrine and metabolic disease)<br><b>P-val:<\/b> 7.7e-66<br><b>PHOSPHO:SKM-GN:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Prkab1, Prkag2<br><b>PROT:SKM-GN:<\/b> Cox5a, Cox5b, Cox6b1, Irs1,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa7, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufs8, Ndufv1,<br> Ndufv2, Ndufv3, Prkag2, Sdha, Sdhb, Uqcr10<br><b>TRNSCRPT:SKM-GN:<\/b> Cox4i1, Cox5a, Cox6b1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Il6r, Mt-co1,<br> Mt-co3, Mt-cyb, Ndufa12, Ndufa3, Ndufa6, Ndufb11,<br> Ndufb4, Ndufb9, Ndufs3, Ndufv1, Uqcrq<br><b>TRNSCRPT:SKM-VL:<\/b> Cox4i1, Cox5a, Cox5b,<br> Cox6b1, Cox6c, Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox7c,<br> Cox8b, Cycs, Il6r, Mt-co1, Mt-co3, Mt-cyb,<br> Ndufa12, Ndufa2, Ndufa3, Ndufa4, Ndufa5, Ndufa6, Ndufa7,<br> Ndufa9, Ndufab1, Ndufb10, Ndufb11, Ndufb4,<br> Ndufb5, Ndufb6, Ndufb7, Ndufb8, Ndufb9, Ndufc1, Ndufc2,<br> Ndufs3, Ndufs4, Ndufs5, Ndufs8, Ndufv1, Sdhb,<br> Uqcr10, Uqcr11, Uqcrb, Uqcrc1, Uqcrq","<b>Mitochondrial translation<\/b> (Translation)<br><b>P-val:<\/b> 1e-14<br><b>PROT:SKM-GN:<\/b> Dap3, Gfm1, Mrpl12, Mrpl16,<br> Mrpl17, mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55,<br> Mrps12, Mrps15, Mrps16, Mrps17, Mrps21, Mrps22,<br> Mrps23, Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3<br><b>TRNSCRPT:SKM-VL:<\/b> Mrpl12, Mrpl38, Mrpl4,<br> Mrpl40, Mrpl45, Mrps34, Mrrf","<b>Mitochondrial translation termination<\/b> <br>(Mitochondrial translation)<br><b>P-val:<\/b> 5.2e-14<br><b>PROT:SKM-GN:<\/b> Dap3, Mrpl12, Mrpl16, Mrpl17,<br> mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55, Mrps12,<br> Mrps15, Mrps16, Mrps17, Mrps21, Mrps22, Mrps23,<br> Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3<br><b>TRNSCRPT:SKM-VL:<\/b> Mrpl12, Mrpl38, Mrpl4,<br> Mrpl40, Mrpl45, Mrps34, Mrrf","<b>Cardiac muscle contraction<\/b> (Circulatory <br>system)<br><b>P-val:<\/b> 1.2e-24<br><b>TRNSCRPT:SKM-GN:<\/b> Atp1a2, Cox4i1, Cox5a,<br> Cox6b1, Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b,<br> Mt-co1, Mt-co3, Mt-cyb, Uqcrq<br><b>TRNSCRPT:SKM-VL:<\/b> Atp1b1, Cox4i1, Cox5a,<br> Cox5b, Cox6b1, Cox6c, Cox7a1, Cox7a2, Cox7a2l2,<br> Cox7b, Cox7c, Cox8b, Mt-co1, Mt-co3, Mt-cyb, Uqcr10,<br> Uqcr11, Uqcrb, Uqcrc1, Uqcrq","<b>Formation of ATP by chemiosmotic coupling<\/b> <br>(Respiratory electron transport, ATP synthesis by <br>chemiosmotic coupling, and heat production by <br>uncoupling proteins.)<br><b>P-val:<\/b> 9.5e-15<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Mt-atp6<br><b>TRNSCRPT:SKM-VL:<\/b> Atp5mc1, Atp5mf, Atp5pb,<br> Atp5pd, Atp5pf, Atp5po, Dmac2l, Mt-atp6","<b>Cristae formation<\/b> (Mitochondrial <br>biogenesis)<br><b>P-val:<\/b> 9.5e-15<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Mt-atp6<br><b>TRNSCRPT:SKM-VL:<\/b> Atp5mc1, Atp5mf, Atp5pb,<br> Atp5pd, Atp5pf, Atp5po, Dmac2l, Mt-atp6"],"color.highlight.background":["yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow"],"color.border":["black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black"],"shadow":[false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],"physics":[false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],"borderWidth":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],"borderWidthSelected":[3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3],"pathway_subclass":["NULL","Carbohydrate metabolism","Carbohydrate metabolism","NULL","Amino acid metabolism","Lipid metabolism","NULL","Carbohydrate metabolism","Carbohydrate metabolism","NULL","Carbohydrate metabolism","Metabolism","The citric acid (TCA) cycle and respiratory electron transport","Pyruvate metabolism and Citric Acid (TCA) cycle","Fatty acid metabolism","Metabolism of lipids","Metabolism of amino acids and derivatives","Pyruvate metabolism and Citric Acid (TCA) cycle","Metabolism of amino acids and derivatives","The citric acid (TCA) cycle and respiratory electron transport","Cell growth and death","Organelle biogenesis and maintenance","Amino acid metabolism","Cellular response to heat stress","Cellular responses to stress","Cellular responses to stress","Amino acid metabolism","Carbohydrate metabolism","Endocrine system","HSF1-dependent transactivation","Glucose metabolism","Cardiovascular disease","Energy metabolism","Transport and catabolism","Nervous system","Amino acid metabolism","Respiratory electron transport, ATP synthesis by chemiosmotic coupling, and heat production by uncoupling proteins.","Respiratory electron transport","Digestive system","Endocrine and metabolic disease","Translation","Mitochondrial translation","Circulatory system","Respiratory electron transport, ATP synthesis by chemiosmotic coupling, and heat production by uncoupling proteins.","Mitochondrial biogenesis"],"group":["1: Carbohydrate metabolism (3/7)","1: Carbohydrate metabolism (3/7)","0: Carbohydrate metabolism (3/11)","3: Lipid metabolism (1/1)","0: Carbohydrate metabolism (3/11)","3: Lipid metabolism (1/1)","1: Carbohydrate metabolism (3/7)","1: Carbohydrate metabolism (3/7)","0: Carbohydrate metabolism (3/11)","1: Carbohydrate metabolism (3/7)","0: Carbohydrate metabolism (3/11)","7: Metabolism (1/4)","1: Carbohydrate metabolism (3/7)","1: Carbohydrate metabolism (3/7)","8: Fatty acid metabolism (1/2)","8: Fatty acid metabolism (1/2)","0: Carbohydrate metabolism (3/11)","1: Carbohydrate metabolism (3/7)","0: Carbohydrate metabolism (3/11)","11: Cardiovascular disease (1/8)","0: Carbohydrate metabolism (3/11)","7: Metabolism (1/4)","13: Amino acid metabolism (2/2)","14: Cellular responses to stress (2/4)","14: Cellular responses to stress (2/4)","14: Cellular responses to stress (2/4)","0: Carbohydrate metabolism (3/11)","1: Carbohydrate metabolism (3/7)","0: Carbohydrate metabolism (3/11)","14: Cellular responses to stress (2/4)","1: Carbohydrate metabolism (3/7)","11: Cardiovascular disease (1/8)","11: Cardiovascular disease (1/8)","0: Carbohydrate metabolism (3/11)","11: Cardiovascular disease (1/8)","13: Amino acid metabolism (2/2)","11: Cardiovascular disease (1/8)","11: Cardiovascular disease (1/8)","0: Carbohydrate metabolism (3/11)","11: Cardiovascular disease (1/8)","19: Mitochondrial translation (1/2)","19: Mitochondrial translation (1/2)","11: Cardiovascular disease (1/8)","7: Metabolism (1/4)","7: Metabolism (1/4)"],"group_number":["1","1","0","3","0","3","1","1","0","1","0","7","1","1","8","8","0","1","0","11","0","7","13","14","14","14","0","1","0","14","1","11","11","0","11","13","11","11","0","11","19","19","11","7","7"],"color.background":["#8DD3C7","#8DD3C7","#FFFFB3","#BEBADA","#FFFFB3","#BEBADA","#8DD3C7","#8DD3C7","#FFFFB3","#8DD3C7","#FFFFB3","#FB8072","#8DD3C7","#8DD3C7","#80B1D3","#80B1D3","#FFFFB3","#8DD3C7","#FFFFB3","#FDB462","#FFFFB3","#FB8072","#B3DE69","#FCCDE5","#FCCDE5","#FCCDE5","#FFFFB3","#8DD3C7","#FFFFB3","#FCCDE5","#8DD3C7","#FDB462","#FDB462","#FFFFB3","#FDB462","#B3DE69","#FDB462","#FDB462","#FFFFB3","#FDB462","#D9D9D9","#D9D9D9","#FDB462","#FB8072","#FB8072"],"color.highlight.border":["#8DD3C7","#8DD3C7","#FFFFB3","#BEBADA","#FFFFB3","#BEBADA","#8DD3C7","#8DD3C7","#FFFFB3","#8DD3C7","#FFFFB3","#FB8072","#8DD3C7","#8DD3C7","#80B1D3","#80B1D3","#FFFFB3","#8DD3C7","#FFFFB3","#FDB462","#FFFFB3","#FB8072","#B3DE69","#FCCDE5","#FCCDE5","#FCCDE5","#FFFFB3","#8DD3C7","#FFFFB3","#FCCDE5","#8DD3C7","#FDB462","#FDB462","#FFFFB3","#FDB462","#B3DE69","#FDB462","#FDB462","#FFFFB3","#FDB462","#D9D9D9","#D9D9D9","#FDB462","#FB8072","#FB8072"],"x":[0.0770050854760573,0.0201617344119303,0.998620632016922,0.780935064282209,-0.954257577010682,0.65817463330946,0.431744716384872,-0.160519116486648,-0.770429337089608,0.237408994500496,-0.916044948981252,0.417501515424252,0.103751651263869,0.197113376016246,-0.0718135672611934,-0.229649527030526,-0.546855790100997,-0.113108246565844,-0.556565988716728,0.633660583710385,-0.490015267291309,0.438311629035877,-0.779361699329372,-0.454421228275762,-0.601368584389613,-0.696103659249356,-1,-0.10201556020866,0.105477652415654,-0.608793141940965,0.0513795251659392,0.829844675986021,0.805596382983207,-0.352791201911871,0.625101453243222,-0.625051786815341,0.615074743591968,0.68926454711054,-0.110005946547761,0.826493392031937,0.265014984819471,0.376736591914081,1,0.535768362567297,0.416428911721658],"y":[-0.635758803229871,-0.409027586854558,0.434383110060193,0.685371897667179,0.34305434065354,0.789580569683623,-0.906022272044304,-0.669423081879417,0.641505103553033,-0.831736163239333,-0.302806235973945,-0.102411273743569,-0.181308690343166,-0.357783120795137,0.994788044024284,0.968969332688282,-0.295794486622084,-0.0652368398515243,0.856227524998535,-0.015594602974575,-0.862371449996332,0.372439091570306,-0.576504256704897,0.272497704832983,0.35438985788643,0.00931138449660018,-0.0131149281596535,-1,0.699722805617247,0.173555493529404,-0.869208679869647,-0.142989730981308,-0.242053732868728,0.678556071480762,-0.272776299691231,-0.603842761791469,-0.130080409933299,-0.195310640845712,0.481999637010462,-0.375224106814464,1,0.888294688118239,-0.287277921027057,0.18306401262812,0.152131293768767]},"edges":{"from":["KEGG:01200","KEGG:01212","KEGG:05415","REAC:R-RNO-163200","KEGG:00330","KEGG:01200","KEGG:01200","KEGG:01210","KEGG:00190","KEGG:04932","KEGG:05415","KEGG:00190","KEGG:05415","REAC:R-RNO-1428517","REAC:R-RNO-163200","KEGG:00190","KEGG:04723","KEGG:05415","REAC:R-RNO-6799198","REAC:R-RNO-163200","REAC:R-RNO-1428517","REAC:R-RNO-1428517","REAC:R-RNO-1592230","REAC:R-RNO-163200","REAC:R-RNO-3371571","REAC:R-RNO-3371497","REAC:R-RNO-3371556","REAC:R-RNO-3371571","REAC:R-RNO-5368287","KEGG:00190","KEGG:04723","KEGG:05415","REAC:R-RNO-1428517","REAC:R-RNO-163200","KEGG:00190","KEGG:04723","KEGG:05415","REAC:R-RNO-1428517","REAC:R-RNO-163200","REAC:R-RNO-611105","KEGG:00051","KEGG:01200","KEGG:01230","REAC:R-RNO-71406","KEGG:00020","KEGG:01200","REAC:R-RNO-1428517","REAC:R-RNO-71406","KEGG:00020","REAC:R-RNO-1428517","REAC:R-RNO-1428517","REAC:R-RNO-1592230","REAC:R-RNO-163200","REAC:R-RNO-163210","REAC:R-RNO-77289"],"to":["KEGG:00020","KEGG:00071","KEGG:00190","KEGG:00190","KEGG:00220","KEGG:00630","KEGG:01230","KEGG:01230","KEGG:04260","KEGG:04260","KEGG:04260","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04932","KEGG:04932","KEGG:04932","KEGG:04932","KEGG:05415","REAC:R-RNO-163200","REAC:R-RNO-163210","REAC:R-RNO-163210","REAC:R-RNO-163210","REAC:R-RNO-3371556","REAC:R-RNO-3371568","REAC:R-RNO-3371568","REAC:R-RNO-3371568","REAC:R-RNO-5419276","REAC:R-RNO-611105","REAC:R-RNO-611105","REAC:R-RNO-611105","REAC:R-RNO-611105","REAC:R-RNO-611105","REAC:R-RNO-6799198","REAC:R-RNO-6799198","REAC:R-RNO-6799198","REAC:R-RNO-6799198","REAC:R-RNO-6799198","REAC:R-RNO-6799198","REAC:R-RNO-70171","REAC:R-RNO-70171","REAC:R-RNO-70171","REAC:R-RNO-70268","REAC:R-RNO-71403","REAC:R-RNO-71403","REAC:R-RNO-71403","REAC:R-RNO-71403","REAC:R-RNO-71406","REAC:R-RNO-71406","REAC:R-RNO-8949613","REAC:R-RNO-8949613","REAC:R-RNO-8949613","REAC:R-RNO-8949613","REAC:R-RNO-8978868"],"value":[22.1077127659574,25.5434782608696,31.36611589849,21.5034965034965,20.0892857142857,21.905637254902,23.0245944531659,21.5201465201465,20.437181663837,22.1432476763259,19.6297653958944,25.0411560270715,24.1757362355954,22.3376846444521,25.9062572154237,29.4556962025316,23.5752979414951,27.9722222222222,19.5343580470163,19.5163747143945,31.3132591093117,21.7105263157895,34.8214285714286,22.8409090909091,23.6538461538462,31.8181818181818,19.6969696969697,31.8181818181818,46.9208211143695,20.2409638554217,28.6384976525822,19.0038504533598,28.3568484464172,34.0087623220153,21.1253473294227,29.6801643192488,19.8020468623754,27.1069287141905,32.3820483314154,36.6244725738397,28.2738095238095,20.9313725490196,21,19.6428571428571,25.7065217391304,21.1224489795918,20.5263157894737,32.5,22.4404761904762,23.2456140350877,21.7105263157895,34.8214285714286,22.8409090909091,43.75,31.7460317460317],"title":["<b>Similarity score:<\/b> 0.44<br><b>Intersection:<\/b> Aco2, Cs, Dlat, Dld, Fh,<br> Idh1, Idh2, Idh3a, Mdh1, Pdhb, Sucla2, Suclg2, Pdha1,<br> Idh3b, Idh3g, Dlst, Mdh2, Ogdh, Sdha, Sdhb,<br> Suclg1","<b>Similarity score:<\/b> 0.51<br><b>Intersection:<\/b> Acaa2, Acads, Acadvl, Acox1,<br> Echs1, Hadh, Hadhb, Acox3, Ehhadh, Acadl, Acadm,<br> Acsl1","<b>Similarity score:<\/b> 0.63<br><b>Intersection:<\/b> Atp5f1a, Cox7a1, Mt-nd1,<br> Mt-nd2, Mt-nd4, Mt-nd5, Ndufa12, Ndufb4, Atp5f1c,<br> Ndufa7, Ndufb10, Ndufs4, Ndufs7, Ndufv1, Taf3,<br> Atp5f1b, Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b,<br> Cox6b1, Ndufa10, Ndufa10l1, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb,<br> Uqcr10, Atp5mc1, Cox4i1, Cox7a2, Cox7a2l2, Cox7b, Cox8b,<br> Mt-atp6, Mt-co1, Mt-co3, Mt-cyb, Mt-nd6, Ndufa3,<br> Ndufb11, Uqcrq, Atp5mc3, Atp5pb, Atp5pd, Cox6c,<br> Cox7c, Ndufa5, Ndufab1, Ndufb5, Ndufb6, Ndufb7,<br> Ndufb8, Ndufc1, Ndufc2, Uqcr11, Uqcrb, Uqcrc1","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Atp5po, Ndufa2, Ndufs1,<br> Ndufv1, Mt-nd1, Mt-nd2, Mt-nd4, Mt-nd5, Ndufa12,<br> Ndufb4, Ndufa7, Ndufb10, Ndufs4, Ndufs7, Atp5pf,<br> Ndufa10, Ndufa10l1, Ndufa6, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3, Atp5mc1,<br> Atp5mf, Mt-atp6, Mt-nd6, Ndufa3, Ndufb11, Atp5pb,<br> Atp5pd, Ndufa5, Ndufab1, Ndufb5, Ndufb6, Ndufb7, Ndufb8,<br> Ndufc1, Ndufc2","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Got1, Got2, Nos3","<b>Similarity score:<\/b> 0.44<br><b>Intersection:<\/b> Aco2, Cs, Dld, Mdh1, Mmut,<br> Glyctk, Mcee, Acss2, Gcsh, Mdh2, Shmt2","<b>Similarity score:<\/b> 0.46<br><b>Intersection:<\/b> Aco2, Cs, Idh1, Idh2, Idh3a,<br> Pkm, Eno1, Eno3, Got1, Prps1, Aldoa, Gapdh, Got2,<br> Idh3b, Idh3g, Pfkp, Pgam1, Pgk1, Tpi1, Gpt","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Aco2, Cs, Idh1, Idh2, Idh3a,<br> Got2, Idh3b, Idh3g, Got1, Gpt","<b>Similarity score:<\/b> 0.41<br><b>Intersection:<\/b> Cox7a1, Cox5a, Cox5b, Cox6b1,<br> Uqcr10, Cox4i1, Cox7a2, Cox7a2l2, Cox7b, Cox8b,<br> Mt-co1, Mt-co3, Mt-cyb, Uqcrq, Cox6c, Cox7c,<br> Uqcr11, Uqcrb, Uqcrc1","<b>Similarity score:<\/b> 0.44<br><b>Intersection:<\/b> Cox5a, Cox5b, Cox6b1, Uqcr10,<br> Cox4i1, Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b,<br> Mt-co1, Mt-co3, Mt-cyb, Uqcrq, Cox6c, Cox7c,<br> Uqcr11, Uqcrb, Uqcrc1","<b>Similarity score:<\/b> 0.39<br><b>Intersection:<\/b> Cox7a1, Cox5a, Cox5b, Cox6b1,<br> Uqcr10, Cox4i1, Cox7a2, Cox7a2l2, Cox7b, Cox8b,<br> Mt-co1, Mt-co3, Mt-cyb, Uqcrq, Cox6c, Cox7c,<br> Uqcr11, Uqcrb, Uqcrc1","<b>Similarity score:<\/b> 0.5<br><b>Intersection:<\/b> Mt-nd1, Mt-nd2, Mt-nd4,<br> Mt-nd5, Ndufa12, Ndufb4, Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Mt-nd6, Ndufa3,<br> Ndufb11, Ndufa5, Ndufab1, Ndufb5, Ndufb6, Ndufb7,<br> Ndufb8, Ndufc1, Ndufc2","<b>Similarity score:<\/b> 0.48<br><b>Intersection:<\/b> Mapk9, Mt-nd1, Mt-nd2,<br> Mt-nd4, Mt-nd5, Ndufa12, Ndufb4, Ndufa7, Ndufb10,<br> Ndufs4, Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa2,<br> Ndufa4, Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Mt-nd6,<br> Ndufa3, Ndufb11, Ndufa5, Ndufab1, Ndufb5, Ndufb6,<br> Ndufb7, Ndufb8, Ndufc1, Ndufc2","<b>Similarity score:<\/b> 0.45<br><b>Intersection:<\/b> Ndufa2, Ndufs1, Ndufv1,<br> Mt-nd1, Mt-nd2, Mt-nd4, Mt-nd5, Ndufa12, Ndufb4,<br> Ndufa7, Ndufb10, Ndufs4, Ndufs7, Ndufa10, Ndufa10l1,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Mt-nd6, Ndufa3, Ndufb11,<br> Ndufa5, Ndufab1, Ndufb5, Ndufb6, Ndufb7, Ndufb8, Ndufc1,<br> Ndufc2","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Ndufa2, Ndufs1, Ndufv1,<br> Mt-nd1, Mt-nd2, Mt-nd4, Mt-nd5, Ndufa12, Ndufb4,<br> Ndufa7, Ndufb10, Ndufs4, Ndufs7, Ndufa10, Ndufa10l1,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Mt-nd6, Ndufa3, Ndufb11,<br> Ndufa5, Ndufab1, Ndufb5, Ndufb6, Ndufb7, Ndufb8, Ndufc1,<br> Ndufc2","<b>Similarity score:<\/b> 0.59<br><b>Intersection:<\/b> Cox7a1, Ndufa12, Ndufb4,<br> Ndufa7, Ndufb10, Ndufs4, Ndufs7, Ndufv1, Cox5a, Cox5b,<br> Cox6b1, Ndufa10, Ndufa10l1, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb,<br> Uqcr10, Cox4i1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Mt-co1,<br> Mt-co3, Mt-cyb, Ndufa3, Ndufb11, Uqcrq, Cox6c,<br> Cox7c, Ndufa5, Ndufab1, Ndufb5, Ndufb6, Ndufb7,<br> Ndufb8, Ndufc1, Ndufc2, Uqcr11, Uqcrb, Uqcrc1","<b>Similarity score:<\/b> 0.47<br><b>Intersection:<\/b> Ndufa12, Ndufb4, Ndufa10,<br> Ndufa10l1, Ndufa2, Ndufa4, Ndufa6, Ndufa7, Ndufa8,<br> Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4,<br> Ndufs5, Ndufs7, Ndufs8, Ndufv1, Ndufv2, Ndufv3,<br> Ndufa3, Ndufb11, Ndufa5, Ndufab1, Ndufb10, Ndufb5,<br> Ndufb6, Ndufb7, Ndufb8, Ndufc1, Ndufc2","<b>Similarity score:<\/b> 0.56<br><b>Intersection:<\/b> Cox7a1, Ndufa12, Ndufb4,<br> Ndufa7, Ndufb10, Ndufs4, Ndufs7, Ndufv1, Cox5a, Cox5b,<br> Cox6b1, Irs1, Ndufa10, Ndufa10l1, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb,<br> Uqcr10, Cox4i1, Cox7a2, Cox7a2l2, Cox7b, Cox8b,<br> Mt-co1, Mt-co3, Mt-cyb, Ndufa3, Ndufb11, Uqcrq,<br> Cox6c, Cox7c, Ndufa5, Ndufab1, Ndufb5, Ndufb6,<br> Ndufb7, Ndufb8, Ndufc1, Ndufc2, Uqcr11, Uqcrb, Uqcrc1","<b>Similarity score:<\/b> 0.39<br><b>Intersection:<\/b> Ndufa12, Ndufb4, Ndufa7,<br> Ndufb10, Ndufs4, Ndufs7, Ndufv1, Ndufa10, Ndufa10l1,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3, Ndufa3,<br> Ndufb11, Ndufa5, Ndufab1, Ndufb5, Ndufb6, Ndufb7,<br> Ndufb8, Ndufc1, Ndufc2","<b>Similarity score:<\/b> 0.39<br><b>Intersection:<\/b> Atp5po, Ndufa2, Ndufs1,<br> Ndufv1, Mt-nd1, Mt-nd2, Mt-nd4, Mt-nd5, Ndufa12,<br> Ndufb4, Ndufa7, Ndufb10, Ndufs4, Ndufs7, Atp5pf,<br> Ndufa10, Ndufa10l1, Ndufa6, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3, Atp5mc1,<br> Mt-atp6, Mt-nd6, Ndufa3, Ndufb11, Atp5pb, Atp5pd,<br> Ndufa5, Ndufab1, Ndufb5, Ndufb6, Ndufb7, Ndufb8, Ndufc1,<br> Ndufc2","<b>Similarity score:<\/b> 0.63<br><b>Intersection:<\/b> Acad9, Atp5po, Etfb, Ndufa2,<br> Ndufb3, Ndufs1, Ndufv1, Coq10a, Mt-nd1, Mt-nd2,<br> Mt-nd4, Mt-nd5, Ndufa12, Ndufb4, Ndufa7, Ndufb10,<br> Ndufs4, Ndufs7, Atp5pf, Etfa, Ndufa10, Ndufa10l1,<br> Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7,<br> Ndufb9, Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3,<br> Ndufv3-ps1, Atp5mc1, Atp5mf, Mt-atp6, Mt-nd6, Ndufa3,<br> Ndufb11, Atp5pb, Atp5pd, Dmac2l, Ndufa5, Ndufab1,<br> Ndufb5, Ndufb6, Ndufb7, Ndufb8, Ndufc1, Ndufc2","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Atp5po, Atp5pf, Atp5mc1,<br> Atp5mf, Mt-atp6, Atp5pb, Atp5pd, Dmac2l","<b>Similarity score:<\/b> 0.7<br><b>Intersection:<\/b> Atp5pf, Atp5po, Atp5mc1,<br> Atp5mf, Mt-atp6, Atp5pb, Atp5pd, Dmac2l","<b>Similarity score:<\/b> 0.46<br><b>Intersection:<\/b> Atp5po, Atp5pf, Atp5mc1,<br> Atp5mf, Mt-atp6, Atp5pb, Atp5pd, Dmac2l","<b>Similarity score:<\/b> 0.47<br><b>Intersection:<\/b> Cryab, Hsp90aa1, Hspa1a,<br> Hspa1b, Camk2d, Hspa2, Hspb8, Camk2g","<b>Similarity score:<\/b> 0.64<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b,<br> Hspa2, Dnajb1, Fkbp4","<b>Similarity score:<\/b> 0.39<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b,<br> Hspa2","<b>Similarity score:<\/b> 0.64<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b,<br> Hspa2, Dnajb1, Fkbp4","<b>Similarity score:<\/b> 0.94<br><b>Intersection:<\/b> Dap3, Mrpl12, Mrpl16, Mrpl17,<br> mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55, Mrps12,<br> Mrps15, Mrps16, Mrps17, Mrps21, Mrps22, Mrps23,<br> Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3, Mrpl4, Mrpl40, Mrpl45, Mrrf","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Mt-nd1, Mt-nd2, Mt-nd4,<br> Mt-nd5, Ndufa12, Ndufb4, Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Mt-nd6, Ndufa3, Ndufb11,<br> Ndufa5, Ndufab1, Ndufb5, Ndufb6, Ndufb7, Ndufb8, Ndufc1,<br> Ndufc2","<b>Similarity score:<\/b> 0.57<br><b>Intersection:<\/b> Mt-nd1, Mt-nd2, Mt-nd4,<br> Mt-nd5, Ndufa12, Ndufb4, Ndufa10, Ndufa10l1, Ndufa2,<br> Ndufa6, Ndufa7, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufv1,<br> Ndufv2, Ndufv3, Mt-nd6, Ndufa3, Ndufb11, Ndufa5,<br> Ndufab1, Ndufb10, Ndufb5, Ndufb6, Ndufb7, Ndufb8, Ndufc1,<br> Ndufc2","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Mt-nd1, Mt-nd2, Mt-nd4,<br> Mt-nd5, Ndufa12, Ndufb4, Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Mt-nd6, Ndufa3, Ndufb11,<br> Ndufa5, Ndufab1, Ndufb5, Ndufb6, Ndufb7, Ndufb8, Ndufc1,<br> Ndufc2","<b>Similarity score:<\/b> 0.57<br><b>Intersection:<\/b> Acad9, Etfb, Ndufa2, Ndufs1,<br> Ndufv1, Coq10a, Mt-nd1, Mt-nd2, Mt-nd4, Mt-nd5,<br> Ndufa12, Ndufb4, Ndufa7, Ndufb10, Ndufs4, Ndufs7,<br> Etfa, Ndufa10, Ndufa10l1, Ndufa6, Ndufa8, Ndufa9,<br> Ndufaf3, Ndufaf4, Ndufaf7, Ndufb9, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Ndufv3-ps1, Mt-nd6, Ndufa3,<br> Ndufb11, Ndufa5, Ndufab1, Ndufb5, Ndufb6, Ndufb7,<br> Ndufb8, Ndufc1, Ndufc2","<b>Similarity score:<\/b> 0.68<br><b>Intersection:<\/b> Acad9, Etfb, Ndufa2, Ndufs1,<br> Ndufv1, Coq10a, Mt-nd1, Mt-nd2, Mt-nd4, Mt-nd5,<br> Ndufa12, Ndufb4, Ndufa7, Ndufb10, Ndufs4, Ndufs7,<br> Etfa, Ndufa10, Ndufa10l1, Ndufa6, Ndufa8, Ndufa9,<br> Ndufaf3, Ndufaf4, Ndufaf7, Ndufb9, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Ndufv3-ps1, Mt-nd6, Ndufa3,<br> Ndufb11, Ndufa5, Ndufab1, Ndufb5, Ndufb6, Ndufb7,<br> Ndufb8, Ndufc1, Ndufc2","<b>Similarity score:<\/b> 0.42<br><b>Intersection:<\/b> Mt-nd1, Mt-nd2, Mt-nd4,<br> Mt-nd5, Ndufa12, Ndufb4, Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Mt-nd6, Ndufa3, Ndufb11,<br> Ndufa5, Ndufab1, Ndufb5, Ndufb6, Ndufb7, Ndufb8, Ndufc1,<br> Ndufc2","<b>Similarity score:<\/b> 0.59<br><b>Intersection:<\/b> Mt-nd1, Mt-nd2, Mt-nd4,<br> Mt-nd5, Ndufa12, Ndufb4, Ndufa10, Ndufa10l1, Ndufa2,<br> Ndufa6, Ndufa7, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufv1,<br> Ndufv2, Ndufv3, Mt-nd6, Ndufa3, Ndufb11, Ndufa5,<br> Ndufab1, Ndufb10, Ndufb5, Ndufb6, Ndufb7, Ndufb8, Ndufc1,<br> Ndufc2","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Mt-nd1, Mt-nd2, Mt-nd4,<br> Mt-nd5, Ndufa12, Ndufb4, Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Mt-nd6, Ndufa3, Ndufb11,<br> Ndufa5, Ndufab1, Ndufb5, Ndufb6, Ndufb7, Ndufb8, Ndufc1,<br> Ndufc2","<b>Similarity score:<\/b> 0.54<br><b>Intersection:<\/b> Acad9, Ndufa2, Ndufs1,<br> Ndufv1, Mt-nd1, Mt-nd2, Mt-nd4, Mt-nd5, Ndufa12, Ndufb4,<br> Ndufa7, Ndufb10, Ndufs4, Ndufs7, Ndufa10,<br> Ndufa10l1, Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4,<br> Ndufaf7, Ndufb9, Ndufs2, Ndufs3, Ndufs5, Ndufv2,<br> Ndufv3, Ndufv3-ps1, Mt-nd6, Ndufa3, Ndufb11, Ndufa5,<br> Ndufab1, Ndufb5, Ndufb6, Ndufb7, Ndufb8, Ndufc1,<br> Ndufc2","<b>Similarity score:<\/b> 0.65<br><b>Intersection:<\/b> Acad9, Ndufa2, Ndufs1,<br> Ndufv1, Mt-nd1, Mt-nd2, Mt-nd4, Mt-nd5, Ndufa12, Ndufb4,<br> Ndufa7, Ndufb10, Ndufs4, Ndufs7, Ndufa10,<br> Ndufa10l1, Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4,<br> Ndufaf7, Ndufb9, Ndufs2, Ndufs3, Ndufs5, Ndufv2,<br> Ndufv3, Ndufv3-ps1, Mt-nd6, Ndufa3, Ndufb11, Ndufa5,<br> Ndufab1, Ndufb5, Ndufb6, Ndufb7, Ndufb8, Ndufc1,<br> Ndufc2","<b>Similarity score:<\/b> 0.73<br><b>Intersection:<\/b> Mt-nd1, Mt-nd2, Mt-nd4,<br> Mt-nd5, Ndufa12, Ndufb4, Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Ndufa10, Ndufa10l1, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2,<br> Ndufv3, Ndufv3-ps1, Mt-nd6, Ndufa3, Ndufb11, Ndufa5,<br> Ndufab1, Ndufb5, Ndufb6, Ndufb7, Ndufb8, Ndufc1,<br> Ndufc2","<b>Similarity score:<\/b> 0.57<br><b>Intersection:<\/b> Aldoa, Hk2, Tpi1, Pfkfb1,<br> Pfkfb3","<b>Similarity score:<\/b> 0.42<br><b>Intersection:<\/b> Eno3, Aldoa, Gapdh, Hk2,<br> Pgam1, Pgk1, Tpi1","<b>Similarity score:<\/b> 0.42<br><b>Intersection:<\/b> Eno3, Aldoa, Gapdh, Pgam1,<br> Pgk1, Tpi1","<b>Similarity score:<\/b> 0.39<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdpr, Bsg,<br> Pdhx","<b>Similarity score:<\/b> 0.51<br><b>Intersection:<\/b> Aco2, Dld, Fh, Idh2, Sucla2,<br> Suclg2, Idh3b, Mdh2, Ogdh, Sdha, Sdhb","<b>Similarity score:<\/b> 0.42<br><b>Intersection:<\/b> Aco2, Dld, Fh, Idh2, Sucla2,<br> Suclg2, Idh3b, Mdh2, Me3, Ogdh, Sdha, Sdhb","<b>Similarity score:<\/b> 0.41<br><b>Intersection:<\/b> Aco2, Dld, Fh, Idh2, Sucla2,<br> Suclg2, Idh3b, Mdh2, Me3, Ogdh, Sdha, Sdhb, Nnt","<b>Similarity score:<\/b> 0.65<br><b>Intersection:<\/b> Aco2, Dld, Fh, Idh2, Sucla2,<br> Suclg2, Idh3b, Mdh2, Me3, Ogdh, Sdha, Sdhb, Nnt","<b>Similarity score:<\/b> 0.45<br><b>Intersection:<\/b> Aco2, Dlat, Dld, Fh, Idh2,<br> Pdhb, Sucla2, Suclg2, Idh3b, Mdh2, Ogdh, Sdha, Sdhb","<b>Similarity score:<\/b> 0.46<br><b>Intersection:<\/b> Aco2, Dlat, Dld, Fh, Idh2,<br> Pdhb, Pdpr, Sucla2, Suclg2, Bsg, Idh3b, L2hgdh,<br> Mdh2, Me3, Ogdh, Pdhx, Sdha, Sdhb, Nnt, Slc16a1","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Atp5po, Atp5pf, Atp5mc1,<br> Atp5mf, Mt-atp6, Atp5pb, Atp5pd, Dmac2l","<b>Similarity score:<\/b> 0.7<br><b>Intersection:<\/b> Atp5pf, Atp5po, Atp5mc1,<br> Atp5mf, Mt-atp6, Atp5pb, Atp5pd, Dmac2l","<b>Similarity score:<\/b> 0.46<br><b>Intersection:<\/b> Atp5po, Atp5pf, Atp5mc1,<br> Atp5mf, Mt-atp6, Atp5pb, Atp5pd, Dmac2l","<b>Similarity score:<\/b> 0.88<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6,<br> Atp5pb, Atp5pd, Atp5pf, Atp5po, Dmac2l","<b>Similarity score:<\/b> 0.63<br><b>Intersection:<\/b> Acsf2, Echs1, Hadh, Hadhb,<br> Mmut, Acad11, Acot11, Dbi, Mcee, Acad10, Acadl,<br> Acot13, Acot9, Them4"],"color.color":["#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484"],"physics":[false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],"color.highlight":["#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454"],"labelHighlightBold":[true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot","physics":false},"manipulation":{"enabled":false},"interaction":{"hideEdgesOnZoom":true,"tooltipDelay":10,"zoomSpeed":1},"edges":{"smooth":false},"physics":{"stabilization":false}},"groups":["1: Carbohydrate metabolism (3/7)","0: Carbohydrate metabolism (3/11)","3: Lipid metabolism (1/1)","7: Metabolism (1/4)","8: Fatty acid metabolism (1/2)","11: Cardiovascular disease (1/8)","13: Amino acid metabolism (2/2)","14: Cellular responses to stress (2/4)","19: Mitochondrial translation (1/2)"],"width":null,"height":null,"idselection":{"enabled":false},"byselection":{"enabled":false},"main":null,"submain":null,"footer":null,"background":"rgba(0, 0, 0, 0)","legend":{"width":0.3,"useGroups":false,"position":"left","ncol":1,"stepX":100,"stepY":40,"zoom":false,"nodes":{"label":["0: Carbohydrate metabolism (3/11)","1: Carbohydrate metabolism (3/7)","3: Lipid metabolism (1/1)","7: Metabolism (1/4)","8: Fatty acid metabolism (1/2)","11: Cardiovascular disease (1/8)","13: Amino acid metabolism (2/2)","14: Cellular responses to stress (2/4)","19: Mitochondrial translation (1/2)"],"shape":["box","box","box","box","box","box","box","box","box"],"color":["#FFFFB3","#8DD3C7","#BEBADA","#FB8072","#80B1D3","#FDB462","#B3DE69","#FCCDE5","#D9D9D9"],"physics":[false,false,false,false,false,false,false,false,false]},"nodesToDataframe":true},"tooltipStay":null,"tooltipStyle":"position: fixed;visibility:hidden;padding: 5px;white-space: nowrap;font-family: verdana;font-size:14px;font-color:#000000;background-color: #f5f4ed;-moz-border-radius: 3px;-webkit-border-radius: 3px;border-radius: 3px;border: 1px solid #808074;box-shadow: 3px 3px 10px rgba(0, 0, 0, 0.2);","igraphlayout":{"type":"square"}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="section level3">
<h3 id="examples-with-other-parameters">Examples with other parameters<a class="anchor" aria-label="anchor" href="#examples-with-other-parameters"></a>
</h3>
<p>Let’s go back to our simple example for input pathway enrichment
results.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pw</span> <span class="op">=</span> <span class="va">GRAPH_PW_ENRICH</span><span class="op">[</span><span class="va">GRAPH_PW_ENRICH</span><span class="op">$</span><span class="va">cluster</span> <span class="op">==</span> <span class="st">"SKM-GN:8w_F1_M1"</span>,<span class="op">]</span></span></code></pre></div>
<div class="section level4">
<h4 id="move-or-remove-group-labels">Move or remove group labels<a class="anchor" aria-label="anchor" href="#move-or-remove-group-labels"></a>
</h4>
<p>By default, the color-coded node group labels are shown on the left
side of the plot. To make it easier to match labels to groups of
pathways, we can set the <code>add_group_label_nodes</code> argument to
<code>TRUE</code>. If you’re zoomed out, hover over a label to display
it in larger font.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/enrichment_network_vis.html">enrichment_network_vis</a></span><span class="op">(</span><span class="va">pw</span>, add_group_label_nodes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Formatting inputs...</span></span>
<span><span class="co">#&gt; Subsetting feature-to-gene map...</span></span>
<span><span class="co">#&gt; Calculating similarity metric between pairs of enriched pathways...</span></span>
<span><span class="co">#&gt; Constructing graph...</span></span>
<span><span class="co">#&gt; Elapsed time: 6.099s</span></span></code></pre></div>
<div id="htmlwidget-c6396e9e472209106200" style="width:960px;height:432.632880098888px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-c6396e9e472209106200">{"x":{"nodes":{"id":["KEGG:00051","KEGG:04137","REAC:R-RNO-70171","KEGG:00564","KEGG:00565","KEGG:04217","KEGG:04977","KEGG:04714","KEGG:05415","KEGG:00190","KEGG:04932","REAC:R-RNO-3371571","REAC:R-RNO-6799198","REAC:R-RNO-611105","REAC:R-RNO-163200","REAC:R-RNO-3371497","REAC:R-RNO-3371556","REAC:R-RNO-1428517","KEGG:00020","KEGG:01200","KEGG:05012","KEGG:05020","KEGG:04723","KEGG:00640","KEGG:00280","KEGG:01212","KEGG:00630","KEGG:03320","KEGG:00620","KEGG:00061","KEGG:00650","KEGG:01210","KEGG:00071","KEGG:00062","REAC:R-RNO-5389840","REAC:R-RNO-5368287","REAC:R-RNO-5419276","REAC:R-RNO-71406","REAC:R-RNO-71403","REAC:R-RNO-8978868","REAC:R-RNO-77289","REAC:R-RNO-389661","REAC:R-RNO-3371568","REAC:R-RNO-556833","REAC:R-RNO-70895","REAC:R-RNO-71291","REAC:R-RNO-70268","REAC:R-RNO-1592230","REAC:R-RNO-5362517","KEGG:04260","KEGG:04670","KEGG:00250","REAC:R-RNO-163210","REAC:R-RNO-8949613","REAC:R-RNO-9663891","REAC:R-RNO-5205647","1: Carbohydrate metabolism (1/2)","3: Lipid metabolism (2/2)","6: Neurodegenerative disease (2/15)","7: Cellular responses to stress (2/4)","8: Carbohydrate metabolism (4/13)","9: Lipid metabolism (3/6)","12: Mitochondrial translation (2/3)","14: Macroautophagy (1/2)"],"value":[100,100,100,100,100,100,100,400,900,900,900,400,900,900,900,400,100,900,100,100,100,100,400,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,400,100,100,100,100,100,100,100,100,1800,1800,1800,1800,1800,1800,1800,1800],"title":["<b>Fructose and mannose metabolism<\/b> <br>(Carbohydrate metabolism)<br><b>P-val:<\/b> 0.00079<br><b>ATAC:SKM-GN:<\/b> Pfkfb1, Pfkfb3","<b>Mitophagy - animal<\/b> (Transport and <br>catabolism)<br><b>P-val:<\/b> 0.0036<br><b>ATAC:SKM-GN:<\/b> Jun, Ubc","<b>Glycolysis<\/b> (Glucose metabolism)<br><b>P-val:<\/b> 0.0027<br><b>ATAC:SKM-GN:<\/b> Pfkfb1, Pfkfb3","<b>Glycerophospholipid metabolism<\/b> (Lipid <br>metabolism)<br><b>P-val:<\/b> 0.00011<br><b>METAB:SKM-GN:<\/b> unknown (METAB)","<b>Ether lipid metabolism<\/b> (Lipid metabolism)<br><b>P-val:<\/b> 8.2e-05<br><b>METAB:SKM-GN:<\/b> unknown (METAB)","<b>Necroptosis<\/b> (Cell growth and death)<br><b>P-val:<\/b> 0.00064<br><b>METAB:SKM-GN:<\/b> unknown (METAB)","<b>Vitamin digestion and absorption<\/b> <br>(Digestive system)<br><b>P-val:<\/b> 0.0016<br><b>METAB:SKM-GN:<\/b> unknown (METAB)","<b>Thermogenesis<\/b> (Environmental adaptation)<br><b>P-val:<\/b> 9.5e-19<br><b>PHOSPHO:SKM-GN:<\/b> Acsl1, Atp5f1a, Atp5f1c,<br> Ndufa7, Ndufb10, Ndufs4, Ndufs7, Ndufv1, Prkab1,<br> Prkag2, Taf3<br><b>PROT:SKM-GN:<\/b> Acsl1, Atp5f1a, Atp5f1b,<br> Atp5f1c, Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Coa5, Coa7,<br> Cox17, Cox19, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6, Ndufa7,<br> Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7, Ndufaf8,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufs8, Ndufv1, Ndufv2, Ndufv3, Prkag2, Sdha,<br> Sdhb, Taf3, Uqcr10","<b>Diabetic cardiomyopathy<\/b> (Cardiovascular <br>disease)<br><b>P-val:<\/b> 1e-35<br><b>PHOSPHO:SKM-GN:<\/b> Atp5f1a, Atp5f1c, Camk2g,<br> Gys1, Ndufa7, Ndufb10, Ndufs4, Ndufs7, Ndufv1,<br> Slc2a4, Taf3<br><b>PROT:SKM-GN:<\/b> Ace, Atp5f1a, Atp5f1b,<br> Atp5f1c, Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b,<br> Cox6b1, Irs1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7,<br> Ndufs8, Ndufv1, Ndufv2, Ndufv3, Pdha1, Pdhb, Ppif,<br> Sdha, Sdhb, Slc2a4, Taf3, Uqcr10<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Cox4i1, Cox5a,<br> Cox6b1, Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b,<br> Mt-atp6, Mt-co1, Mt-co3, Mt-cyb, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa12, Ndufa3, Ndufa6, Ndufb11,<br> Ndufb4, Ndufb9, Ndufs3, Ndufv1, Tbc1d4, Uqcrq","<b>Oxidative phosphorylation<\/b> (Energy <br>metabolism)<br><b>P-val:<\/b> 6.2e-44<br><b>PHOSPHO:SKM-GN:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufb10, Ndufs4, Ndufs7, Ndufv1, Taf3<br><b>PROT:SKM-GN:<\/b> Atp5f1a, Atp5f1b, Atp5f1c,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox17, Cox5a,<br> Cox5b, Cox6b1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufs8,<br> Ndufv1, Ndufv2, Ndufv3, Ppa2, Sdha, Sdhb, Taf3,<br> Uqcr10<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Cox4i1,<br> Cox5a, Cox6b1, Cox7a1, Cox7a2, Cox7a2l2, Cox7b,<br> Cox8b, Mt-atp6, Mt-co1, Mt-co3, Mt-cyb, Mt-nd1,<br> Mt-nd2, Mt-nd5, Mt-nd6, Ndufa12, Ndufa3, Ndufa6,<br> Ndufb11, Ndufb4, Ndufb9, Ndufs3, Ndufv1, Uqcrq","<b>Non-alcoholic fatty liver disease<\/b> <br>(Endocrine and metabolic disease)<br><b>P-val:<\/b> 2.4e-28<br><b>PHOSPHO:SKM-GN:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Prkab1, Prkag2<br><b>PROT:SKM-GN:<\/b> Cox5a, Cox5b, Cox6b1, Irs1,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa7, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufs8, Ndufv1,<br> Ndufv2, Ndufv3, Prkag2, Sdha, Sdhb, Uqcr10<br><b>TRNSCRPT:SKM-GN:<\/b> Cox4i1, Cox5a, Cox6b1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Il6r, Mt-co1,<br> Mt-co3, Mt-cyb, Ndufa12, Ndufa3, Ndufa6, Ndufb11,<br> Ndufb4, Ndufb9, Ndufs3, Ndufv1, Uqcrq","<b>HSF1-dependent transactivation<\/b> (Cellular <br>response to heat stress)<br><b>P-val:<\/b> 1.2e-07<br><b>PHOSPHO:SKM-GN:<\/b> Camk2g, Hsp90aa1, Hspa1a,<br> Hspa1b<br><b>PROT:SKM-GN:<\/b> Dnajb1, Fkbp4, Hsp90aa1,<br> Hspa1a, Hspa1b, Hspb8","<b>Complex I biogenesis<\/b> (Respiratory electron <br>transport)<br><b>P-val:<\/b> 9.9e-33<br><b>PHOSPHO:SKM-GN:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1<br><b>PROT:SKM-GN:<\/b> Acad9, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa7, Ndufa8, Ndufa9,<br> Ndufaf3, Ndufaf4, Ndufaf7, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufv1, Ndufv2,<br> Ndufv3, Ndufv3-ps1<br><b>TRNSCRPT:SKM-GN:<\/b> Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa12, Ndufa3, Ndufa6, Ndufb11, Ndufb4,<br> Ndufb9, Ndufs3, Ndufv1","<b>Respiratory electron transport<\/b> <br>(Respiratory electron transport, ATP synthesis by <br>chemiosmotic coupling, and heat production by uncoupling <br>proteins.)<br><b>P-val:<\/b> 7.9e-34<br><b>PHOSPHO:SKM-GN:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1<br><b>PROT:SKM-GN:<\/b> Acad9, Etfa, Etfb, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa6, Ndufa7, Ndufa8,<br> Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufv1,<br> Ndufv2, Ndufv3, Ndufv3-ps1<br><b>TRNSCRPT:SKM-GN:<\/b> Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa12, Ndufa3, Ndufa6, Ndufb11, Ndufb4,<br> Ndufb9, Ndufs3, Ndufv1","<b>Respiratory electron transport, ATP synthesis <br>by chemiosmotic coupling, and heat production by <br>uncoupling proteins.<\/b> (The citric acid (TCA) <br>cycle and respiratory electron transport)<br><b>P-val:<\/b> 1.6e-35<br><b>PHOSPHO:SKM-GN:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1<br><b>PROT:SKM-GN:<\/b> Acad9, Atp5pf, Atp5po, Etfa,<br> Etfb, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa7, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufv1, Ndufv2, Ndufv3, Ndufv3-ps1<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Mt-atp6,<br> Mt-nd1, Mt-nd2, Mt-nd5, Mt-nd6, Ndufa12, Ndufa3,<br> Ndufa6, Ndufb11, Ndufb4, Ndufb9, Ndufs3, Ndufv1","<b>HSP90 chaperone cycle for steroid hormone <br>receptors (SHR)<\/b> (Cellular responses to stress)<br><b>P-val:<\/b> 7.5e-06<br><b>PHOSPHO:SKM-GN:<\/b> Dync1li2, Hsp90aa1, Hspa1a,<br> Hspa1b<br><b>PROT:SKM-GN:<\/b> Dnaja1, Dnajb1, Fkbp4, Fkbp5,<br> Hsp90aa1, Hspa1a, Hspa1b, Stip1","<b>Cellular response to heat stress<\/b> (Cellular <br>responses to stress)<br><b>P-val:<\/b> 0.001<br><b>PHOSPHO:SKM-GN:<\/b> Bag3, Camk2g, Hsp90aa1,<br> Hspa1a, Hspa1b","<b>The citric acid (TCA) cycle and respiratory <br>electron transport<\/b> (Metabolism)<br><b>P-val:<\/b> 5e-44<br><b>PHOSPHO:SKM-GN:<\/b> Dld, Ndufa7, Ndufb10,<br> Ndufs4, Ndufs7, Ndufv1<br><b>PROT:SKM-GN:<\/b> Acad9, Atp5pf, Atp5po, Bsg,<br> Dlat, Dld, Etfa, Etfb, Fh, Idh3b, L2hgdh, Mdh2, Me3,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa7, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufv1, Ndufv2, Ndufv3, Ndufv3-ps1, Ogdh, Pdhb,<br> Pdhx, Sdha, Sdhb, Sucla2, Suclg2<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Me3,<br> Mt-atp6, Mt-nd1, Mt-nd2, Mt-nd5, Mt-nd6, Ndufa12,<br> Ndufa3, Ndufa6, Ndufb11, Ndufb4, Ndufb9, Ndufs3,<br> Ndufv1, Pdhx","<b>Citrate cycle (TCA cycle)<\/b> (Carbohydrate <br>metabolism)<br><b>P-val:<\/b> 5.5e-16<br><b>PROT:SKM-GN:<\/b> Cs, Dlat, Dld, Dlst, Fh,<br> Idh3a, Idh3b, Idh3g, Mdh1, Mdh2, Ogdh, Pdha1, Pdhb,<br> Sdha, Sdhb, Sucla2, Suclg1, Suclg2","<b>Carbon metabolism<\/b> (NULL)<br><b>P-val:<\/b> 5.7e-14<br><b>PROT:SKM-GN:<\/b> Acads, Acss2, Cs, Dlat, Dld,<br> Dlst, Fh, Gcsh, Got1, Got2, Hibch, Hk2, Idh3a,<br> Idh3b, Idh3g, Mcee, Mdh1, Mdh2, Me3, Mmut, Ogdh,<br> Pdha1, Pdhb, Sdha, Sdhb, Shmt2, Sucla2, Suclg1, Suclg2","<b>Parkinson disease<\/b> (Neurodegenerative <br>disease)<br><b>P-val:<\/b> 1.3e-10<br><b>PROT:SKM-GN:<\/b> Atp5f1a, Atp5f1b, Atp5f1c,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b,<br> Cox6b1, Htra2, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufs8,<br> Ndufv1, Ndufv2, Ndufv3, Ppif, Sdha, Sdhb, Taf3,<br> Trap1, Uqcr10","<b>Prion disease<\/b> (Neurodegenerative disease)<br><b>P-val:<\/b> 1.1e-09<br><b>PROT:SKM-GN:<\/b> Atp5f1a, Atp5f1b, Atp5f1c,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b,<br> Cox6b1, Hspa1a, Hspa1b, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufs8, Ndufv1, Ndufv2, Ndufv3, Ppif, Sdha,<br> Sdhb, Stip1, Taf3, Uqcr10","<b>Retrograde endocannabinoid signaling<\/b> <br>(Nervous system)<br><b>P-val:<\/b> 4.3e-16<br><b>PROT:SKM-GN:<\/b> Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufs8, Ndufv1, Ndufv2, Ndufv3<br><b>TRNSCRPT:SKM-GN:<\/b> Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa12, Ndufa3, Ndufa6, Ndufb11, Ndufb4,<br> Ndufb9, Ndufs3, Ndufv1","<b>Propanoate metabolism<\/b> (Carbohydrate <br>metabolism)<br><b>P-val:<\/b> 6.2e-08<br><b>PROT:SKM-GN:<\/b> Acads, Acss2, Bckdha, Bckdhb,<br> Dld, Hibch, Mcee, Mlycd, Mmut, Sucla2, Suclg1,<br> Suclg2","<b>Valine, leucine and isoleucine degradation<\/b> <br>(Amino acid metabolism)<br><b>P-val:<\/b> 1.5e-07<br><b>PROT:SKM-GN:<\/b> Acaa2, Acadm, Acads, Bckdha,<br> Bckdhb, Dld, Hadh, Hibadh, Hibch, Hsd17b10, Mcee,<br> Mmut, Oxct1","<b>Fatty acid metabolism<\/b> (NULL)<br><b>P-val:<\/b> 2.3e-06<br><b>PROT:SKM-GN:<\/b> Acaa2, Acadl, Acadm, Acads,<br> Acadvl, Acsl1, Cbr4, Fasn, Hadh, Hsd17b8, Mecr,<br> Oxsm","<b>Glyoxylate and dicarboxylate metabolism<\/b> <br>(Carbohydrate metabolism)<br><b>P-val:<\/b> 3.7e-06<br><b>PROT:SKM-GN:<\/b> Acss2, Cs, Dld, Gcsh, Mcee,<br> Mdh1, Mdh2, Mmut, Shmt2","<b>PPAR signaling pathway<\/b> (Endocrine system)<br><b>P-val:<\/b> 3.3e-05<br><b>PROT:SKM-GN:<\/b> Acadl, Acadm, Acsl1, Fabp4,<br> Fabp5, Gk, Me3, Plin2, Plin4, Plin5, Sorbs1","<b>Pyruvate metabolism<\/b> (Carbohydrate <br>metabolism)<br><b>P-val:<\/b> 3.5e-05<br><b>PROT:SKM-GN:<\/b> Acss2, Dlat, Dld, Fh, Ldhd,<br> Mdh1, Mdh2, Me3, Pdha1, Pdhb","<b>Fatty acid biosynthesis<\/b> (Lipid metabolism)<br><b>P-val:<\/b> 9.1e-05<br><b>PROT:SKM-GN:<\/b> Acsl1, Cbr4, Fasn, Hsd17b8,<br> Mecr, Oxsm","<b>Butanoate metabolism<\/b> (Carbohydrate <br>metabolism)<br><b>P-val:<\/b> 9.1e-05<br><b>PROT:SKM-GN:<\/b> Acads, Aldh5a1, Bdh1, Hadh,<br> L2hgdh, Oxct1","<b>2-Oxocarboxylic acid metabolism<\/b> (NULL)<br><b>P-val:<\/b> 9.1e-05<br><b>PROT:SKM-GN:<\/b> Cs, Got1, Got2, Idh3a, Idh3b,<br> Idh3g","<b>Fatty acid degradation<\/b> (Lipid metabolism)<br><b>P-val:<\/b> 0.00013<br><b>PROT:SKM-GN:<\/b> Acaa2, Acadl, Acadm, Acads,<br> Acadvl, Acsl1, Eci2, Eci3, Gcdh, Hadh","<b>Fatty acid elongation<\/b> (Lipid metabolism)<br><b>P-val:<\/b> 0.0014<br><b>PROT:SKM-GN:<\/b> Acaa2, Acot2, Hadh, Mecr,<br> Them4","<b>Mitochondrial translation elongation<\/b> <br>(Mitochondrial translation)<br><b>P-val:<\/b> 5.1e-14<br><b>PROT:SKM-GN:<\/b> Dap3, Gfm1, Mrpl12, Mrpl16,<br> Mrpl17, mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55,<br> Mrps12, Mrps15, Mrps16, Mrps17, Mrps21, Mrps22,<br> Mrps23, Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Mitochondrial translation<\/b> (Translation)<br><b>P-val:<\/b> 1e-13<br><b>PROT:SKM-GN:<\/b> Dap3, Gfm1, Mrpl12, Mrpl16,<br> Mrpl17, mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55,<br> Mrps12, Mrps15, Mrps16, Mrps17, Mrps21, Mrps22,<br> Mrps23, Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Mitochondrial translation termination<\/b> <br>(Mitochondrial translation)<br><b>P-val:<\/b> 5.9e-13<br><b>PROT:SKM-GN:<\/b> Dap3, Mrpl12, Mrpl16, Mrpl17,<br> mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55, Mrps12,<br> Mrps15, Mrps16, Mrps17, Mrps21, Mrps22, Mrps23,<br> Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Pyruvate metabolism and Citric Acid (TCA) <br>cycle<\/b> (The citric acid (TCA) cycle and respiratory <br>electron transport)<br><b>P-val:<\/b> 2.5e-10<br><b>PROT:SKM-GN:<\/b> Bsg, Dlat, Dld, Fh, Idh3b,<br> L2hgdh, Mdh2, Me3, Ogdh, Pdhb, Pdhx, Sdha, Sdhb,<br> Sucla2, Suclg2","<b>Citric acid cycle (TCA cycle)<\/b> (Pyruvate <br>metabolism and Citric Acid (TCA) cycle)<br><b>P-val:<\/b> 7.5e-09<br><b>PROT:SKM-GN:<\/b> Dld, Fh, Idh3b, Mdh2, Me3,<br> Ogdh, Sdha, Sdhb, Sucla2, Suclg2","<b>Fatty acid metabolism<\/b> (Metabolism of <br>lipids)<br><b>P-val:<\/b> 1.2e-08<br><b>PROT:SKM-GN:<\/b> Acad10, Acadl, Acot11, Acot13,<br> Acot9, Acsl1, Cbr4, Eci2, Eci3, Fasn, Hadh,<br> Hsd17b8, Mcee, Mlycd, Mmut, Phyh, Prkag2, Ptges2,<br> Them4","<b>Mitochondrial Fatty Acid Beta-Oxidation<\/b> <br>(Fatty acid metabolism)<br><b>P-val:<\/b> 2.3e-06<br><b>PROT:SKM-GN:<\/b> Acad10, Acadl, Acot11, Acot13,<br> Acot9, Hadh, Mcee, Mmut, Them4","<b>Glyoxylate metabolism and glycine <br>degradation<\/b> (Metabolism of amino acids and derivatives)<br><b>P-val:<\/b> 6.5e-06<br><b>PROT:SKM-GN:<\/b> Dlat, Dld, Gcsh, Got2, Ogdh,<br> Pdhb, Pdhx","<b>Attenuation phase<\/b> (HSF1-dependent <br>transactivation)<br><b>P-val:<\/b> 3.1e-05<br><b>PROT:SKM-GN:<\/b> Dnajb1, Fkbp4, Hsp90aa1,<br> Hspa1a, Hspa1b","<b>Metabolism of lipids<\/b> (Metabolism)<br><b>P-val:<\/b> 0.00027<br><b>PROT:SKM-GN:<\/b> Acad10, Acadl, Acot11, Acot13,<br> Acot9, Acp6, Acsl1, Cbr4, Eci2, Eci3, Fabp5,<br> Fasn, Fdxr, Gpd1l, Hadh, Hsd17b8, Lpin1, Mcee, Mlycd,<br> Mmut, Oxct1, Phyh, Plin3, Prkag2, Ptges2, Them4","<b>Branched-chain amino acid catabolism<\/b> <br>(Metabolism of amino acids and derivatives)<br><b>P-val:<\/b> 0.00026<br><b>PROT:SKM-GN:<\/b> Bckdk, Dld, Hibadh, Hibch,<br> Ppm1k","<b>Metabolism of amino acids and derivatives<\/b> <br>(Metabolism)<br><b>P-val:<\/b> 0.00049<br><b>PROT:SKM-GN:<\/b> Bckdk, Dlat, Dld, Ethe1, Gcdh,<br> Gcsh, Got1, Got2, Hibadh, Hibch, Oat, Ogdh, Pdhb,<br> Pdhx, Ppm1k","<b>Pyruvate metabolism<\/b> (Pyruvate metabolism <br>and Citric Acid (TCA) cycle)<br><b>P-val:<\/b> 0.0014<br><b>PROT:SKM-GN:<\/b> Bsg, Dlat, Dld, Pdhb, Pdhx","<b>Mitochondrial biogenesis<\/b> (Organelle <br>biogenesis and maintenance)<br><b>P-val:<\/b> 3.7e-05<br><b>PROT:SKM-GN:<\/b> Acss2, Atp5pf, Atp5po, Sirt3,<br> Sod2<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Signaling by Retinoic Acid<\/b> (Signaling by <br>Nuclear Receptors)<br><b>P-val:<\/b> 0.0035<br><b>PROT:SKM-GN:<\/b> Dlat, Dld, Fabp5, Pdhb, Pdhx","<b>Cardiac muscle contraction<\/b> (Circulatory <br>system)<br><b>P-val:<\/b> 1.5e-11<br><b>TRNSCRPT:SKM-GN:<\/b> Atp1a2, Cox4i1, Cox5a,<br> Cox6b1, Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b,<br> Mt-co1, Mt-co3, Mt-cyb, Uqcrq","<b>Leukocyte transendothelial migration<\/b> <br>(Immune system)<br><b>P-val:<\/b> 0.0024<br><b>TRNSCRPT:SKM-GN:<\/b> Cdh5, Cxcl12, Jam2,<br> Rapgef4, Vcl","<b>Alanine, aspartate and glutamate <br>metabolism<\/b> (Amino acid metabolism)<br><b>P-val:<\/b> 0.0029<br><b>TRNSCRPT:SKM-GN:<\/b> Aldh5a1, Got1, Got2","<b>Formation of ATP by chemiosmotic coupling<\/b> <br>(Respiratory electron transport, ATP synthesis by <br>chemiosmotic coupling, and heat production by <br>uncoupling proteins.)<br><b>P-val:<\/b> 0.00031<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Cristae formation<\/b> (Mitochondrial <br>biogenesis)<br><b>P-val:<\/b> 0.00031<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Selective autophagy<\/b> (Macroautophagy)<br><b>P-val:<\/b> 0.0014<br><b>TRNSCRPT:SKM-GN:<\/b> Map1lc3a, Map1lc3b, Src","<b>Mitophagy<\/b> (Selective autophagy)<br><b>P-val:<\/b> 0.0014<br><b>TRNSCRPT:SKM-GN:<\/b> Map1lc3a, Map1lc3b, Src","1: Carbohydrate metabolism (1/2)","3: Lipid metabolism (2/2)","6: Neurodegenerative disease (2/15)","7: Cellular responses to stress (2/4)","8: Carbohydrate metabolism (4/13)","9: Lipid metabolism (3/6)","12: Mitochondrial translation (2/3)","14: Macroautophagy (1/2)"],"color.highlight.background":["yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","#8DD3C7","#BEBADA","#FB8072","#80B1D3","#FDB462","#B3DE69","#FCCDE5","#D9D9D9"],"color.border":["black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","#8DD3C7","#BEBADA","#FB8072","#80B1D3","#FDB462","#B3DE69","#FCCDE5","#D9D9D9"],"shadow":[false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,true,true,true,true,true,true,true],"physics":[false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],"borderWidth":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],"borderWidthSelected":[3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,0,0,0,0,0,0,0,0],"pathway_subclass":["Carbohydrate metabolism","Transport and catabolism","Glucose metabolism","Lipid metabolism","Lipid metabolism","Cell growth and death","Digestive system","Environmental adaptation","Cardiovascular disease","Energy metabolism","Endocrine and metabolic disease","Cellular response to heat stress","Respiratory electron transport","Respiratory electron transport, ATP synthesis by chemiosmotic coupling, and heat production by uncoupling proteins.","The citric acid (TCA) cycle and respiratory electron transport","Cellular responses to stress","Cellular responses to stress","Metabolism","Carbohydrate metabolism","NULL","Neurodegenerative disease","Neurodegenerative disease","Nervous system","Carbohydrate metabolism","Amino acid metabolism","NULL","Carbohydrate metabolism","Endocrine system","Carbohydrate metabolism","Lipid metabolism","Carbohydrate metabolism","NULL","Lipid metabolism","Lipid metabolism","Mitochondrial translation","Translation","Mitochondrial translation","The citric acid (TCA) cycle and respiratory electron transport","Pyruvate metabolism and Citric Acid (TCA) cycle","Metabolism of lipids","Fatty acid metabolism","Metabolism of amino acids and derivatives","HSF1-dependent transactivation","Metabolism","Metabolism of amino acids and derivatives","Metabolism","Pyruvate metabolism and Citric Acid (TCA) cycle","Organelle biogenesis and maintenance","Signaling by Nuclear Receptors","Circulatory system","Immune system","Amino acid metabolism","Respiratory electron transport, ATP synthesis by chemiosmotic coupling, and heat production by uncoupling proteins.","Mitochondrial biogenesis","Macroautophagy","Selective autophagy",null,null,null,null,null,null,null,null],"group":["1: Carbohydrate metabolism (1/2)","0: Carbohydrate metabolism (1/6)","1: Carbohydrate metabolism (1/2)","3: Lipid metabolism (2/2)","3: Lipid metabolism (2/2)","0: Carbohydrate metabolism (1/6)","0: Carbohydrate metabolism (1/6)","6: Neurodegenerative disease (2/15)","6: Neurodegenerative disease (2/15)","6: Neurodegenerative disease (2/15)","6: Neurodegenerative disease (2/15)","7: Cellular responses to stress (2/4)","6: Neurodegenerative disease (2/15)","6: Neurodegenerative disease (2/15)","6: Neurodegenerative disease (2/15)","7: Cellular responses to stress (2/4)","7: Cellular responses to stress (2/4)","6: Neurodegenerative disease (2/15)","8: Carbohydrate metabolism (4/13)","8: Carbohydrate metabolism (4/13)","6: Neurodegenerative disease (2/15)","6: Neurodegenerative disease (2/15)","6: Neurodegenerative disease (2/15)","8: Carbohydrate metabolism (4/13)","8: Carbohydrate metabolism (4/13)","9: Lipid metabolism (3/6)","8: Carbohydrate metabolism (4/13)","0: Carbohydrate metabolism (1/6)","8: Carbohydrate metabolism (4/13)","9: Lipid metabolism (3/6)","0: Carbohydrate metabolism (1/6)","8: Carbohydrate metabolism (4/13)","9: Lipid metabolism (3/6)","9: Lipid metabolism (3/6)","12: Mitochondrial translation (2/3)","12: Mitochondrial translation (2/3)","12: Mitochondrial translation (2/3)","8: Carbohydrate metabolism (4/13)","8: Carbohydrate metabolism (4/13)","9: Lipid metabolism (3/6)","9: Lipid metabolism (3/6)","8: Carbohydrate metabolism (4/13)","7: Cellular responses to stress (2/4)","9: Lipid metabolism (3/6)","8: Carbohydrate metabolism (4/13)","8: Carbohydrate metabolism (4/13)","8: Carbohydrate metabolism (4/13)","6: Neurodegenerative disease (2/15)","8: Carbohydrate metabolism (4/13)","6: Neurodegenerative disease (2/15)","0: Carbohydrate metabolism (1/6)","8: Carbohydrate metabolism (4/13)","6: Neurodegenerative disease (2/15)","6: Neurodegenerative disease (2/15)","14: Macroautophagy (1/2)","14: Macroautophagy (1/2)","1: Carbohydrate metabolism (1/2)","3: Lipid metabolism (2/2)","6: Neurodegenerative disease (2/15)","7: Cellular responses to stress (2/4)","8: Carbohydrate metabolism (4/13)","9: Lipid metabolism (3/6)","12: Mitochondrial translation (2/3)","14: Macroautophagy (1/2)"],"group_number":["1","0","1","3","3","0","0","6","6","6","6","7","6","6","6","7","7","6","8","8","6","6","6","8","8","9","8","0","8","9","0","8","9","9","12","12","12","8","8","9","9","8","7","9","8","8","8","6","8","6","0","8","6","6","14","14",null,null,null,null,null,null,null,null],"color.background":["#8DD3C7","#FFFFB3","#8DD3C7","#BEBADA","#BEBADA","#FFFFB3","#FFFFB3","#FB8072","#FB8072","#FB8072","#FB8072","#80B1D3","#FB8072","#FB8072","#FB8072","#80B1D3","#80B1D3","#FB8072","#FDB462","#FDB462","#FB8072","#FB8072","#FB8072","#FDB462","#FDB462","#B3DE69","#FDB462","#FFFFB3","#FDB462","#B3DE69","#FFFFB3","#FDB462","#B3DE69","#B3DE69","#FCCDE5","#FCCDE5","#FCCDE5","#FDB462","#FDB462","#B3DE69","#B3DE69","#FDB462","#80B1D3","#B3DE69","#FDB462","#FDB462","#FDB462","#FB8072","#FDB462","#FB8072","#FFFFB3","#FDB462","#FB8072","#FB8072","#D9D9D9","#D9D9D9","#8DD3C7","#BEBADA","#FB8072","#80B1D3","#FDB462","#B3DE69","#FCCDE5","#D9D9D9"],"color.highlight.border":["#8DD3C7","#FFFFB3","#8DD3C7","#BEBADA","#BEBADA","#FFFFB3","#FFFFB3","#FB8072","#FB8072","#FB8072","#FB8072","#80B1D3","#FB8072","#FB8072","#FB8072","#80B1D3","#80B1D3","#FB8072","#FDB462","#FDB462","#FB8072","#FB8072","#FB8072","#FDB462","#FDB462","#B3DE69","#FDB462","#FFFFB3","#FDB462","#B3DE69","#FFFFB3","#FDB462","#B3DE69","#B3DE69","#FCCDE5","#FCCDE5","#FCCDE5","#FDB462","#FDB462","#B3DE69","#B3DE69","#FDB462","#80B1D3","#B3DE69","#FDB462","#FDB462","#FDB462","#FB8072","#FDB462","#FB8072","#FFFFB3","#FDB462","#FB8072","#FB8072","#D9D9D9","#D9D9D9","#8DD3C7","#BEBADA","#FB8072","#80B1D3","#FDB462","#B3DE69","#FCCDE5","#D9D9D9"],"shape":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"box","box","box","box","box","box","box","box"],"label":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"1: Carbohydrate metabolism (1/2)","3: Lipid metabolism (2/2)","6: Neurodegenerative disease (2/15)","7: Cellular responses to stress (2/4)","8: Carbohydrate metabolism (4/13)","9: Lipid metabolism (3/6)","12: Mitochondrial translation (2/3)","14: Macroautophagy (1/2)"],"labelHighlightBold":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,false,false,false,false,false,false,false,false],"x":[0.173984612301516,0.465452546250392,0.0340940217954018,-0.81377279350663,-0.697144719326789,-1,-0.372094451397653,-0.469952899395405,-0.59710937756877,-0.541418091402422,-0.649523905047501,0.776506472024078,-0.462211751638996,-0.537193026829272,-0.4658304916754,0.90533095533982,0.793432605654155,-0.282859408570303,0.0694184460623797,0.24327341728451,-0.531136567799147,-0.549829171610287,-0.426334201423413,0.533920028104157,0.652384830787197,-0.207522494303231,0.443537387782406,0.725379434842207,0.139697989091703,-0.407048936478496,0.14726348133504,0.315359967913409,-0.0990638514512721,-0.0407737299437426,0.366662268357213,0.447876392251479,0.511036252007607,-0.0201971160734438,-0.0222865816684146,-0.301065427128867,-0.319070941796182,0.05442809753609,0.908360008913254,-0.430251903022758,0.493994990841304,0.215107841518719,0.0110614999569434,-0.428593138648665,-0.0323254377771625,-0.783742650515278,0.92691187079475,0.341237182213693,-0.4684007114913,-0.386099587724517,0.850805408628213,0.740263183015976,0.314401526384102,-0.908879254054077,-0.659852668029811,1,0.160421850993565,-0.587298121310813,0.239256235712773,0.969616512081186],"y":[-0.493387642296504,-0.300559123011595,-0.458504027675886,-0.399198278700595,-0.468185259679223,-0.168682464392372,-0.365947963540199,0.327410400066763,0.353906583742683,0.416145449780549,0.279671841176295,-0.312954110433751,0.224905099582078,0.311068283916227,0.404264272464346,-0.277401847717382,-0.422236789570583,0.374426759694142,0.319787903217676,0.3601624213913,0.185370000312284,0.249426479658882,0.278728972236645,0.426459280726623,0.576776944038992,-0.876421749464771,0.199766529446969,-0.68348616202815,0.377946030277813,-0.843539416201648,1,0.485614956937015,-1,-0.820813303109505,-0.852381094537589,-0.736308132189728,-0.844436602556837,0.358919955637807,0.230379235826783,-0.752969013939359,-0.602277143302159,0.501749265187857,-0.404844066089878,-0.691692902225967,0.672505926944541,0.587654586118244,0.445471016287283,0.779102054296141,0.504552447134754,0.396177261514475,0.433095952473293,0.702644197599744,0.592082705440146,0.585535295621086,0.0984399165038652,0.00721083679254142,-0.503065340108407,-0.485045957065669,-0.0504051970300216,-0.149750083660785,0.054632590542524,-0.823402946771917,-0.930723849057681,0.174814646935512]},"edges":{"from":["REAC:R-RNO-1428517","KEGG:01212","KEGG:01212","KEGG:01212","KEGG:04714","KEGG:05415","KEGG:01210","REAC:R-RNO-71291","KEGG:00640","KEGG:00564","KEGG:00020","KEGG:01200","KEGG:01200","KEGG:01200","KEGG:00020","KEGG:00020","KEGG:01200","KEGG:00190","KEGG:04932","KEGG:05415","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05012","KEGG:05020","KEGG:05415","REAC:R-RNO-1428517","REAC:R-RNO-163200","REAC:R-RNO-611105","REAC:R-RNO-6799198","KEGG:00190","KEGG:04714","KEGG:05415","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05415","REAC:R-RNO-1428517","REAC:R-RNO-163200","REAC:R-RNO-611105","REAC:R-RNO-6799198","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05012","KEGG:05415","REAC:R-RNO-1428517","REAC:R-RNO-163200","REAC:R-RNO-611105","REAC:R-RNO-6799198","KEGG:04714","KEGG:00190","KEGG:04714","KEGG:05415","REAC:R-RNO-163200","REAC:R-RNO-611105","REAC:R-RNO-6799198","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05415","REAC:R-RNO-611105","REAC:R-RNO-6799198","KEGG:00190","REAC:R-RNO-1428517","REAC:R-RNO-1592230","REAC:R-RNO-163200","REAC:R-RNO-3371571","REAC:R-RNO-3371497","REAC:R-RNO-3371571","REAC:R-RNO-3371497","REAC:R-RNO-3371556","REAC:R-RNO-3371571","KEGG:00020","KEGG:01200","REAC:R-RNO-1428517","REAC:R-RNO-71406","REAC:R-RNO-9663891","KEGG:00020","KEGG:00620","REAC:R-RNO-1428517","REAC:R-RNO-389661","REAC:R-RNO-70268","REAC:R-RNO-71291","REAC:R-RNO-71406","REAC:R-RNO-5389840","REAC:R-RNO-5368287","REAC:R-RNO-5389840","KEGG:00061","REAC:R-RNO-77289","REAC:R-RNO-8978868","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05415","REAC:R-RNO-6799198","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05415","KEGG:00051","KEGG:00020","KEGG:00620","REAC:R-RNO-1428517","REAC:R-RNO-389661","REAC:R-RNO-71291","REAC:R-RNO-71406","KEGG:00280","KEGG:01200","REAC:R-RNO-389661","REAC:R-RNO-70895","KEGG:00020","KEGG:01200","REAC:R-RNO-1428517","REAC:R-RNO-71406","KEGG:00020","KEGG:00620","KEGG:01200","REAC:R-RNO-1428517","REAC:R-RNO-8978868","KEGG:00190","REAC:R-RNO-1428517","REAC:R-RNO-1592230","REAC:R-RNO-163200","REAC:R-RNO-163210","KEGG:00061","KEGG:01212","REAC:R-RNO-3371497","KEGG:00051","KEGG:00564","KEGG:00020","KEGG:05012","KEGG:00061","REAC:R-RNO-5389840","REAC:R-RNO-9663891"],"to":["KEGG:00020","KEGG:00061","KEGG:00062","KEGG:00071","KEGG:00190","KEGG:00190","KEGG:00250","KEGG:00250","KEGG:00280","KEGG:00565","KEGG:00620","KEGG:00620","KEGG:00630","KEGG:00640","KEGG:01200","KEGG:01210","KEGG:01210","KEGG:04260","KEGG:04260","KEGG:04260","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04932","KEGG:04932","KEGG:04932","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05415","REAC:R-RNO-1428517","REAC:R-RNO-1428517","REAC:R-RNO-1428517","REAC:R-RNO-1428517","REAC:R-RNO-1428517","REAC:R-RNO-1428517","REAC:R-RNO-163200","REAC:R-RNO-163200","REAC:R-RNO-163200","REAC:R-RNO-163200","REAC:R-RNO-163200","REAC:R-RNO-163200","REAC:R-RNO-163210","REAC:R-RNO-163210","REAC:R-RNO-163210","REAC:R-RNO-163210","REAC:R-RNO-3371497","REAC:R-RNO-3371556","REAC:R-RNO-3371556","REAC:R-RNO-3371568","REAC:R-RNO-3371568","REAC:R-RNO-3371568","REAC:R-RNO-389661","REAC:R-RNO-389661","REAC:R-RNO-389661","REAC:R-RNO-389661","REAC:R-RNO-5205647","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5368287","REAC:R-RNO-5419276","REAC:R-RNO-5419276","REAC:R-RNO-556833","REAC:R-RNO-556833","REAC:R-RNO-556833","REAC:R-RNO-611105","REAC:R-RNO-611105","REAC:R-RNO-611105","REAC:R-RNO-611105","REAC:R-RNO-611105","REAC:R-RNO-6799198","REAC:R-RNO-6799198","REAC:R-RNO-6799198","REAC:R-RNO-6799198","REAC:R-RNO-70171","REAC:R-RNO-70268","REAC:R-RNO-70268","REAC:R-RNO-70268","REAC:R-RNO-70268","REAC:R-RNO-70268","REAC:R-RNO-70268","REAC:R-RNO-70895","REAC:R-RNO-71291","REAC:R-RNO-71291","REAC:R-RNO-71291","REAC:R-RNO-71403","REAC:R-RNO-71403","REAC:R-RNO-71403","REAC:R-RNO-71403","REAC:R-RNO-71406","REAC:R-RNO-71406","REAC:R-RNO-71406","REAC:R-RNO-71406","REAC:R-RNO-77289","REAC:R-RNO-8949613","REAC:R-RNO-8949613","REAC:R-RNO-8949613","REAC:R-RNO-8949613","REAC:R-RNO-8949613","REAC:R-RNO-8978868","REAC:R-RNO-8978868","7: Cellular responses to stress (2/4)","1: Carbohydrate metabolism (1/2)","3: Lipid metabolism (2/2)","8: Carbohydrate metabolism (4/13)","6: Neurodegenerative disease (2/15)","9: Lipid metabolism (3/6)","12: Mitochondrial translation (2/3)","14: Macroautophagy (1/2)"],"value":[19.7132616487455,37.5,20.3571428571429,31.9444444444444,29.9709724238026,39.4301470588235,23.8095238095238,19.7916666666667,24.3055555555556,18.75,25.8333333333333,30,32.7586206896552,25.78125,40.5172413793103,21.6666666666667,30.1724137931034,27.827380952381,29.1666666666667,27.3521505376344,33.3664772727273,25.2403846153846,30.2027925531915,27.2529069767442,27.2529069767442,32.1593237704918,29.0948275862069,32.703488372093,34.5394736842105,35.4166666666667,34.2526964560863,25.9673808762392,33.8068181818182,37.1408045977011,40.1041666666667,30.3197064989518,37.1031746031746,22.0833333333333,21.5277777777778,19.6759259259259,19.9786324786325,37.1408045977011,40.1041666666667,30.3197064989518,45.9795321637427,37.1031746031746,22.0833333333333,21.5277777777778,19.6759259259259,19.9786324786325,27.8432914046122,22.0211161387632,20.125786163522,21.1180124223602,37.7896613190731,34.9308300395257,33.75,26.2443438914027,20.5487033523087,18.7618595825427,24.2851307189542,40.8967391304348,39.375,26.3636363636364,26.3636363636364,34.375,26.875,25,25,34.7222222222222,40,29.1666666666667,43.75,19.047619047619,26.4285714285714,20.0501253132832,25.2100840336134,50,18.75,21.25,21.7857142857143,32.5,36.6666666666667,26.25,26.25,50,49.0740740740741,49.0740740740741,20.3703703703704,34,43,23.2023411371237,20.3113485630066,20.6235697940503,22.3897734231476,42.3214285714286,24.1251803751804,21.1423444976077,21.4772727272727,23.2625164690382,50,18.75,21.25,27.2727272727273,32.5,26.25,33.3333333333333,20,18.8888888888889,36.6666666666667,33.3333333333333,34.3421052631579,33.6206896551724,29.5454545454545,41.6666666666667,30.8333333333333,22.8947368421053,29.375,31.8181818181818,37.5,26.3636363636364,26.3636363636364,34.375,26.875,50,21.6666666666667,18.75,50,50,50,50,50,50,50,50],"title":["<b>Similarity score:<\/b> 0.39<br><b>Intersection:<\/b> Dld, Dlat, Fh, Idh3b, Mdh2,<br> Ogdh, Pdhb, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.75<br><b>Intersection:<\/b> Acsl1, Cbr4, Fasn, Hsd17b8,<br> Mecr, Oxsm","<b>Similarity score:<\/b> 0.41<br><b>Intersection:<\/b> Acaa2, Hadh, Mecr","<b>Similarity score:<\/b> 0.64<br><b>Intersection:<\/b> Acaa2, Acadl, Acadm, Acads,<br> Acadvl, Acsl1, Hadh","<b>Similarity score:<\/b> 0.6<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufb10, Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox17, Cox5a, Cox5b,<br> Cox6b1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa4, Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Sdha,<br> Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.79<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufb10, Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb,<br> Uqcr10, Atp5mc1, Cox4i1, Cox7a1, Cox7a2, Cox7a2l2,<br> Cox7b, Cox8b, Mt-atp6, Mt-co1, Mt-co3, Mt-cyb,<br> Mt-nd1, Mt-nd2, Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4,<br> Uqcrq","<b>Similarity score:<\/b> 0.48<br><b>Intersection:<\/b> Got1, Got2","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Got1, Got2","<b>Similarity score:<\/b> 0.49<br><b>Intersection:<\/b> Acads, Bckdha, Bckdhb, Dld,<br> Hibch, Mcee, Mmut","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> unknown (METAB)","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Dlat, Dld, Fh, Mdh1, Mdh2,<br> Pdha1, Pdhb","<b>Similarity score:<\/b> 0.6<br><b>Intersection:<\/b> Acss2, Dlat, Dld, Fh, Mdh1,<br> Mdh2, Me3, Pdha1, Pdhb","<b>Similarity score:<\/b> 0.66<br><b>Intersection:<\/b> Acss2, Cs, Dld, Gcsh, Mcee,<br> Mdh1, Mdh2, Mmut, Shmt2","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Acads, Acss2, Dld, Hibch,<br> Mcee, Mmut, Sucla2, Suclg1, Suclg2","<b>Similarity score:<\/b> 0.81<br><b>Intersection:<\/b> Cs, Dlat, Dld, Dlst, Fh,<br> Idh3a, Idh3b, Idh3g, Mdh1, Mdh2, Ogdh, Pdha1, Pdhb,<br> Sdha, Sdhb, Sucla2, Suclg1, Suclg2","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Cs, Idh3a, Idh3b, Idh3g","<b>Similarity score:<\/b> 0.6<br><b>Intersection:<\/b> Cs, Got1, Got2, Idh3a, Idh3b,<br> Idh3g","<b>Similarity score:<\/b> 0.56<br><b>Intersection:<\/b> Cox5a, Cox6b1, Cox4i1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Mt-co1, Mt-co3,<br> Mt-cyb, Uqcrq","<b>Similarity score:<\/b> 0.58<br><b>Intersection:<\/b> Cox5a, Cox6b1, Cox4i1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Mt-co1, Mt-co3,<br> Mt-cyb, Uqcrq","<b>Similarity score:<\/b> 0.55<br><b>Intersection:<\/b> Cox5a, Cox6b1, Cox4i1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Mt-co1, Mt-co3,<br> Mt-cyb, Uqcrq","<b>Similarity score:<\/b> 0.67<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Mt-nd1,<br> Mt-nd2, Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.5<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.6<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Ndufa3,<br> Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.55<br><b>Intersection:<\/b> Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufs8, Ndufv1, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.55<br><b>Intersection:<\/b> Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufs8, Ndufv1, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.64<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Mt-nd1,<br> Mt-nd2, Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.58<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.65<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.71<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6, Ndufa8,<br> Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufs8,<br> Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10, Cox4i1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Mt-co1, Mt-co3,<br> Mt-cyb, Ndufa3, Ndufb11, Ndufb4, Uqcrq","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Prkab1, Prkag2, Cox5a, Cox5b, Cox6b1,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb,<br> Uqcr10","<b>Similarity score:<\/b> 0.68<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Cox5a, Cox5b, Cox6b1, Irs1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6, Ndufa8,<br> Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10, Cox4i1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Mt-co1,<br> Mt-co3, Mt-cyb, Ndufa3, Ndufb11, Ndufb4, Uqcrq","<b>Similarity score:<\/b> 0.74<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.8<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.61<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Cox5a, Cox5b, Cox6b1, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa4, Ndufa6, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufs8, Ndufv2,<br> Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.74<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Ppif, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.44<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3, Sdha, Sdhb","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.39<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.74<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.8<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.61<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Cox5a, Cox5b, Cox6b1, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa4, Ndufa6, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufs8, Ndufv2,<br> Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.92<br><b>Intersection:<\/b> Atp5f1a, Atp5f1b, Atp5f1c,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b,<br> Cox6b1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa7, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufs8,<br> Ndufv1, Ndufv2, Ndufv3, Ppif, Sdha, Sdhb, Taf3, Uqcr10","<b>Similarity score:<\/b> 0.74<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Ppif, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.44<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3, Sdha, Sdhb","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.39<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.56<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufb10, Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb,<br> Uqcr10","<b>Similarity score:<\/b> 0.44<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3, Sdha,<br> Sdhb, Atp5mc1, Atp5mf, Mt-atp6, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3,<br> Ndufaf4, Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Sdha, Sdhb","<b>Similarity score:<\/b> 0.42<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3, Pdhb,<br> Sdha, Sdhb, Atp5mc1, Mt-atp6, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.76<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Atp5pf, Atp5po, Etfa, Etfb,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6, Ndufa8,<br> Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3,<br> Ndufv3-ps1, Atp5mc1, Atp5mf, Mt-atp6, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.7<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Etfa, Etfb, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3,<br> Ndufaf4, Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Ndufv3-ps1, Mt-nd1,<br> Mt-nd2, Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.68<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4,<br> Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufv2, Ndufv3, Ndufv3-ps1, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3,<br> Atp5mc1, Atp5mf, Mt-atp6, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.41<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3,<br> Ndufaf4, Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Ndufa3, Ndufb11,<br> Ndufb4","<b>Similarity score:<\/b> 0.49<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3,<br> Atp5mc1, Mt-atp6, Mt-nd1, Mt-nd2, Mt-nd5, Mt-nd6,<br> Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.82<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Etfa, Etfb, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3,<br> Ndufaf4, Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Ndufv3-ps1, Mt-nd1,<br> Mt-nd2, Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.79<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4,<br> Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufv2, Ndufv3, Ndufv3-ps1, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.53<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.53<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.54<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.5<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b,<br> Dnajb1, Fkbp4","<b>Similarity score:<\/b> 0.5<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Camk2g, Hsp90aa1, Hspa1a,<br> Hspa1b","<b>Similarity score:<\/b> 0.8<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b,<br> Dnajb1, Fkbp4","<b>Similarity score:<\/b> 0.58<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b","<b>Similarity score:<\/b> 0.88<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b,<br> Dnajb1, Fkbp4","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Dlat, Dld, Ogdh, Pdhb","<b>Similarity score:<\/b> 0.53<br><b>Intersection:<\/b> Dlat, Dld, Gcsh, Got2, Ogdh,<br> Pdhb","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Dld, Dlat, Ogdh, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.5<br><b>Intersection:<\/b> Dlat, Dld, Ogdh, Pdhb, Pdhx","<b>Similarity score:<\/b> 1<br><b>Intersection:<\/b> Map1lc3a, Map1lc3b, Src","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Dlat, Dld, Pdhb","<b>Similarity score:<\/b> 0.42<br><b>Intersection:<\/b> Dlat, Dld, Pdhb","<b>Similarity score:<\/b> 0.44<br><b>Intersection:<\/b> Dld, Dlat, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.65<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.73<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 1<br><b>Intersection:<\/b> Dap3, Gfm1, Mrpl12, Mrpl16,<br> Mrpl17, mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55,<br> Mrps12, Mrps15, Mrps16, Mrps17, Mrps21, Mrps22,<br> Mrps23, Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Similarity score:<\/b> 0.98<br><b>Intersection:<\/b> Dap3, Mrpl12, Mrpl16, Mrpl17,<br> mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55, Mrps12,<br> Mrps15, Mrps16, Mrps17, Mrps21, Mrps22, Mrps23,<br> Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Similarity score:<\/b> 0.98<br><b>Intersection:<\/b> Dap3, Mrpl12, Mrpl16, Mrpl17,<br> mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55, Mrps12,<br> Mrps15, Mrps16, Mrps17, Mrps21, Mrps22, Mrps23,<br> Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Similarity score:<\/b> 0.41<br><b>Intersection:<\/b> Acsl1, Cbr4, Fasn, Hsd17b8","<b>Similarity score:<\/b> 0.68<br><b>Intersection:<\/b> Acad10, Acadl, Acot11,<br> Acot13, Acot9, Hadh, Mcee, Mmut, Them4","<b>Similarity score:<\/b> 0.86<br><b>Intersection:<\/b> Acad10, Acadl, Acot11,<br> Acot13, Acot9, Acsl1, Cbr4, Eci2, Eci3, Fasn, Hadh,<br> Hsd17b8, Mcee, Mlycd, Mmut, Phyh, Prkag2, Ptges2,<br> Them4","<b>Similarity score:<\/b> 0.46<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.41<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2,<br> Ndufv3","<b>Similarity score:<\/b> 0.41<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Ndufa3, Ndufb11,<br> Ndufb4","<b>Similarity score:<\/b> 0.45<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.85<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4,<br> Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufv2, Ndufv3, Ndufv3-ps1, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.48<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.42<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2,<br> Ndufv3","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Ndufa3, Ndufb11,<br> Ndufb4","<b>Similarity score:<\/b> 0.47<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 1<br><b>Intersection:<\/b> Pfkfb1, Pfkfb3","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Dlat, Dld, Pdhb","<b>Similarity score:<\/b> 0.42<br><b>Intersection:<\/b> Dlat, Dld, Pdhb","<b>Similarity score:<\/b> 0.55<br><b>Intersection:<\/b> Dld, Bsg, Dlat, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.65<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.67<br><b>Intersection:<\/b> Bsg, Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Dld, Hibadh, Hibch","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Dlat, Dld, Gcsh, Got1, Got2,<br> Hibch, Ogdh, Pdhb","<b>Similarity score:<\/b> 0.73<br><b>Intersection:<\/b> Dlat, Dld, Gcsh, Got2, Ogdh,<br> Pdhb, Pdhx","<b>Similarity score:<\/b> 0.67<br><b>Intersection:<\/b> Bckdk, Dld, Hibadh, Hibch,<br> Ppm1k","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Dld, Fh, Idh3b, Mdh2, Ogdh,<br> Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.67<br><b>Intersection:<\/b> Dld, Fh, Idh3b, Mdh2, Me3,<br> Ogdh, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.59<br><b>Intersection:<\/b> Dld, Fh, Idh3b, Mdh2, Me3,<br> Ogdh, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.83<br><b>Intersection:<\/b> Dld, Fh, Idh3b, Mdh2, Me3,<br> Ogdh, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.62<br><b>Intersection:<\/b> Dlat, Dld, Fh, Idh3b, Mdh2,<br> Ogdh, Pdhb, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.46<br><b>Intersection:<\/b> Dlat, Dld, Fh, Mdh2, Me3,<br> Pdhb","<b>Similarity score:<\/b> 0.59<br><b>Intersection:<\/b> Dlat, Dld, Fh, Idh3b, Mdh2,<br> Me3, Ogdh, Pdhb, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.64<br><b>Intersection:<\/b> Dld, Bsg, Dlat, Fh, Idh3b,<br> L2hgdh, Mdh2, Me3, Ogdh, Pdhb, Pdhx, Sdha, Sdhb,<br> Sucla2, Suclg2","<b>Similarity score:<\/b> 0.75<br><b>Intersection:<\/b> Acad10, Acadl, Acot11,<br> Acot13, Acot9, Hadh, Mcee, Mmut, Them4","<b>Similarity score:<\/b> 0.53<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.53<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.54<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 1<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Acsl1, Cbr4, Fasn, Hsd17b8","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Acadl, Acsl1, Cbr4, Fasn,<br> Hadh, Hsd17b8",null,null,null,null,null,null,null,null],"color.color":["#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484",null,null,null,null,null,null,null,null],"physics":[false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],"color.highlight":["#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454",null,null,null,null,null,null,null,null],"labelHighlightBold":[true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,false,false,false,false,false,false,false,false],"color":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"#80B1D3","#8DD3C7","#BEBADA","#FDB462","#FB8072","#B3DE69","#FCCDE5","#D9D9D9"],"labelHighlightBold.1":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,false,false,false,false,false,false,false,false],"shadow":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,true,true,true,true,true,true,true,true]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot","physics":false},"manipulation":{"enabled":false},"interaction":{"hideEdgesOnZoom":true,"tooltipDelay":10,"zoomSpeed":1},"edges":{"smooth":false},"physics":{"stabilization":false}},"groups":["1: Carbohydrate metabolism (1/2)","0: Carbohydrate metabolism (1/6)","3: Lipid metabolism (2/2)","6: Neurodegenerative disease (2/15)","7: Cellular responses to stress (2/4)","8: Carbohydrate metabolism (4/13)","9: Lipid metabolism (3/6)","12: Mitochondrial translation (2/3)","14: Macroautophagy (1/2)"],"width":null,"height":null,"idselection":{"enabled":false},"byselection":{"enabled":false},"main":null,"submain":null,"footer":null,"background":"rgba(0, 0, 0, 0)","tooltipStay":null,"tooltipStyle":"position: fixed;visibility:hidden;padding: 5px;white-space: nowrap;font-family: verdana;font-size:14px;font-color:#000000;background-color: #f5f4ed;-moz-border-radius: 3px;-webkit-border-radius: 3px;border-radius: 3px;border: 1px solid #808074;box-shadow: 3px 3px 10px rgba(0, 0, 0, 0.2);","igraphlayout":{"type":"square"}},"evals":[],"jsHooks":[]}</script><p>You can also remove group labels entirely with
<code>label_nodes = FALSE</code>.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/enrichment_network_vis.html">enrichment_network_vis</a></span><span class="op">(</span><span class="va">pw</span>, label_nodes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Formatting inputs...</span></span>
<span><span class="co">#&gt; Subsetting feature-to-gene map...</span></span>
<span><span class="co">#&gt; Calculating similarity metric between pairs of enriched pathways...</span></span>
<span><span class="co">#&gt; Constructing graph...</span></span>
<span><span class="co">#&gt; Elapsed time: 6.082s</span></span></code></pre></div>
<div id="htmlwidget-b6f129550fac42de0b35" style="width:960px;height:432.632880098888px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-b6f129550fac42de0b35">{"x":{"nodes":{"id":["KEGG:00051","KEGG:04137","REAC:R-RNO-70171","KEGG:00564","KEGG:00565","KEGG:04217","KEGG:04977","KEGG:04714","KEGG:05415","KEGG:00190","KEGG:04932","REAC:R-RNO-3371571","REAC:R-RNO-6799198","REAC:R-RNO-611105","REAC:R-RNO-163200","REAC:R-RNO-3371497","REAC:R-RNO-3371556","REAC:R-RNO-1428517","KEGG:00020","KEGG:01200","KEGG:05012","KEGG:05020","KEGG:04723","KEGG:00640","KEGG:00280","KEGG:01212","KEGG:00630","KEGG:03320","KEGG:00620","KEGG:00061","KEGG:00650","KEGG:01210","KEGG:00071","KEGG:00062","REAC:R-RNO-5389840","REAC:R-RNO-5368287","REAC:R-RNO-5419276","REAC:R-RNO-71406","REAC:R-RNO-71403","REAC:R-RNO-8978868","REAC:R-RNO-77289","REAC:R-RNO-389661","REAC:R-RNO-3371568","REAC:R-RNO-556833","REAC:R-RNO-70895","REAC:R-RNO-71291","REAC:R-RNO-70268","REAC:R-RNO-1592230","REAC:R-RNO-5362517","KEGG:04260","KEGG:04670","KEGG:00250","REAC:R-RNO-163210","REAC:R-RNO-8949613","REAC:R-RNO-9663891","REAC:R-RNO-5205647"],"value":[100,100,100,100,100,100,100,400,900,900,900,400,900,900,900,400,100,900,100,100,100,100,400,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,400,100,100,100,100,100,100,100,100],"title":["<b>Fructose and mannose metabolism<\/b> <br>(Carbohydrate metabolism)<br><b>P-val:<\/b> 0.00079<br><b>ATAC:SKM-GN:<\/b> Pfkfb1, Pfkfb3","<b>Mitophagy - animal<\/b> (Transport and <br>catabolism)<br><b>P-val:<\/b> 0.0036<br><b>ATAC:SKM-GN:<\/b> Jun, Ubc","<b>Glycolysis<\/b> (Glucose metabolism)<br><b>P-val:<\/b> 0.0027<br><b>ATAC:SKM-GN:<\/b> Pfkfb1, Pfkfb3","<b>Glycerophospholipid metabolism<\/b> (Lipid <br>metabolism)<br><b>P-val:<\/b> 0.00011<br><b>METAB:SKM-GN:<\/b> unknown (METAB)","<b>Ether lipid metabolism<\/b> (Lipid metabolism)<br><b>P-val:<\/b> 8.2e-05<br><b>METAB:SKM-GN:<\/b> unknown (METAB)","<b>Necroptosis<\/b> (Cell growth and death)<br><b>P-val:<\/b> 0.00064<br><b>METAB:SKM-GN:<\/b> unknown (METAB)","<b>Vitamin digestion and absorption<\/b> <br>(Digestive system)<br><b>P-val:<\/b> 0.0016<br><b>METAB:SKM-GN:<\/b> unknown (METAB)","<b>Thermogenesis<\/b> (Environmental adaptation)<br><b>P-val:<\/b> 9.5e-19<br><b>PHOSPHO:SKM-GN:<\/b> Acsl1, Atp5f1a, Atp5f1c,<br> Ndufa7, Ndufb10, Ndufs4, Ndufs7, Ndufv1, Prkab1,<br> Prkag2, Taf3<br><b>PROT:SKM-GN:<\/b> Acsl1, Atp5f1a, Atp5f1b,<br> Atp5f1c, Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Coa5, Coa7,<br> Cox17, Cox19, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6, Ndufa7,<br> Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7, Ndufaf8,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufs8, Ndufv1, Ndufv2, Ndufv3, Prkag2, Sdha,<br> Sdhb, Taf3, Uqcr10","<b>Diabetic cardiomyopathy<\/b> (Cardiovascular <br>disease)<br><b>P-val:<\/b> 1e-35<br><b>PHOSPHO:SKM-GN:<\/b> Atp5f1a, Atp5f1c, Camk2g,<br> Gys1, Ndufa7, Ndufb10, Ndufs4, Ndufs7, Ndufv1,<br> Slc2a4, Taf3<br><b>PROT:SKM-GN:<\/b> Ace, Atp5f1a, Atp5f1b,<br> Atp5f1c, Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b,<br> Cox6b1, Irs1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7,<br> Ndufs8, Ndufv1, Ndufv2, Ndufv3, Pdha1, Pdhb, Ppif,<br> Sdha, Sdhb, Slc2a4, Taf3, Uqcr10<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Cox4i1, Cox5a,<br> Cox6b1, Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b,<br> Mt-atp6, Mt-co1, Mt-co3, Mt-cyb, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa12, Ndufa3, Ndufa6, Ndufb11,<br> Ndufb4, Ndufb9, Ndufs3, Ndufv1, Tbc1d4, Uqcrq","<b>Oxidative phosphorylation<\/b> (Energy <br>metabolism)<br><b>P-val:<\/b> 6.2e-44<br><b>PHOSPHO:SKM-GN:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufb10, Ndufs4, Ndufs7, Ndufv1, Taf3<br><b>PROT:SKM-GN:<\/b> Atp5f1a, Atp5f1b, Atp5f1c,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox17, Cox5a,<br> Cox5b, Cox6b1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufs8,<br> Ndufv1, Ndufv2, Ndufv3, Ppa2, Sdha, Sdhb, Taf3,<br> Uqcr10<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Cox4i1,<br> Cox5a, Cox6b1, Cox7a1, Cox7a2, Cox7a2l2, Cox7b,<br> Cox8b, Mt-atp6, Mt-co1, Mt-co3, Mt-cyb, Mt-nd1,<br> Mt-nd2, Mt-nd5, Mt-nd6, Ndufa12, Ndufa3, Ndufa6,<br> Ndufb11, Ndufb4, Ndufb9, Ndufs3, Ndufv1, Uqcrq","<b>Non-alcoholic fatty liver disease<\/b> <br>(Endocrine and metabolic disease)<br><b>P-val:<\/b> 2.4e-28<br><b>PHOSPHO:SKM-GN:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Prkab1, Prkag2<br><b>PROT:SKM-GN:<\/b> Cox5a, Cox5b, Cox6b1, Irs1,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa7, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufs8, Ndufv1,<br> Ndufv2, Ndufv3, Prkag2, Sdha, Sdhb, Uqcr10<br><b>TRNSCRPT:SKM-GN:<\/b> Cox4i1, Cox5a, Cox6b1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Il6r, Mt-co1,<br> Mt-co3, Mt-cyb, Ndufa12, Ndufa3, Ndufa6, Ndufb11,<br> Ndufb4, Ndufb9, Ndufs3, Ndufv1, Uqcrq","<b>HSF1-dependent transactivation<\/b> (Cellular <br>response to heat stress)<br><b>P-val:<\/b> 1.2e-07<br><b>PHOSPHO:SKM-GN:<\/b> Camk2g, Hsp90aa1, Hspa1a,<br> Hspa1b<br><b>PROT:SKM-GN:<\/b> Dnajb1, Fkbp4, Hsp90aa1,<br> Hspa1a, Hspa1b, Hspb8","<b>Complex I biogenesis<\/b> (Respiratory electron <br>transport)<br><b>P-val:<\/b> 9.9e-33<br><b>PHOSPHO:SKM-GN:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1<br><b>PROT:SKM-GN:<\/b> Acad9, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa7, Ndufa8, Ndufa9,<br> Ndufaf3, Ndufaf4, Ndufaf7, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufv1, Ndufv2,<br> Ndufv3, Ndufv3-ps1<br><b>TRNSCRPT:SKM-GN:<\/b> Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa12, Ndufa3, Ndufa6, Ndufb11, Ndufb4,<br> Ndufb9, Ndufs3, Ndufv1","<b>Respiratory electron transport<\/b> <br>(Respiratory electron transport, ATP synthesis by <br>chemiosmotic coupling, and heat production by uncoupling <br>proteins.)<br><b>P-val:<\/b> 7.9e-34<br><b>PHOSPHO:SKM-GN:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1<br><b>PROT:SKM-GN:<\/b> Acad9, Etfa, Etfb, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa6, Ndufa7, Ndufa8,<br> Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufv1,<br> Ndufv2, Ndufv3, Ndufv3-ps1<br><b>TRNSCRPT:SKM-GN:<\/b> Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa12, Ndufa3, Ndufa6, Ndufb11, Ndufb4,<br> Ndufb9, Ndufs3, Ndufv1","<b>Respiratory electron transport, ATP synthesis <br>by chemiosmotic coupling, and heat production by <br>uncoupling proteins.<\/b> (The citric acid (TCA) <br>cycle and respiratory electron transport)<br><b>P-val:<\/b> 1.6e-35<br><b>PHOSPHO:SKM-GN:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1<br><b>PROT:SKM-GN:<\/b> Acad9, Atp5pf, Atp5po, Etfa,<br> Etfb, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa7, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufv1, Ndufv2, Ndufv3, Ndufv3-ps1<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Mt-atp6,<br> Mt-nd1, Mt-nd2, Mt-nd5, Mt-nd6, Ndufa12, Ndufa3,<br> Ndufa6, Ndufb11, Ndufb4, Ndufb9, Ndufs3, Ndufv1","<b>HSP90 chaperone cycle for steroid hormone <br>receptors (SHR)<\/b> (Cellular responses to stress)<br><b>P-val:<\/b> 7.5e-06<br><b>PHOSPHO:SKM-GN:<\/b> Dync1li2, Hsp90aa1, Hspa1a,<br> Hspa1b<br><b>PROT:SKM-GN:<\/b> Dnaja1, Dnajb1, Fkbp4, Fkbp5,<br> Hsp90aa1, Hspa1a, Hspa1b, Stip1","<b>Cellular response to heat stress<\/b> (Cellular <br>responses to stress)<br><b>P-val:<\/b> 0.001<br><b>PHOSPHO:SKM-GN:<\/b> Bag3, Camk2g, Hsp90aa1,<br> Hspa1a, Hspa1b","<b>The citric acid (TCA) cycle and respiratory <br>electron transport<\/b> (Metabolism)<br><b>P-val:<\/b> 5e-44<br><b>PHOSPHO:SKM-GN:<\/b> Dld, Ndufa7, Ndufb10,<br> Ndufs4, Ndufs7, Ndufv1<br><b>PROT:SKM-GN:<\/b> Acad9, Atp5pf, Atp5po, Bsg,<br> Dlat, Dld, Etfa, Etfb, Fh, Idh3b, L2hgdh, Mdh2, Me3,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa7, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufv1, Ndufv2, Ndufv3, Ndufv3-ps1, Ogdh, Pdhb,<br> Pdhx, Sdha, Sdhb, Sucla2, Suclg2<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Me3,<br> Mt-atp6, Mt-nd1, Mt-nd2, Mt-nd5, Mt-nd6, Ndufa12,<br> Ndufa3, Ndufa6, Ndufb11, Ndufb4, Ndufb9, Ndufs3,<br> Ndufv1, Pdhx","<b>Citrate cycle (TCA cycle)<\/b> (Carbohydrate <br>metabolism)<br><b>P-val:<\/b> 5.5e-16<br><b>PROT:SKM-GN:<\/b> Cs, Dlat, Dld, Dlst, Fh,<br> Idh3a, Idh3b, Idh3g, Mdh1, Mdh2, Ogdh, Pdha1, Pdhb,<br> Sdha, Sdhb, Sucla2, Suclg1, Suclg2","<b>Carbon metabolism<\/b> (NULL)<br><b>P-val:<\/b> 5.7e-14<br><b>PROT:SKM-GN:<\/b> Acads, Acss2, Cs, Dlat, Dld,<br> Dlst, Fh, Gcsh, Got1, Got2, Hibch, Hk2, Idh3a,<br> Idh3b, Idh3g, Mcee, Mdh1, Mdh2, Me3, Mmut, Ogdh,<br> Pdha1, Pdhb, Sdha, Sdhb, Shmt2, Sucla2, Suclg1, Suclg2","<b>Parkinson disease<\/b> (Neurodegenerative <br>disease)<br><b>P-val:<\/b> 1.3e-10<br><b>PROT:SKM-GN:<\/b> Atp5f1a, Atp5f1b, Atp5f1c,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b,<br> Cox6b1, Htra2, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufs8,<br> Ndufv1, Ndufv2, Ndufv3, Ppif, Sdha, Sdhb, Taf3,<br> Trap1, Uqcr10","<b>Prion disease<\/b> (Neurodegenerative disease)<br><b>P-val:<\/b> 1.1e-09<br><b>PROT:SKM-GN:<\/b> Atp5f1a, Atp5f1b, Atp5f1c,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b,<br> Cox6b1, Hspa1a, Hspa1b, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufs8, Ndufv1, Ndufv2, Ndufv3, Ppif, Sdha,<br> Sdhb, Stip1, Taf3, Uqcr10","<b>Retrograde endocannabinoid signaling<\/b> <br>(Nervous system)<br><b>P-val:<\/b> 4.3e-16<br><b>PROT:SKM-GN:<\/b> Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufs8, Ndufv1, Ndufv2, Ndufv3<br><b>TRNSCRPT:SKM-GN:<\/b> Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa12, Ndufa3, Ndufa6, Ndufb11, Ndufb4,<br> Ndufb9, Ndufs3, Ndufv1","<b>Propanoate metabolism<\/b> (Carbohydrate <br>metabolism)<br><b>P-val:<\/b> 6.2e-08<br><b>PROT:SKM-GN:<\/b> Acads, Acss2, Bckdha, Bckdhb,<br> Dld, Hibch, Mcee, Mlycd, Mmut, Sucla2, Suclg1,<br> Suclg2","<b>Valine, leucine and isoleucine degradation<\/b> <br>(Amino acid metabolism)<br><b>P-val:<\/b> 1.5e-07<br><b>PROT:SKM-GN:<\/b> Acaa2, Acadm, Acads, Bckdha,<br> Bckdhb, Dld, Hadh, Hibadh, Hibch, Hsd17b10, Mcee,<br> Mmut, Oxct1","<b>Fatty acid metabolism<\/b> (NULL)<br><b>P-val:<\/b> 2.3e-06<br><b>PROT:SKM-GN:<\/b> Acaa2, Acadl, Acadm, Acads,<br> Acadvl, Acsl1, Cbr4, Fasn, Hadh, Hsd17b8, Mecr,<br> Oxsm","<b>Glyoxylate and dicarboxylate metabolism<\/b> <br>(Carbohydrate metabolism)<br><b>P-val:<\/b> 3.7e-06<br><b>PROT:SKM-GN:<\/b> Acss2, Cs, Dld, Gcsh, Mcee,<br> Mdh1, Mdh2, Mmut, Shmt2","<b>PPAR signaling pathway<\/b> (Endocrine system)<br><b>P-val:<\/b> 3.3e-05<br><b>PROT:SKM-GN:<\/b> Acadl, Acadm, Acsl1, Fabp4,<br> Fabp5, Gk, Me3, Plin2, Plin4, Plin5, Sorbs1","<b>Pyruvate metabolism<\/b> (Carbohydrate <br>metabolism)<br><b>P-val:<\/b> 3.5e-05<br><b>PROT:SKM-GN:<\/b> Acss2, Dlat, Dld, Fh, Ldhd,<br> Mdh1, Mdh2, Me3, Pdha1, Pdhb","<b>Fatty acid biosynthesis<\/b> (Lipid metabolism)<br><b>P-val:<\/b> 9.1e-05<br><b>PROT:SKM-GN:<\/b> Acsl1, Cbr4, Fasn, Hsd17b8,<br> Mecr, Oxsm","<b>Butanoate metabolism<\/b> (Carbohydrate <br>metabolism)<br><b>P-val:<\/b> 9.1e-05<br><b>PROT:SKM-GN:<\/b> Acads, Aldh5a1, Bdh1, Hadh,<br> L2hgdh, Oxct1","<b>2-Oxocarboxylic acid metabolism<\/b> (NULL)<br><b>P-val:<\/b> 9.1e-05<br><b>PROT:SKM-GN:<\/b> Cs, Got1, Got2, Idh3a, Idh3b,<br> Idh3g","<b>Fatty acid degradation<\/b> (Lipid metabolism)<br><b>P-val:<\/b> 0.00013<br><b>PROT:SKM-GN:<\/b> Acaa2, Acadl, Acadm, Acads,<br> Acadvl, Acsl1, Eci2, Eci3, Gcdh, Hadh","<b>Fatty acid elongation<\/b> (Lipid metabolism)<br><b>P-val:<\/b> 0.0014<br><b>PROT:SKM-GN:<\/b> Acaa2, Acot2, Hadh, Mecr,<br> Them4","<b>Mitochondrial translation elongation<\/b> <br>(Mitochondrial translation)<br><b>P-val:<\/b> 5.1e-14<br><b>PROT:SKM-GN:<\/b> Dap3, Gfm1, Mrpl12, Mrpl16,<br> Mrpl17, mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55,<br> Mrps12, Mrps15, Mrps16, Mrps17, Mrps21, Mrps22,<br> Mrps23, Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Mitochondrial translation<\/b> (Translation)<br><b>P-val:<\/b> 1e-13<br><b>PROT:SKM-GN:<\/b> Dap3, Gfm1, Mrpl12, Mrpl16,<br> Mrpl17, mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55,<br> Mrps12, Mrps15, Mrps16, Mrps17, Mrps21, Mrps22,<br> Mrps23, Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Mitochondrial translation termination<\/b> <br>(Mitochondrial translation)<br><b>P-val:<\/b> 5.9e-13<br><b>PROT:SKM-GN:<\/b> Dap3, Mrpl12, Mrpl16, Mrpl17,<br> mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55, Mrps12,<br> Mrps15, Mrps16, Mrps17, Mrps21, Mrps22, Mrps23,<br> Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Pyruvate metabolism and Citric Acid (TCA) <br>cycle<\/b> (The citric acid (TCA) cycle and respiratory <br>electron transport)<br><b>P-val:<\/b> 2.5e-10<br><b>PROT:SKM-GN:<\/b> Bsg, Dlat, Dld, Fh, Idh3b,<br> L2hgdh, Mdh2, Me3, Ogdh, Pdhb, Pdhx, Sdha, Sdhb,<br> Sucla2, Suclg2","<b>Citric acid cycle (TCA cycle)<\/b> (Pyruvate <br>metabolism and Citric Acid (TCA) cycle)<br><b>P-val:<\/b> 7.5e-09<br><b>PROT:SKM-GN:<\/b> Dld, Fh, Idh3b, Mdh2, Me3,<br> Ogdh, Sdha, Sdhb, Sucla2, Suclg2","<b>Fatty acid metabolism<\/b> (Metabolism of <br>lipids)<br><b>P-val:<\/b> 1.2e-08<br><b>PROT:SKM-GN:<\/b> Acad10, Acadl, Acot11, Acot13,<br> Acot9, Acsl1, Cbr4, Eci2, Eci3, Fasn, Hadh,<br> Hsd17b8, Mcee, Mlycd, Mmut, Phyh, Prkag2, Ptges2,<br> Them4","<b>Mitochondrial Fatty Acid Beta-Oxidation<\/b> <br>(Fatty acid metabolism)<br><b>P-val:<\/b> 2.3e-06<br><b>PROT:SKM-GN:<\/b> Acad10, Acadl, Acot11, Acot13,<br> Acot9, Hadh, Mcee, Mmut, Them4","<b>Glyoxylate metabolism and glycine <br>degradation<\/b> (Metabolism of amino acids and derivatives)<br><b>P-val:<\/b> 6.5e-06<br><b>PROT:SKM-GN:<\/b> Dlat, Dld, Gcsh, Got2, Ogdh,<br> Pdhb, Pdhx","<b>Attenuation phase<\/b> (HSF1-dependent <br>transactivation)<br><b>P-val:<\/b> 3.1e-05<br><b>PROT:SKM-GN:<\/b> Dnajb1, Fkbp4, Hsp90aa1,<br> Hspa1a, Hspa1b","<b>Metabolism of lipids<\/b> (Metabolism)<br><b>P-val:<\/b> 0.00027<br><b>PROT:SKM-GN:<\/b> Acad10, Acadl, Acot11, Acot13,<br> Acot9, Acp6, Acsl1, Cbr4, Eci2, Eci3, Fabp5,<br> Fasn, Fdxr, Gpd1l, Hadh, Hsd17b8, Lpin1, Mcee, Mlycd,<br> Mmut, Oxct1, Phyh, Plin3, Prkag2, Ptges2, Them4","<b>Branched-chain amino acid catabolism<\/b> <br>(Metabolism of amino acids and derivatives)<br><b>P-val:<\/b> 0.00026<br><b>PROT:SKM-GN:<\/b> Bckdk, Dld, Hibadh, Hibch,<br> Ppm1k","<b>Metabolism of amino acids and derivatives<\/b> <br>(Metabolism)<br><b>P-val:<\/b> 0.00049<br><b>PROT:SKM-GN:<\/b> Bckdk, Dlat, Dld, Ethe1, Gcdh,<br> Gcsh, Got1, Got2, Hibadh, Hibch, Oat, Ogdh, Pdhb,<br> Pdhx, Ppm1k","<b>Pyruvate metabolism<\/b> (Pyruvate metabolism <br>and Citric Acid (TCA) cycle)<br><b>P-val:<\/b> 0.0014<br><b>PROT:SKM-GN:<\/b> Bsg, Dlat, Dld, Pdhb, Pdhx","<b>Mitochondrial biogenesis<\/b> (Organelle <br>biogenesis and maintenance)<br><b>P-val:<\/b> 3.7e-05<br><b>PROT:SKM-GN:<\/b> Acss2, Atp5pf, Atp5po, Sirt3,<br> Sod2<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Signaling by Retinoic Acid<\/b> (Signaling by <br>Nuclear Receptors)<br><b>P-val:<\/b> 0.0035<br><b>PROT:SKM-GN:<\/b> Dlat, Dld, Fabp5, Pdhb, Pdhx","<b>Cardiac muscle contraction<\/b> (Circulatory <br>system)<br><b>P-val:<\/b> 1.5e-11<br><b>TRNSCRPT:SKM-GN:<\/b> Atp1a2, Cox4i1, Cox5a,<br> Cox6b1, Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b,<br> Mt-co1, Mt-co3, Mt-cyb, Uqcrq","<b>Leukocyte transendothelial migration<\/b> <br>(Immune system)<br><b>P-val:<\/b> 0.0024<br><b>TRNSCRPT:SKM-GN:<\/b> Cdh5, Cxcl12, Jam2,<br> Rapgef4, Vcl","<b>Alanine, aspartate and glutamate <br>metabolism<\/b> (Amino acid metabolism)<br><b>P-val:<\/b> 0.0029<br><b>TRNSCRPT:SKM-GN:<\/b> Aldh5a1, Got1, Got2","<b>Formation of ATP by chemiosmotic coupling<\/b> <br>(Respiratory electron transport, ATP synthesis by <br>chemiosmotic coupling, and heat production by <br>uncoupling proteins.)<br><b>P-val:<\/b> 0.00031<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Cristae formation<\/b> (Mitochondrial <br>biogenesis)<br><b>P-val:<\/b> 0.00031<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Selective autophagy<\/b> (Macroautophagy)<br><b>P-val:<\/b> 0.0014<br><b>TRNSCRPT:SKM-GN:<\/b> Map1lc3a, Map1lc3b, Src","<b>Mitophagy<\/b> (Selective autophagy)<br><b>P-val:<\/b> 0.0014<br><b>TRNSCRPT:SKM-GN:<\/b> Map1lc3a, Map1lc3b, Src"],"color.highlight.background":["yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow"],"color.border":["black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black"],"shadow":[false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],"physics":[false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],"borderWidth":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],"borderWidthSelected":[3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3],"pathway_subclass":["Carbohydrate metabolism","Transport and catabolism","Glucose metabolism","Lipid metabolism","Lipid metabolism","Cell growth and death","Digestive system","Environmental adaptation","Cardiovascular disease","Energy metabolism","Endocrine and metabolic disease","Cellular response to heat stress","Respiratory electron transport","Respiratory electron transport, ATP synthesis by chemiosmotic coupling, and heat production by uncoupling proteins.","The citric acid (TCA) cycle and respiratory electron transport","Cellular responses to stress","Cellular responses to stress","Metabolism","Carbohydrate metabolism","NULL","Neurodegenerative disease","Neurodegenerative disease","Nervous system","Carbohydrate metabolism","Amino acid metabolism","NULL","Carbohydrate metabolism","Endocrine system","Carbohydrate metabolism","Lipid metabolism","Carbohydrate metabolism","NULL","Lipid metabolism","Lipid metabolism","Mitochondrial translation","Translation","Mitochondrial translation","The citric acid (TCA) cycle and respiratory electron transport","Pyruvate metabolism and Citric Acid (TCA) cycle","Metabolism of lipids","Fatty acid metabolism","Metabolism of amino acids and derivatives","HSF1-dependent transactivation","Metabolism","Metabolism of amino acids and derivatives","Metabolism","Pyruvate metabolism and Citric Acid (TCA) cycle","Organelle biogenesis and maintenance","Signaling by Nuclear Receptors","Circulatory system","Immune system","Amino acid metabolism","Respiratory electron transport, ATP synthesis by chemiosmotic coupling, and heat production by uncoupling proteins.","Mitochondrial biogenesis","Macroautophagy","Selective autophagy"],"group":["1","0","1","3","3","0","0","6","6","6","6","7","6","6","6","7","7","6","8","8","6","6","6","8","8","9","8","0","8","9","0","8","9","9","12","12","12","8","8","9","9","8","7","9","8","8","8","6","8","6","0","8","6","6","14","14"],"group_number":["1","0","1","3","3","0","0","6","6","6","6","7","6","6","6","7","7","6","8","8","6","6","6","8","8","9","8","0","8","9","0","8","9","9","12","12","12","8","8","9","9","8","7","9","8","8","8","6","8","6","0","8","6","6","14","14"],"color.background":["#8DD3C7","#FFFFB3","#8DD3C7","#BEBADA","#BEBADA","#FFFFB3","#FFFFB3","#FB8072","#FB8072","#FB8072","#FB8072","#80B1D3","#FB8072","#FB8072","#FB8072","#80B1D3","#80B1D3","#FB8072","#FDB462","#FDB462","#FB8072","#FB8072","#FB8072","#FDB462","#FDB462","#B3DE69","#FDB462","#FFFFB3","#FDB462","#B3DE69","#FFFFB3","#FDB462","#B3DE69","#B3DE69","#FCCDE5","#FCCDE5","#FCCDE5","#FDB462","#FDB462","#B3DE69","#B3DE69","#FDB462","#80B1D3","#B3DE69","#FDB462","#FDB462","#FDB462","#FB8072","#FDB462","#FB8072","#FFFFB3","#FDB462","#FB8072","#FB8072","#D9D9D9","#D9D9D9"],"color.highlight.border":["#8DD3C7","#FFFFB3","#8DD3C7","#BEBADA","#BEBADA","#FFFFB3","#FFFFB3","#FB8072","#FB8072","#FB8072","#FB8072","#80B1D3","#FB8072","#FB8072","#FB8072","#80B1D3","#80B1D3","#FB8072","#FDB462","#FDB462","#FB8072","#FB8072","#FB8072","#FDB462","#FDB462","#B3DE69","#FDB462","#FFFFB3","#FDB462","#B3DE69","#FFFFB3","#FDB462","#B3DE69","#B3DE69","#FCCDE5","#FCCDE5","#FCCDE5","#FDB462","#FDB462","#B3DE69","#B3DE69","#FDB462","#80B1D3","#B3DE69","#FDB462","#FDB462","#FDB462","#FB8072","#FDB462","#FB8072","#FFFFB3","#FDB462","#FB8072","#FB8072","#D9D9D9","#D9D9D9"],"x":[-0.11026967701262,-0.982861196625032,-0.0574475172782052,-1,-0.907816591517361,0.172208487605309,-0.321089439480344,-0.397983806881813,-0.425887580187571,-0.357376847671733,-0.48435699534822,0.66327124446585,-0.254595885009632,-0.297698652326017,-0.275367407565433,0.743314369574789,0.647907398410176,-0.0883022248998029,0.282255493202449,0.469040364820352,-0.276043682293548,-0.359697169745887,-0.349203157223405,0.718108437330633,0.75295604907962,-0.641467915662642,0.740015514147539,-0.991251062621519,0.34994770256412,-0.474778551903129,0.486021932172035,0.550804757336745,-0.638404926742745,-0.813770650786042,0.198329969084393,0.139857018718515,0.273983601856271,0.196194210193302,0.223013448043693,-0.571792836110292,-0.539222376524774,0.247130484104703,0.567153847677941,-0.426981323057316,0.566724570214044,0.41699846687339,0.183814627978322,-0.162992896749402,0.156381795173481,-0.634707363896377,-0.134351283106032,0.644011003654512,-0.244345075790664,-0.15333839620616,0.94762434366122,1],"y":[-0.831989638823115,-0.322034854898085,-0.977035579458035,0.212166899784794,0.0973110459401187,-0.380406150393419,-1,0.559513618258743,0.450140193024031,0.625366334587401,0.503712084742127,-0.635015153057943,0.448640605952243,0.388354889791006,0.608883245712365,-0.533095546075824,-0.443289277591293,0.526167125843414,0.541381686953654,0.484897537485571,0.521748875236027,0.404439088054035,0.495314542956331,0.262613346317959,0.0512281523182951,-0.457842372043466,0.591205847785105,-0.104884450043368,0.39805548109104,-0.515853133903684,-0.91810063252879,0.619193734759778,-0.270419131308773,-0.507546831302382,-0.868755929988269,-0.715716737338843,-0.742736958905191,0.517027314965256,0.665440489174645,-0.634483206144121,-0.799628018055225,0.408100049120196,-0.543907823621292,-0.681491067855321,0.0343043043155746,0.267806407980252,0.319757766218906,1,0.395637887910044,0.560186094837954,-0.38251101638016,0.409907514985502,0.809194285672352,0.785631370206028,-0.24750305742147,-0.109983309507107]},"edges":{"from":["REAC:R-RNO-1428517","KEGG:01212","KEGG:01212","KEGG:01212","KEGG:04714","KEGG:05415","KEGG:01210","REAC:R-RNO-71291","KEGG:00640","KEGG:00564","KEGG:00020","KEGG:01200","KEGG:01200","KEGG:01200","KEGG:00020","KEGG:00020","KEGG:01200","KEGG:00190","KEGG:04932","KEGG:05415","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05012","KEGG:05020","KEGG:05415","REAC:R-RNO-1428517","REAC:R-RNO-163200","REAC:R-RNO-611105","REAC:R-RNO-6799198","KEGG:00190","KEGG:04714","KEGG:05415","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05415","REAC:R-RNO-1428517","REAC:R-RNO-163200","REAC:R-RNO-611105","REAC:R-RNO-6799198","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05012","KEGG:05415","REAC:R-RNO-1428517","REAC:R-RNO-163200","REAC:R-RNO-611105","REAC:R-RNO-6799198","KEGG:04714","KEGG:00190","KEGG:04714","KEGG:05415","REAC:R-RNO-163200","REAC:R-RNO-611105","REAC:R-RNO-6799198","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05415","REAC:R-RNO-611105","REAC:R-RNO-6799198","KEGG:00190","REAC:R-RNO-1428517","REAC:R-RNO-1592230","REAC:R-RNO-163200","REAC:R-RNO-3371571","REAC:R-RNO-3371497","REAC:R-RNO-3371571","REAC:R-RNO-3371497","REAC:R-RNO-3371556","REAC:R-RNO-3371571","KEGG:00020","KEGG:01200","REAC:R-RNO-1428517","REAC:R-RNO-71406","REAC:R-RNO-9663891","KEGG:00020","KEGG:00620","REAC:R-RNO-1428517","REAC:R-RNO-389661","REAC:R-RNO-70268","REAC:R-RNO-71291","REAC:R-RNO-71406","REAC:R-RNO-5389840","REAC:R-RNO-5368287","REAC:R-RNO-5389840","KEGG:00061","REAC:R-RNO-77289","REAC:R-RNO-8978868","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05415","REAC:R-RNO-6799198","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05415","KEGG:00051","KEGG:00020","KEGG:00620","REAC:R-RNO-1428517","REAC:R-RNO-389661","REAC:R-RNO-71291","REAC:R-RNO-71406","KEGG:00280","KEGG:01200","REAC:R-RNO-389661","REAC:R-RNO-70895","KEGG:00020","KEGG:01200","REAC:R-RNO-1428517","REAC:R-RNO-71406","KEGG:00020","KEGG:00620","KEGG:01200","REAC:R-RNO-1428517","REAC:R-RNO-8978868","KEGG:00190","REAC:R-RNO-1428517","REAC:R-RNO-1592230","REAC:R-RNO-163200","REAC:R-RNO-163210","KEGG:00061","KEGG:01212"],"to":["KEGG:00020","KEGG:00061","KEGG:00062","KEGG:00071","KEGG:00190","KEGG:00190","KEGG:00250","KEGG:00250","KEGG:00280","KEGG:00565","KEGG:00620","KEGG:00620","KEGG:00630","KEGG:00640","KEGG:01200","KEGG:01210","KEGG:01210","KEGG:04260","KEGG:04260","KEGG:04260","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04932","KEGG:04932","KEGG:04932","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05415","REAC:R-RNO-1428517","REAC:R-RNO-1428517","REAC:R-RNO-1428517","REAC:R-RNO-1428517","REAC:R-RNO-1428517","REAC:R-RNO-1428517","REAC:R-RNO-163200","REAC:R-RNO-163200","REAC:R-RNO-163200","REAC:R-RNO-163200","REAC:R-RNO-163200","REAC:R-RNO-163200","REAC:R-RNO-163210","REAC:R-RNO-163210","REAC:R-RNO-163210","REAC:R-RNO-163210","REAC:R-RNO-3371497","REAC:R-RNO-3371556","REAC:R-RNO-3371556","REAC:R-RNO-3371568","REAC:R-RNO-3371568","REAC:R-RNO-3371568","REAC:R-RNO-389661","REAC:R-RNO-389661","REAC:R-RNO-389661","REAC:R-RNO-389661","REAC:R-RNO-5205647","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5368287","REAC:R-RNO-5419276","REAC:R-RNO-5419276","REAC:R-RNO-556833","REAC:R-RNO-556833","REAC:R-RNO-556833","REAC:R-RNO-611105","REAC:R-RNO-611105","REAC:R-RNO-611105","REAC:R-RNO-611105","REAC:R-RNO-611105","REAC:R-RNO-6799198","REAC:R-RNO-6799198","REAC:R-RNO-6799198","REAC:R-RNO-6799198","REAC:R-RNO-70171","REAC:R-RNO-70268","REAC:R-RNO-70268","REAC:R-RNO-70268","REAC:R-RNO-70268","REAC:R-RNO-70268","REAC:R-RNO-70268","REAC:R-RNO-70895","REAC:R-RNO-71291","REAC:R-RNO-71291","REAC:R-RNO-71291","REAC:R-RNO-71403","REAC:R-RNO-71403","REAC:R-RNO-71403","REAC:R-RNO-71403","REAC:R-RNO-71406","REAC:R-RNO-71406","REAC:R-RNO-71406","REAC:R-RNO-71406","REAC:R-RNO-77289","REAC:R-RNO-8949613","REAC:R-RNO-8949613","REAC:R-RNO-8949613","REAC:R-RNO-8949613","REAC:R-RNO-8949613","REAC:R-RNO-8978868","REAC:R-RNO-8978868"],"value":[19.7132616487455,37.5,20.3571428571429,31.9444444444444,29.9709724238026,39.4301470588235,23.8095238095238,19.7916666666667,24.3055555555556,18.75,25.8333333333333,30,32.7586206896552,25.78125,40.5172413793103,21.6666666666667,30.1724137931034,27.827380952381,29.1666666666667,27.3521505376344,33.3664772727273,25.2403846153846,30.2027925531915,27.2529069767442,27.2529069767442,32.1593237704918,29.0948275862069,32.703488372093,34.5394736842105,35.4166666666667,34.2526964560863,25.9673808762392,33.8068181818182,37.1408045977011,40.1041666666667,30.3197064989518,37.1031746031746,22.0833333333333,21.5277777777778,19.6759259259259,19.9786324786325,37.1408045977011,40.1041666666667,30.3197064989518,45.9795321637427,37.1031746031746,22.0833333333333,21.5277777777778,19.6759259259259,19.9786324786325,27.8432914046122,22.0211161387632,20.125786163522,21.1180124223602,37.7896613190731,34.9308300395257,33.75,26.2443438914027,20.5487033523087,18.7618595825427,24.2851307189542,40.8967391304348,39.375,26.3636363636364,26.3636363636364,34.375,26.875,25,25,34.7222222222222,40,29.1666666666667,43.75,19.047619047619,26.4285714285714,20.0501253132832,25.2100840336134,50,18.75,21.25,21.7857142857143,32.5,36.6666666666667,26.25,26.25,50,49.0740740740741,49.0740740740741,20.3703703703704,34,43,23.2023411371237,20.3113485630066,20.6235697940503,22.3897734231476,42.3214285714286,24.1251803751804,21.1423444976077,21.4772727272727,23.2625164690382,50,18.75,21.25,27.2727272727273,32.5,26.25,33.3333333333333,20,18.8888888888889,36.6666666666667,33.3333333333333,34.3421052631579,33.6206896551724,29.5454545454545,41.6666666666667,30.8333333333333,22.8947368421053,29.375,31.8181818181818,37.5,26.3636363636364,26.3636363636364,34.375,26.875,50,21.6666666666667,18.75],"title":["<b>Similarity score:<\/b> 0.39<br><b>Intersection:<\/b> Dld, Dlat, Fh, Idh3b, Mdh2,<br> Ogdh, Pdhb, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.75<br><b>Intersection:<\/b> Acsl1, Cbr4, Fasn, Hsd17b8,<br> Mecr, Oxsm","<b>Similarity score:<\/b> 0.41<br><b>Intersection:<\/b> Acaa2, Hadh, Mecr","<b>Similarity score:<\/b> 0.64<br><b>Intersection:<\/b> Acaa2, Acadl, Acadm, Acads,<br> Acadvl, Acsl1, Hadh","<b>Similarity score:<\/b> 0.6<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufb10, Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox17, Cox5a, Cox5b,<br> Cox6b1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa4, Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Sdha,<br> Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.79<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufb10, Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb,<br> Uqcr10, Atp5mc1, Cox4i1, Cox7a1, Cox7a2, Cox7a2l2,<br> Cox7b, Cox8b, Mt-atp6, Mt-co1, Mt-co3, Mt-cyb,<br> Mt-nd1, Mt-nd2, Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4,<br> Uqcrq","<b>Similarity score:<\/b> 0.48<br><b>Intersection:<\/b> Got1, Got2","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Got1, Got2","<b>Similarity score:<\/b> 0.49<br><b>Intersection:<\/b> Acads, Bckdha, Bckdhb, Dld,<br> Hibch, Mcee, Mmut","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> unknown (METAB)","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Dlat, Dld, Fh, Mdh1, Mdh2,<br> Pdha1, Pdhb","<b>Similarity score:<\/b> 0.6<br><b>Intersection:<\/b> Acss2, Dlat, Dld, Fh, Mdh1,<br> Mdh2, Me3, Pdha1, Pdhb","<b>Similarity score:<\/b> 0.66<br><b>Intersection:<\/b> Acss2, Cs, Dld, Gcsh, Mcee,<br> Mdh1, Mdh2, Mmut, Shmt2","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Acads, Acss2, Dld, Hibch,<br> Mcee, Mmut, Sucla2, Suclg1, Suclg2","<b>Similarity score:<\/b> 0.81<br><b>Intersection:<\/b> Cs, Dlat, Dld, Dlst, Fh,<br> Idh3a, Idh3b, Idh3g, Mdh1, Mdh2, Ogdh, Pdha1, Pdhb,<br> Sdha, Sdhb, Sucla2, Suclg1, Suclg2","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Cs, Idh3a, Idh3b, Idh3g","<b>Similarity score:<\/b> 0.6<br><b>Intersection:<\/b> Cs, Got1, Got2, Idh3a, Idh3b,<br> Idh3g","<b>Similarity score:<\/b> 0.56<br><b>Intersection:<\/b> Cox5a, Cox6b1, Cox4i1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Mt-co1, Mt-co3,<br> Mt-cyb, Uqcrq","<b>Similarity score:<\/b> 0.58<br><b>Intersection:<\/b> Cox5a, Cox6b1, Cox4i1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Mt-co1, Mt-co3,<br> Mt-cyb, Uqcrq","<b>Similarity score:<\/b> 0.55<br><b>Intersection:<\/b> Cox5a, Cox6b1, Cox4i1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Mt-co1, Mt-co3,<br> Mt-cyb, Uqcrq","<b>Similarity score:<\/b> 0.67<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Mt-nd1,<br> Mt-nd2, Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.5<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.6<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Ndufa3,<br> Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.55<br><b>Intersection:<\/b> Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufs8, Ndufv1, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.55<br><b>Intersection:<\/b> Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufs8, Ndufv1, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.64<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Mt-nd1,<br> Mt-nd2, Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.58<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.65<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.71<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6, Ndufa8,<br> Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufs8,<br> Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10, Cox4i1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Mt-co1, Mt-co3,<br> Mt-cyb, Ndufa3, Ndufb11, Ndufb4, Uqcrq","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Prkab1, Prkag2, Cox5a, Cox5b, Cox6b1,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb,<br> Uqcr10","<b>Similarity score:<\/b> 0.68<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Cox5a, Cox5b, Cox6b1, Irs1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6, Ndufa8,<br> Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10, Cox4i1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Mt-co1,<br> Mt-co3, Mt-cyb, Ndufa3, Ndufb11, Ndufb4, Uqcrq","<b>Similarity score:<\/b> 0.74<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.8<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.61<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Cox5a, Cox5b, Cox6b1, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa4, Ndufa6, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufs8, Ndufv2,<br> Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.74<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Ppif, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.44<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3, Sdha, Sdhb","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.39<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.74<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.8<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.61<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Cox5a, Cox5b, Cox6b1, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa4, Ndufa6, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufs8, Ndufv2,<br> Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.92<br><b>Intersection:<\/b> Atp5f1a, Atp5f1b, Atp5f1c,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b,<br> Cox6b1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa7, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufs8,<br> Ndufv1, Ndufv2, Ndufv3, Ppif, Sdha, Sdhb, Taf3, Uqcr10","<b>Similarity score:<\/b> 0.74<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Ppif, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.44<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3, Sdha, Sdhb","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.39<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.56<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufb10, Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb,<br> Uqcr10","<b>Similarity score:<\/b> 0.44<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3, Sdha,<br> Sdhb, Atp5mc1, Atp5mf, Mt-atp6, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3,<br> Ndufaf4, Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Sdha, Sdhb","<b>Similarity score:<\/b> 0.42<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3, Pdhb,<br> Sdha, Sdhb, Atp5mc1, Mt-atp6, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.76<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Atp5pf, Atp5po, Etfa, Etfb,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6, Ndufa8,<br> Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3,<br> Ndufv3-ps1, Atp5mc1, Atp5mf, Mt-atp6, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.7<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Etfa, Etfb, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3,<br> Ndufaf4, Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Ndufv3-ps1, Mt-nd1,<br> Mt-nd2, Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.68<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4,<br> Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufv2, Ndufv3, Ndufv3-ps1, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3,<br> Atp5mc1, Atp5mf, Mt-atp6, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.41<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3,<br> Ndufaf4, Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Ndufa3, Ndufb11,<br> Ndufb4","<b>Similarity score:<\/b> 0.49<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3,<br> Atp5mc1, Mt-atp6, Mt-nd1, Mt-nd2, Mt-nd5, Mt-nd6,<br> Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.82<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Etfa, Etfb, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3,<br> Ndufaf4, Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Ndufv3-ps1, Mt-nd1,<br> Mt-nd2, Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.79<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4,<br> Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufv2, Ndufv3, Ndufv3-ps1, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.53<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.53<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.54<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.5<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b,<br> Dnajb1, Fkbp4","<b>Similarity score:<\/b> 0.5<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Camk2g, Hsp90aa1, Hspa1a,<br> Hspa1b","<b>Similarity score:<\/b> 0.8<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b,<br> Dnajb1, Fkbp4","<b>Similarity score:<\/b> 0.58<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b","<b>Similarity score:<\/b> 0.88<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b,<br> Dnajb1, Fkbp4","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Dlat, Dld, Ogdh, Pdhb","<b>Similarity score:<\/b> 0.53<br><b>Intersection:<\/b> Dlat, Dld, Gcsh, Got2, Ogdh,<br> Pdhb","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Dld, Dlat, Ogdh, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.5<br><b>Intersection:<\/b> Dlat, Dld, Ogdh, Pdhb, Pdhx","<b>Similarity score:<\/b> 1<br><b>Intersection:<\/b> Map1lc3a, Map1lc3b, Src","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Dlat, Dld, Pdhb","<b>Similarity score:<\/b> 0.42<br><b>Intersection:<\/b> Dlat, Dld, Pdhb","<b>Similarity score:<\/b> 0.44<br><b>Intersection:<\/b> Dld, Dlat, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.65<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.73<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 1<br><b>Intersection:<\/b> Dap3, Gfm1, Mrpl12, Mrpl16,<br> Mrpl17, mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55,<br> Mrps12, Mrps15, Mrps16, Mrps17, Mrps21, Mrps22,<br> Mrps23, Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Similarity score:<\/b> 0.98<br><b>Intersection:<\/b> Dap3, Mrpl12, Mrpl16, Mrpl17,<br> mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55, Mrps12,<br> Mrps15, Mrps16, Mrps17, Mrps21, Mrps22, Mrps23,<br> Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Similarity score:<\/b> 0.98<br><b>Intersection:<\/b> Dap3, Mrpl12, Mrpl16, Mrpl17,<br> mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55, Mrps12,<br> Mrps15, Mrps16, Mrps17, Mrps21, Mrps22, Mrps23,<br> Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Similarity score:<\/b> 0.41<br><b>Intersection:<\/b> Acsl1, Cbr4, Fasn, Hsd17b8","<b>Similarity score:<\/b> 0.68<br><b>Intersection:<\/b> Acad10, Acadl, Acot11,<br> Acot13, Acot9, Hadh, Mcee, Mmut, Them4","<b>Similarity score:<\/b> 0.86<br><b>Intersection:<\/b> Acad10, Acadl, Acot11,<br> Acot13, Acot9, Acsl1, Cbr4, Eci2, Eci3, Fasn, Hadh,<br> Hsd17b8, Mcee, Mlycd, Mmut, Phyh, Prkag2, Ptges2,<br> Them4","<b>Similarity score:<\/b> 0.46<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.41<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2,<br> Ndufv3","<b>Similarity score:<\/b> 0.41<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Ndufa3, Ndufb11,<br> Ndufb4","<b>Similarity score:<\/b> 0.45<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.85<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4,<br> Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufv2, Ndufv3, Ndufv3-ps1, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.48<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.42<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2,<br> Ndufv3","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Ndufa3, Ndufb11,<br> Ndufb4","<b>Similarity score:<\/b> 0.47<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 1<br><b>Intersection:<\/b> Pfkfb1, Pfkfb3","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Dlat, Dld, Pdhb","<b>Similarity score:<\/b> 0.42<br><b>Intersection:<\/b> Dlat, Dld, Pdhb","<b>Similarity score:<\/b> 0.55<br><b>Intersection:<\/b> Dld, Bsg, Dlat, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.65<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.67<br><b>Intersection:<\/b> Bsg, Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Dld, Hibadh, Hibch","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Dlat, Dld, Gcsh, Got1, Got2,<br> Hibch, Ogdh, Pdhb","<b>Similarity score:<\/b> 0.73<br><b>Intersection:<\/b> Dlat, Dld, Gcsh, Got2, Ogdh,<br> Pdhb, Pdhx","<b>Similarity score:<\/b> 0.67<br><b>Intersection:<\/b> Bckdk, Dld, Hibadh, Hibch,<br> Ppm1k","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Dld, Fh, Idh3b, Mdh2, Ogdh,<br> Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.67<br><b>Intersection:<\/b> Dld, Fh, Idh3b, Mdh2, Me3,<br> Ogdh, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.59<br><b>Intersection:<\/b> Dld, Fh, Idh3b, Mdh2, Me3,<br> Ogdh, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.83<br><b>Intersection:<\/b> Dld, Fh, Idh3b, Mdh2, Me3,<br> Ogdh, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.62<br><b>Intersection:<\/b> Dlat, Dld, Fh, Idh3b, Mdh2,<br> Ogdh, Pdhb, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.46<br><b>Intersection:<\/b> Dlat, Dld, Fh, Mdh2, Me3,<br> Pdhb","<b>Similarity score:<\/b> 0.59<br><b>Intersection:<\/b> Dlat, Dld, Fh, Idh3b, Mdh2,<br> Me3, Ogdh, Pdhb, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.64<br><b>Intersection:<\/b> Dld, Bsg, Dlat, Fh, Idh3b,<br> L2hgdh, Mdh2, Me3, Ogdh, Pdhb, Pdhx, Sdha, Sdhb,<br> Sucla2, Suclg2","<b>Similarity score:<\/b> 0.75<br><b>Intersection:<\/b> Acad10, Acadl, Acot11,<br> Acot13, Acot9, Hadh, Mcee, Mmut, Them4","<b>Similarity score:<\/b> 0.53<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.53<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.54<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 1<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Acsl1, Cbr4, Fasn, Hsd17b8","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Acadl, Acsl1, Cbr4, Fasn,<br> Hadh, Hsd17b8"],"color.color":["#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484"],"physics":[false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],"color.highlight":["#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454"],"labelHighlightBold":[true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot","physics":false},"manipulation":{"enabled":false},"interaction":{"hideEdgesOnZoom":true,"tooltipDelay":10,"zoomSpeed":1},"edges":{"smooth":false},"physics":{"stabilization":false}},"groups":["1","0","3","6","7","8","9","12","14"],"width":null,"height":null,"idselection":{"enabled":false},"byselection":{"enabled":false},"main":null,"submain":null,"footer":null,"background":"rgba(0, 0, 0, 0)","tooltipStay":null,"tooltipStyle":"position: fixed;visibility:hidden;padding: 5px;white-space: nowrap;font-family: verdana;font-size:14px;font-color:#000000;background-color: #f5f4ed;-moz-border-radius: 3px;-webkit-border-radius: 3px;border-radius: 3px;border: 1px solid #808074;box-shadow: 3px 3px 10px rgba(0, 0, 0, 0.2);","igraphlayout":{"type":"square"}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="section level4">
<h4 id="save-similarity-scores">Save similarity scores<a class="anchor" aria-label="anchor" href="#save-similarity-scores"></a>
</h4>
<p>Notice that each time we run this function, it takes more than 10
seconds to execute. If you want to play with the parameters of
<code><a href="../reference/enrichment_network_vis.html">enrichment_network_vis()</a></code> for the same pathway enrichment
results, you can save the pathway similarity scores in a file to avoid
re-computing them each time, which is the most computationally expensive
step in this function.</p>
<p><code>similarity_scores_file</code> must specify an RDS file (suffix
.rds or .RDS). The first time the function is run with a non-null
<code>similarity_scores_file</code> argument, the scores are computed
and saved to the given RDS file. Then future calls to this function with
<code>similarity_scores_file</code> set to an existing file will load
the scores from the file instead of re-computing them. Notice the
dramatic difference in elapsed time for these two function calls.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="../reference/enrichment_network_vis.html">enrichment_network_vis</a></span><span class="op">(</span><span class="va">pw</span>, similarity_scores_file <span class="op">=</span> <span class="st">"/tmp/SKM-GN_8w_F1_M1_scores.RDS"</span>, verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt;    user  system elapsed </span></span>
<span><span class="co">#&gt;   4.944   0.178   5.127</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="../reference/enrichment_network_vis.html">enrichment_network_vis</a></span><span class="op">(</span><span class="va">pw</span>, similarity_scores_file <span class="op">=</span> <span class="st">"/tmp/SKM-GN_8w_F1_M1_scores.RDS"</span>, verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt;    user  system elapsed </span></span>
<span><span class="co">#&gt;   0.385   0.114   0.498</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="remove-metab-only-enrichments">Remove METAB-only enrichments<a class="anchor" aria-label="anchor" href="#remove-metab-only-enrichments"></a>
</h4>
<p>Since metabolites are not readily mapped to genes, they cannot easily
be incorporated into this network where pathway similarity is calculated
using the overlap of gene symbols. Therefore, pathway enrichments driven
exclusively by metabolites are treated differently. If two pathways are
driven exclusively by metabolites, an artificial edge is drawn between
them if at least one non-stopword overlaps between the pathway names or
parent pathway names. Otherwise, the pathways are represented as
singleton nodes. To exclude pathways enriched only by differential
metabolites set <code>include_metab_singletons</code> to
<code>FALSE</code>.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/enrichment_network_vis.html">enrichment_network_vis</a></span><span class="op">(</span><span class="va">pw</span>, include_metab_singletons<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Formatting inputs...</span></span>
<span><span class="co">#&gt; Subsetting feature-to-gene map...</span></span>
<span><span class="co">#&gt; Calculating similarity metric between pairs of enriched pathways...</span></span>
<span><span class="co">#&gt; Constructing graph...</span></span>
<span><span class="co">#&gt; Elapsed time: 4.487s</span></span></code></pre></div>
<div id="htmlwidget-f0b3650be7813f38d129" style="width:960px;height:432.632880098888px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-f0b3650be7813f38d129">{"x":{"nodes":{"id":["KEGG:00051","KEGG:04137","REAC:R-RNO-70171","KEGG:04714","KEGG:05415","KEGG:00190","KEGG:04932","REAC:R-RNO-3371571","REAC:R-RNO-6799198","REAC:R-RNO-611105","REAC:R-RNO-163200","REAC:R-RNO-3371497","REAC:R-RNO-3371556","REAC:R-RNO-1428517","KEGG:00020","KEGG:01200","KEGG:05012","KEGG:05020","KEGG:04723","KEGG:00640","KEGG:00280","KEGG:01212","KEGG:00630","KEGG:03320","KEGG:00620","KEGG:00061","KEGG:00650","KEGG:01210","KEGG:00071","KEGG:00062","REAC:R-RNO-5389840","REAC:R-RNO-5368287","REAC:R-RNO-5419276","REAC:R-RNO-71406","REAC:R-RNO-71403","REAC:R-RNO-8978868","REAC:R-RNO-77289","REAC:R-RNO-389661","REAC:R-RNO-3371568","REAC:R-RNO-556833","REAC:R-RNO-70895","REAC:R-RNO-71291","REAC:R-RNO-70268","REAC:R-RNO-1592230","REAC:R-RNO-5362517","KEGG:04260","KEGG:04670","KEGG:00250","REAC:R-RNO-163210","REAC:R-RNO-8949613","REAC:R-RNO-9663891","REAC:R-RNO-5205647"],"value":[100,100,100,400,900,900,900,400,900,900,900,400,100,900,100,100,100,100,400,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,100,400,100,100,100,100,100,100,100,100],"title":["<b>Fructose and mannose metabolism<\/b> <br>(Carbohydrate metabolism)<br><b>P-val:<\/b> 0.00079<br><b>ATAC:SKM-GN:<\/b> Pfkfb1, Pfkfb3","<b>Mitophagy - animal<\/b> (Transport and <br>catabolism)<br><b>P-val:<\/b> 0.0036<br><b>ATAC:SKM-GN:<\/b> Jun, Ubc","<b>Glycolysis<\/b> (Glucose metabolism)<br><b>P-val:<\/b> 0.0027<br><b>ATAC:SKM-GN:<\/b> Pfkfb1, Pfkfb3","<b>Thermogenesis<\/b> (Environmental adaptation)<br><b>P-val:<\/b> 9.5e-19<br><b>PHOSPHO:SKM-GN:<\/b> Acsl1, Atp5f1a, Atp5f1c,<br> Ndufa7, Ndufb10, Ndufs4, Ndufs7, Ndufv1, Prkab1,<br> Prkag2, Taf3<br><b>PROT:SKM-GN:<\/b> Acsl1, Atp5f1a, Atp5f1b,<br> Atp5f1c, Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Coa5, Coa7,<br> Cox17, Cox19, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6, Ndufa7,<br> Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7, Ndufaf8,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufs8, Ndufv1, Ndufv2, Ndufv3, Prkag2, Sdha,<br> Sdhb, Taf3, Uqcr10","<b>Diabetic cardiomyopathy<\/b> (Cardiovascular <br>disease)<br><b>P-val:<\/b> 1e-35<br><b>PHOSPHO:SKM-GN:<\/b> Atp5f1a, Atp5f1c, Camk2g,<br> Gys1, Ndufa7, Ndufb10, Ndufs4, Ndufs7, Ndufv1,<br> Slc2a4, Taf3<br><b>PROT:SKM-GN:<\/b> Ace, Atp5f1a, Atp5f1b,<br> Atp5f1c, Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b,<br> Cox6b1, Irs1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7,<br> Ndufs8, Ndufv1, Ndufv2, Ndufv3, Pdha1, Pdhb, Ppif,<br> Sdha, Sdhb, Slc2a4, Taf3, Uqcr10<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Cox4i1, Cox5a,<br> Cox6b1, Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b,<br> Mt-atp6, Mt-co1, Mt-co3, Mt-cyb, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa12, Ndufa3, Ndufa6, Ndufb11,<br> Ndufb4, Ndufb9, Ndufs3, Ndufv1, Tbc1d4, Uqcrq","<b>Oxidative phosphorylation<\/b> (Energy <br>metabolism)<br><b>P-val:<\/b> 6.2e-44<br><b>PHOSPHO:SKM-GN:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufb10, Ndufs4, Ndufs7, Ndufv1, Taf3<br><b>PROT:SKM-GN:<\/b> Atp5f1a, Atp5f1b, Atp5f1c,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox17, Cox5a,<br> Cox5b, Cox6b1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufs8,<br> Ndufv1, Ndufv2, Ndufv3, Ppa2, Sdha, Sdhb, Taf3,<br> Uqcr10<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Cox4i1,<br> Cox5a, Cox6b1, Cox7a1, Cox7a2, Cox7a2l2, Cox7b,<br> Cox8b, Mt-atp6, Mt-co1, Mt-co3, Mt-cyb, Mt-nd1,<br> Mt-nd2, Mt-nd5, Mt-nd6, Ndufa12, Ndufa3, Ndufa6,<br> Ndufb11, Ndufb4, Ndufb9, Ndufs3, Ndufv1, Uqcrq","<b>Non-alcoholic fatty liver disease<\/b> <br>(Endocrine and metabolic disease)<br><b>P-val:<\/b> 2.4e-28<br><b>PHOSPHO:SKM-GN:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Prkab1, Prkag2<br><b>PROT:SKM-GN:<\/b> Cox5a, Cox5b, Cox6b1, Irs1,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa7, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufs8, Ndufv1,<br> Ndufv2, Ndufv3, Prkag2, Sdha, Sdhb, Uqcr10<br><b>TRNSCRPT:SKM-GN:<\/b> Cox4i1, Cox5a, Cox6b1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Il6r, Mt-co1,<br> Mt-co3, Mt-cyb, Ndufa12, Ndufa3, Ndufa6, Ndufb11,<br> Ndufb4, Ndufb9, Ndufs3, Ndufv1, Uqcrq","<b>HSF1-dependent transactivation<\/b> (Cellular <br>response to heat stress)<br><b>P-val:<\/b> 1.2e-07<br><b>PHOSPHO:SKM-GN:<\/b> Camk2g, Hsp90aa1, Hspa1a,<br> Hspa1b<br><b>PROT:SKM-GN:<\/b> Dnajb1, Fkbp4, Hsp90aa1,<br> Hspa1a, Hspa1b, Hspb8","<b>Complex I biogenesis<\/b> (Respiratory electron <br>transport)<br><b>P-val:<\/b> 9.9e-33<br><b>PHOSPHO:SKM-GN:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1<br><b>PROT:SKM-GN:<\/b> Acad9, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa7, Ndufa8, Ndufa9,<br> Ndufaf3, Ndufaf4, Ndufaf7, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufv1, Ndufv2,<br> Ndufv3, Ndufv3-ps1<br><b>TRNSCRPT:SKM-GN:<\/b> Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa12, Ndufa3, Ndufa6, Ndufb11, Ndufb4,<br> Ndufb9, Ndufs3, Ndufv1","<b>Respiratory electron transport<\/b> <br>(Respiratory electron transport, ATP synthesis by <br>chemiosmotic coupling, and heat production by uncoupling <br>proteins.)<br><b>P-val:<\/b> 7.9e-34<br><b>PHOSPHO:SKM-GN:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1<br><b>PROT:SKM-GN:<\/b> Acad9, Etfa, Etfb, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa6, Ndufa7, Ndufa8,<br> Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufv1,<br> Ndufv2, Ndufv3, Ndufv3-ps1<br><b>TRNSCRPT:SKM-GN:<\/b> Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa12, Ndufa3, Ndufa6, Ndufb11, Ndufb4,<br> Ndufb9, Ndufs3, Ndufv1","<b>Respiratory electron transport, ATP synthesis <br>by chemiosmotic coupling, and heat production by <br>uncoupling proteins.<\/b> (The citric acid (TCA) <br>cycle and respiratory electron transport)<br><b>P-val:<\/b> 1.6e-35<br><b>PHOSPHO:SKM-GN:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1<br><b>PROT:SKM-GN:<\/b> Acad9, Atp5pf, Atp5po, Etfa,<br> Etfb, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa7, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufv1, Ndufv2, Ndufv3, Ndufv3-ps1<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Mt-atp6,<br> Mt-nd1, Mt-nd2, Mt-nd5, Mt-nd6, Ndufa12, Ndufa3,<br> Ndufa6, Ndufb11, Ndufb4, Ndufb9, Ndufs3, Ndufv1","<b>HSP90 chaperone cycle for steroid hormone <br>receptors (SHR)<\/b> (Cellular responses to stress)<br><b>P-val:<\/b> 7.5e-06<br><b>PHOSPHO:SKM-GN:<\/b> Dync1li2, Hsp90aa1, Hspa1a,<br> Hspa1b<br><b>PROT:SKM-GN:<\/b> Dnaja1, Dnajb1, Fkbp4, Fkbp5,<br> Hsp90aa1, Hspa1a, Hspa1b, Stip1","<b>Cellular response to heat stress<\/b> (Cellular <br>responses to stress)<br><b>P-val:<\/b> 0.001<br><b>PHOSPHO:SKM-GN:<\/b> Bag3, Camk2g, Hsp90aa1,<br> Hspa1a, Hspa1b","<b>The citric acid (TCA) cycle and respiratory <br>electron transport<\/b> (Metabolism)<br><b>P-val:<\/b> 5e-44<br><b>PHOSPHO:SKM-GN:<\/b> Dld, Ndufa7, Ndufb10,<br> Ndufs4, Ndufs7, Ndufv1<br><b>PROT:SKM-GN:<\/b> Acad9, Atp5pf, Atp5po, Bsg,<br> Dlat, Dld, Etfa, Etfb, Fh, Idh3b, L2hgdh, Mdh2, Me3,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa7, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufv1, Ndufv2, Ndufv3, Ndufv3-ps1, Ogdh, Pdhb,<br> Pdhx, Sdha, Sdhb, Sucla2, Suclg2<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Me3,<br> Mt-atp6, Mt-nd1, Mt-nd2, Mt-nd5, Mt-nd6, Ndufa12,<br> Ndufa3, Ndufa6, Ndufb11, Ndufb4, Ndufb9, Ndufs3,<br> Ndufv1, Pdhx","<b>Citrate cycle (TCA cycle)<\/b> (Carbohydrate <br>metabolism)<br><b>P-val:<\/b> 5.5e-16<br><b>PROT:SKM-GN:<\/b> Cs, Dlat, Dld, Dlst, Fh,<br> Idh3a, Idh3b, Idh3g, Mdh1, Mdh2, Ogdh, Pdha1, Pdhb,<br> Sdha, Sdhb, Sucla2, Suclg1, Suclg2","<b>Carbon metabolism<\/b> (NULL)<br><b>P-val:<\/b> 5.7e-14<br><b>PROT:SKM-GN:<\/b> Acads, Acss2, Cs, Dlat, Dld,<br> Dlst, Fh, Gcsh, Got1, Got2, Hibch, Hk2, Idh3a,<br> Idh3b, Idh3g, Mcee, Mdh1, Mdh2, Me3, Mmut, Ogdh,<br> Pdha1, Pdhb, Sdha, Sdhb, Shmt2, Sucla2, Suclg1, Suclg2","<b>Parkinson disease<\/b> (Neurodegenerative <br>disease)<br><b>P-val:<\/b> 1.3e-10<br><b>PROT:SKM-GN:<\/b> Atp5f1a, Atp5f1b, Atp5f1c,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b,<br> Cox6b1, Htra2, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufs8,<br> Ndufv1, Ndufv2, Ndufv3, Ppif, Sdha, Sdhb, Taf3,<br> Trap1, Uqcr10","<b>Prion disease<\/b> (Neurodegenerative disease)<br><b>P-val:<\/b> 1.1e-09<br><b>PROT:SKM-GN:<\/b> Atp5f1a, Atp5f1b, Atp5f1c,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b,<br> Cox6b1, Hspa1a, Hspa1b, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufs8, Ndufv1, Ndufv2, Ndufv3, Ppif, Sdha,<br> Sdhb, Stip1, Taf3, Uqcr10","<b>Retrograde endocannabinoid signaling<\/b> <br>(Nervous system)<br><b>P-val:<\/b> 4.3e-16<br><b>PROT:SKM-GN:<\/b> Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufs8, Ndufv1, Ndufv2, Ndufv3<br><b>TRNSCRPT:SKM-GN:<\/b> Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa12, Ndufa3, Ndufa6, Ndufb11, Ndufb4,<br> Ndufb9, Ndufs3, Ndufv1","<b>Propanoate metabolism<\/b> (Carbohydrate <br>metabolism)<br><b>P-val:<\/b> 6.2e-08<br><b>PROT:SKM-GN:<\/b> Acads, Acss2, Bckdha, Bckdhb,<br> Dld, Hibch, Mcee, Mlycd, Mmut, Sucla2, Suclg1,<br> Suclg2","<b>Valine, leucine and isoleucine degradation<\/b> <br>(Amino acid metabolism)<br><b>P-val:<\/b> 1.5e-07<br><b>PROT:SKM-GN:<\/b> Acaa2, Acadm, Acads, Bckdha,<br> Bckdhb, Dld, Hadh, Hibadh, Hibch, Hsd17b10, Mcee,<br> Mmut, Oxct1","<b>Fatty acid metabolism<\/b> (NULL)<br><b>P-val:<\/b> 2.3e-06<br><b>PROT:SKM-GN:<\/b> Acaa2, Acadl, Acadm, Acads,<br> Acadvl, Acsl1, Cbr4, Fasn, Hadh, Hsd17b8, Mecr,<br> Oxsm","<b>Glyoxylate and dicarboxylate metabolism<\/b> <br>(Carbohydrate metabolism)<br><b>P-val:<\/b> 3.7e-06<br><b>PROT:SKM-GN:<\/b> Acss2, Cs, Dld, Gcsh, Mcee,<br> Mdh1, Mdh2, Mmut, Shmt2","<b>PPAR signaling pathway<\/b> (Endocrine system)<br><b>P-val:<\/b> 3.3e-05<br><b>PROT:SKM-GN:<\/b> Acadl, Acadm, Acsl1, Fabp4,<br> Fabp5, Gk, Me3, Plin2, Plin4, Plin5, Sorbs1","<b>Pyruvate metabolism<\/b> (Carbohydrate <br>metabolism)<br><b>P-val:<\/b> 3.5e-05<br><b>PROT:SKM-GN:<\/b> Acss2, Dlat, Dld, Fh, Ldhd,<br> Mdh1, Mdh2, Me3, Pdha1, Pdhb","<b>Fatty acid biosynthesis<\/b> (Lipid metabolism)<br><b>P-val:<\/b> 9.1e-05<br><b>PROT:SKM-GN:<\/b> Acsl1, Cbr4, Fasn, Hsd17b8,<br> Mecr, Oxsm","<b>Butanoate metabolism<\/b> (Carbohydrate <br>metabolism)<br><b>P-val:<\/b> 9.1e-05<br><b>PROT:SKM-GN:<\/b> Acads, Aldh5a1, Bdh1, Hadh,<br> L2hgdh, Oxct1","<b>2-Oxocarboxylic acid metabolism<\/b> (NULL)<br><b>P-val:<\/b> 9.1e-05<br><b>PROT:SKM-GN:<\/b> Cs, Got1, Got2, Idh3a, Idh3b,<br> Idh3g","<b>Fatty acid degradation<\/b> (Lipid metabolism)<br><b>P-val:<\/b> 0.00013<br><b>PROT:SKM-GN:<\/b> Acaa2, Acadl, Acadm, Acads,<br> Acadvl, Acsl1, Eci2, Eci3, Gcdh, Hadh","<b>Fatty acid elongation<\/b> (Lipid metabolism)<br><b>P-val:<\/b> 0.0014<br><b>PROT:SKM-GN:<\/b> Acaa2, Acot2, Hadh, Mecr,<br> Them4","<b>Mitochondrial translation elongation<\/b> <br>(Mitochondrial translation)<br><b>P-val:<\/b> 5.1e-14<br><b>PROT:SKM-GN:<\/b> Dap3, Gfm1, Mrpl12, Mrpl16,<br> Mrpl17, mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55,<br> Mrps12, Mrps15, Mrps16, Mrps17, Mrps21, Mrps22,<br> Mrps23, Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Mitochondrial translation<\/b> (Translation)<br><b>P-val:<\/b> 1e-13<br><b>PROT:SKM-GN:<\/b> Dap3, Gfm1, Mrpl12, Mrpl16,<br> Mrpl17, mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55,<br> Mrps12, Mrps15, Mrps16, Mrps17, Mrps21, Mrps22,<br> Mrps23, Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Mitochondrial translation termination<\/b> <br>(Mitochondrial translation)<br><b>P-val:<\/b> 5.9e-13<br><b>PROT:SKM-GN:<\/b> Dap3, Mrpl12, Mrpl16, Mrpl17,<br> mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55, Mrps12,<br> Mrps15, Mrps16, Mrps17, Mrps21, Mrps22, Mrps23,<br> Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Pyruvate metabolism and Citric Acid (TCA) <br>cycle<\/b> (The citric acid (TCA) cycle and respiratory <br>electron transport)<br><b>P-val:<\/b> 2.5e-10<br><b>PROT:SKM-GN:<\/b> Bsg, Dlat, Dld, Fh, Idh3b,<br> L2hgdh, Mdh2, Me3, Ogdh, Pdhb, Pdhx, Sdha, Sdhb,<br> Sucla2, Suclg2","<b>Citric acid cycle (TCA cycle)<\/b> (Pyruvate <br>metabolism and Citric Acid (TCA) cycle)<br><b>P-val:<\/b> 7.5e-09<br><b>PROT:SKM-GN:<\/b> Dld, Fh, Idh3b, Mdh2, Me3,<br> Ogdh, Sdha, Sdhb, Sucla2, Suclg2","<b>Fatty acid metabolism<\/b> (Metabolism of <br>lipids)<br><b>P-val:<\/b> 1.2e-08<br><b>PROT:SKM-GN:<\/b> Acad10, Acadl, Acot11, Acot13,<br> Acot9, Acsl1, Cbr4, Eci2, Eci3, Fasn, Hadh,<br> Hsd17b8, Mcee, Mlycd, Mmut, Phyh, Prkag2, Ptges2,<br> Them4","<b>Mitochondrial Fatty Acid Beta-Oxidation<\/b> <br>(Fatty acid metabolism)<br><b>P-val:<\/b> 2.3e-06<br><b>PROT:SKM-GN:<\/b> Acad10, Acadl, Acot11, Acot13,<br> Acot9, Hadh, Mcee, Mmut, Them4","<b>Glyoxylate metabolism and glycine <br>degradation<\/b> (Metabolism of amino acids and derivatives)<br><b>P-val:<\/b> 6.5e-06<br><b>PROT:SKM-GN:<\/b> Dlat, Dld, Gcsh, Got2, Ogdh,<br> Pdhb, Pdhx","<b>Attenuation phase<\/b> (HSF1-dependent <br>transactivation)<br><b>P-val:<\/b> 3.1e-05<br><b>PROT:SKM-GN:<\/b> Dnajb1, Fkbp4, Hsp90aa1,<br> Hspa1a, Hspa1b","<b>Metabolism of lipids<\/b> (Metabolism)<br><b>P-val:<\/b> 0.00027<br><b>PROT:SKM-GN:<\/b> Acad10, Acadl, Acot11, Acot13,<br> Acot9, Acp6, Acsl1, Cbr4, Eci2, Eci3, Fabp5,<br> Fasn, Fdxr, Gpd1l, Hadh, Hsd17b8, Lpin1, Mcee, Mlycd,<br> Mmut, Oxct1, Phyh, Plin3, Prkag2, Ptges2, Them4","<b>Branched-chain amino acid catabolism<\/b> <br>(Metabolism of amino acids and derivatives)<br><b>P-val:<\/b> 0.00026<br><b>PROT:SKM-GN:<\/b> Bckdk, Dld, Hibadh, Hibch,<br> Ppm1k","<b>Metabolism of amino acids and derivatives<\/b> <br>(Metabolism)<br><b>P-val:<\/b> 0.00049<br><b>PROT:SKM-GN:<\/b> Bckdk, Dlat, Dld, Ethe1, Gcdh,<br> Gcsh, Got1, Got2, Hibadh, Hibch, Oat, Ogdh, Pdhb,<br> Pdhx, Ppm1k","<b>Pyruvate metabolism<\/b> (Pyruvate metabolism <br>and Citric Acid (TCA) cycle)<br><b>P-val:<\/b> 0.0014<br><b>PROT:SKM-GN:<\/b> Bsg, Dlat, Dld, Pdhb, Pdhx","<b>Mitochondrial biogenesis<\/b> (Organelle <br>biogenesis and maintenance)<br><b>P-val:<\/b> 3.7e-05<br><b>PROT:SKM-GN:<\/b> Acss2, Atp5pf, Atp5po, Sirt3,<br> Sod2<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Signaling by Retinoic Acid<\/b> (Signaling by <br>Nuclear Receptors)<br><b>P-val:<\/b> 0.0035<br><b>PROT:SKM-GN:<\/b> Dlat, Dld, Fabp5, Pdhb, Pdhx","<b>Cardiac muscle contraction<\/b> (Circulatory <br>system)<br><b>P-val:<\/b> 1.5e-11<br><b>TRNSCRPT:SKM-GN:<\/b> Atp1a2, Cox4i1, Cox5a,<br> Cox6b1, Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b,<br> Mt-co1, Mt-co3, Mt-cyb, Uqcrq","<b>Leukocyte transendothelial migration<\/b> <br>(Immune system)<br><b>P-val:<\/b> 0.0024<br><b>TRNSCRPT:SKM-GN:<\/b> Cdh5, Cxcl12, Jam2,<br> Rapgef4, Vcl","<b>Alanine, aspartate and glutamate <br>metabolism<\/b> (Amino acid metabolism)<br><b>P-val:<\/b> 0.0029<br><b>TRNSCRPT:SKM-GN:<\/b> Aldh5a1, Got1, Got2","<b>Formation of ATP by chemiosmotic coupling<\/b> <br>(Respiratory electron transport, ATP synthesis by <br>chemiosmotic coupling, and heat production by <br>uncoupling proteins.)<br><b>P-val:<\/b> 0.00031<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Cristae formation<\/b> (Mitochondrial <br>biogenesis)<br><b>P-val:<\/b> 0.00031<br><b>TRNSCRPT:SKM-GN:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Selective autophagy<\/b> (Macroautophagy)<br><b>P-val:<\/b> 0.0014<br><b>TRNSCRPT:SKM-GN:<\/b> Map1lc3a, Map1lc3b, Src","<b>Mitophagy<\/b> (Selective autophagy)<br><b>P-val:<\/b> 0.0014<br><b>TRNSCRPT:SKM-GN:<\/b> Map1lc3a, Map1lc3b, Src"],"color.highlight.background":["yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow"],"color.border":["black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black"],"shadow":[false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],"physics":[false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],"borderWidth":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],"borderWidthSelected":[3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3],"pathway_subclass":["Carbohydrate metabolism","Transport and catabolism","Glucose metabolism","Environmental adaptation","Cardiovascular disease","Energy metabolism","Endocrine and metabolic disease","Cellular response to heat stress","Respiratory electron transport","Respiratory electron transport, ATP synthesis by chemiosmotic coupling, and heat production by uncoupling proteins.","The citric acid (TCA) cycle and respiratory electron transport","Cellular responses to stress","Cellular responses to stress","Metabolism","Carbohydrate metabolism","NULL","Neurodegenerative disease","Neurodegenerative disease","Nervous system","Carbohydrate metabolism","Amino acid metabolism","NULL","Carbohydrate metabolism","Endocrine system","Carbohydrate metabolism","Lipid metabolism","Carbohydrate metabolism","NULL","Lipid metabolism","Lipid metabolism","Mitochondrial translation","Translation","Mitochondrial translation","The citric acid (TCA) cycle and respiratory electron transport","Pyruvate metabolism and Citric Acid (TCA) cycle","Metabolism of lipids","Fatty acid metabolism","Metabolism of amino acids and derivatives","HSF1-dependent transactivation","Metabolism","Metabolism of amino acids and derivatives","Metabolism","Pyruvate metabolism and Citric Acid (TCA) cycle","Organelle biogenesis and maintenance","Signaling by Nuclear Receptors","Circulatory system","Immune system","Amino acid metabolism","Respiratory electron transport, ATP synthesis by chemiosmotic coupling, and heat production by uncoupling proteins.","Mitochondrial biogenesis","Macroautophagy","Selective autophagy"],"group":["1: Carbohydrate metabolism (1/2)","0: Carbohydrate metabolism (1/4)","1: Carbohydrate metabolism (1/2)","3: Neurodegenerative disease (2/12)","3: Neurodegenerative disease (2/12)","3: Neurodegenerative disease (2/12)","3: Neurodegenerative disease (2/12)","4: Cellular responses to stress (2/4)","3: Neurodegenerative disease (2/12)","3: Neurodegenerative disease (2/12)","3: Neurodegenerative disease (2/12)","4: Cellular responses to stress (2/4)","4: Cellular responses to stress (2/4)","3: Neurodegenerative disease (2/12)","5: Carbohydrate metabolism (4/13)","5: Carbohydrate metabolism (4/13)","3: Neurodegenerative disease (2/12)","3: Neurodegenerative disease (2/12)","3: Neurodegenerative disease (2/12)","5: Carbohydrate metabolism (4/13)","5: Carbohydrate metabolism (4/13)","6: Lipid metabolism (3/6)","5: Carbohydrate metabolism (4/13)","0: Carbohydrate metabolism (1/4)","5: Carbohydrate metabolism (4/13)","6: Lipid metabolism (3/6)","0: Carbohydrate metabolism (1/4)","5: Carbohydrate metabolism (4/13)","6: Lipid metabolism (3/6)","6: Lipid metabolism (3/6)","9: Mitochondrial translation (2/3)","9: Mitochondrial translation (2/3)","9: Mitochondrial translation (2/3)","5: Carbohydrate metabolism (4/13)","5: Carbohydrate metabolism (4/13)","6: Lipid metabolism (3/6)","6: Lipid metabolism (3/6)","5: Carbohydrate metabolism (4/13)","4: Cellular responses to stress (2/4)","6: Lipid metabolism (3/6)","5: Carbohydrate metabolism (4/13)","5: Carbohydrate metabolism (4/13)","5: Carbohydrate metabolism (4/13)","10: Mitochondrial biogenesis (1/3)","5: Carbohydrate metabolism (4/13)","3: Neurodegenerative disease (2/12)","0: Carbohydrate metabolism (1/4)","5: Carbohydrate metabolism (4/13)","10: Mitochondrial biogenesis (1/3)","10: Mitochondrial biogenesis (1/3)","12: Macroautophagy (1/2)","12: Macroautophagy (1/2)"],"group_number":["1","0","1","3","3","3","3","4","3","3","3","4","4","3","5","5","3","3","3","5","5","6","5","0","5","6","0","5","6","6","9","9","9","5","5","6","6","5","4","6","5","5","5","10","5","3","0","5","10","10","12","12"],"color.background":["#8DD3C7","#FFFFB3","#8DD3C7","#BEBADA","#BEBADA","#BEBADA","#BEBADA","#FB8072","#BEBADA","#BEBADA","#BEBADA","#FB8072","#FB8072","#BEBADA","#80B1D3","#80B1D3","#BEBADA","#BEBADA","#BEBADA","#80B1D3","#80B1D3","#FDB462","#80B1D3","#FFFFB3","#80B1D3","#FDB462","#FFFFB3","#80B1D3","#FDB462","#FDB462","#B3DE69","#B3DE69","#B3DE69","#80B1D3","#80B1D3","#FDB462","#FDB462","#80B1D3","#FB8072","#FDB462","#80B1D3","#80B1D3","#80B1D3","#FCCDE5","#80B1D3","#BEBADA","#FFFFB3","#80B1D3","#FCCDE5","#FCCDE5","#D9D9D9","#D9D9D9"],"color.highlight.border":["#8DD3C7","#FFFFB3","#8DD3C7","#BEBADA","#BEBADA","#BEBADA","#BEBADA","#FB8072","#BEBADA","#BEBADA","#BEBADA","#FB8072","#FB8072","#BEBADA","#80B1D3","#80B1D3","#BEBADA","#BEBADA","#BEBADA","#80B1D3","#80B1D3","#FDB462","#80B1D3","#FFFFB3","#80B1D3","#FDB462","#FFFFB3","#80B1D3","#FDB462","#FDB462","#B3DE69","#B3DE69","#B3DE69","#80B1D3","#80B1D3","#FDB462","#FDB462","#80B1D3","#FB8072","#FDB462","#80B1D3","#80B1D3","#80B1D3","#FCCDE5","#80B1D3","#BEBADA","#FFFFB3","#80B1D3","#FCCDE5","#FCCDE5","#D9D9D9","#D9D9D9"],"x":[-1,-0.743836889408865,-0.93720105409491,-0.389304679477786,-0.469009929239331,-0.396148982657184,-0.537717632548201,0.124723644670541,-0.452039997435162,-0.380084536259528,-0.308035421972906,0.0615886824861551,-0.00867839762790046,-0.11270068837114,0.284077799933344,0.469061521466689,-0.300714904469577,-0.309595248502214,-0.38419923394782,0.805973009649766,0.901718851672665,-0.0583914706584879,0.664695681333776,0.433118340300832,0.351262628577501,-0.223416071717299,0.431536330131871,0.592956275211667,0.0941532667828828,0.113310718336078,0.980243487917006,0.86225124534708,1,0.184986599830582,0.218945366724953,-0.270888087309871,-0.464466952168874,0.23390725455284,-0.080732121552482,-0.398446345197597,0.716507988093471,0.436381652729479,0.146341800600573,-0.213990325821101,0.184704448350647,-0.674847444955762,-0.767448400087322,0.656212510373648,-0.186195457338006,-0.279142101980501,0.674310137957092,0.729134040831238],"y":[0.220276752852224,-0.676930720253997,0.361761039459626,-0.201306438675938,-0.104028243483145,-0.0317363975283015,-0.163815758823814,-0.960339541676258,-0.23518013735292,-0.280938341091951,-0.0554831020890324,-0.856921024101778,-1,-0.14277614405461,-0.160778192448307,-0.142869912840374,-0.161428742137251,-0.241499037337326,-0.126775508827228,-0.188833192401747,-0.391267955508458,0.879924542867613,0.0652325452694416,0.941320583998033,-0.287246588486268,0.945425683629148,0.629584585194919,-0.214115600977902,1,0.792456958159077,0.450633932490091,0.378556641820294,0.298726683729666,-0.15187186684891,-0.0063561152680105,0.81621086118093,0.77155463453964,-0.256699302845263,-0.901118700522727,0.904024622480485,-0.541317616527026,-0.404651589585392,-0.293150033614805,0.35915383981233,-0.359952195971371,-0.0209819384242558,0.631836863823891,-0.407921603262297,0.128939764803656,0.151940597391915,0.800323506603139,0.673858302038751]},"edges":{"from":["REAC:R-RNO-1428517","KEGG:01212","KEGG:01212","KEGG:01212","KEGG:04714","KEGG:05415","KEGG:01210","REAC:R-RNO-71291","KEGG:00640","KEGG:00020","KEGG:01200","KEGG:01200","KEGG:01200","KEGG:00020","KEGG:00020","KEGG:01200","KEGG:00190","KEGG:04932","KEGG:05415","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05012","KEGG:05020","KEGG:05415","REAC:R-RNO-1428517","REAC:R-RNO-163200","REAC:R-RNO-611105","REAC:R-RNO-6799198","KEGG:00190","KEGG:04714","KEGG:05415","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05415","REAC:R-RNO-1428517","REAC:R-RNO-163200","REAC:R-RNO-611105","REAC:R-RNO-6799198","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05012","KEGG:05415","REAC:R-RNO-1428517","REAC:R-RNO-163200","REAC:R-RNO-611105","REAC:R-RNO-6799198","KEGG:04714","KEGG:00190","KEGG:04714","KEGG:05415","REAC:R-RNO-163200","REAC:R-RNO-611105","REAC:R-RNO-6799198","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05415","REAC:R-RNO-611105","REAC:R-RNO-6799198","KEGG:00190","REAC:R-RNO-1428517","REAC:R-RNO-1592230","REAC:R-RNO-163200","REAC:R-RNO-3371571","REAC:R-RNO-3371497","REAC:R-RNO-3371571","REAC:R-RNO-3371497","REAC:R-RNO-3371556","REAC:R-RNO-3371571","KEGG:00020","KEGG:01200","REAC:R-RNO-1428517","REAC:R-RNO-71406","REAC:R-RNO-9663891","KEGG:00020","KEGG:00620","REAC:R-RNO-1428517","REAC:R-RNO-389661","REAC:R-RNO-70268","REAC:R-RNO-71291","REAC:R-RNO-71406","REAC:R-RNO-5389840","REAC:R-RNO-5368287","REAC:R-RNO-5389840","KEGG:00061","REAC:R-RNO-77289","REAC:R-RNO-8978868","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05415","REAC:R-RNO-6799198","KEGG:00190","KEGG:04714","KEGG:04932","KEGG:05415","KEGG:00051","KEGG:00020","KEGG:00620","REAC:R-RNO-1428517","REAC:R-RNO-389661","REAC:R-RNO-71291","REAC:R-RNO-71406","KEGG:00280","KEGG:01200","REAC:R-RNO-389661","REAC:R-RNO-70895","KEGG:00020","KEGG:01200","REAC:R-RNO-1428517","REAC:R-RNO-71406","KEGG:00020","KEGG:00620","KEGG:01200","REAC:R-RNO-1428517","REAC:R-RNO-8978868","KEGG:00190","REAC:R-RNO-1428517","REAC:R-RNO-1592230","REAC:R-RNO-163200","REAC:R-RNO-163210","KEGG:00061","KEGG:01212"],"to":["KEGG:00020","KEGG:00061","KEGG:00062","KEGG:00071","KEGG:00190","KEGG:00190","KEGG:00250","KEGG:00250","KEGG:00280","KEGG:00620","KEGG:00620","KEGG:00630","KEGG:00640","KEGG:01200","KEGG:01210","KEGG:01210","KEGG:04260","KEGG:04260","KEGG:04260","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04723","KEGG:04932","KEGG:04932","KEGG:04932","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05012","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05020","KEGG:05415","REAC:R-RNO-1428517","REAC:R-RNO-1428517","REAC:R-RNO-1428517","REAC:R-RNO-1428517","REAC:R-RNO-1428517","REAC:R-RNO-1428517","REAC:R-RNO-163200","REAC:R-RNO-163200","REAC:R-RNO-163200","REAC:R-RNO-163200","REAC:R-RNO-163200","REAC:R-RNO-163200","REAC:R-RNO-163210","REAC:R-RNO-163210","REAC:R-RNO-163210","REAC:R-RNO-163210","REAC:R-RNO-3371497","REAC:R-RNO-3371556","REAC:R-RNO-3371556","REAC:R-RNO-3371568","REAC:R-RNO-3371568","REAC:R-RNO-3371568","REAC:R-RNO-389661","REAC:R-RNO-389661","REAC:R-RNO-389661","REAC:R-RNO-389661","REAC:R-RNO-5205647","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5362517","REAC:R-RNO-5368287","REAC:R-RNO-5419276","REAC:R-RNO-5419276","REAC:R-RNO-556833","REAC:R-RNO-556833","REAC:R-RNO-556833","REAC:R-RNO-611105","REAC:R-RNO-611105","REAC:R-RNO-611105","REAC:R-RNO-611105","REAC:R-RNO-611105","REAC:R-RNO-6799198","REAC:R-RNO-6799198","REAC:R-RNO-6799198","REAC:R-RNO-6799198","REAC:R-RNO-70171","REAC:R-RNO-70268","REAC:R-RNO-70268","REAC:R-RNO-70268","REAC:R-RNO-70268","REAC:R-RNO-70268","REAC:R-RNO-70268","REAC:R-RNO-70895","REAC:R-RNO-71291","REAC:R-RNO-71291","REAC:R-RNO-71291","REAC:R-RNO-71403","REAC:R-RNO-71403","REAC:R-RNO-71403","REAC:R-RNO-71403","REAC:R-RNO-71406","REAC:R-RNO-71406","REAC:R-RNO-71406","REAC:R-RNO-71406","REAC:R-RNO-77289","REAC:R-RNO-8949613","REAC:R-RNO-8949613","REAC:R-RNO-8949613","REAC:R-RNO-8949613","REAC:R-RNO-8949613","REAC:R-RNO-8978868","REAC:R-RNO-8978868"],"value":[19.7132616487455,37.5,20.3571428571429,31.9444444444444,29.9709724238026,39.4301470588235,23.8095238095238,19.7916666666667,24.3055555555556,25.8333333333333,30,32.7586206896552,25.78125,40.5172413793103,21.6666666666667,30.1724137931034,27.827380952381,29.1666666666667,27.3521505376344,33.3664772727273,25.2403846153846,30.2027925531915,27.2529069767442,27.2529069767442,32.1593237704918,29.0948275862069,32.703488372093,34.5394736842105,35.4166666666667,34.2526964560863,25.9673808762392,33.8068181818182,37.1408045977011,40.1041666666667,30.3197064989518,37.1031746031746,22.0833333333333,21.5277777777778,19.6759259259259,19.9786324786325,37.1408045977011,40.1041666666667,30.3197064989518,45.9795321637427,37.1031746031746,22.0833333333333,21.5277777777778,19.6759259259259,19.9786324786325,27.8432914046122,22.0211161387632,20.125786163522,21.1180124223602,37.7896613190731,34.9308300395257,33.75,26.2443438914027,20.5487033523087,18.7618595825427,24.2851307189542,40.8967391304348,39.375,26.3636363636364,26.3636363636364,34.375,26.875,25,25,34.7222222222222,40,29.1666666666667,43.75,19.047619047619,26.4285714285714,20.0501253132832,25.2100840336134,50,18.75,21.25,21.7857142857143,32.5,36.6666666666667,26.25,26.25,50,49.0740740740741,49.0740740740741,20.3703703703704,34,43,23.2023411371237,20.3113485630066,20.6235697940503,22.3897734231476,42.3214285714286,24.1251803751804,21.1423444976077,21.4772727272727,23.2625164690382,50,18.75,21.25,27.2727272727273,32.5,26.25,33.3333333333333,20,18.8888888888889,36.6666666666667,33.3333333333333,34.3421052631579,33.6206896551724,29.5454545454545,41.6666666666667,30.8333333333333,22.8947368421053,29.375,31.8181818181818,37.5,26.3636363636364,26.3636363636364,34.375,26.875,50,21.6666666666667,18.75],"title":["<b>Similarity score:<\/b> 0.39<br><b>Intersection:<\/b> Dld, Dlat, Fh, Idh3b, Mdh2,<br> Ogdh, Pdhb, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.75<br><b>Intersection:<\/b> Acsl1, Cbr4, Fasn, Hsd17b8,<br> Mecr, Oxsm","<b>Similarity score:<\/b> 0.41<br><b>Intersection:<\/b> Acaa2, Hadh, Mecr","<b>Similarity score:<\/b> 0.64<br><b>Intersection:<\/b> Acaa2, Acadl, Acadm, Acads,<br> Acadvl, Acsl1, Hadh","<b>Similarity score:<\/b> 0.6<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufb10, Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox17, Cox5a, Cox5b,<br> Cox6b1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa4, Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Sdha,<br> Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.79<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufb10, Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb,<br> Uqcr10, Atp5mc1, Cox4i1, Cox7a1, Cox7a2, Cox7a2l2,<br> Cox7b, Cox8b, Mt-atp6, Mt-co1, Mt-co3, Mt-cyb,<br> Mt-nd1, Mt-nd2, Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4,<br> Uqcrq","<b>Similarity score:<\/b> 0.48<br><b>Intersection:<\/b> Got1, Got2","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Got1, Got2","<b>Similarity score:<\/b> 0.49<br><b>Intersection:<\/b> Acads, Bckdha, Bckdhb, Dld,<br> Hibch, Mcee, Mmut","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Dlat, Dld, Fh, Mdh1, Mdh2,<br> Pdha1, Pdhb","<b>Similarity score:<\/b> 0.6<br><b>Intersection:<\/b> Acss2, Dlat, Dld, Fh, Mdh1,<br> Mdh2, Me3, Pdha1, Pdhb","<b>Similarity score:<\/b> 0.66<br><b>Intersection:<\/b> Acss2, Cs, Dld, Gcsh, Mcee,<br> Mdh1, Mdh2, Mmut, Shmt2","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Acads, Acss2, Dld, Hibch,<br> Mcee, Mmut, Sucla2, Suclg1, Suclg2","<b>Similarity score:<\/b> 0.81<br><b>Intersection:<\/b> Cs, Dlat, Dld, Dlst, Fh,<br> Idh3a, Idh3b, Idh3g, Mdh1, Mdh2, Ogdh, Pdha1, Pdhb,<br> Sdha, Sdhb, Sucla2, Suclg1, Suclg2","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Cs, Idh3a, Idh3b, Idh3g","<b>Similarity score:<\/b> 0.6<br><b>Intersection:<\/b> Cs, Got1, Got2, Idh3a, Idh3b,<br> Idh3g","<b>Similarity score:<\/b> 0.56<br><b>Intersection:<\/b> Cox5a, Cox6b1, Cox4i1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Mt-co1, Mt-co3,<br> Mt-cyb, Uqcrq","<b>Similarity score:<\/b> 0.58<br><b>Intersection:<\/b> Cox5a, Cox6b1, Cox4i1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Mt-co1, Mt-co3,<br> Mt-cyb, Uqcrq","<b>Similarity score:<\/b> 0.55<br><b>Intersection:<\/b> Cox5a, Cox6b1, Cox4i1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Mt-co1, Mt-co3,<br> Mt-cyb, Uqcrq","<b>Similarity score:<\/b> 0.67<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Mt-nd1,<br> Mt-nd2, Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.5<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.6<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Ndufa3,<br> Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.55<br><b>Intersection:<\/b> Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufs8, Ndufv1, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.55<br><b>Intersection:<\/b> Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa4, Ndufa6, Ndufa7, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs4, Ndufs5,<br> Ndufs7, Ndufs8, Ndufv1, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.64<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Mt-nd1,<br> Mt-nd2, Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.58<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.65<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.71<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6, Ndufa8,<br> Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufs8,<br> Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10, Cox4i1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Mt-co1, Mt-co3,<br> Mt-cyb, Ndufa3, Ndufb11, Ndufb4, Uqcrq","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Prkab1, Prkag2, Cox5a, Cox5b, Cox6b1,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb,<br> Uqcr10","<b>Similarity score:<\/b> 0.68<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Cox5a, Cox5b, Cox6b1, Irs1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6, Ndufa8,<br> Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10, Cox4i1,<br> Cox7a1, Cox7a2, Cox7a2l2, Cox7b, Cox8b, Mt-co1,<br> Mt-co3, Mt-cyb, Ndufa3, Ndufb11, Ndufb4, Uqcrq","<b>Similarity score:<\/b> 0.74<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.8<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.61<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Cox5a, Cox5b, Cox6b1, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa4, Ndufa6, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufs8, Ndufv2,<br> Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.74<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Ppif, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.44<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3, Sdha, Sdhb","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.39<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.74<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.8<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.61<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Cox5a, Cox5b, Cox6b1, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa4, Ndufa6, Ndufa8, Ndufa9,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufs8, Ndufv2,<br> Ndufv3, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.92<br><b>Intersection:<\/b> Atp5f1a, Atp5f1b, Atp5f1c,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b,<br> Cox6b1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa7, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs4, Ndufs5, Ndufs7, Ndufs8,<br> Ndufv1, Ndufv2, Ndufv3, Ppif, Sdha, Sdhb, Taf3, Uqcr10","<b>Similarity score:<\/b> 0.74<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b, Atp5f1d,<br> Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1, Ndufa10,<br> Ndufa10l1, Ndufa12, Ndufa2, Ndufa4, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufs8, Ndufv2, Ndufv3, Ppif, Sdha, Sdhb, Uqcr10","<b>Similarity score:<\/b> 0.44<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3, Sdha, Sdhb","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.39<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Ndufa7, Ndufs4, Ndufs7,<br> Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6,<br> Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.56<br><b>Intersection:<\/b> Atp5f1a, Atp5f1c, Ndufa7,<br> Ndufb10, Ndufs4, Ndufs7, Ndufv1, Taf3, Atp5f1b,<br> Atp5f1d, Atp5f1e, Atp5pf, Atp5po, Cox5a, Cox5b, Cox6b1,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa4,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufs8, Ndufv2, Ndufv3, Sdha, Sdhb,<br> Uqcr10","<b>Similarity score:<\/b> 0.44<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3, Sdha,<br> Sdhb, Atp5mc1, Atp5mf, Mt-atp6, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3,<br> Ndufaf4, Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Sdha, Sdhb","<b>Similarity score:<\/b> 0.42<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3, Pdhb,<br> Sdha, Sdhb, Atp5mc1, Mt-atp6, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.76<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Atp5pf, Atp5po, Etfa, Etfb,<br> Ndufa10, Ndufa10l1, Ndufa12, Ndufa2, Ndufa6, Ndufa8,<br> Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7, Ndufb9, Ndufs1,<br> Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3,<br> Ndufv3-ps1, Atp5mc1, Atp5mf, Mt-atp6, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.7<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Etfa, Etfb, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3,<br> Ndufaf4, Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Ndufv3-ps1, Mt-nd1,<br> Mt-nd2, Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.68<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4,<br> Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufv2, Ndufv3, Ndufv3-ps1, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3,<br> Atp5mc1, Atp5mf, Mt-atp6, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.41<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3,<br> Ndufaf4, Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Ndufa3, Ndufb11,<br> Ndufb4","<b>Similarity score:<\/b> 0.49<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Atp5pf, Atp5po, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufb9,<br> Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2, Ndufv3,<br> Atp5mc1, Mt-atp6, Mt-nd1, Mt-nd2, Mt-nd5, Mt-nd6,<br> Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.82<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Etfa, Etfb, Ndufa10, Ndufa10l1,<br> Ndufa12, Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3,<br> Ndufaf4, Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3,<br> Ndufs5, Ndufv2, Ndufv3, Ndufv3-ps1, Mt-nd1,<br> Mt-nd2, Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.79<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4,<br> Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufv2, Ndufv3, Ndufv3-ps1, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.53<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.53<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.54<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.5<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b,<br> Dnajb1, Fkbp4","<b>Similarity score:<\/b> 0.5<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Camk2g, Hsp90aa1, Hspa1a,<br> Hspa1b","<b>Similarity score:<\/b> 0.8<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b,<br> Dnajb1, Fkbp4","<b>Similarity score:<\/b> 0.58<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b","<b>Similarity score:<\/b> 0.88<br><b>Intersection:<\/b> Hsp90aa1, Hspa1a, Hspa1b,<br> Dnajb1, Fkbp4","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Dlat, Dld, Ogdh, Pdhb","<b>Similarity score:<\/b> 0.53<br><b>Intersection:<\/b> Dlat, Dld, Gcsh, Got2, Ogdh,<br> Pdhb","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Dld, Dlat, Ogdh, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.5<br><b>Intersection:<\/b> Dlat, Dld, Ogdh, Pdhb, Pdhx","<b>Similarity score:<\/b> 1<br><b>Intersection:<\/b> Map1lc3a, Map1lc3b, Src","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Dlat, Dld, Pdhb","<b>Similarity score:<\/b> 0.42<br><b>Intersection:<\/b> Dlat, Dld, Pdhb","<b>Similarity score:<\/b> 0.44<br><b>Intersection:<\/b> Dld, Dlat, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.65<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.73<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 1<br><b>Intersection:<\/b> Dap3, Gfm1, Mrpl12, Mrpl16,<br> Mrpl17, mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55,<br> Mrps12, Mrps15, Mrps16, Mrps17, Mrps21, Mrps22,<br> Mrps23, Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Similarity score:<\/b> 0.98<br><b>Intersection:<\/b> Dap3, Mrpl12, Mrpl16, Mrpl17,<br> mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55, Mrps12,<br> Mrps15, Mrps16, Mrps17, Mrps21, Mrps22, Mrps23,<br> Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Similarity score:<\/b> 0.98<br><b>Intersection:<\/b> Dap3, Mrpl12, Mrpl16, Mrpl17,<br> mrpl24, Mrpl38, Mrpl47, Mrpl53, Mrpl55, Mrps12,<br> Mrps15, Mrps16, Mrps17, Mrps21, Mrps22, Mrps23,<br> Mrps24, Mrps27, Mrps31, Mrps33, Mrps34, Mrps35,<br> Mrps36, Mrps5, Ptcd3","<b>Similarity score:<\/b> 0.41<br><b>Intersection:<\/b> Acsl1, Cbr4, Fasn, Hsd17b8","<b>Similarity score:<\/b> 0.68<br><b>Intersection:<\/b> Acad10, Acadl, Acot11,<br> Acot13, Acot9, Hadh, Mcee, Mmut, Them4","<b>Similarity score:<\/b> 0.86<br><b>Intersection:<\/b> Acad10, Acadl, Acot11,<br> Acot13, Acot9, Acsl1, Cbr4, Eci2, Eci3, Fasn, Hadh,<br> Hsd17b8, Mcee, Mlycd, Mmut, Phyh, Prkag2, Ptges2,<br> Them4","<b>Similarity score:<\/b> 0.46<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.41<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2,<br> Ndufv3","<b>Similarity score:<\/b> 0.41<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Ndufa3, Ndufb11,<br> Ndufb4","<b>Similarity score:<\/b> 0.45<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.85<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Acad9, Ndufa10, Ndufa10l1, Ndufa12,<br> Ndufa2, Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4,<br> Ndufaf7, Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5,<br> Ndufv2, Ndufv3, Ndufv3-ps1, Mt-nd1, Mt-nd2, Mt-nd5,<br> Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.48<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 0.42<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufaf3, Ndufaf4, Ndufaf7,<br> Ndufb9, Ndufs1, Ndufs2, Ndufs3, Ndufs5, Ndufv2,<br> Ndufv3","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Ndufa3, Ndufb11,<br> Ndufb4","<b>Similarity score:<\/b> 0.47<br><b>Intersection:<\/b> Ndufa7, Ndufb10, Ndufs4,<br> Ndufs7, Ndufv1, Ndufa10, Ndufa10l1, Ndufa12, Ndufa2,<br> Ndufa6, Ndufa8, Ndufa9, Ndufb9, Ndufs1, Ndufs2,<br> Ndufs3, Ndufs5, Ndufv2, Ndufv3, Mt-nd1, Mt-nd2,<br> Mt-nd5, Mt-nd6, Ndufa3, Ndufb11, Ndufb4","<b>Similarity score:<\/b> 1<br><b>Intersection:<\/b> Pfkfb1, Pfkfb3","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Dlat, Dld, Pdhb","<b>Similarity score:<\/b> 0.42<br><b>Intersection:<\/b> Dlat, Dld, Pdhb","<b>Similarity score:<\/b> 0.55<br><b>Intersection:<\/b> Dld, Bsg, Dlat, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.65<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.52<br><b>Intersection:<\/b> Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.67<br><b>Intersection:<\/b> Bsg, Dlat, Dld, Pdhb, Pdhx","<b>Similarity score:<\/b> 0.4<br><b>Intersection:<\/b> Dld, Hibadh, Hibch","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Dlat, Dld, Gcsh, Got1, Got2,<br> Hibch, Ogdh, Pdhb","<b>Similarity score:<\/b> 0.73<br><b>Intersection:<\/b> Dlat, Dld, Gcsh, Got2, Ogdh,<br> Pdhb, Pdhx","<b>Similarity score:<\/b> 0.67<br><b>Intersection:<\/b> Bckdk, Dld, Hibadh, Hibch,<br> Ppm1k","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Dld, Fh, Idh3b, Mdh2, Ogdh,<br> Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.67<br><b>Intersection:<\/b> Dld, Fh, Idh3b, Mdh2, Me3,<br> Ogdh, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.59<br><b>Intersection:<\/b> Dld, Fh, Idh3b, Mdh2, Me3,<br> Ogdh, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.83<br><b>Intersection:<\/b> Dld, Fh, Idh3b, Mdh2, Me3,<br> Ogdh, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.62<br><b>Intersection:<\/b> Dlat, Dld, Fh, Idh3b, Mdh2,<br> Ogdh, Pdhb, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.46<br><b>Intersection:<\/b> Dlat, Dld, Fh, Mdh2, Me3,<br> Pdhb","<b>Similarity score:<\/b> 0.59<br><b>Intersection:<\/b> Dlat, Dld, Fh, Idh3b, Mdh2,<br> Me3, Ogdh, Pdhb, Sdha, Sdhb, Sucla2, Suclg2","<b>Similarity score:<\/b> 0.64<br><b>Intersection:<\/b> Dld, Bsg, Dlat, Fh, Idh3b,<br> L2hgdh, Mdh2, Me3, Ogdh, Pdhb, Pdhx, Sdha, Sdhb,<br> Sucla2, Suclg2","<b>Similarity score:<\/b> 0.75<br><b>Intersection:<\/b> Acad10, Acadl, Acot11,<br> Acot13, Acot9, Hadh, Mcee, Mmut, Them4","<b>Similarity score:<\/b> 0.53<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.53<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.69<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.54<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 1<br><b>Intersection:<\/b> Atp5mc1, Atp5mf, Mt-atp6","<b>Similarity score:<\/b> 0.43<br><b>Intersection:<\/b> Acsl1, Cbr4, Fasn, Hsd17b8","<b>Similarity score:<\/b> 0.38<br><b>Intersection:<\/b> Acadl, Acsl1, Cbr4, Fasn,<br> Hadh, Hsd17b8"],"color.color":["#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484","#848484"],"physics":[false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],"color.highlight":["#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454","#545454"],"labelHighlightBold":[true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot","physics":false},"manipulation":{"enabled":false},"interaction":{"hideEdgesOnZoom":true,"tooltipDelay":10,"zoomSpeed":1},"edges":{"smooth":false},"physics":{"stabilization":false}},"groups":["1: Carbohydrate metabolism (1/2)","0: Carbohydrate metabolism (1/4)","3: Neurodegenerative disease (2/12)","4: Cellular responses to stress (2/4)","5: Carbohydrate metabolism (4/13)","6: Lipid metabolism (3/6)","9: Mitochondrial translation (2/3)","10: Mitochondrial biogenesis (1/3)","12: Macroautophagy (1/2)"],"width":null,"height":null,"idselection":{"enabled":false},"byselection":{"enabled":false},"main":null,"submain":null,"footer":null,"background":"rgba(0, 0, 0, 0)","legend":{"width":0.3,"useGroups":false,"position":"left","ncol":1,"stepX":100,"stepY":40,"zoom":false,"nodes":{"label":["0: Carbohydrate metabolism (1/4)","1: Carbohydrate metabolism (1/2)","3: Neurodegenerative disease (2/12)","4: Cellular responses to stress (2/4)","5: Carbohydrate metabolism (4/13)","6: Lipid metabolism (3/6)","9: Mitochondrial translation (2/3)","10: Mitochondrial biogenesis (1/3)","12: Macroautophagy (1/2)"],"shape":["box","box","box","box","box","box","box","box","box"],"color":["#FFFFB3","#8DD3C7","#BEBADA","#FB8072","#80B1D3","#FDB462","#B3DE69","#FCCDE5","#D9D9D9"],"physics":[false,false,false,false,false,false,false,false,false]},"nodesToDataframe":true},"tooltipStay":null,"tooltipStyle":"position: fixed;visibility:hidden;padding: 5px;white-space: nowrap;font-family: verdana;font-size:14px;font-color:#000000;background-color: #f5f4ed;-moz-border-radius: 3px;-webkit-border-radius: 3px;border-radius: 3px;border: 1px solid #808074;box-shadow: 3px 3px 10px rgba(0, 0, 0, 0.2);","igraphlayout":{"type":"square"}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="section level4">
<h4 id="export-a-network-to-cytoscape">Export a network to Cytoscape<a class="anchor" aria-label="anchor" href="#export-a-network-to-cytoscape"></a>
</h4>
<p>The output of <code><a href="../reference/enrichment_network_vis.html">enrichment_network_vis()</a></code> is nice for
exploration, but it comes short of publication-ready figures. You can
export a network to Cytoscape to manually polish it. For this code to
work, you need to install the <a href="https://bioconductor.org/packages/release/bioc/html/RCy3.html" class="external-link">RCy3
R package</a> and have Cytoscape running locally.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/cytoscape/RCy3" class="external-link">RCy3</a></span><span class="op">)</span></span>
<span><span class="va">vn</span> <span class="op">=</span> <span class="fu"><a href="../reference/enrichment_network_vis.html">enrichment_network_vis</a></span><span class="op">(</span><span class="va">pw</span>,</span>
<span>                            title <span class="op">=</span> <span class="st">"SKM-GN:8w_F1_M1"</span>,</span>
<span>                            return_graph_for_cytoscape <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/pkg/RCy3/man/cytoscapePing.html" class="external-link">cytoscapePing</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/RCy3/man/createNetworkFromIgraph.html" class="external-link">createNetworkFromIgraph</a></span><span class="op">(</span><span class="va">vn</span>, new.title<span class="op">=</span><span class="st">'gastroc-8w-both-up'</span><span class="op">)</span></span></code></pre></div>
<p>Once you’ve imported the network into Cytoscape, you can import the
style sheet found <a href="https://github.com/MoTrPAC/MotrpacRatTraining6mo/blob/vignette/cytoscape/enrichment_network_style.xml" class="external-link">here</a>
and follow <a href="https://docs.google.com/presentation/d/1RvEluLJ1RfAEGsWyiAet1xTuDzxiMOmYqNdWHnxedhI/edit?usp=sharing" class="external-link">these
instructions</a> to quickly make beautiful network with nodes colored by
either tissue or ome. For example: <img src="example.png"></p>
</div>
</div>
<div class="section level3">
<h3 id="printing-networks-within-a-loop">Printing networks within a loop<a class="anchor" aria-label="anchor" href="#printing-networks-within-a-loop"></a>
</h3>
<p>Rendering <code>visNetwork</code> objects within a loop has some
known challenges. See solutions <a href="https://stackoverflow.com/questions/47577894/r-markdown-visnetwork-not-displayed-in-html-output-when-using-for-loops-or-fun/" class="external-link">here</a>.</p>
</div>
</div>
<div class="section level2">
<h2 id="getting-help">Getting help<a class="anchor" aria-label="anchor" href="#getting-help"></a>
</h2>
<p>For questions, bug reporting, and feature requests for this package,
please <a href="https://github.com/MoTrPAC/MotrpacRatTraining6mo/issues" class="external-link">submit a
new issue</a> and include as many details as possible.</p>
<p>If the concern is related to data provided in the
<code>MotrpacRatTraining6moData</code> package, please submit an issue
<a href="https://github.com/MoTrPAC/MotrpacRatTraining6moData/issues" class="external-link">here</a>
instead.</p>
</div>
<div class="section level2">
<h2 id="acknowledgements">Acknowledgements<a class="anchor" aria-label="anchor" href="#acknowledgements"></a>
</h2>
<p>MoTrPAC is supported by the National Institutes of Health (NIH)
Common Fund through cooperative agreements managed by the National
Institute of Diabetes and Digestive and Kidney Diseases (NIDDK),
National Institute of Arthritis and Musculoskeletal Diseases (NIAMS),
and National Institute on Aging (NIA).</p>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: macOS Big Sur ... 10.16</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] KEGGREST_1.36.3                 FELLA_1.16.0                   </span></span>
<span><span class="co">#&gt; [3] foreach_1.5.2                   gprofiler2_0.2.1               </span></span>
<span><span class="co">#&gt; [5] repfdr_1.2.3                    MotrpacRatTraining6moData_1.3.3</span></span>
<span><span class="co">#&gt; [7] MotrpacRatTraining6mo_1.0.1    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] TH.data_1.1-1          colorspace_2.0-3       ellipsis_0.3.2        </span></span>
<span><span class="co">#&gt;   [4] rprojroot_2.0.3        XVector_0.36.0         fs_1.5.2              </span></span>
<span><span class="co">#&gt;   [7] rstudioapi_0.14        farver_2.1.1           graphlayouts_0.8.1    </span></span>
<span><span class="co">#&gt;  [10] ggrepel_0.9.1          fansi_1.0.3            mvtnorm_1.1-3         </span></span>
<span><span class="co">#&gt;  [13] mathjaxr_1.6-0         codetools_0.2-18       splines_4.2.1         </span></span>
<span><span class="co">#&gt;  [16] mnormt_2.1.0           cachem_1.0.6           knitr_1.40            </span></span>
<span><span class="co">#&gt;  [19] TFisher_0.2.0          polyclip_1.10-0        jsonlite_1.8.0        </span></span>
<span><span class="co">#&gt;  [22] png_0.1-7              ggforce_0.3.4          compiler_4.2.1        </span></span>
<span><span class="co">#&gt;  [25] httr_1.4.4             assertthat_0.2.1       Matrix_1.5-1          </span></span>
<span><span class="co">#&gt;  [28] fastmap_1.1.0          lazyeval_0.2.2         limma_3.52.3          </span></span>
<span><span class="co">#&gt;  [31] cli_3.4.0              tweenr_2.0.2           visNetwork_2.1.0      </span></span>
<span><span class="co">#&gt;  [34] htmltools_0.5.3        tools_4.2.1            igraph_1.3.4          </span></span>
<span><span class="co">#&gt;  [37] GenomeInfoDbData_1.2.8 gtable_0.3.1           glue_1.6.2            </span></span>
<span><span class="co">#&gt;  [40] dplyr_1.0.10           Rcpp_1.0.9             slam_0.1-50           </span></span>
<span><span class="co">#&gt;  [43] Biobase_2.56.0         jquerylib_0.1.4        pkgdown_2.0.6         </span></span>
<span><span class="co">#&gt;  [46] vctrs_0.4.1            Biostrings_2.64.1      multtest_2.52.0       </span></span>
<span><span class="co">#&gt;  [49] iterators_1.0.14       ggraph_2.0.6           xfun_0.33             </span></span>
<span><span class="co">#&gt;  [52] stringr_1.4.1          rbibutils_2.2.9        lifecycle_1.0.2       </span></span>
<span><span class="co">#&gt;  [55] qqconf_1.3.0           edgeR_3.38.4           zlibbioc_1.42.0       </span></span>
<span><span class="co">#&gt;  [58] MASS_7.3-58.1          zoo_1.8-11             scales_1.2.1          </span></span>
<span><span class="co">#&gt;  [61] tidygraph_1.2.2        ragg_1.2.2             sandwich_3.0-2        </span></span>
<span><span class="co">#&gt;  [64] RColorBrewer_1.1-3     curl_4.3.2             yaml_2.3.5            </span></span>
<span><span class="co">#&gt;  [67] memoise_2.0.1          gridExtra_2.3          ggplot2_3.3.6         </span></span>
<span><span class="co">#&gt;  [70] sass_0.4.2             stringi_1.7.8          highr_0.9             </span></span>
<span><span class="co">#&gt;  [73] S4Vectors_0.34.0       mutoss_0.1-12          desc_1.4.2            </span></span>
<span><span class="co">#&gt;  [76] plotrix_3.8-2          BiocGenerics_0.42.0    GenomeInfoDb_1.32.4   </span></span>
<span><span class="co">#&gt;  [79] bitops_1.0-7           Rdpack_2.4             rlang_1.0.5           </span></span>
<span><span class="co">#&gt;  [82] pkgconfig_2.0.3        systemfonts_1.0.4      evaluate_0.16         </span></span>
<span><span class="co">#&gt;  [85] lattice_0.20-45        lpsymphony_1.24.0      purrr_0.3.4           </span></span>
<span><span class="co">#&gt;  [88] labeling_0.4.2         htmlwidgets_1.5.4      tidyselect_1.1.2      </span></span>
<span><span class="co">#&gt;  [91] plyr_1.8.7             magrittr_2.0.3         R6_2.5.1              </span></span>
<span><span class="co">#&gt;  [94] IRanges_2.30.1         IHW_1.24.0             generics_0.1.3        </span></span>
<span><span class="co">#&gt;  [97] multcomp_1.4-20        DBI_1.1.3              pillar_1.8.1          </span></span>
<span><span class="co">#&gt; [100] withr_2.5.0            sn_2.1.0               survival_3.4-0        </span></span>
<span><span class="co">#&gt; [103] RCurl_1.98-1.8         tibble_3.1.8           crayon_1.5.1          </span></span>
<span><span class="co">#&gt; [106] fdrtool_1.2.17         utf8_1.2.2             plotly_4.10.0         </span></span>
<span><span class="co">#&gt; [109] rmarkdown_2.16         viridis_0.6.2          locfit_1.5-9.6        </span></span>
<span><span class="co">#&gt; [112] grid_4.2.1             data.table_1.14.2      metap_1.8             </span></span>
<span><span class="co">#&gt; [115] digest_0.6.29          tidyr_1.2.1            numDeriv_2016.8-1.1   </span></span>
<span><span class="co">#&gt; [118] textshaping_0.3.6      stats4_4.2.1           munsell_0.5.0         </span></span>
<span><span class="co">#&gt; [121] viridisLite_0.4.1      bslib_0.4.0</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Nicole Gay, David Amar, Pierre Jean Beltran, MoTrPAC Study Group, NIH Common Fund, National Institute of Diabetes and Digestive and Kidney Diseases (NIDDK), National Institute of Arthritis and Musculoskeletal Diseases (NIAMS), National Institute on Aging (NIA), National Institute of Biomedical Imaging and Bioengineering.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
