#' Wrapper for [DESeq2::DESeq()]
#' 
#' Wrapper to perform multiple pairwise contrasts with [DESeq2::DESeq()] 
#' 
#' @param counts data frame of raw filtered RNA-seq counts. Row names are gene IDs and 
#'   column names are sample IDs. Column names must correspond to values in \code{meta$viallabel}
#' @param meta data frame of metadata with columns \code{c(outcome_of_interest, covar, 'viallabel')} at a minimum. 
#'   \code{counts} are subset to \code{meta$viallabel}.
#' @param covar character vector, adjustment variables to include in the DESeq model
#' @param outcome_of_interest character, outcome of interest to include in the model. 
#'   \code{meta} must include this variable as column. 
#'   \code{contrasts} must include levels in this variable. 
#' @param contrasts list of vectors, where each vector is in the form 
#'   \code{c(outcome_of_interest, numerator, denominator)}, e.g. \code{c('sex_group','female.1w','female.control')}
#' @param dds optional [DESeq2::DESeqResults()] object if it was previously generated
#' @param shrink bool, whether to apply [DESeq2::lfcShrink()]
#' @param verbose bool, whether to print the design string 
#' 
#' @return named list of two items: 
#' \describe{
#'   \item{\code{res}}{data frame of results}
#'   \item{\code{dds}}{[DESeq2::DESeqResults()] object}
#' }
#' 
#' @export 
#' @importFrom DESeq2 DESeqDataSetFromMatrix DESeq lfcShrink
#' @importFrom ashr ash
#' 
#' @examples 
#' \dontrun{
#' # Get 1- and 2- week training effects in female gastrocnemius
#' data = transcript_prep_data("SKM-GN", sex = "female")
#' deseq_res = run_deseq(data$filt_counts, 
#'                       data$metadata,
#'                       data$covariates,
#'                       "group",
#'                       list(c("group","1w","control"), c("group","2w","control")))
#' head(deseq_res$res)
#'                       
#' # Get shrunken effects using the DESeqResults objects generated in the previous step
#' deseq_res_shrunk = run_deseq(data$filt_counts, 
#'                             data$metadata,
#'                             data$covariates,
#'                             "group",
#'                             list(c("group","1w","control"), c("group","2w","control")),
#'                             dds = deseq_res$dds,
#'                             shrink = TRUE)
#' }                           
run_deseq = function(counts, meta, covar, outcome_of_interest, contrasts, dds=NULL, shrink=FALSE, verbose=FALSE){
  
  meta = as.data.table(meta)
  meta[,(outcome_of_interest) := as.factor(get(outcome_of_interest))]
  counts = counts[,as.character(meta[,viallabel])]
  
  # coerce to counts (RSEM does something weird)
  counts = as.data.frame(apply(counts, c(1,2), as.integer)) 
  
  # remove missing values; center and scale covariates
  new = fix_covariates(covar, meta, center_scale = TRUE)
  covar = new$covariates
  meta = data.table(new$meta)
  
  # make contrast
  contrast = paste0('~', paste0(c(outcome_of_interest, covar), collapse=' + '))
  if(verbose) message(contrast)
  
  if(is.null(dds)){
    # run DESeq
    dds = DESeq2::DESeqDataSetFromMatrix(countData = counts,
                                 colData = meta,
                                 design = eval(parse(text=contrast)))
    dds = DESeq2::DESeq(dds, quiet = TRUE)
  }
  
  # get results for each contrast 
  res_list = list()
  for (c in contrasts){
    if(!shrink){
      res = DESeq2::results(dds, contrast = c)
    }else{
      # suppress warnings generated by genes with some 0 counts, e.g.:
      # warning: solve(): system is singular (rcond: 1.45838e-17); attempting approx solution
      # not suppressing warnings anyway?
      
      # make sure ashr is a dependency
      res = suppressWarnings(DESeq2::lfcShrink(dds, 
                                               contrast = c, 
                                               type = 'ashr', 
                                               quiet = TRUE, 
                                               control = list(numiter.em=1000), 
                                               optmethod = 'mixSQP')) # "apeglm" doesn't work with contrasts in this form 
    }
    res_dt = data.table::data.table(gene_id = rownames(counts), 
                                    log2FoldChange = res$log2FoldChange,
                                    lfcSE = res$lfcSE,
                                    pvalue = res$pvalue,
                                    numerator = c[2],
                                    denominator = c[3],
                                    covariates = paste0(covar, collapse=','))
    if(!shrink){
      res_dt[,stat := res$stat]
    }
    res_list[[paste0(c, collapse=' ')]] = res_dt
  }
  all_res = data.table::rbindlist(res_list)
  return(list(res=as.data.frame(all_res),
              dds=dds))
}


#' RNA-seq timewise differential analysis 
#' 
#' Use DESeq2 to perform pairwise contrasts between each group of trained animals
#' and the sex-matched control group for a single tissue. Analysis is performed separately for males and 
#' females. 
#'
#' @param tissue `r tissue()`
#' @param covariates character vector of covariates that correspond to column names of [MotrpacRatTraining6moData::TRNSCRPT_META].
#'   Defaults to covariates that were used for the manuscript. 
#' @param outliers vector of viallabels to exclude during differential analysis. Defaults
#'   to \code{[MotrpacRatTraining6moData::OUTLIERS\$viallabel[[MotrpacRatTraining6moData::OUTLIERS]$assay == "TRNSCRPT"]}
#' @param add_shrunk_logfc boolean, whether to calculate shrunk log fold-changes in addition to standard log fold-changes
#' @param rdata_outfile NULL or path in which to save DESeq2 objects in an RData file 
#' @param overwrite boolean, whether to overwrite the file if \code{rdata_outfile} exists
#' @param verbose boolean, whether to print messages 
#'
#' @return a data frame with one row per gene per contrast (usually 8 rows per gene):
#' \describe{
#'   \item{\code{feature_ID}}{`r feature_ID()`}
#'   \item{\code{sex}}{`r sex()`}
#'   \item{\code{comparison_group}}{`r comparison_group()`}
#'   \item{\code{assay}}{`r assay()`}
#'   \item{\code{assay_code}}{`r assay_code()`}
#'   \item{\code{tissue}}{`r tissue()`}
#'   \item{\code{tissue_code}}{`r tissue_code()`}
#'   \item{\code{covariates}}{character, comma-separated list of adjustment variables}
#'   \item{\code{removed_samples}}{character, comma-separated list of outliers (vial labels) removed from differential analysis}
#'   \item{\code{logFC}}{`r logFC()`}
#'   \item{\code{logFC_se}}{`r logFC_se()`}
#'   \item{\code{shrunk_logFC}}{double, log fold-change shrunk with \code{type = 'ashr'} and \code{optmethod = 'mixSQP'}, 
#'     only if \code{add_shrunk_logfc = TRUE}}
#'   \item{\code{shrunk_logFC_se}}{double, standard error of \code{shrunk_logFC}, only if \code{add_shrunk_logfc = TRUE}}
#'   \item{\code{zscore}}{double, Wald statistic}
#'   \item{\code{p_value}}{`r p_value_da()`}
#'   \item{\code{comparison_average_intensity}}{`r comparison_average_intensity()`}
#'   \item{\code{comparison_average_intensity_se}}{`r comparison_average_intensity_se()`}
#'   \item{\code{reference_average_intensity}}{`r reference_average_intensity()`}
#'   \item{\code{reference_average_intensity_se}}{`r reference_average_intensity_se()`}
#' }
#' 
#' @export 
#'
#' @examples
#' \dontrun{
#' # Perform differential analysis for expressed genes in brown adipose tissue 
#' # with default parameters, i.e., outliers and covariates used for the manuscript; 
#' # calculate both standard and shrunk log fold-changes
#' da = transcript_timewise_da("BAT")
#' 
#' # Same as above but don't calculate shrunk log fold-changes
#' da = transcript_timewise_da("BAT", add_shrunk_logfc = FALSE)
#' 
#' # Same as the first example but save the [DESeq2::DESeq2()] DESeqResults objects in an RData file 
#' da = transcript_timewise_da("BAT", rdata_outfile = "~/test/BAT_RNA_DA.RData", overwrite = TRUE)
#' }
transcript_timewise_da = function(tissue, 
                                   covariates = c('pct_globin', 'RIN', 'pct_umi_dup', 'median_5_3_bias'), 
                                   outliers = na.omit(MotrpacRatTraining6moData::OUTLIERS$viallabel),
                                   add_shrunk_logfc = TRUE, 
                                   rdata_outfile = NULL,
                                   overwrite = FALSE,
                                   verbose = FALSE){
  .tissue = tissue # data.table workaround
  
  check_da_args(.tissue, rdata_outfile, overwrite)
  
  if(verbose) message("Loading data...")
  data = transcript_prep_data(tissue, 
                              covariates = covariates, 
                              outliers = outliers, 
                              center_scale = TRUE, 
                              adjust_covariates = TRUE)
  meta = data.table::as.data.table(data$metadata)
  counts = data$filt_counts
  covariates = data$covariates
  
  sex_res = list()
  dds_list = list()
  for(SEX in unique(meta[,sex])){
    
    if(verbose) message(sprintf("Performing differential expression analysis for %s %ss...", .tissue, SEX))
    
    # subset counts and meta
    curr_meta = meta[sex == SEX]
    curr_counts = counts[,meta[,viallabel]]
    curr_outliers = filter_outliers(TISSUE=tissue, SEX=SEX, outliers=data$outliers)
    
    contrasts = list()
    i = 1
    for (tp in c('1w','2w','4w','8w')){
      contrasts[[i]] = c('group', tp, 'control')
      i = i+1
    }

    if(verbose) message("Calculating standard fold-changes...")
    # standard results 
    deseq_res = run_deseq(curr_counts, # filtered counts
                          curr_meta, # metadata
                          covariates, # covariates
                          outcome_of_interest = 'group', # outcome of interest
                          contrasts = contrasts, # list of contrasts in format c(outcome_of_interest, numerator_level, denominator_level)
                          shrink = FALSE,
                          verbose = verbose)
    
    if(!is.null(rdata_outfile)){
      dds_list[[SEX]] = deseq_res$dds
    }
    
    # shrunk results 
    if(add_shrunk_logfc){
      if(verbose) message("Calculating shrunk fold-changes...")
      deseq_res_shrunk = run_deseq(curr_counts, # filtered counts
                                   curr_meta, # metadata
                                   covariates, # covariates
                                   outcome_of_interest = 'group', # outcome of interest
                                   contrasts = contrasts, # list of contrasts in format c(outcome_of_interest, numerator_level, denominator_level)
                                   shrink = TRUE,
                                   verbose = verbose,
                                   dds = deseq_res$dds)
    }
    
    
    # collect res
    if(add_shrunk_logfc){
      res_shrunk = data.table::data.table(deseq_res_shrunk$res)
      res_nonshrunk = data.table::data.table(deseq_res$res)
      setnames(res_shrunk, c("log2FoldChange", "lfcSE"), c("shrunk_logFC","shrunk_logFC_se"))
      setnames(res_nonshrunk, c("log2FoldChange", "lfcSE", "stat"), c("logFC","logFC_se", "zscore"))
      res_shrunk = res_shrunk[,.(gene_id, shrunk_logFC, shrunk_logFC_se, numerator, denominator)]
      res = merge(res_nonshrunk, res_shrunk, by=c("gene_id","numerator","denominator"))
    }else{
      res = data.table::data.table(deseq_res$res)
      setnames(res, c("log2FoldChange", "lfcSE", "stat"), c("logFC","logFC_se", "zscore"))
    }
    
    setnames(res, c("numerator","pvalue","gene_id"), c("comparison_group","p_value","feature_ID"))
    res[,denominator := NULL]
    
    # add some columns
    res[,sex := SEX]
    res[,removed_samples = ifelse(length(curr_outliers)>0, paste0(curr_outliers, collapse=','), NA_character_)]
    # res[,covariates := paste0(covariates, collapse=',')] added within run_deseq()
    
    # add average intensities 
    norm_counts = as.data.frame(counts(deseq_res$dds, normalized=TRUE))
    ref_sub = norm_counts[,as.character(curr_meta[group == 'control', viallabel])]
    ref_means = rowMeans(ref_sub, na.rm=TRUE)
    ref_se = apply(ref_sub, 1, function(x) stats::sd(x)/sqrt(sum(!is.na(x))) )
    mlist = list()
    i = 1
    for(tp in unique(res[,comparison_group])){
      # get average values
      counts_sub = norm_counts[,as.character(curr_meta[group == tp, viallabel])]
      counts_means = data.table::data.table(sex=SEX,
                                            comparison_group=tp,
                                            comparison_average_intensity=rowMeans(counts_sub, na.rm=TRUE),
                                            comparison_average_intensity_se=apply(counts_sub, 1, 
                                                                                  function(x) stats::sd(x)/sqrt(sum(!is.na(x))) ),
                                            reference_average_intensity=ref_means,
                                            reference_average_intensity_se=ref_se,
                                            feature_ID=rownames(counts_sub))
      mlist[[i]] = counts_means
      i = i+1
    }
    cmeans = data.table::rbindlist(mlist)
    dt = merge(res, cmeans, by=c('feature_ID', 'sex', 'comparison_group'))
    
    sex_res[[SEX]] = dt
  }
  
  dt = data.table::rbindlist(sex_res)
  
  # save DESeq2 RData
  if(!is.null(rdata_outfile)){
    save(dds_list, file=rdata_outfile)
    if(verbose) message(sprintf("'dds_list' saved in 'rdata_outfile': %s", rdata_outfile))
  }
  
  # add columns
  dt[,tissue := .tissue]
  dt[,tissue_code := MotrpacRatTraining6moData::TISSUE_ABBREV_TO_CODE[[.tissue]]]
  dt[,assay_code := 'transcript-rna-seq']
  dt[,assay := 'TRNSCRPT']

  if(add_shrunk_logfc){
    dt = dt[,.(
      feature_ID,
      sex,
      comparison_group,
      assay,      
      assay_code,
      tissue,
      tissue_code,
      covariates,
      removed_samples,
      logFC,
      logFC_se,
      shrunk_logFC,
      shrunk_logFC_se,
      zscore,
      p_value,
      comparison_average_intensity,
      comparison_average_intensity_se,
      reference_average_intensity,
      reference_average_intensity_se
    )]
  }else{
    dt = dt[,.(
      feature_ID,
      sex,
      comparison_group,
      assay,      
      assay_code,
      tissue,
      tissue_code,
      covariates,
      removed_samples,
      logFC,
      logFC_se,
      zscore,
      p_value,
      comparison_average_intensity,
      comparison_average_intensity_se,
      reference_average_intensity,
      reference_average_intensity_se
    )]
  }
  if(verbose) message("Done.")
  return(as.data.frame(dt))
}


#' RNA-seq training differential analysis 
#' 
#' Use DESeq2 to perform a likelihood ratio test to test the effect of training
#' across time points. Analysis is performed separately for males and females. 
#'
#' @param tissue `r tissue()`
#' @param covariates character vector of covariates that correspond to column names of [MotrpacRatTraining6moData::TRNSCRPT_META].
#'   Defaults to covariates that were used for the manuscript. 
#' @param outliers vector of viallabels to exclude during differential analysis. Defaults
#'   to \code{[MotrpacRatTraining6moData::OUTLIERS]$viallabel}
#' @param rdata_outfile NULL or path in which to save DESeq2 objects in an RData file 
#' @param overwrite boolean, whether to overwrite the file if \code{rdata_outfile} exists
#' @param verbose boolean, whether to print messages
#'
#' @return a data frame with one row per gene:
#' \describe{
#'   \item{\code{feature_ID}}{`r feature_ID()`}
#'   \item{\code{assay}}{`r assay()`}
#'   \item{\code{assay_code}}{`r assay_code()`}
#'   \item{\code{tissue}}{`r tissue()`}
#'   \item{\code{tissue_code}}{`r tissue_code()`}
#'   \item{\code{removed_samples_male}}{character, comma-separated list of male outliers (vial labels) removed from differential analysis}
#'   \item{\code{removed_samples_female}}{character, comma-separated list of female outliers (vial labels) removed from differential analysis}
#'   \item{\code{lrt_male}}{double, likelihood ratio test statistic for males}
#'   \item{\code{lrt_female}}{double, likelihood ratio test statistic for females}
#'   \item{\code{p_value_male}}{double, nominal LRT p-value for males}
#'   \item{\code{p_value_female}}{double, nominal LRT p-value for females}
#'   \item{\code{full_model_male}}{character, full model used in LRT for males}
#'   \item{\code{full_model_female}}{character, full model used in LRT for females}
#'   \item{\code{reduced_model_male}}{character, reduced model used in LRT for males}
#'   \item{\code{reduced_model_female}}{character, reduced model used in LRT for females}
#'   \item{\code{p_value}}{double, combined male and female nominal p-value using the sum of logs}
#' }
#' 
#' @export
#' @importFrom metap sumlog
#' @importFrom DESeq2 DESeqDataSetFromMatrix DESeq estimateSizeFactors estimateDispersions nbinomLRT results
#'
#' @examples
#' \dontrun{
#' # Perform differential analysis for expressed genes in brown adipose tissue 
#' # with default parameters, i.e., outliers and covariates used for the manuscript
#' da = transcript_training_da("BAT")
#' 
#' # Same as above but save the [DESeq2::DESeq2()] DESeqResults objects in an RData file 
#' da = transcript_training_da("BAT", 
#'                             rdata_outfile = "~/test/BAT_RNA_training-da.RData", 
#'                             overwrite = TRUE)
#' }
transcript_training_da = function(tissue, 
                                   covariates = c('pct_globin', 'RIN', 'pct_umi_dup', 'median_5_3_bias'), 
                                   outliers = na.omit(MotrpacRatTraining6moData::OUTLIERS$viallabel),
                                   rdata_outfile = NULL,
                                   overwrite = FALSE,
                                   verbose = FALSE){
  
  .tissue = tissue # data.table workaround
  
  check_da_args(.tissue, rdata_outfile, overwrite)
  
  if(verbose) message("Loading data...")
  data = transcript_prep_data(tissue, covariates = covariates, outliers = outliers, center_scale = TRUE, adjust_covariates = TRUE)
  meta = data.table::as.data.table(data$metadata)
  counts = data$filt_counts
  covariates = data$covariates
  
  # refactor group
  meta[,group := factor(group, levels=c("control","1w","2w","4w","8w"))]
  
  sex_res = list()
  dds_list = list()
  for(SEX in unique(meta[,sex])){
    
    if(verbose) message(sprintf("Performing LRTs for %s %ss...", .tissue, SEX))
    
    # subset counts and meta
    curr_meta = meta[sex == SEX]
    curr_counts = counts[,curr_meta[,viallabel]]
    curr_outliers = filter_outliers(TISSUE=tissue, SEX=SEX, outliers=data$outliers)
    curr_meta[,group := as.factor(group)]
    
    full = paste0('~', paste0(c(covariates, 'group'), collapse=' + '))
    reduced = paste0('~', paste0(covariates, collapse=' + '))
    
    dds = DESeq2::DESeqDataSetFromMatrix(countData = curr_counts,
                                 colData = curr_meta,
                                 design = eval(parse(text=full)))
    dds = DESeq2::estimateSizeFactors(dds)
    dds = DESeq2::estimateDispersions(dds)
    dds = DESeq2::nbinomLRT(dds, reduced=eval(parse(text=reduced)), maxit=500)
    dds_list[[SEX]] = dds
    
    res = DESeq2::results(dds)
    res_dt = data.table::data.table(feature_ID = rownames(res), 
                                    lrt = res$stat,
                                    p_value = res$pvalue,
                                    removed_samples = ifelse(length(curr_outliers)>0, paste0(curr_outliers, collapse=','), NA_character_),
                                    full_model=gsub(' ','',full),
                                    reduced_model=gsub(' ','',reduced))
    # add some columns
    res_dt[,assay := "TRNSCRPT"]
    res_dt[,assay_code := "transcript-rna-seq"]
    res_dt[,tissue := .tissue]
    res_dt[,tissue_code := MotrpacRatTraining6moData::TISSUE_ABBREV_TO_CODE[[.tissue]]]
    
    res_dt = res_dt[,.(
      feature_ID,
      assay,      
      assay_code,
      tissue,
      tissue_code,
      removed_samples,
      lrt,
      p_value,
      full_model,
      reduced_model
    )]
    
    sex_res[[SEX]] = res_dt
  }
  
  if(!is.null(rdata_outfile)){
    if(overwrite | (!overwrite & !file.exists(rdata_outfile))){
      save(dds_list, file=rdata_outfile)
      if(verbose) message(sprintf("'dds_list' saved in 'rdata_outfile': %s", rdata_outfile))
    }
  }
  
  if(length(sex_res) > 1){
    male = sex_res[['male']]
    female = sex_res[['female']]
    merged = merge(male, female, by=c('feature_ID','assay','assay_code','tissue','tissue_code'),
                   suffixes=c("_male","_female"), all=TRUE)
    missing = merged[is.na(p_value_male) | is.na(p_value_female)]
    complete = merged[!is.na(p_value_male) & !is.na(p_value_female)]
    complete[, p_value := metap::sumlog(c(p_value_male, p_value_female))$p, by=seq(1, nrow(complete))]
    missing[, p_value := ifelse(is.na(p_value_male), p_value_female, p_value_male)]
    res_dt = rbindlist(list(complete, missing))
  }else{
    res_dt = sex_res[[1]]
    cols = c('removed_samples','lrt','p_value','full_model','reduced_model')
    data.table::setnames(res_dt, old=cols, new=paste0(cols, "_", names(sex_res)[1]))
    res_dt[,p_value := get(sprintf("p_value_%s", names(sex_res)[1]))]
  }
  
  if(verbose) message("Done.")
  return(as.data.frame(res_dt))
}
